{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 13__fimo_v_mind\n",
    "# comparing ChIP overlap b/w FIMO and MIND\n",
    "\n",
    "in this notebook, i compare the accuracy of FIMO and MIND by comparing how often hits from either one overlap with known ChIP-seq peaks. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from os import walk\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from plotting_utils import *\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "chip_dir = \"../../misc/07__chip\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_motif_dir = \"../../data/06__mind_results/HepG2/files/1__motif_scores\"\n",
    "k562_motif_dir = \"../../data/06__mind_results/K562/files/1__motif_scores\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fimo_f = \"../../misc/05__fimo/pool2.fimo.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "index_f = \"../../data/00__index/dels_oligo_pool.index.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del_buffer = 10\n",
    "fdr_cutoff = 0.05"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['enhancer_NEAT1__pX__tile2__minus__peak1.loss__results.txt',\n",
       " 'enhancer_NEAT1__pX__tile2__minus__peak2.loss__results.txt',\n",
       " 'enhancer_MALAT1__pX__tile2__minus__peak1.loss__results.txt',\n",
       " 'enhancer_MALAT1__pX__tile2__minus__peak2.loss__results.txt',\n",
       " 'enhancer_MALAT1__pX__tile2__minus__peak3.loss__results.txt']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_files = []\n",
    "for (dirpath, dirnames, filenames) in walk(hepg2_motif_dir):\n",
    "    hepg2_files.extend(filenames)\n",
    "    break\n",
    "hepg2_files = [x for x in hepg2_files if \"loss__results\" in x and \"expr_filt\" not in x]\n",
    "hepg2_files[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['enhancer_MALAT1__pX__tile1__minus__peak1.loss__results.txt',\n",
       " 'enhancer_MALAT1__pX__tile2__minus__peak1.loss__results.txt',\n",
       " 'enhancer_MALAT1__pX__tile2__minus__peak2.loss__results.txt',\n",
       " 'FALEC__p1__tile2__plus__peak1.loss__results.txt',\n",
       " 'GAS5__p1__tile2__minus__peak1.loss__results.txt']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k562_files = []\n",
    "for (dirpath, dirnames, filenames) in walk(k562_motif_dir):\n",
    "    k562_files.extend(filenames)\n",
    "    break\n",
    "k562_files = [x for x in k562_files if \"loss__results\" in x and \"expr_filt\" not in x]\n",
    "k562_files[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....</td>\n",
       "      <td>39</td>\n",
       "      <td>59</td>\n",
       "      <td>-</td>\n",
       "      <td>27.0204</td>\n",
       "      <td>4.070000e-11</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>GGAGGAGGGAGGGGAGGGGAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__mrna.noflip.NA__p1@IRX5__c...</td>\n",
       "      <td>24</td>\n",
       "      <td>44</td>\n",
       "      <td>+</td>\n",
       "      <td>26.6735</td>\n",
       "      <td>6.730000e-11</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>GGAGGAGGAAGAGGGAGAGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RREB1</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__mrna.flip.sense...</td>\n",
       "      <td>24</td>\n",
       "      <td>43</td>\n",
       "      <td>-</td>\n",
       "      <td>27.0550</td>\n",
       "      <td>3.310000e-10</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>CCCCACCCCACCCCCACCCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RREB1</td>\n",
       "      <td>FLIPPED_DELETION__mrna.flip.antisense__p1@CDKN...</td>\n",
       "      <td>71</td>\n",
       "      <td>90</td>\n",
       "      <td>+</td>\n",
       "      <td>27.0550</td>\n",
       "      <td>3.310000e-10</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>CCCCACCCCACCCCCACCCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....</td>\n",
       "      <td>38</td>\n",
       "      <td>58</td>\n",
       "      <td>-</td>\n",
       "      <td>25.3878</td>\n",
       "      <td>3.590000e-10</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>GAGGGAGGGAGGGGAGGGGGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name                                      sequence name  start  \\\n",
       "0        ZNF263  WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....     39   \n",
       "1        ZNF263  WILDTYPE_SNP_INDIV__mrna.noflip.NA__p1@IRX5__c...     24   \n",
       "2         RREB1  WILDTYPE_BUT_HAS_SNP_DELETION__mrna.flip.sense...     24   \n",
       "3         RREB1  FLIPPED_DELETION__mrna.flip.antisense__p1@CDKN...     71   \n",
       "4        ZNF263  WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....     38   \n",
       "\n",
       "   stop strand    score       p-value   q-value       matched sequence  \n",
       "0    59      -  27.0204  4.070000e-11  0.000024  GGAGGAGGGAGGGGAGGGGAG  \n",
       "1    44      +  26.6735  6.730000e-11  0.000024  GGAGGAGGAAGAGGGAGAGGG  \n",
       "2    43      -  27.0550  3.310000e-10  0.000043   CCCCACCCCACCCCCACCCC  \n",
       "3    90      +  27.0550  3.310000e-10  0.000043   CCCCACCCCACCCCCACCCC  \n",
       "4    58      -  25.3878  3.590000e-10  0.000024  GAGGGAGGGAGGGGAGGGGGG  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = pd.read_table(fimo_f, sep=\"\\t\")\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>full_oligo</th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>tile_number</th>\n",
       "      <th>local_start</th>\n",
       "      <th>local_end</th>\n",
       "      <th>RE_count_1</th>\n",
       "      <th>RE_count_2</th>\n",
       "      <th>RE_count_3</th>\n",
       "      <th>SNP</th>\n",
       "      <th>barcode</th>\n",
       "      <th>element</th>\n",
       "      <th>unique_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>GATTCGTCGTG</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.2</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CGAGATAAGTT</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.3</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CTCGATTAAGG</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.4</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGAATAAGTCA</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.5</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CCTAATCGACA</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          full_oligo            oligo_type  \\\n",
       "0  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "1  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "2  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "3  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "4  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "\n",
       "        oligo_id dupe_info                                           seq_name  \\\n",
       "0  1.2.0.0.0.0.1      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  1.2.0.0.0.0.2      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  1.2.0.0.0.0.3      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  1.2.0.0.0.0.4      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  1.2.0.0.0.0.5      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand                        ...                          \\\n",
       "0  65187718.0      -                        ...                           \n",
       "1  65187718.0      -                        ...                           \n",
       "2  65187718.0      -                        ...                           \n",
       "3  65187718.0      -                        ...                           \n",
       "4  65187718.0      -                        ...                           \n",
       "\n",
       "   tile_number  local_start  local_end  RE_count_1  RE_count_2  RE_count_3  \\\n",
       "0          2.0         94.0      208.0           1           1           0   \n",
       "1          2.0         94.0      208.0           1           1           0   \n",
       "2          2.0         94.0      208.0           1           1           0   \n",
       "3          2.0         94.0      208.0           1           1           0   \n",
       "4          2.0         94.0      208.0           1           1           0   \n",
       "\n",
       "    SNP      barcode                                            element  \\\n",
       "0  none  GATTCGTCGTG  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "1  none  CGAGATAAGTT  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "2  none  CTCGATTAAGG  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "3  none  TGAATAAGTCA  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "4  none  CCTAATCGACA  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "\n",
       "                                           unique_id  \n",
       "0  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "1  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "2  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "3  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "4  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index = pd.read_table(index_f, sep=\"\\t\")\n",
    "index.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. find motifs < 0.05 FDR & put in terms of absolute coords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>padj</th>\n",
       "      <th>neg_log_pval</th>\n",
       "      <th>fdr_cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CREB1</td>\n",
       "      <td>20</td>\n",
       "      <td>28</td>\n",
       "      <td>6.794534</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.002256</td>\n",
       "      <td>2.646604</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gmeb1</td>\n",
       "      <td>15</td>\n",
       "      <td>32</td>\n",
       "      <td>6.216823</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.020562</td>\n",
       "      <td>1.686933</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Creb5</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "      <td>6.029732</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.002256</td>\n",
       "      <td>2.646604</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Atf3</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>6.001356</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.002256</td>\n",
       "      <td>2.646604</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Crem</td>\n",
       "      <td>20</td>\n",
       "      <td>30</td>\n",
       "      <td>5.974204</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.002256</td>\n",
       "      <td>2.646604</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif  start  end     score    pval tile_chr   tile_start     tile_end  \\\n",
       "0  CREB1     20   28  6.794534  0.0001     chr1  173837092.0  173837206.0   \n",
       "1  Gmeb1     15   32  6.216823  0.0020     chr1  173837092.0  173837206.0   \n",
       "2  Creb5     18   30  6.029732  0.0001     chr1  173837092.0  173837206.0   \n",
       "3   Atf3     22   30  6.001356  0.0001     chr1  173837092.0  173837206.0   \n",
       "4   Crem     20   30  5.974204  0.0001     chr1  173837092.0  173837206.0   \n",
       "\n",
       "      strand      padj  neg_log_pval  fdr_cutoff  \n",
       "0  antisense  0.002256      2.646604        0.05  \n",
       "1  antisense  0.020562      1.686933        0.05  \n",
       "2  antisense  0.002256      2.646604        0.05  \n",
       "3  antisense  0.002256      2.646604        0.05  \n",
       "4  antisense  0.002256      2.646604        0.05  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_data = {}\n",
    "for f in hepg2_files:\n",
    "    name = f.split(\".\")[0]\n",
    "    df = pd.read_table(\"%s/%s\" % (hepg2_motif_dir, f), sep=\"\\t\")\n",
    "    hepg2_data[name] = df\n",
    "    #print(\"%s: min motif nuc: %s, max motif nuc: %s\" % (name, df[\"start\"].min(), df[\"end\"].max()))\n",
    "hepg2_data[\"GAS5__p1__tile2__minus__peak1\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>padj</th>\n",
       "      <th>neg_log_pval</th>\n",
       "      <th>fdr_cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CREB1</td>\n",
       "      <td>20</td>\n",
       "      <td>28</td>\n",
       "      <td>6.356575</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.003707</td>\n",
       "      <td>2.431004</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gmeb1</td>\n",
       "      <td>15</td>\n",
       "      <td>32</td>\n",
       "      <td>5.702321</td>\n",
       "      <td>0.0028</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.044032</td>\n",
       "      <td>1.356232</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Creb5</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "      <td>5.651505</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.003707</td>\n",
       "      <td>2.431004</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Atf3</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>5.539003</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.003707</td>\n",
       "      <td>2.431004</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATF7</td>\n",
       "      <td>17</td>\n",
       "      <td>31</td>\n",
       "      <td>5.538545</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>chr1</td>\n",
       "      <td>173837092.0</td>\n",
       "      <td>173837206.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>0.003707</td>\n",
       "      <td>2.431004</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif  start  end     score    pval tile_chr   tile_start     tile_end  \\\n",
       "0  CREB1     20   28  6.356575  0.0001     chr1  173837092.0  173837206.0   \n",
       "1  Gmeb1     15   32  5.702321  0.0028     chr1  173837092.0  173837206.0   \n",
       "2  Creb5     18   30  5.651505  0.0001     chr1  173837092.0  173837206.0   \n",
       "3   Atf3     22   30  5.539003  0.0001     chr1  173837092.0  173837206.0   \n",
       "4   ATF7     17   31  5.538545  0.0001     chr1  173837092.0  173837206.0   \n",
       "\n",
       "      strand      padj  neg_log_pval  fdr_cutoff  \n",
       "0  antisense  0.003707      2.431004        0.05  \n",
       "1  antisense  0.044032      1.356232        0.05  \n",
       "2  antisense  0.003707      2.431004        0.05  \n",
       "3  antisense  0.003707      2.431004        0.05  \n",
       "4  antisense  0.003707      2.431004        0.05  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k562_data = {}\n",
    "for f in k562_files:\n",
    "    name = f.split(\".\")[0]\n",
    "    df = pd.read_table(\"%s/%s\" % (k562_motif_dir, f), sep=\"\\t\")\n",
    "    k562_data[name] = df\n",
    "    #print(\"%s: min motif nuc: %s, max motif nuc: %s\" % (name, df[\"start\"].min(), df[\"end\"].max()))\n",
    "k562_data[\"GAS5__p1__tile2__minus__peak1\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_strand(row):\n",
    "    if row[\"strand\"] == \"sense\":\n",
    "        return \"+\"\n",
    "    else:\n",
    "        return \"-\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>motif</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>73164056</td>\n",
       "      <td>73164070</td>\n",
       "      <td>Tcf21</td>\n",
       "      <td>2.599636</td>\n",
       "      <td>+</td>\n",
       "      <td>JPX__p1__tile1__plus__peak1</td>\n",
       "      <td>Tcf21__chrX__73164056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173837165</td>\n",
       "      <td>173837173</td>\n",
       "      <td>FOXO3</td>\n",
       "      <td>2.235314</td>\n",
       "      <td>+</td>\n",
       "      <td>GAS5__p1__tile2__minus__peak1</td>\n",
       "      <td>FOXO3__chr1__173837165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556062</td>\n",
       "      <td>211556072</td>\n",
       "      <td>ETV4</td>\n",
       "      <td>4.871526</td>\n",
       "      <td>+</td>\n",
       "      <td>LINC00467__p1__tile1__plus__peak1</td>\n",
       "      <td>ETV4__chr1__211556062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3</td>\n",
       "      <td>169482983</td>\n",
       "      <td>169482993</td>\n",
       "      <td>NFATC1</td>\n",
       "      <td>5.952807</td>\n",
       "      <td>-</td>\n",
       "      <td>TERC__p1__tile1__minus__peak1</td>\n",
       "      <td>NFATC1__chr3__169482983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr14</td>\n",
       "      <td>101292392</td>\n",
       "      <td>101292407</td>\n",
       "      <td>JUND(var.2)</td>\n",
       "      <td>4.699084</td>\n",
       "      <td>-</td>\n",
       "      <td>MEG3__p1__tile2__plus__peak1</td>\n",
       "      <td>JUND(var.2)__chr14__101292392</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    tile_chr  motif_start  motif_end        motif     score strand  \\\n",
       "1       chrX     73164056   73164070        Tcf21  2.599636      +   \n",
       "132     chr1    173837165  173837173        FOXO3  2.235314      +   \n",
       "6       chr1    211556062  211556072         ETV4  4.871526      +   \n",
       "16      chr3    169482983  169482993       NFATC1  5.952807      -   \n",
       "9      chr14    101292392  101292407  JUND(var.2)  4.699084      -   \n",
       "\n",
       "                                  gene                             id  \n",
       "1          JPX__p1__tile1__plus__peak1          Tcf21__chrX__73164056  \n",
       "132      GAS5__p1__tile2__minus__peak1         FOXO3__chr1__173837165  \n",
       "6    LINC00467__p1__tile1__plus__peak1          ETV4__chr1__211556062  \n",
       "16       TERC__p1__tile1__minus__peak1        NFATC1__chr3__169482983  \n",
       "9         MEG3__p1__tile2__plus__peak1  JUND(var.2)__chr14__101292392  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_all_motifs = pd.DataFrame()\n",
    "hepg2_top5_motifs = pd.DataFrame()\n",
    "for name in hepg2_data:\n",
    "    df = hepg2_data[name]\n",
    "\n",
    "    if \"plus\" in name:\n",
    "        df[\"motif_start\"] = df[\"tile_start\"].astype(int) + del_buffer + df[\"start\"].astype(int) - 1\n",
    "        df[\"motif_end\"] = df[\"tile_start\"].astype(int) + del_buffer + df[\"start\"].astype(int) + (df[\"end\"]-df[\"start\"]).astype(int) - 1\n",
    "    elif \"minus\" in name:\n",
    "        df[\"motif_end\"] = df[\"tile_end\"].astype(int) - del_buffer - df[\"start\"].astype(int) + 1\n",
    "        df[\"motif_start\"] = df[\"motif_end\"].astype(int) - (df[\"end\"]-df[\"start\"]).astype(int)\n",
    "        \n",
    "    # limit to motifs < fdr cutoff\n",
    "    fdr = df[\"fdr_cutoff\"].iloc[0]\n",
    "    sig_motifs = df[df[\"padj\"] < fdr]\n",
    "    sig_motifs[\"gene\"] = name\n",
    "    sig_motifs[\"id\"] = sig_motifs[\"motif\"].astype(str) + \"__\" + sig_motifs[\"tile_chr\"].astype(str) + \"__\" + sig_motifs[\"motif_start\"].astype(str)\n",
    "    \n",
    "    # for other analysis - limit to top 5\n",
    "    top5_motifs = sig_motifs.sort_values(by=\"score\", ascending=False).head(5)\n",
    "    \n",
    "    hepg2_all_motifs = hepg2_all_motifs.append(sig_motifs)\n",
    "    hepg2_top5_motifs = hepg2_top5_motifs.append(top5_motifs)\n",
    "\n",
    "hepg2_all_motifs[\"strand\"] = hepg2_all_motifs.apply(get_strand, axis=1)\n",
    "hepg2_all_motifs = hepg2_all_motifs[[\"tile_chr\", \"motif_start\", \"motif_end\", \"motif\", \"score\", \"strand\", \"gene\", \"id\"]]\n",
    "hepg2_all_motifs = hepg2_all_motifs.drop_duplicates()\n",
    "hepg2_all_motifs.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>motif</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173837495</td>\n",
       "      <td>173837508</td>\n",
       "      <td>HSF1</td>\n",
       "      <td>2.835706</td>\n",
       "      <td>+</td>\n",
       "      <td>ZBTB37__p1__tile2__plus__peak2</td>\n",
       "      <td>HSF1__chr1__173837495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr7</td>\n",
       "      <td>130794785</td>\n",
       "      <td>130794796</td>\n",
       "      <td>JUNB</td>\n",
       "      <td>4.021405</td>\n",
       "      <td>+</td>\n",
       "      <td>LINC_PINT__p2__tile1__minus__peak1</td>\n",
       "      <td>JUNB__chr7__130794785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5</td>\n",
       "      <td>90610261</td>\n",
       "      <td>90610269</td>\n",
       "      <td>Pax2</td>\n",
       "      <td>5.744323</td>\n",
       "      <td>+</td>\n",
       "      <td>LUCAT1__p1__tile2__minus__peak1</td>\n",
       "      <td>Pax2__chr5__90610261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr3</td>\n",
       "      <td>70048830</td>\n",
       "      <td>70048842</td>\n",
       "      <td>GLI2</td>\n",
       "      <td>2.816638</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMMSON__p1__tile2__plus__peak1</td>\n",
       "      <td>GLI2__chr3__70048830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173837493</td>\n",
       "      <td>173837504</td>\n",
       "      <td>Stat5a::Stat5b</td>\n",
       "      <td>4.157354</td>\n",
       "      <td>+</td>\n",
       "      <td>ZBTB37__p1__tile2__plus__peak2</td>\n",
       "      <td>Stat5a::Stat5b__chr1__173837493</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   tile_chr  motif_start  motif_end           motif     score strand  \\\n",
       "53     chr1    173837495  173837508            HSF1  2.835706      +   \n",
       "8      chr7    130794785  130794796            JUNB  4.021405      +   \n",
       "10     chr5     90610261   90610269            Pax2  5.744323      +   \n",
       "37     chr3     70048830   70048842            GLI2  2.816638      +   \n",
       "25     chr1    173837493  173837504  Stat5a::Stat5b  4.157354      +   \n",
       "\n",
       "                                  gene                               id  \n",
       "53      ZBTB37__p1__tile2__plus__peak2            HSF1__chr1__173837495  \n",
       "8   LINC_PINT__p2__tile1__minus__peak1            JUNB__chr7__130794785  \n",
       "10     LUCAT1__p1__tile2__minus__peak1             Pax2__chr5__90610261  \n",
       "37     SAMMSON__p1__tile2__plus__peak1             GLI2__chr3__70048830  \n",
       "25      ZBTB37__p1__tile2__plus__peak2  Stat5a::Stat5b__chr1__173837493  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k562_all_motifs = pd.DataFrame()\n",
    "k562_top5_motifs = pd.DataFrame()\n",
    "\n",
    "for name in k562_data:\n",
    "    df = k562_data[name]\n",
    "\n",
    "    if \"plus\" in name:\n",
    "        df[\"motif_start\"] = df[\"tile_start\"].astype(int) + del_buffer + df[\"start\"].astype(int) - 1\n",
    "        df[\"motif_end\"] = df[\"tile_start\"].astype(int) + del_buffer + df[\"start\"].astype(int) + (df[\"end\"]-df[\"start\"]).astype(int) - 1\n",
    "    elif \"minus\" in name:\n",
    "        df[\"motif_end\"] = df[\"tile_end\"].astype(int) - del_buffer - df[\"start\"].astype(int) + 1\n",
    "        df[\"motif_start\"] = df[\"motif_end\"].astype(int) - (df[\"end\"]-df[\"start\"]).astype(int)\n",
    "        \n",
    "    # limit to motifs < 0.05\n",
    "    fdr = df[\"fdr_cutoff\"].iloc[0]\n",
    "    sig_motifs = df[df[\"padj\"] < fdr]\n",
    "    sig_motifs[\"gene\"] = name\n",
    "    sig_motifs[\"id\"] = sig_motifs[\"motif\"].astype(str) + \"__\" + sig_motifs[\"tile_chr\"].astype(str) + \"__\" + sig_motifs[\"motif_start\"].astype(str)\n",
    "    \n",
    "    # for other analysis - limit to top 5\n",
    "    top5_motifs = sig_motifs.sort_values(by=\"score\", ascending=False).head(5)\n",
    "    \n",
    "    k562_all_motifs = k562_all_motifs.append(sig_motifs)\n",
    "    k562_top5_motifs = k562_top5_motifs.append(top5_motifs)\n",
    "\n",
    "k562_all_motifs[\"strand\"] = k562_all_motifs.apply(get_strand, axis=1)\n",
    "k562_all_motifs = k562_all_motifs[[\"tile_chr\", \"motif_start\", \"motif_end\", \"motif\", \"score\", \"strand\", \"gene\", \"id\"]]\n",
    "k562_all_motifs = k562_all_motifs.drop_duplicates()\n",
    "k562_all_motifs.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_all_motifs[\"cell_type\"] = \"HepG2\"\n",
    "k562_all_motifs[\"cell_type\"] = \"K562\"\n",
    "all_motifs = hepg2_all_motifs.append(k562_all_motifs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1063"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_motifs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## same for fimo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>105564</th>\n",
       "      <td>RREB1</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...</td>\n",
       "      <td>64</td>\n",
       "      <td>83</td>\n",
       "      <td>-</td>\n",
       "      <td>2.63303</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>0.0181</td>\n",
       "      <td>CCCCCCCTTCCCCCATCCCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56914</th>\n",
       "      <td>TFAP2A(var.3)</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...</td>\n",
       "      <td>92</td>\n",
       "      <td>104</td>\n",
       "      <td>+</td>\n",
       "      <td>11.65570</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.0401</td>\n",
       "      <td>CGCCTGCGGGGCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90986</th>\n",
       "      <td>NR2C2</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>54</td>\n",
       "      <td>68</td>\n",
       "      <td>-</td>\n",
       "      <td>9.05714</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>0.0823</td>\n",
       "      <td>GGGGGTGAAGGTGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102940</th>\n",
       "      <td>TFAP2A(var.3)</td>\n",
       "      <td>FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...</td>\n",
       "      <td>66</td>\n",
       "      <td>78</td>\n",
       "      <td>-</td>\n",
       "      <td>9.60656</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>0.0484</td>\n",
       "      <td>TGCCCCTGTGGGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102979</th>\n",
       "      <td>KLF14</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...</td>\n",
       "      <td>37</td>\n",
       "      <td>50</td>\n",
       "      <td>-</td>\n",
       "      <td>8.01724</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>0.0871</td>\n",
       "      <td>CCTCGCGCCCCCTC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        #pattern name                                      sequence name  \\\n",
       "105564          RREB1  WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...   \n",
       "56914   TFAP2A(var.3)  WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...   \n",
       "90986           NR2C2  FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...   \n",
       "102940  TFAP2A(var.3)  FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...   \n",
       "102979          KLF14  WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...   \n",
       "\n",
       "        start  stop strand     score   p-value  q-value      matched sequence  \n",
       "105564     64    83      -   2.63303  0.000094   0.0181  CCCCCCCTTCCCCCATCCCC  \n",
       "56914      92   104      +  11.65570  0.000038   0.0401         CGCCTGCGGGGCG  \n",
       "90986      54    68      -   9.05714  0.000077   0.0823       GGGGGTGAAGGTGGG  \n",
       "102940     66    78      -   9.60656  0.000090   0.0484         TGCCCCTGTGGGC  \n",
       "102979     37    50      -   8.01724  0.000090   0.0871        CCTCGCGCCCCCTC  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_wt = fimo[~(fimo[\"sequence name\"].str.contains(\"DELETION\")) & \n",
    "               ~(fimo[\"sequence name\"].str.contains(\"SNP_INDIV\"))]\n",
    "fimo_wt.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>KLF16</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@L...</td>\n",
       "      <td>21</td>\n",
       "      <td>31</td>\n",
       "      <td>-</td>\n",
       "      <td>10.4138</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.01850</td>\n",
       "      <td>CCAACTCCCCC</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@L...</td>\n",
       "      <td>chr16</td>\n",
       "      <td>86542457.0</td>\n",
       "      <td>86542571.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>FOXD1</td>\n",
       "      <td>FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...</td>\n",
       "      <td>58</td>\n",
       "      <td>65</td>\n",
       "      <td>-</td>\n",
       "      <td>13.3300</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.34000</td>\n",
       "      <td>GTAAACAA</td>\n",
       "      <td>FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...</td>\n",
       "      <td>chr16</td>\n",
       "      <td>86542551.0</td>\n",
       "      <td>86542665.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1738</th>\n",
       "      <td>ZBTB7C</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>102</td>\n",
       "      <td>113</td>\n",
       "      <td>-</td>\n",
       "      <td>10.7419</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>0.17000</td>\n",
       "      <td>GCCACCACCCCT</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>chr3</td>\n",
       "      <td>169482812.0</td>\n",
       "      <td>169482926.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1725</th>\n",
       "      <td>NFYB</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>14</td>\n",
       "      <td>28</td>\n",
       "      <td>+</td>\n",
       "      <td>11.5192</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.30000</td>\n",
       "      <td>AGCTTGGCCAATCCG</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>chr3</td>\n",
       "      <td>169482812.0</td>\n",
       "      <td>169482926.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>EGR1</td>\n",
       "      <td>WILDTYPE__Antisense.noflip.NA__p2@FLJ43663__ch...</td>\n",
       "      <td>16</td>\n",
       "      <td>29</td>\n",
       "      <td>+</td>\n",
       "      <td>15.9615</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.00168</td>\n",
       "      <td>CCGCCGCCGCCTCC</td>\n",
       "      <td>WILDTYPE__Antisense.noflip.NA__p2@FLJ43663__ch...</td>\n",
       "      <td>chr7</td>\n",
       "      <td>130794716.0</td>\n",
       "      <td>130794830.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #pattern name                                      sequence name  start  \\\n",
       "898          KLF16  WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@L...     21   \n",
       "1042         FOXD1  FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...     58   \n",
       "1738        ZBTB7C  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...    102   \n",
       "1725          NFYB  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...     14   \n",
       "1083          EGR1  WILDTYPE__Antisense.noflip.NA__p2@FLJ43663__ch...     16   \n",
       "\n",
       "      stop strand    score   p-value  q-value matched sequence  \\\n",
       "898     31      -  10.4138  0.000071  0.01850      CCAACTCCCCC   \n",
       "1042    65      -  13.3300  0.000044  0.34000         GTAAACAA   \n",
       "1738   113      -  10.7419  0.000092  0.17000     GCCACCACCCCT   \n",
       "1725    28      +  11.5192  0.000046  0.30000  AGCTTGGCCAATCCG   \n",
       "1083    29      +  15.9615  0.000002  0.00168   CCGCCGCCGCCTCC   \n",
       "\n",
       "                                              unique_id    chr   tile_start  \\\n",
       "898   WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@L...  chr16   86542457.0   \n",
       "1042  FLIPPED__Lincrna.flip.antisense__p1@LOC400550_...  chr16   86542551.0   \n",
       "1738  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...   chr3  169482812.0   \n",
       "1725  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...   chr3  169482812.0   \n",
       "1083  WILDTYPE__Antisense.noflip.NA__p2@FLJ43663__ch...   chr7  130794716.0   \n",
       "\n",
       "         tile_end  \n",
       "898    86542571.0  \n",
       "1042   86542665.0  \n",
       "1738  169482926.0  \n",
       "1725  169482926.0  \n",
       "1083  130794830.0  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_wt = fimo_wt.merge(index[[\"unique_id\", \"chr\", \"tile_start\", \"tile_end\"]].drop_duplicates(), \n",
    "                        left_on=\"sequence name\", right_on=\"unique_id\")\n",
    "fimo_wt.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1730</th>\n",
       "      <td>ZIC3</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>72</td>\n",
       "      <td>86</td>\n",
       "      <td>-</td>\n",
       "      <td>8.6000</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.05910</td>\n",
       "      <td>CGCAACCCGGTGCGC</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>chr3</td>\n",
       "      <td>169482812.0</td>\n",
       "      <td>169482926.0</td>\n",
       "      <td>169482883</td>\n",
       "      <td>169482898</td>\n",
       "      <td>ZIC3__chr3__169482883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1595</th>\n",
       "      <td>STAT1::STAT2</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@MA...</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>-</td>\n",
       "      <td>11.1818</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.14400</td>\n",
       "      <td>CTAGTTTTACTTTTC</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@MA...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65266377.0</td>\n",
       "      <td>65266491.0</td>\n",
       "      <td>65266379</td>\n",
       "      <td>65266394</td>\n",
       "      <td>STAT1::STAT2__chr11__65266379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>NRF1</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>44</td>\n",
       "      <td>54</td>\n",
       "      <td>+</td>\n",
       "      <td>11.5455</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.00923</td>\n",
       "      <td>GCGCCCGCCCC</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>chr9</td>\n",
       "      <td>21994446.0</td>\n",
       "      <td>21994560.0</td>\n",
       "      <td>21994489</td>\n",
       "      <td>21994500</td>\n",
       "      <td>NRF1__chr9__21994489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TFAP2C(var.2)</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>75</td>\n",
       "      <td>85</td>\n",
       "      <td>-</td>\n",
       "      <td>14.3077</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.01930</td>\n",
       "      <td>AGCCTTAGGCA</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>211555971.0</td>\n",
       "      <td>211556085.0</td>\n",
       "      <td>211556045</td>\n",
       "      <td>211556056</td>\n",
       "      <td>TFAP2C(var.2)__chr1__211556045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>SP1</td>\n",
       "      <td>WILDTYPE__Antisense.flip.sense__p1@CDKN2B-AS,p...</td>\n",
       "      <td>66</td>\n",
       "      <td>76</td>\n",
       "      <td>+</td>\n",
       "      <td>15.8462</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.00220</td>\n",
       "      <td>CCCCCACCCCC</td>\n",
       "      <td>WILDTYPE__Antisense.flip.sense__p1@CDKN2B-AS,p...</td>\n",
       "      <td>chr9</td>\n",
       "      <td>21994622.0</td>\n",
       "      <td>21994736.0</td>\n",
       "      <td>21994687</td>\n",
       "      <td>21994698</td>\n",
       "      <td>SP1__chr9__21994687</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      #pattern name                                      sequence name  start  \\\n",
       "1730           ZIC3  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...     72   \n",
       "1595   STAT1::STAT2  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@MA...      3   \n",
       "179            NRF1  FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...     44   \n",
       "16    TFAP2C(var.2)  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...     75   \n",
       "557             SP1  WILDTYPE__Antisense.flip.sense__p1@CDKN2B-AS,p...     66   \n",
       "\n",
       "      stop strand    score   p-value  q-value matched sequence  \\\n",
       "1730    86      -   8.6000  0.000056  0.05910  CGCAACCCGGTGCGC   \n",
       "1595    17      -  11.1818  0.000033  0.14400  CTAGTTTTACTTTTC   \n",
       "179     54      +  11.5455  0.000036  0.00923      GCGCCCGCCCC   \n",
       "16      85      -  14.3077  0.000007  0.01930      AGCCTTAGGCA   \n",
       "557     76      +  15.8462  0.000002  0.00220      CCCCCACCCCC   \n",
       "\n",
       "                                              unique_id    chr   tile_start  \\\n",
       "1730  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...   chr3  169482812.0   \n",
       "1595  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@MA...  chr11   65266377.0   \n",
       "179   FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...   chr9   21994446.0   \n",
       "16    WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...   chr1  211555971.0   \n",
       "557   WILDTYPE__Antisense.flip.sense__p1@CDKN2B-AS,p...   chr9   21994622.0   \n",
       "\n",
       "         tile_end  motif_start  motif_end                              id  \n",
       "1730  169482926.0    169482883  169482898           ZIC3__chr3__169482883  \n",
       "1595   65266491.0     65266379   65266394   STAT1::STAT2__chr11__65266379  \n",
       "179    21994560.0     21994489   21994500            NRF1__chr9__21994489  \n",
       "16    211556085.0    211556045  211556056  TFAP2C(var.2)__chr1__211556045  \n",
       "557    21994736.0     21994687   21994698             SP1__chr9__21994687  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_wt[\"motif_start\"] = fimo_wt[\"tile_start\"].astype(int) + fimo_wt[\"start\"].astype(int) - 1\n",
    "fimo_wt[\"motif_end\"] = fimo_wt[\"tile_start\"].astype(int) + fimo_wt[\"stop\"].astype(int)\n",
    "fimo_wt[\"id\"] = fimo_wt[\"#pattern name\"] + \"__\" + fimo_wt[\"chr\"] + \"__\" + fimo_wt[\"motif_start\"].astype(str)\n",
    "fimo_wt.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>motif</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556009</td>\n",
       "      <td>211556030</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>23.8367</td>\n",
       "      <td>-</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>ZNF263__chr1__211556009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556005</td>\n",
       "      <td>211556026</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>22.1429</td>\n",
       "      <td>-</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>ZNF263__chr1__211556005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556000</td>\n",
       "      <td>211556015</td>\n",
       "      <td>SP2</td>\n",
       "      <td>18.7241</td>\n",
       "      <td>+</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>SP2__chr1__211556000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556000</td>\n",
       "      <td>211556021</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>18.4082</td>\n",
       "      <td>-</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>ZNF263__chr1__211556000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556000</td>\n",
       "      <td>211556011</td>\n",
       "      <td>SP1</td>\n",
       "      <td>16.7692</td>\n",
       "      <td>+</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...</td>\n",
       "      <td>SP1__chr1__211556000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  tile_chr  motif_start  motif_end   motif    score strand  \\\n",
       "0     chr1    211556009  211556030  ZNF263  23.8367      -   \n",
       "1     chr1    211556005  211556026  ZNF263  22.1429      -   \n",
       "2     chr1    211556000  211556015     SP2  18.7241      +   \n",
       "3     chr1    211556000  211556021  ZNF263  18.4082      -   \n",
       "4     chr1    211556000  211556011     SP1  16.7692      +   \n",
       "\n",
       "                                                gene                       id  \n",
       "0  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...  ZNF263__chr1__211556009  \n",
       "1  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...  ZNF263__chr1__211556005  \n",
       "2  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...     SP2__chr1__211556000  \n",
       "3  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...  ZNF263__chr1__211556000  \n",
       "4  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@LI...     SP1__chr1__211556000  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_bed = fimo_wt[[\"chr\", \"motif_start\", \"motif_end\", \"#pattern name\", \"score\", \"strand\", \"sequence name\", \"id\"]]\n",
    "fimo_bed.columns = [\"tile_chr\", \"motif_start\", \"motif_end\", \"motif\", \"score\", \"strand\", \"gene\", \"id\"]\n",
    "fimo_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1759"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fimo_bed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# make df w/ a cell column\n",
    "# since fimo = same across cell types, obviously, just repeat df\n",
    "hepg2_fimo = fimo_bed.copy()\n",
    "hepg2_fimo[\"cell_type\"] = \"HepG2\"\n",
    "k562_fimo = fimo_bed.copy()\n",
    "k562_fimo[\"cell_type\"] = \"K562\"\n",
    "all_fimo = hepg2_fimo.append(k562_fimo)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. intersect w/ bed files for cell type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['K562__ARID3A__narrowPeak_clean.bed.gz',\n",
       " 'K562__ATF1__narrowPeak_clean.bed.gz',\n",
       " 'K562__ATF3__narrowPeak_clean.bed.gz',\n",
       " 'K562__BACH1__narrowPeak_clean.bed.gz',\n",
       " 'K562__BCL3__narrowPeak_clean.bed.gz']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_files = []\n",
    "for (dirpath, dirnames, filenames) in walk(chip_dir):\n",
    "    chip_files.extend(filenames)\n",
    "    break\n",
    "\n",
    "chip_files = [x for x in chip_files if \"__\" in x]\n",
    "hepg2_chip_files = [x for x in chip_files if \"HepG2\" in x]\n",
    "k562_chip_files = [x for x in chip_files if \"K562\" in x]\n",
    "k562_chip_files[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total chip files: 539\n",
      "total HepG2 chip files: 59\n",
      "total K562 chip files: 100\n"
     ]
    }
   ],
   "source": [
    "print(\"total chip files: %s\" % (len(chip_files)))\n",
    "print(\"total HepG2 chip files: %s\" % (len(hepg2_chip_files)))\n",
    "print(\"total K562 chip files: %s\" % (len(k562_chip_files)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ATF3', 'BCL3', 'BHLHE40', 'CEBPB', 'CREB1']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_chip_motifs = [x.split(\"__\")[1].upper() for x in hepg2_chip_files]\n",
    "k562_chip_motifs = [x.split(\"__\")[1].upper() for x in k562_chip_files]\n",
    "chip_motifs = [x.split(\"__\")[1].upper() for x in chip_files]\n",
    "chip_motifs[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "333"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniq_motifs = sorted(list(all_motifs[\"motif\"].unique()))\n",
    "len(uniq_motifs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_overlap(row, int_ids):\n",
    "    if row[\"id\"] in int_ids:\n",
    "        return \"ChIP_overlap\"\n",
    "    else:\n",
    "        return \"no_overlap\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def intersect_chip_files(sub_df, chip_filename, colname):\n",
    "    sub_df.to_csv(\"tmp.motifs.bed\", sep=\"\\t\", index=False, header=False)\n",
    "    !bedtools intersect -u -a tmp.motifs.bed -b $chip_filename > tmp.txt\n",
    "    \n",
    "    try:\n",
    "        results = pd.read_table(\"tmp.txt\", sep=\"\\t\", header=None)\n",
    "        int_ids = list(results[7])\n",
    "        sub_df[colname] = sub_df.apply(get_overlap, int_ids=int_ids, axis=1)\n",
    "    except:\n",
    "        sub_df[colname] = \"no_overlap\"\n",
    "    return sub_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def merge_and_intersect_chip_files(chip_files, motif, chip_dir, sub_df, colname):\n",
    "    chip_files_sub = [x for x in chip_files if motif in x]\n",
    "    chip_paths = [\"%s/%s\" % (chip_dir, x) for x in chip_files_sub]\n",
    "    all_chip = pd.DataFrame()\n",
    "    for f in chip_paths:\n",
    "        tmp = pd.read_table(f, sep=\"\\t\", header=None)\n",
    "        all_chip = all_chip.append(tmp)\n",
    "    all_chip.to_csv(\"tmp.chip.bed\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "    # intersect bed files\n",
    "    sub_df = intersect_chip_files(sub_df, \"tmp.chip.bed\", colname)\n",
    "    return sub_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== ARNT::HIF1A ===\n",
      "=== ATF4 ===\n",
      "=== ATF7 ===\n",
      "=== Ahr::Arnt ===\n",
      "=== Arid3a ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Arid5a ===\n",
      "=== Arnt ===\n",
      "=== Arntl ===\n",
      "=== Ascl2 ===\n",
      "=== Atf1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== Atf3 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== BARHL2 ===\n",
      "=== BATF3 ===\n",
      "=== BATF::JUN ===\n",
      "=== BHLHE22 ===\n",
      "=== BHLHE40 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== BHLHE41 ===\n",
      "=== Bach1::Mafk ===\n",
      "=== Barhl1 ===\n",
      "=== Bcl6 ===\n",
      "=== CDX1 ===\n",
      "=== CDX2 ===\n",
      "=== CEBPB ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== CEBPD ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== CEBPE ===\n",
      "=== CEBPG ===\n",
      "=== CENPB ===\n",
      "=== CREB1 ===\n",
      "has been chipped\n",
      "=== CREB3 ===\n",
      "=== CREB3L1 ===\n",
      "=== Creb3l2 ===\n",
      "=== Creb5 ===\n",
      "=== Crem ===\n",
      "=== DBP ===\n",
      "=== DLX6 ===\n",
      "=== DUX4 ===\n",
      "=== DUXA ===\n",
      "=== Dlx3 ===\n",
      "=== Dlx4 ===\n",
      "=== Dux ===\n",
      "=== E2F4 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== E2F6 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== E2F7 ===\n",
      "=== E2F8 ===\n",
      "=== EBF1 ===\n",
      "has been chipped\n",
      "=== EGR1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== EGR2 ===\n",
      "=== EGR3 ===\n",
      "=== EGR4 ===\n",
      "=== EHF ===\n",
      "=== ELF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== ELF3 ===\n",
      "=== ELF4 ===\n",
      "=== ELF5 ===\n",
      "=== ELK1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ELK3 ===\n",
      "=== ELK4 ===\n",
      "has been chipped\n",
      "=== EMX2 ===\n",
      "=== EN1 ===\n",
      "=== EN2 ===\n",
      "=== ERF ===\n",
      "=== ERG ===\n",
      "=== ESR2 ===\n",
      "=== ETS1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ETV1 ===\n",
      "=== ETV2 ===\n",
      "=== ETV3 ===\n",
      "=== ETV4 ===\n",
      "=== ETV5 ===\n",
      "=== ETV6 ===\n",
      "=== FEV ===\n",
      "=== FIGLA ===\n",
      "=== FLI1 ===\n",
      "=== FOS ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== FOS::JUN ===\n",
      "=== FOSL1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== FOSL2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== FOXC2 ===\n",
      "=== FOXD1 ===\n",
      "=== FOXD2 ===\n",
      "=== FOXF2 ===\n",
      "=== FOXG1 ===\n",
      "=== FOXI1 ===\n",
      "=== FOXL1 ===\n",
      "=== FOXO3 ===\n",
      "=== FOXO4 ===\n",
      "=== FOXO6 ===\n",
      "=== Foxa2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== Foxj2 ===\n",
      "=== Foxj3 ===\n",
      "=== Foxk1 ===\n",
      "=== Foxo1 ===\n",
      "=== Foxq1 ===\n",
      "=== GATA3 ===\n",
      "has been chipped\n",
      "=== GATA5 ===\n",
      "=== GBX2 ===\n",
      "=== GCM1 ===\n",
      "=== GCM2 ===\n",
      "=== GLI2 ===\n",
      "=== GLIS2 ===\n",
      "=== GLIS3 ===\n",
      "=== GMEB2 ===\n",
      "=== GSC ===\n",
      "=== GSX1 ===\n",
      "=== GSX2 ===\n",
      "=== Gabpa ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Gata4 ===\n",
      "=== Gfi1 ===\n",
      "=== Gmeb1 ===\n",
      "=== HES5 ===\n",
      "=== HES7 ===\n",
      "=== HESX1 ===\n",
      "=== HEY1 ===\n",
      "=== HEY2 ===\n",
      "=== HINFP ===\n",
      "=== HLF ===\n",
      "=== HLTF ===\n",
      "=== HNF1A ===\n",
      "=== HNF1B ===\n",
      "=== HOXA10 ===\n",
      "=== HOXA13 ===\n",
      "=== HOXA5 ===\n",
      "=== HOXB13 ===\n",
      "=== HOXB2 ===\n",
      "=== HOXC10 ===\n",
      "=== HOXC11 ===\n",
      "=== HOXC12 ===\n",
      "=== HOXD11 ===\n",
      "=== HOXD12 ===\n",
      "=== HOXD13 ===\n",
      "=== HSF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== HSF2 ===\n",
      "=== HSF4 ===\n",
      "=== Hand1::Tcf3 ===\n",
      "=== Hes1 ===\n",
      "=== Hes2 ===\n",
      "=== Hnf4a ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== Hoxa11 ===\n",
      "=== Hoxd3 ===\n",
      "=== Hoxd9 ===\n",
      "=== ID4 ===\n",
      "=== INSM1 ===\n",
      "=== IRF1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== Id2 ===\n",
      "=== JDP2 ===\n",
      "=== JDP2(var.2) ===\n",
      "=== JUN ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== JUN(var.2) ===\n",
      "=== JUNB ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== JUND ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== JUND(var.2) ===\n",
      "=== KLF13 ===\n",
      "=== KLF14 ===\n",
      "=== KLF16 ===\n",
      "=== KLF5 ===\n",
      "=== Klf1 ===\n",
      "=== Klf12 ===\n",
      "=== Klf4 ===\n",
      "=== LBX2 ===\n",
      "=== LEF1 ===\n",
      "=== LHX2 ===\n",
      "=== LHX6 ===\n",
      "=== LHX9 ===\n",
      "=== Lhx8 ===\n",
      "=== MAF::NFE2 ===\n",
      "=== MAFF ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== MAFG ===\n",
      "=== MAFG::NFE2L1 ===\n",
      "=== MAFK ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== MAX ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== MAX::MYC ===\n",
      "=== MEF2A ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== MEF2B ===\n",
      "=== MEF2C ===\n",
      "has been chipped\n",
      "=== MEIS1 ===\n",
      "=== MEIS2 ===\n",
      "=== MEIS3 ===\n",
      "=== MEOX2 ===\n",
      "=== MGA ===\n",
      "=== MLX ===\n",
      "=== MLXIPL ===\n",
      "=== MNT ===\n",
      "=== MNX1 ===\n",
      "=== MSC ===\n",
      "=== MSX1 ===\n",
      "=== MSX2 ===\n",
      "=== MTF1 ===\n",
      "=== MYBL1 ===\n",
      "=== MYBL2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== MYF6 ===\n",
      "=== MZF1 ===\n",
      "=== MZF1(var.2) ===\n",
      "=== Mafb ===\n",
      "=== Mecom ===\n",
      "=== Mitf ===\n",
      "=== Mlxip ===\n",
      "=== Msx3 ===\n",
      "=== Myb ===\n",
      "=== Myc ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Mycn ===\n",
      "=== Myod1 ===\n",
      "=== Myog ===\n",
      "=== NFAT5 ===\n",
      "=== NFATC1 ===\n",
      "has been chipped\n",
      "=== NFATC2 ===\n",
      "=== NFATC3 ===\n",
      "=== NFE2 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NFIC ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== NFIL3 ===\n",
      "=== NFIX ===\n",
      "=== NFKB1 ===\n",
      "=== NFKB2 ===\n",
      "=== NFYA ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NFYB ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NHLH1 ===\n",
      "=== NKX2-8 ===\n",
      "=== NKX6-1 ===\n",
      "=== NOTO ===\n",
      "=== NR2C2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== NR4A2 ===\n",
      "=== NRF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== NRL ===\n",
      "=== Neurog1 ===\n",
      "=== Nfe2l2 ===\n",
      "=== Nobox ===\n",
      "=== Npas2 ===\n",
      "=== Nr2e1 ===\n",
      "=== Nr2f6 ===\n",
      "=== OLIG3 ===\n",
      "=== OTX1 ===\n",
      "=== PAX4 ===\n",
      "=== PAX5 ===\n",
      "has been chipped\n",
      "=== PBX1 ===\n",
      "=== PITX3 ===\n",
      "=== PKNOX1 ===\n",
      "=== PKNOX2 ===\n",
      "=== POU6F2 ===\n",
      "=== PPARG ===\n",
      "=== PRDM1 ===\n",
      "has been chipped\n",
      "=== PROX1 ===\n",
      "=== Pax2 ===\n",
      "=== Pparg::Rxra ===\n",
      "=== Prrx2 ===\n",
      "=== RARA::RXRA ===\n",
      "=== RAX ===\n",
      "=== RAX2 ===\n",
      "=== REL ===\n",
      "=== RELA ===\n",
      "has been chipped\n",
      "=== RFX4 ===\n",
      "=== RORA ===\n",
      "=== RUNX1 ===\n",
      "=== RUNX2 ===\n",
      "=== Rarg(var.2) ===\n",
      "=== Rfx1 ===\n",
      "=== SCRT1 ===\n",
      "=== SCRT2 ===\n",
      "=== SHOX ===\n",
      "=== SMAD2::SMAD3::SMAD4 ===\n",
      "=== SMAD3 ===\n",
      "=== SNAI2 ===\n",
      "=== SOX10 ===\n",
      "=== SOX21 ===\n",
      "=== SOX4 ===\n",
      "=== SP1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== SP2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== SP3 ===\n",
      "=== SP4 ===\n",
      "has been chipped\n",
      "=== SP8 ===\n",
      "=== SPDEF ===\n",
      "=== SPI1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== SPIB ===\n",
      "=== SPIC ===\n",
      "=== SREBF1 ===\n",
      "=== SREBF2 ===\n",
      "=== SREBF2(var.2) ===\n",
      "=== SRF ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== STAT1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== STAT1::STAT2 ===\n",
      "=== STAT3 ===\n",
      "has been chipped\n",
      "=== Sox11 ===\n",
      "=== Sox17 ===\n",
      "=== Spz1 ===\n",
      "=== Srebf1(var.2) ===\n",
      "=== Stat4 ===\n",
      "=== Stat5a::Stat5b ===\n",
      "=== Stat6 ===\n",
      "=== TAL1::TCF3 ===\n",
      "=== TBP ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== TBR1 ===\n",
      "=== TBX1 ===\n",
      "=== TBX15 ===\n",
      "=== TBX21 ===\n",
      "=== TBX4 ===\n",
      "=== TBX5 ===\n",
      "=== TCF3 ===\n",
      "has been chipped\n",
      "=== TCF4 ===\n",
      "=== TCF7L2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== TEAD1 ===\n",
      "=== TEAD3 ===\n",
      "=== TEAD4 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== TEF ===\n",
      "=== TFAP2A ===\n",
      "has been chipped\n",
      "=== TFAP2A(var.3) ===\n",
      "=== TFAP2B(var.3) ===\n",
      "=== TFAP2C(var.3) ===\n",
      "=== TFAP4 ===\n",
      "=== TFE3 ===\n",
      "=== TFEB ===\n",
      "=== TFEC ===\n",
      "=== TGIF2 ===\n",
      "=== THAP1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== Tcf12 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== Tcf21 ===\n",
      "=== Tcf7 ===\n",
      "=== Tcfl5 ===\n",
      "=== Twist2 ===\n",
      "=== USF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== USF2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== VSX2 ===\n",
      "=== XBP1 ===\n",
      "=== YY1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== YY2 ===\n",
      "=== ZBTB18 ===\n",
      "=== ZBTB33 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== ZEB1 ===\n",
      "has been chipped\n",
      "=== ZIC1 ===\n",
      "=== ZIC3 ===\n",
      "=== ZIC4 ===\n",
      "=== ZNF263 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ZNF354C ===\n",
      "=== ZNF740 ===\n",
      "=== Zfx ===\n",
      "=== Znf423 ===\n"
     ]
    }
   ],
   "source": [
    "all_motifs_int = pd.DataFrame()\n",
    "n_chip = 0\n",
    "n_cell_chip = 0\n",
    "\n",
    "for motif in uniq_motifs:\n",
    "    print(\"=== %s ===\" % motif)\n",
    "    sub_df = all_motifs[all_motifs[\"motif\"] == motif]\n",
    "    \n",
    "    # first check if these motifs overlap *any* chip peaks\n",
    "    # sub_df = intersect_chip_files(sub_df, \"%s/all.chip.bed.gz\" % chip_dir, \"any_chip_overlap\")\n",
    "    \n",
    "    # force motif to upper\n",
    "    motif_up = motif.upper()\n",
    "    \n",
    "    # see if that motif has been chipped at all\n",
    "    if motif_up in chip_motifs:\n",
    "        print(\"has been chipped\")\n",
    "        n_chip += 1\n",
    "\n",
    "        sub_df = merge_and_intersect_chip_files(chip_files, motif_up, chip_dir, sub_df, \"all_chip_overlap\")\n",
    "            \n",
    "        # do both k562 & hepg2\n",
    "        for cell, cell_chip_motifs, cell_chip_files in zip([\"HepG2\", \"K562\"], [hepg2_chip_motifs, k562_chip_motifs], [hepg2_chip_files, k562_chip_files]):\n",
    "            cell_df = sub_df[sub_df[\"cell_type\"] == cell]\n",
    "\n",
    "            # now see if that motif has been chipped in cell type\n",
    "            if motif_up in cell_chip_motifs:\n",
    "                print(\"in %s\" % cell)\n",
    "                n_cell_chip += 1\n",
    "\n",
    "                # find chip files and merge again\n",
    "                cell_df = merge_and_intersect_chip_files(cell_chip_files, motif_up, chip_dir, cell_df, \n",
    "                                                        \"cell_chip_overlap\")\n",
    "            else:\n",
    "                cell_df[\"cell_chip_overlap\"] = \"no_ChIP\"\n",
    "            \n",
    "            all_motifs_int = all_motifs_int.append(cell_df)\n",
    "    else:\n",
    "        sub_df[\"all_chip_overlap\"] = \"no_ChIP\"\n",
    "        sub_df[\"cell_chip_overlap\"] = \"no_ChIP\"\n",
    "        all_motifs_int = all_motifs_int.append(sub_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>motif</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>id</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_chip_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150488212</td>\n",
       "      <td>150488223</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>7.971965</td>\n",
       "      <td>-</td>\n",
       "      <td>FALEC__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr1__150488212</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556129</td>\n",
       "      <td>211556140</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>4.725896</td>\n",
       "      <td>+</td>\n",
       "      <td>LINC00467__p1__tile2__plus__peak5</td>\n",
       "      <td>NRF1__chr1__211556129</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr13</td>\n",
       "      <td>50656265</td>\n",
       "      <td>50656276</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>5.437629</td>\n",
       "      <td>+</td>\n",
       "      <td>DLEU1__p1__tile1__plus__peak2</td>\n",
       "      <td>NRF1__chr13__50656265</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr13</td>\n",
       "      <td>50656306</td>\n",
       "      <td>50656317</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>5.326211</td>\n",
       "      <td>+</td>\n",
       "      <td>DLEU1__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr13__50656306</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr13</td>\n",
       "      <td>50656376</td>\n",
       "      <td>50656387</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>0.956409</td>\n",
       "      <td>+</td>\n",
       "      <td>DLEU1__p1__tile2__plus__peak2</td>\n",
       "      <td>NRF1__chr13__50656376</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr16</td>\n",
       "      <td>86543256</td>\n",
       "      <td>86543267</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>1.719780</td>\n",
       "      <td>-</td>\n",
       "      <td>FOXF1__p__tile2__plus__peak2</td>\n",
       "      <td>NRF1__chr16__86543256</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr20</td>\n",
       "      <td>47895120</td>\n",
       "      <td>47895131</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>9.380756</td>\n",
       "      <td>-</td>\n",
       "      <td>ZFAS1__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr20__47895120</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150488212</td>\n",
       "      <td>150488223</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>3.878904</td>\n",
       "      <td>-</td>\n",
       "      <td>FALEC__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr1__150488212</td>\n",
       "      <td>K562</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>211556129</td>\n",
       "      <td>211556140</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>3.518599</td>\n",
       "      <td>+</td>\n",
       "      <td>LINC00467__p1__tile2__plus__peak2</td>\n",
       "      <td>NRF1__chr1__211556129</td>\n",
       "      <td>K562</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr13</td>\n",
       "      <td>50656265</td>\n",
       "      <td>50656276</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>3.648975</td>\n",
       "      <td>+</td>\n",
       "      <td>DLEU1__p1__tile1__plus__peak1</td>\n",
       "      <td>NRF1__chr13__50656265</td>\n",
       "      <td>K562</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr13</td>\n",
       "      <td>50656306</td>\n",
       "      <td>50656317</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>5.194976</td>\n",
       "      <td>+</td>\n",
       "      <td>DLEU1__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr13__50656306</td>\n",
       "      <td>K562</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr20</td>\n",
       "      <td>47895120</td>\n",
       "      <td>47895131</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>7.273011</td>\n",
       "      <td>-</td>\n",
       "      <td>ZFAS1__p1__tile2__plus__peak1</td>\n",
       "      <td>NRF1__chr20__47895120</td>\n",
       "      <td>K562</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>ChIP_overlap</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   tile_chr  motif_start  motif_end motif     score strand  \\\n",
       "0      chr1    150488212  150488223  NRF1  7.971965      -   \n",
       "2      chr1    211556129  211556140  NRF1  4.725896      +   \n",
       "1     chr13     50656265   50656276  NRF1  5.437629      +   \n",
       "1     chr13     50656306   50656317  NRF1  5.326211      +   \n",
       "8     chr13     50656376   50656387  NRF1  0.956409      +   \n",
       "10    chr16     86543256   86543267  NRF1  1.719780      -   \n",
       "0     chr20     47895120   47895131  NRF1  9.380756      -   \n",
       "3      chr1    150488212  150488223  NRF1  3.878904      -   \n",
       "2      chr1    211556129  211556140  NRF1  3.518599      +   \n",
       "1     chr13     50656265   50656276  NRF1  3.648975      +   \n",
       "1     chr13     50656306   50656317  NRF1  5.194976      +   \n",
       "0     chr20     47895120   47895131  NRF1  7.273011      -   \n",
       "\n",
       "                                 gene                     id cell_type  \\\n",
       "0       FALEC__p1__tile2__plus__peak1  NRF1__chr1__150488212     HepG2   \n",
       "2   LINC00467__p1__tile2__plus__peak5  NRF1__chr1__211556129     HepG2   \n",
       "1       DLEU1__p1__tile1__plus__peak2  NRF1__chr13__50656265     HepG2   \n",
       "1       DLEU1__p1__tile2__plus__peak1  NRF1__chr13__50656306     HepG2   \n",
       "8       DLEU1__p1__tile2__plus__peak2  NRF1__chr13__50656376     HepG2   \n",
       "10       FOXF1__p__tile2__plus__peak2  NRF1__chr16__86543256     HepG2   \n",
       "0       ZFAS1__p1__tile2__plus__peak1  NRF1__chr20__47895120     HepG2   \n",
       "3       FALEC__p1__tile2__plus__peak1  NRF1__chr1__150488212      K562   \n",
       "2   LINC00467__p1__tile2__plus__peak2  NRF1__chr1__211556129      K562   \n",
       "1       DLEU1__p1__tile1__plus__peak1  NRF1__chr13__50656265      K562   \n",
       "1       DLEU1__p1__tile2__plus__peak1  NRF1__chr13__50656306      K562   \n",
       "0       ZFAS1__p1__tile2__plus__peak1  NRF1__chr20__47895120      K562   \n",
       "\n",
       "   all_chip_overlap cell_chip_overlap  \n",
       "0      ChIP_overlap      ChIP_overlap  \n",
       "2      ChIP_overlap      ChIP_overlap  \n",
       "1      ChIP_overlap      ChIP_overlap  \n",
       "1      ChIP_overlap      ChIP_overlap  \n",
       "8      ChIP_overlap      ChIP_overlap  \n",
       "10       no_overlap        no_overlap  \n",
       "0      ChIP_overlap      ChIP_overlap  \n",
       "3      ChIP_overlap      ChIP_overlap  \n",
       "2      ChIP_overlap      ChIP_overlap  \n",
       "1      ChIP_overlap        no_overlap  \n",
       "1      ChIP_overlap      ChIP_overlap  \n",
       "0      ChIP_overlap      ChIP_overlap  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_motifs_int[all_motifs_int[\"motif\"] == \"NRF1\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## same for fimo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "281"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_uniq_motifs = sorted(list(all_fimo[\"motif\"].unique()))\n",
    "len(fimo_uniq_motifs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== ARNT::HIF1A ===\n",
      "=== ATF4 ===\n",
      "=== ATF7 ===\n",
      "=== Arntl ===\n",
      "=== Ascl2 ===\n",
      "=== Atf1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== Atf3 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== BATF3 ===\n",
      "=== BATF::JUN ===\n",
      "=== Bach1::Mafk ===\n",
      "=== Barhl1 ===\n",
      "=== CDX1 ===\n",
      "=== CDX2 ===\n",
      "=== CEBPG ===\n",
      "=== CREB1 ===\n",
      "has been chipped\n",
      "=== CREB3 ===\n",
      "=== CREB3L1 ===\n",
      "=== CTCF ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== CUX1 ===\n",
      "=== CUX2 ===\n",
      "=== Creb5 ===\n",
      "=== Crem ===\n",
      "=== Crx ===\n",
      "=== DBP ===\n",
      "=== Ddit3::Cebpa ===\n",
      "=== Dlx4 ===\n",
      "=== Dux ===\n",
      "=== E2F4 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== E2F6 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== E2F7 ===\n",
      "=== E2F8 ===\n",
      "=== EBF1 ===\n",
      "has been chipped\n",
      "=== EGR1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== EGR2 ===\n",
      "=== EGR3 ===\n",
      "=== EGR4 ===\n",
      "=== EHF ===\n",
      "=== ELF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== ELF3 ===\n",
      "=== ELF4 ===\n",
      "=== ELF5 ===\n",
      "=== ELK1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ELK3 ===\n",
      "=== ELK4 ===\n",
      "has been chipped\n",
      "=== EOMES ===\n",
      "=== ERF ===\n",
      "=== ERG ===\n",
      "=== ESR2 ===\n",
      "=== ETS1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ETV1 ===\n",
      "=== ETV2 ===\n",
      "=== ETV3 ===\n",
      "=== ETV4 ===\n",
      "=== ETV5 ===\n",
      "=== ETV6 ===\n",
      "=== EWSR1-FLI1 ===\n",
      "=== FEV ===\n",
      "=== FIGLA ===\n",
      "=== FLI1 ===\n",
      "=== FOS ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== FOS::JUN ===\n",
      "=== FOSL1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== FOSL2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== FOXC1 ===\n",
      "=== FOXC2 ===\n",
      "=== FOXD1 ===\n",
      "=== FOXD2 ===\n",
      "=== FOXF2 ===\n",
      "=== FOXG1 ===\n",
      "=== FOXH1 ===\n",
      "=== FOXI1 ===\n",
      "=== FOXL1 ===\n",
      "=== FOXO3 ===\n",
      "=== FOXO4 ===\n",
      "=== FOXO6 ===\n",
      "=== FOXP1 ===\n",
      "=== FOXP3 ===\n",
      "=== Foxd3 ===\n",
      "=== Foxj2 ===\n",
      "=== Foxj3 ===\n",
      "=== Foxk1 ===\n",
      "=== Foxo1 ===\n",
      "=== GATA1::TAL1 ===\n",
      "=== GCM1 ===\n",
      "=== GCM2 ===\n",
      "=== GLI2 ===\n",
      "=== GLIS1 ===\n",
      "=== GLIS2 ===\n",
      "=== GLIS3 ===\n",
      "=== GRHL1 ===\n",
      "=== Gabpa ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Gfi1 ===\n",
      "=== Gmeb1 ===\n",
      "=== HEY1 ===\n",
      "=== HEY2 ===\n",
      "=== HIC2 ===\n",
      "=== HINFP ===\n",
      "=== HLF ===\n",
      "=== HNF4G ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== HOXA13 ===\n",
      "=== HOXB13 ===\n",
      "=== HOXC13 ===\n",
      "=== HOXD13 ===\n",
      "=== HSF2 ===\n",
      "=== Hes1 ===\n",
      "=== Hic1 ===\n",
      "=== ID4 ===\n",
      "=== INSM1 ===\n",
      "=== IRF1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== IRF2 ===\n",
      "=== IRF7 ===\n",
      "=== IRF8 ===\n",
      "=== IRF9 ===\n",
      "=== JDP2 ===\n",
      "=== JDP2(var.2) ===\n",
      "=== JUN ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== JUN(var.2) ===\n",
      "=== JUNB ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== JUND ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== JUND(var.2) ===\n",
      "=== KLF13 ===\n",
      "=== KLF14 ===\n",
      "=== KLF16 ===\n",
      "=== KLF5 ===\n",
      "=== Klf1 ===\n",
      "=== Klf12 ===\n",
      "=== Klf4 ===\n",
      "=== MAF::NFE2 ===\n",
      "=== MAFF ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== MAFG ===\n",
      "=== MEF2A ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== MEF2B ===\n",
      "=== MEF2C ===\n",
      "has been chipped\n",
      "=== MEF2D ===\n",
      "=== MEIS1 ===\n",
      "=== MEIS2 ===\n",
      "=== MEIS3 ===\n",
      "=== MGA ===\n",
      "=== MSC ===\n",
      "=== MTF1 ===\n",
      "=== MYBL1 ===\n",
      "=== MYF6 ===\n",
      "=== MZF1(var.2) ===\n",
      "=== Mafb ===\n",
      "=== Mitf ===\n",
      "=== Myb ===\n",
      "=== Myc ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Myod1 ===\n",
      "=== Myog ===\n",
      "=== NFAT5 ===\n",
      "=== NFATC1 ===\n",
      "has been chipped\n",
      "=== NFATC2 ===\n",
      "=== NFE2 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NFIA ===\n",
      "=== NFIC::TLX1 ===\n",
      "=== NFIL3 ===\n",
      "=== NFIX ===\n",
      "=== NFKB1 ===\n",
      "=== NFYA ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NFYB ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== NHLH1 ===\n",
      "=== NR1H2::RXRA ===\n",
      "=== NR2C2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== NRF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== NRL ===\n",
      "=== Nfe2l2 ===\n",
      "=== Nr2e1 ===\n",
      "=== Nr2f6 ===\n",
      "=== Nr2f6(var.2) ===\n",
      "=== Nr5a2 ===\n",
      "=== ONECUT1 ===\n",
      "=== ONECUT2 ===\n",
      "=== ONECUT3 ===\n",
      "=== PAX5 ===\n",
      "has been chipped\n",
      "=== PHOX2A ===\n",
      "=== PKNOX1 ===\n",
      "=== PKNOX2 ===\n",
      "=== PLAG1 ===\n",
      "=== POU4F1 ===\n",
      "=== PPARG ===\n",
      "=== PRDM1 ===\n",
      "has been chipped\n",
      "=== PROP1 ===\n",
      "=== PROX1 ===\n",
      "=== Pax2 ===\n",
      "=== Phox2b ===\n",
      "=== Pou5f1::Sox2 ===\n",
      "=== Pparg::Rxra ===\n",
      "=== RARA(var.2) ===\n",
      "=== RARA::RXRA ===\n",
      "=== REL ===\n",
      "=== REST ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== RORA ===\n",
      "=== RORA(var.2) ===\n",
      "=== RREB1 ===\n",
      "=== RUNX1 ===\n",
      "=== RUNX2 ===\n",
      "=== RUNX3 ===\n",
      "has been chipped\n",
      "=== RXRA::VDR ===\n",
      "=== Rarb ===\n",
      "=== Rarg ===\n",
      "=== Rfx1 ===\n",
      "=== SCRT2 ===\n",
      "=== SMAD2::SMAD3::SMAD4 ===\n",
      "=== SMAD3 ===\n",
      "=== SNAI2 ===\n",
      "=== SOX4 ===\n",
      "=== SP1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== SP2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== SP3 ===\n",
      "=== SP4 ===\n",
      "has been chipped\n",
      "=== SP8 ===\n",
      "=== SPDEF ===\n",
      "=== SPI1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== SPIB ===\n",
      "=== SPIC ===\n",
      "=== SREBF1 ===\n",
      "=== SREBF2 ===\n",
      "=== SREBF2(var.2) ===\n",
      "=== STAT1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== STAT1::STAT2 ===\n",
      "=== STAT3 ===\n",
      "has been chipped\n",
      "=== Sox2 ===\n",
      "=== Sox6 ===\n",
      "=== Spz1 ===\n",
      "=== Srebf1(var.2) ===\n",
      "=== Stat4 ===\n",
      "=== Stat5a::Stat5b ===\n",
      "=== Stat6 ===\n",
      "=== TBP ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== TBR1 ===\n",
      "=== TBX1 ===\n",
      "=== TBX15 ===\n",
      "=== TBX2 ===\n",
      "=== TBX20 ===\n",
      "=== TBX21 ===\n",
      "=== TBX4 ===\n",
      "=== TBX5 ===\n",
      "=== TCF3 ===\n",
      "has been chipped\n",
      "=== TCF4 ===\n",
      "=== TCF7L2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== TEF ===\n",
      "=== TFAP2A ===\n",
      "has been chipped\n",
      "=== TFAP2A(var.2) ===\n",
      "=== TFAP2A(var.3) ===\n",
      "=== TFAP2B ===\n",
      "=== TFAP2B(var.2) ===\n",
      "=== TFAP2B(var.3) ===\n",
      "=== TFAP2C ===\n",
      "has been chipped\n",
      "=== TFAP2C(var.2) ===\n",
      "=== TFAP2C(var.3) ===\n",
      "=== TFAP4 ===\n",
      "=== TFCP2 ===\n",
      "=== TFEC ===\n",
      "=== TGIF1 ===\n",
      "=== TGIF2 ===\n",
      "=== THAP1 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== Tcf12 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "=== Tcf21 ===\n",
      "=== Tcfl5 ===\n",
      "=== Twist2 ===\n",
      "=== USF1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== USF2 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== Vdr ===\n",
      "=== XBP1 ===\n",
      "=== YY1 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== YY2 ===\n",
      "=== ZBTB18 ===\n",
      "=== ZBTB33 ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== ZBTB7A ===\n",
      "has been chipped\n",
      "in HepG2\n",
      "in K562\n",
      "=== ZBTB7B ===\n",
      "=== ZBTB7C ===\n",
      "=== ZEB1 ===\n",
      "has been chipped\n",
      "=== ZIC1 ===\n",
      "=== ZIC3 ===\n",
      "=== ZIC4 ===\n",
      "=== ZNF143 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ZNF263 ===\n",
      "has been chipped\n",
      "in K562\n",
      "=== ZNF410 ===\n",
      "=== ZNF740 ===\n",
      "=== Zfx ===\n",
      "=== Znf423 ===\n"
     ]
    }
   ],
   "source": [
    "fimo_motifs_int = pd.DataFrame()\n",
    "n_chip = 0\n",
    "n_cell_chip = 0\n",
    "\n",
    "for motif in fimo_uniq_motifs:\n",
    "    print(\"=== %s ===\" % motif)\n",
    "    sub_df = all_fimo[all_fimo[\"motif\"] == motif]\n",
    "    \n",
    "    # first check if these motifs overlap *any* chip peaks\n",
    "    # sub_df = intersect_chip_files(sub_df, \"%s/all.chip.bed.gz\" % chip_dir, \"any_chip_overlap\")\n",
    "    \n",
    "    # force motif to upper\n",
    "    motif_up = motif.upper()\n",
    "    \n",
    "    # see if that motif has been chipped at all\n",
    "    if motif_up in chip_motifs:\n",
    "        print(\"has been chipped\")\n",
    "        n_chip += 1\n",
    "\n",
    "        sub_df = merge_and_intersect_chip_files(chip_files, motif_up, chip_dir, sub_df, \"all_chip_overlap\")\n",
    "            \n",
    "        # do both k562 & hepg2\n",
    "        for cell, cell_chip_motifs, cell_chip_files in zip([\"HepG2\", \"K562\"], [hepg2_chip_motifs, k562_chip_motifs], [hepg2_chip_files, k562_chip_files]):\n",
    "            cell_df = sub_df[sub_df[\"cell_type\"] == cell]\n",
    "            \n",
    "            # now see if that motif has been chipped in cell type\n",
    "            if motif_up in cell_chip_motifs:\n",
    "                print(\"in %s\" % cell)\n",
    "                n_cell_chip += 1\n",
    "\n",
    "                # find chip files and merge again\n",
    "                cell_df = merge_and_intersect_chip_files(cell_chip_files, motif_up, chip_dir, cell_df, \n",
    "                                                         \"cell_chip_overlap\")\n",
    "            else:\n",
    "                cell_df[\"cell_chip_overlap\"] = \"no_ChIP\"\n",
    "            \n",
    "            fimo_motifs_int = fimo_motifs_int.append(cell_df)\n",
    "    else:\n",
    "        sub_df[\"all_chip_overlap\"] = \"no_ChIP\"\n",
    "        sub_df[\"cell_chip_overlap\"] = \"no_ChIP\"\n",
    "        fimo_motifs_int = fimo_motifs_int.append(sub_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>motif_start</th>\n",
       "      <th>motif_end</th>\n",
       "      <th>motif</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>id</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_chip_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>chr9</td>\n",
       "      <td>21994467</td>\n",
       "      <td>21994478</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>14.9455</td>\n",
       "      <td>-</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>NRF1__chr9__21994467</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>chr9</td>\n",
       "      <td>21994460</td>\n",
       "      <td>21994471</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>12.6727</td>\n",
       "      <td>+</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>NRF1__chr9__21994460</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>chr9</td>\n",
       "      <td>21994489</td>\n",
       "      <td>21994500</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>11.5455</td>\n",
       "      <td>+</td>\n",
       "      <td>FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...</td>\n",
       "      <td>NRF1__chr9__21994489</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>chr9</td>\n",
       "      <td>21994528</td>\n",
       "      <td>21994539</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>14.9455</td>\n",
       "      <td>+</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...</td>\n",
       "      <td>NRF1__chr9__21994528</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>chr9</td>\n",
       "      <td>21994535</td>\n",
       "      <td>21994546</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>12.6727</td>\n",
       "      <td>-</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...</td>\n",
       "      <td>NRF1__chr9__21994535</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>no_overlap</td>\n",
       "      <td>no_overlap</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    tile_chr  motif_start  motif_end motif    score strand  \\\n",
       "131     chr9     21994467   21994478  NRF1  14.9455      -   \n",
       "156     chr9     21994460   21994471  NRF1  12.6727      +   \n",
       "179     chr9     21994489   21994500  NRF1  11.5455      +   \n",
       "248     chr9     21994528   21994539  NRF1  14.9455      +   \n",
       "273     chr9     21994535   21994546  NRF1  12.6727      -   \n",
       "\n",
       "                                                  gene                    id  \\\n",
       "131  FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...  NRF1__chr9__21994467   \n",
       "156  FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...  NRF1__chr9__21994460   \n",
       "179  FLIPPED__mrna.flip.antisense__p1@CDKN2A__chr9:...  NRF1__chr9__21994489   \n",
       "248  WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...  NRF1__chr9__21994528   \n",
       "273  WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__p1@CDKN...  NRF1__chr9__21994535   \n",
       "\n",
       "    cell_type all_chip_overlap cell_chip_overlap  \n",
       "131     HepG2       no_overlap        no_overlap  \n",
       "156     HepG2       no_overlap        no_overlap  \n",
       "179     HepG2       no_overlap        no_overlap  \n",
       "248     HepG2       no_overlap        no_overlap  \n",
       "273     HepG2       no_overlap        no_overlap  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_motifs_int[fimo_motifs_int[\"motif\"] == \"NRF1\"].head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## 4. find numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cell_type</th>\n",
       "      <th>cell_chip_overlap</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>37</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_ChIP</td>\n",
       "      <td>685</td>\n",
       "      <td>224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "cell_type cell_chip_overlap  HepG2  K562\n",
       "0              ChIP_overlap     37    36\n",
       "1                   no_ChIP    685   224\n",
       "2                no_overlap     54    27"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results = pd.pivot_table(all_motifs_int, values=\"id\", index=\"cell_chip_overlap\", columns=\"cell_type\",\n",
    "                              aggfunc=\"count\").reset_index()\n",
    "cell_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cell_type</th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>113</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_ChIP</td>\n",
       "      <td>585</td>\n",
       "      <td>202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>78</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "cell_type all_chip_overlap  HepG2  K562\n",
       "0             ChIP_overlap    113    55\n",
       "1                  no_ChIP    585   202\n",
       "2               no_overlap     78    30"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_results = pd.pivot_table(all_motifs_int, values=\"id\", index=\"all_chip_overlap\", columns=\"cell_type\",\n",
    "                             aggfunc=\"count\").reset_index()\n",
    "all_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  all_chip_overlap cell_type  value\n",
       "0     ChIP_overlap     HepG2    113\n",
       "2       no_overlap     HepG2     78\n",
       "3     ChIP_overlap      K562     55\n",
       "5       no_overlap      K562     30"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_melt = pd.melt(cell_results, id_vars=\"cell_chip_overlap\")\n",
    "all_results_melt = pd.melt(all_results, id_vars=\"all_chip_overlap\")\n",
    "\n",
    "cell_results_melt = cell_results_melt[cell_results_melt[\"cell_chip_overlap\"] != \"no_ChIP\"]\n",
    "all_results_melt = all_results_melt[all_results_melt[\"all_chip_overlap\"] != \"no_ChIP\"]\n",
    "\n",
    "all_results_melt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>value_count</th>\n",
       "      <th>value_total</th>\n",
       "      <th>percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>113</td>\n",
       "      <td>191</td>\n",
       "      <td>59.162304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>78</td>\n",
       "      <td>191</td>\n",
       "      <td>40.837696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>55</td>\n",
       "      <td>85</td>\n",
       "      <td>64.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>30</td>\n",
       "      <td>85</td>\n",
       "      <td>35.294118</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  all_chip_overlap cell_type  value_count  value_total    percent\n",
       "0     ChIP_overlap     HepG2          113          191  59.162304\n",
       "1       no_overlap     HepG2           78          191  40.837696\n",
       "2     ChIP_overlap      K562           55           85  64.705882\n",
       "3       no_overlap      K562           30           85  35.294118"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_counts = cell_results_melt.groupby(\"cell_type\")[\"value\"].agg(\"sum\").reset_index()\n",
    "cell_results_melt = cell_results_melt.merge(cell_results_counts, on=\"cell_type\", suffixes=(\"_count\", \"_total\"))\n",
    "cell_results_melt[\"percent\"] = (cell_results_melt[\"value_count\"]/cell_results_melt[\"value_total\"])*100\n",
    "\n",
    "all_results_counts = all_results_melt.groupby(\"cell_type\")[\"value\"].agg(\"sum\").reset_index()\n",
    "all_results_melt = all_results_melt.merge(all_results_counts, on=\"cell_type\", suffixes=(\"_count\", \"_total\"))\n",
    "all_results_melt[\"percent\"] = (all_results_melt[\"value_count\"]/all_results_melt[\"value_total\"])*100\n",
    "\n",
    "all_results_melt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cell-type ChIP</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>40.659341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cell-type ChIP</td>\n",
       "      <td>K562</td>\n",
       "      <td>57.142857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all ChIP</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>59.162304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>all ChIP</td>\n",
       "      <td>K562</td>\n",
       "      <td>64.705882</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             type cell_type    percent\n",
       "0  cell-type ChIP     HepG2  40.659341\n",
       "2  cell-type ChIP      K562  57.142857\n",
       "0        all ChIP     HepG2  59.162304\n",
       "2        all ChIP      K562  64.705882"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_melt = cell_results_melt[cell_results_melt[\"cell_chip_overlap\"] == \"ChIP_overlap\"]\n",
    "cell_results_melt[\"type\"] = \"cell-type ChIP\"\n",
    "cell_results_melt = cell_results_melt[[\"type\", \"cell_type\", \"percent\"]]\n",
    "\n",
    "all_results_melt = all_results_melt[all_results_melt[\"all_chip_overlap\"] == \"ChIP_overlap\"]\n",
    "all_results_melt[\"type\"] = \"all ChIP\"\n",
    "all_results_melt = all_results_melt[[\"type\", \"cell_type\", \"percent\"]]\n",
    "\n",
    "results_melt = cell_results_melt.append(all_results_melt)\n",
    "results_melt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAJEAAACGCAYAAADdC2gAAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADUlJREFUeJzt3X1MU+ceB/BvoYi8FCggL0LGH2yggaDyGsnYlsKmI46o\nccrGuAOTsckgi2NsERdkbkGQqEzYC4QMehOmRMYIDPaijEoycS7T8TKDIgoK6IDJsFBc29PeP24k\n7oLnKVQOnt7f569pTk+/Xb+ec/r0OU8lRqPRCELMYLXUAYj4UYmI2ahExGxUImI2KhExG5WImI1K\nRMzGLFFpaSkA4OLFi0hPT1/0QER8pKwNQkNDsXPnTgQGBuLgwYNCZCIiI+Ebsc7OzgYAXL58GU5O\nTvD29qYikVl4S3SP0WiEwWBAV1cX1q5d+8Dtbty4gcLCQuTk5ODLL78Ex3HYtWsXjh8/DgDw9/dH\nbGzsw0tPHgnM01l2djb6+vrg6+sLvV6PTz/99IHbOjo6wsPDA6dOnUJycjJu3rwJlUoFAEhLS0Nh\nYeGsEtXU1KCmpgYAMD4+jqefftqMl0MepsHBQVRUVDC3Y5bI2dkZu3btgpeXF9ra2ni3lcvlkMvl\nUKvVMBgMMBgMWL58OXQ6HQBAIpHMesyOHTuwY8cOAEBeXh7y8vKYoYkwTH0vmCWSy+VYsWIFPvnk\nE9jZ2Zm00y1btqCyshI2NjZIT09HT08PPvvsM0RHR5v0eCIyRoaJiQnjr7/+ahweHjZyHMfa3Cz7\n9u1b1P2T+TH1/WCOExUVFaGpqQmDg4M4cOCAEL0mIsMskY2NDeRyOSIiIjA1NSVEJiIyzGui8PBw\nHDt2DN3d3YiJiREiExEZZokiIiIwMjICW1tbxMfHC5GJiAzzdLZ//36sW7cOq1evRm5urhCZiMgw\nj0QrV65ESEgIJBIJPD09hchERIZZouHhYbz55puQSqXQ6/XIzs6GRCKh79DIDGaJSkpKhMhBRIwm\npRGz8Zaop6cHVVVVGBsbEyoPESHeEhUXFyM0NBRFRUVC5SEixFsiuVyOkJAQ2NraCpWHiBBvifr7\n+1FbW4tr166htrYWtbW1QuUiIsL76Wz37t2QSCTIzMyccy4QIQCjRGfOnPnHnyUSCSIiIhY1EBEf\n3hK9+OKLAOaekUjIPbwl8vHxgVKpRGdnJ6ysrOY1Ul1XV4eRkREMDAzAzc1tZtK+k5PTQwlOHh3M\nEeu+vj4cOnRoQTufmJiASqVCfX09bt68ifb2dmzYsOEf29w/Ud/KisY+xYi3RCdOnMDQ0BAqKysh\nk8kAANu2bTNpx93d3cjNzYWTk9PMpH1ra+tZ2/3vRH0iPrwl8vPzw+uvvw7gv9dFxnmszBcSEoLy\n8nJcv379H5P2ieXhLRHHcbh69SqSkpKwd+9evPzyyybvePPmzWaHI+LAexFy/PhxbN++HQCQm5uL\nsrIyQUIRceEtkb29PWxsbAAAtra2M/9NyP14T2dyuRwnTpzAunXr0NnZCblcLlQuIiK8R6J3330X\nbm5uUKlUcHV1xfvvvy9ULiIizHEihUIBhUIhRBYiUvMa3eM4brFyEBFjHom+/vprnD17FgaDAWq1\nGp9//rkQuYiIMEt07tw5hIeHIywsDM3NzUJkIiLDPJ3p9Xr4+PigubkZ169fFyITERnmkSg3NxcS\niQQajYbuxSdzYpbo3hevcXFxix6GiBPNvSBmYx6JKioq0NXVBTs7OwQHB+OVV14RIpfFyfr230sd\ngdeh5/+14McySzQ4OIiPP/4YAPDBBx8s+ImI5WKWaGhoCF988QVsbGxmbh0CTJ+cRiwfs0Svvfba\nzET9FStWwNXV1aQdnz9/Hl1dXejt7YWLiwvNsbZgzBJduHAB58+fh62tLbRarckj1iqVCjKZDKtX\nr0ZcXNwD51gT8WOW6NatW9i0aROCg4PnNWKt0Wjw9ttvIywsDAqF4oFzrGmivvgxS3T37l24ublB\nqVRifHzc5B1HRUWhqqoKzz33HO8ca9ZEfUv+VGMpmCXKz88Hx3Fwdnae16S0Z5991qxgRDx4S3Rv\nab17d3nQMntkLrwlKioqQk1NzczphpC5ME9nDQ0N6Ovrg4ODAwDgrbfeWvRQRFyYJTp48CA0Gg3s\n7e2FyENEiFmi6upqjI6OIiYmBmq1GklJSULkIiLCHJjRarXw8/NDQkICLl++LEQmIjLMErm4uOC3\n337DkSNH5nUvPvn/wTydRUZGYuPGjeA4DoGBgUJkIiLDLNHVq1fR0NAAuVyOv//+GyEhIULkIiLC\nPJ0lJiYiMTEROp2O1rMmc2IeidLT0xEeHo5XX30V3t7eQmQiIsM8EqWkpOD333/H4cOH0dnZKUQm\nIjIm3QF76NAhaLVa7Nu3j66JyCzMEhmNRoyOjsLa2hparVaITERkmCXKzMyEUqmEXq+nNRfJnJgl\n8vHxwTvvvCNEFiJSzBKZQ6VSoaWlBc7OzjRR34Ix17FubW2FQqGA0WhES0uLyRP1Ozo6AADff/89\nGhsbaTF0C8ZcxzogIAB+fn4AgJycHJN33NbWBplMBrVaDb1eT4uhWzDeEkVGRkKn06G9vR0GgwED\nAwM4cuSISTvOzMwEAFy6dAlKpZIWQ7dgzGuiuro6BAQEYM2aNZBK538JdeDAgQUFI+LBvAixtrZG\nUFAQBgYGMDIyIkQmIjImjRPZ29vjypUreOqpp4TIRESGWaKLFy+ira0NBoMBly5dosWuyCzMErW0\ntOCjjz6a85MVIQCjRO3t7dBqtTOLMwDA+vXrBQlGxIO3RLdu3cIzzzwDjUaDqakp+i1YMifeT2d6\nvR4//PAD9Ho99Ho9vvvuO6FyERExacT6scceAwDs2bNHkFBEXJgj1pGRkUJlISJF33gSszFLVF9f\nj/7+fmRlZUGpVAqRiYgMs0Td3d3o6+tDQkICBgYGhMhERIZZIrVajerqanh5eWFiYkKITERkeC+s\nOzo6kJycDH9/f3Ach/z8fKFyERHhLVFZWRkGBwexatUqGI1GWm6PzIm3RKWlpeju7kZwcDCtCEIe\niLdEFRUVuHbt2j/+ztRJZqdOnUJPTw+Ghobg4uICADRR30Lxlmjjxo348ccfERsbC6PRiG+//dbk\nHbu6uiIjIwOBgYFobW3FyMgITdS3ULwlamhowJkzZzA5OQkA8PX1NXnHnp6eKCgoQFZWFiQSCU3U\nt2C8//QzMjKwf/9+/PXXX9BoNAgLCzN5x0ePHoWjoyO0Wi0qKyvR0tJC00gsFHNSWnl5OXJycjA5\nOYnS0lIUFBSYtOPCwkKzwxFxYJbI2toaMpkMMpkMBoNBiExEZJglSkpKQm5uLjiOw/bt24XIRESG\n+XFoenoa3t7e8PLywk8//SREJiIyzCNRY2Mj3njjDRpsJA/ELJGNjQ1UKhWWLVsGgH77lczGLJFa\nrYa/vz8kEglN1CdzMuk26snJSXAcB71eL0QmIjImrah/584d3Llzh45EZE7MEm3ZskWIHETE6BtP\nYjbeI1FKSgqWL1+OuLi4mUlp9OmM/C/eEu3duxdNTU0zy+0RMhfeEj3xxBMICAjAsWPHoNfrsXnz\nZqFyERFhXhOdPXsWhw8fxtGjR6FSqQSIRMSG+elMp9PhwoULkEqlmJqaEiITERlmiXJycvDNN99A\nr9ebtaBDY2MjxsbGAACpqakL3g959DBPZzKZDC+99BKSk5Ph7u6+4Ce6cuUKUlNTafFQC7SoP8tw\nv+npaQCYNep9/0T98fHxWfOsZQ85x71boB6WvJ/zTNpOjK9jcHDQpMdKjCbM8aivr0dvby+kUil2\n794974DAf9fDHhsbg4ODA5KSkha0j4dh69atqKurW7Lnf1gepdfBeyQ6efIkYmNjMTAwgA0bNqC2\ntnbBT7R169YFP5Y82qzzeO7TWblyJU6ePAlvb2+o1Wps2rQJjo6OAsZbHA/zNLCUHpXXwXs66+3t\nRX19PdavX4/bt28jICAAq1atEjIfEQHeT2dfffUVsrKyoFKpkJCQsKDf9iCWj/dI9PPPP+P06dOI\niYmhGw+XSElJCXbu3Inm5mbcvXsXGo0G7u7ueOGFF1BVVYWJiQnEx8ejvb0dOp0OHMchIyND0Iy8\nh5aoqChERUUJlcVs9/6HOzg4oLCwEO+9996c242OjqKqqgrOzs6YmJhARkYGqqurl+xNYCkvL4eH\nhwf++OMPuLu7QyqV4pdffsH4+DhkMhk8PDwQGhqKoKAgfPjhh4Lns7jzU2VlJZYtW4be3l4UFBTA\n1tYW8fHxKC4uRlRUFHQ6Haanp5Geno6Ojg6cO3cOvb29S/omsKxZswZNTU1IT0+Hv78/8vPzsXbt\nWigUCnh5eaGhoQFBQUEoKSmZ+Z05IVncpLTU1FSkpaWhv78fUqkU3t7euHHjBh5//HGkpKTg9u3b\n4DgOVlZWiI6Ohq+vL/78809otdolexNYoqKikJaWhm3btqGqqgoymQzR0dE4ffo0amtr8eSTT6K4\nuBhOTk5obW0VPJ9Jg41icf/pbM+ePbCzs4NMJsPzzz+P4uJiREdHAwAUCgWUSiU8PT0xPDyMmJgY\nlJWVQaFQwMnJCYmJiUv8SsTFokrEh+8aiZjn/6ZEZPFY3DURER6ViJiNSkTMRiUiZqMSEbP9B38Y\n2Q/o5uyUAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2af365dabbe0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "figs_dir = \"figs/0__overlap_bar\"\n",
    "!mkdir -p $figs_dir\n",
    "\n",
    "plt.figure(figsize=(1.8, 1.9))\n",
    "sns.barplot(data=results_melt[results_melt[\"type\"] == \"all ChIP\"], \n",
    "            x=\"cell_type\", y=\"percent\", color=sns.color_palette(\"Set2\")[0])\n",
    "plt.xlabel(\"\")\n",
    "plt.ylabel(\"% of motifs that overlap a ChIP peak\")\n",
    "plt.ylim((0, 100))\n",
    "plt.savefig(\"%s/all_chip_overlap.pdf\" % figs_dir, dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## same with fimo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cell_type</th>\n",
       "      <th>cell_chip_overlap</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>61</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_ChIP</td>\n",
       "      <td>1475</td>\n",
       "      <td>1235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>223</td>\n",
       "      <td>392</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "cell_type cell_chip_overlap  HepG2  K562\n",
       "0              ChIP_overlap     61   132\n",
       "1                   no_ChIP   1475  1235\n",
       "2                no_overlap    223   392"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results = pd.pivot_table(fimo_motifs_int, values=\"id\", index=\"cell_chip_overlap\", columns=\"cell_type\",\n",
    "                              aggfunc=\"count\").reset_index()\n",
    "cell_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cell_type</th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>256</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_ChIP</td>\n",
       "      <td>1140</td>\n",
       "      <td>1140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>363</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "cell_type all_chip_overlap  HepG2  K562\n",
       "0             ChIP_overlap    256   256\n",
       "1                  no_ChIP   1140  1140\n",
       "2               no_overlap    363   363"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_results = pd.pivot_table(fimo_motifs_int, values=\"id\", index=\"all_chip_overlap\", columns=\"cell_type\",\n",
    "                             aggfunc=\"count\").reset_index()\n",
    "all_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  all_chip_overlap cell_type  value\n",
       "0     ChIP_overlap     HepG2    256\n",
       "2       no_overlap     HepG2    363\n",
       "3     ChIP_overlap      K562    256\n",
       "5       no_overlap      K562    363"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_melt = pd.melt(cell_results, id_vars=\"cell_chip_overlap\")\n",
    "all_results_melt = pd.melt(all_results, id_vars=\"all_chip_overlap\")\n",
    "\n",
    "cell_results_melt = cell_results_melt[cell_results_melt[\"cell_chip_overlap\"] != \"no_ChIP\"]\n",
    "all_results_melt = all_results_melt[all_results_melt[\"all_chip_overlap\"] != \"no_ChIP\"]\n",
    "\n",
    "all_results_melt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_chip_overlap</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>value_count</th>\n",
       "      <th>value_total</th>\n",
       "      <th>percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>256</td>\n",
       "      <td>619</td>\n",
       "      <td>41.357027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>363</td>\n",
       "      <td>619</td>\n",
       "      <td>58.642973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ChIP_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>256</td>\n",
       "      <td>619</td>\n",
       "      <td>41.357027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>no_overlap</td>\n",
       "      <td>K562</td>\n",
       "      <td>363</td>\n",
       "      <td>619</td>\n",
       "      <td>58.642973</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  all_chip_overlap cell_type  value_count  value_total    percent\n",
       "0     ChIP_overlap     HepG2          256          619  41.357027\n",
       "1       no_overlap     HepG2          363          619  58.642973\n",
       "2     ChIP_overlap      K562          256          619  41.357027\n",
       "3       no_overlap      K562          363          619  58.642973"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_counts = cell_results_melt.groupby(\"cell_type\")[\"value\"].agg(\"sum\").reset_index()\n",
    "cell_results_melt = cell_results_melt.merge(cell_results_counts, on=\"cell_type\", suffixes=(\"_count\", \"_total\"))\n",
    "cell_results_melt[\"percent\"] = (cell_results_melt[\"value_count\"]/cell_results_melt[\"value_total\"])*100\n",
    "\n",
    "all_results_counts = all_results_melt.groupby(\"cell_type\")[\"value\"].agg(\"sum\").reset_index()\n",
    "all_results_melt = all_results_melt.merge(all_results_counts, on=\"cell_type\", suffixes=(\"_count\", \"_total\"))\n",
    "all_results_melt[\"percent\"] = (all_results_melt[\"value_count\"]/all_results_melt[\"value_total\"])*100\n",
    "\n",
    "all_results_melt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cell-type ChIP</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>21.478873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cell-type ChIP</td>\n",
       "      <td>K562</td>\n",
       "      <td>25.190840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all ChIP</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>41.357027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>all ChIP</td>\n",
       "      <td>K562</td>\n",
       "      <td>41.357027</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             type cell_type    percent\n",
       "0  cell-type ChIP     HepG2  21.478873\n",
       "2  cell-type ChIP      K562  25.190840\n",
       "0        all ChIP     HepG2  41.357027\n",
       "2        all ChIP      K562  41.357027"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_results_melt = cell_results_melt[cell_results_melt[\"cell_chip_overlap\"] == \"ChIP_overlap\"]\n",
    "cell_results_melt[\"type\"] = \"cell-type ChIP\"\n",
    "cell_results_melt = cell_results_melt[[\"type\", \"cell_type\", \"percent\"]]\n",
    "\n",
    "all_results_melt = all_results_melt[all_results_melt[\"all_chip_overlap\"] == \"ChIP_overlap\"]\n",
    "all_results_melt[\"type\"] = \"all ChIP\"\n",
    "all_results_melt = all_results_melt[[\"type\", \"cell_type\", \"percent\"]]\n",
    "\n",
    "results_melt = cell_results_melt.append(all_results_melt)\n",
    "results_melt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAJEAAACGCAYAAADdC2gAAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADUlJREFUeJzt3X1MU+ceB/BvoYi8FCggL0LGH2yggaDyGsnYlsKmI46o\nccrGuAOTsckgi2NsERdkbkGQqEzYC4QMehOmRMYIDPaijEoycS7T8TKDIgoK6IDJsFBc29PeP24k\n7oLnKRSOnt7f5y8lp4dv7ddzTp8+56nEaDQaQYgZrB52ACJ+VCJiNioRMRuViJiNSkTMRiUiZqMS\nEbMxS1RaWgoAuHjxItLT05c8EBEfKWuD0NBQ7Ny5E4GBgTh48KAQmYjISPhGrLOzswEAly9fhpOT\nE7y9valIZBbeEt1jNBphMBjQ1dWFtWvXPnC7GzduoLCwEDk5Ofjyyy/BcRx27dqF48ePAwD8/f0R\nGxu7eOnJI4F5OsvOzkZfXx98fX2h1+vx6aefPnBbR0dHeHh44NSpU0hOTsbNmzehUqkAAGlpaSgs\nLJxVopqaGtTU1AAAxsfH8fTTT5vxdMhiGhwcREVFBXM7ZomcnZ2xa9cueHl5oa2tjXdbuVwOuVwO\ntVoNg8EAg8GA5cuXQ6fTAQAkEsmsx+zYsQM7duwAAOTl5SEvL48ZmgjD1NeCWSK5XI4VK1bgk08+\ngZ2dnUk73bJlCyorK2FjY4P09HT09PTgs88+Q3R0tEmPJyJjZJiYmDD++uuvxuHhYSPHcazNzbJv\n374l3T+ZH1NfD+Y4UVFREZqamjA4OIgDBw4I0WsiMswS2djYQC6XIyIiAlNTU0JkIiLDvCYKDw/H\nsWPH0N3djZiYGCEyEZFhligiIgIjIyOwtbVFfHy8EJmIyDBPZ/v378e6deuwevVq5ObmCpGJiAzz\nSLRy5UqEhIRAIpHA09NTiExEZJglGh4exptvvgmpVAq9Xo/s7GxIJBL6DI3MYJaopKREiBxExGhS\nGjEbb4l6enpQVVWFsbExofIQEeItUXFxMUJDQ1FUVCRUHiJCvCWSy+UICQmBra2tUHmICPGWqL+/\nH7W1tbh27Rpqa2tRW1srVC4iIrzvznbv3g2JRILMzMw55wIRAjBKdObMmX/8XSKRICIiYkkDEfHh\nLdGLL74IYO4ZiYTcw1siHx8fKJVKdHZ2wsrKal4j1XV1dRgZGcHAwADc3NxmJu07OTktSnDy6GCO\nWPf19eHQoUML2vnExARUKhXq6+tx8+ZNtLe3Y8OGDf/Y5v6J+lZWNPYpRrwlOnHiBIaGhlBZWQmZ\nTAYA2LZtm0k77u7uRm5uLpycnGYm7VtbW8/a7n8n6hPx4S2Rn58fXn/9dQD/vS4yzmNlvpCQEJSX\nl+P69ev/mLRPLA9viTiOw9WrV5GUlIS9e/fi5ZdfNnnHmzdvNjscEQfei5Djx49j+/btAIDc3FyU\nlZUJEoqIC2+J7O3tYWNjAwCwtbWd+TMh9+M9ncnlcpw4cQLr1q1DZ2cn5HK5ULmIiPAeid599124\nublBpVLB1dUV77//vlC5iIgwx4kUCgUUCoUQWYhIzWt0j+O4pcpBRIx5JPr6669x9uxZGAwGqNVq\nfP7550LkIiLCLNG5c+cQHh6OsLAwNDc3C5GJiAzzdKbX6+Hj44Pm5mZcv35diExEZJhHotzcXEgk\nEmg0GroXn8yJWaJ7H7zGxcUteRgiTjT3gpiNeSSqqKhAV1cX7OzsEBwcjFdeeUWIXEREmCUaHBzE\nxx9/DAD44IMPljwQER9miYaGhvDFF1/AxsZm5tYhwPTJacTyMUv02muvzUzUX7FiBVxdXU3a8fnz\n59HV1YXe3l64uLjQHGsLxizRhQsXcP78edja2kKr1Zo8Yq1SqSCTybB69WrExcU9cI41ET9miW7d\nuoVNmzYhODh4XiPWGo0Gb7/9NsLCwqBQKB44x5om6osfs0R3796Fm5sblEolxsfHTd5xVFQUqqqq\n8Nxzz/HOsaaJ+uLHLFF+fj44joOzs/O8JqU9++yzZgUj4sFbontL6927y4OW2SNz4S1RUVERampq\nZk43hMyFeTpraGhAX18fHBwcAABvvfXWkoci4sIs0cGDB6HRaGBvby9EHiJCzBJVV1djdHQUMTEx\nUKvVSEpKEiIXERHmwIxWq4Wfnx8SEhJw+fJlITIRkWGWyMXFBb/99huOHDkyr3vxyf8P5uksMjIS\nGzduBMdxCAwMFCITERlmia5evYqGhgbI5XL8/fffCAkJESIXERHm6SwxMRGJiYnQ6XS0njWZE/NI\nlJ6ejvDwcLz66qvw9vYWIhMRGeaRKCUlBb///jsOHz6Mzs5OITIRkTHpDthDhw5Bq9Vi3759dE1E\nZmGWyGg0YnR0FNbW1tBqtUJkIiLDLFFmZiaUSiX0ej2tuUjmxCyRj48P3nnnHSGyEJFilsgcKpUK\nLS0tcHZ2XvBE/axv/71E6RbHoef/ZdJ2lvI85sJcx7q1tRUKhQJGoxEtLS0mT9Tv6OgAAHz//fdo\nbGykxdAtGHMd64CAAPj5+QEAcnJyTN5xW1sbZDIZ1Go19Ho9LYZuwXhLFBkZCZ1Oh/b2dhgMBgwM\nDODIkSMm7TgzMxMAcOnSJSiVSloM3YIxr4nq6uoQEBCANWvWQCqd/yXUgQMHFhSMiAfzIsTa2hpB\nQUEYGBjAyMiIEJmIyJg0TmRvb48rV67gqaeeEiITERlmiS5evIi2tjYYDAZcunSJFrsiszBL1NLS\ngo8++mjOd1aEAIwStbe3Q6vVzizOAADr168XJBgRD94S3bp1C8888ww0Gg2mpqbou2DJnHjfnen1\nevzwww/Q6/XQ6/X47rvvhMpFRMSkEevHHnsMALBnzx5BQhFxYY5YR0ZGCpWFiBR94knMxixRfX09\n+vv7kZWVBaVSKUQmIjLMEnV3d6Ovrw8JCQkYGBgQIhMRGWaJ1Go1qqur4eXlhYmJCSEyEZHhvbDu\n6OhAcnIy/P39wXEc8vPzhcpFRIS3RGVlZRgcHMSqVatgNBppuT0yJ94SlZaWoru7G8HBwbQiCHkg\n3hJVVFTg2rVr//iZqZPMTp06hZ6eHgwNDcHFxQUAaEV9C8Vboo0bN+LHH39EbGwsjEYjvv32W5N3\n7OrqioyMDAQGBqK1tRUjIyM0Ud9C8ZaooaEBZ86cweTkJADA19fX5B17enqioKAAWVlZkEgkNFHf\ngvH+18/IyMD+/fvx119/QaPRICwszOQdHz16FI6OjtBqtaisrERLSwtNI7FQzElp5eXlyMnJweTk\nJEpLS1FQUGDSjgsLC80OR8SBWSJra2vIZDLIZDIYDAYhMhGRYZYoKSkJubm54DgO27dvFyITERnm\n26Hp6Wl4e3vDy8sLP/30kxCZiMgwj0SNjY144403aLCRPBCzRDY2NlCpVFi2bBkA+u5XMhuzRGq1\nGv7+/pBIJDRRn8zJpNuoJycnwXEc9Hq9EJmIyJi0ov6dO3dw584dOhKROTFLtGXLFiFyEBGjTzyJ\n2XiPRCkpKVi+fDni4uJmJqXRuzPyv3hLtHfvXjQ1Nc0st0fIXHhL9MQTTyAgIADHjh2DXq/H5s2b\nhcpFRIR5TXT27FkcPnwYR48ehUqlEiASERvmuzOdTocLFy5AKpViampKiExEZJglysnJwTfffAO9\nXm/Wgg6NjY0YGxsDAKSmpi54P+TRwzydyWQyvPTSS0hOToa7u/uCf9GVK1eQmppKi4daoCX9Wob7\nTU9PA8CsUe/7J+qPj4/PmmctW+Qc926BWix5P+eZtJ0Yn8fg4KBJj5UYTZjjUV9fj97eXkilUuze\nvXveAYH/roc9NjYGBwcHJCUlLWgfi2Hr1q2oq6t7aL9/sTxKz4P3SHTy5EnExsZiYGAAGzZsQG1t\n7YJ/0datWxf8WPJos87juU9n5cqVOHnyJLy9vaFWq7Fp0yY4OjoKGG9pLOZp4GF6VJ4H7+mst7cX\n9fX1WL9+PW7fvo2AgACsWrVKyHxEBHjfnX311VfIysqCSqVCQkLCgr7bg1g+3iPRzz//jNOnTyMm\nJoZuPHxISkpKsHPnTjQ3N+Pu3bvQaDRwd3fHCy+8gKqqKkxMTCA+Ph7t7e3Q6XTgOA4ZGRmCZuQ9\ntERFRSEqKkqoLGa79w/u4OCAwsJCvPfee3NuNzo6iqqqKjg7O2NiYgIZGRmorq5+aC8CS3l5OTw8\nPPDHH3/A3d0dUqkUv/zyC8bHxyGTyeDh4YHQ0FAEBQXhww8/FDyfxZ2fKisrsWzZMvT29qKgoAC2\ntraIj49HcXExoqKioNPpMD09jfT0dHR0dODcuXPo7e19qC8Cy5o1a9DU1IT09HT4+/sjPz8fa9eu\nhUKhgJeXFxoaGhAUFISSkpKZ75kTksVNSktNTUVaWhr6+/shlUrh7e2NGzdu4PHHH0dKSgpu374N\njuNgZWWF6Oho+Pr64s8//4RWq31oLwJLVFQU0tLSsG3bNlRVVUEmkyE6OhqnT59GbW0tnnzySRQX\nF8PJyQmtra2C5zNpsFEs7j+d7dmzB3Z2dpDJZHj++edRXFyM6OhoAIBCoYBSqYSnpyeGh4cRExOD\nsrIyKBQKODk5ITEx8SE/E3GxqBLx4btGIub5vykRWToWd01EhEclImajEhGzUYmI2ahExGz/AXhW\n2Q9UYgvhAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2af365e8af60>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "figs_dir = \"figs/0__overlap_bar\"\n",
    "!mkdir -p $figs_dir\n",
    "\n",
    "plt.figure(figsize=(1.8, 1.9))\n",
    "sns.barplot(data=results_melt[results_melt[\"type\"] == \"all ChIP\"], \n",
    "            x=\"cell_type\", y=\"percent\", color=sns.color_palette(\"Set2\")[0])\n",
    "plt.xlabel(\"\")\n",
    "plt.ylabel(\"% of motifs that overlap a ChIP peak\")\n",
    "plt.ylim((0, 100))\n",
    "plt.savefig(\"%s/all_chip_overlap.fimo.pdf\" % figs_dir, dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
