{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 06__tf_tissue_sp\n",
    "# calculating tissue specificity of TFs (across HepG2, HeLa, and K562)\n",
    "\n",
    "in this notebook, i calculate the tissue specificity of TFs across the 3 cell types in our MPRAs using ENCODE RNA-seq data. then, i correlate motif coverage (# bp covered and maximum coverage) as well as average TF specificity with MPRA activities and specificities.\n",
    "\n",
    "note: the FIMO mappings and coverage calculations were done separately (see methods)\n",
    "\n",
    "------\n",
    "\n",
    "figures in this notebook:\n",
    "- **Fig 3C**: KDE plot of correlations of MPRA activity & specificity with each of the 3 metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from itertools import chain\n",
    "from decimal import Decimal\n",
    "from scipy import stats\n",
    "from scipy import signal\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from plotting_utils import *\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "expr_f = \"../../misc/03__rna_seq_expr/Expression.all.cells.txt\"\n",
    "tf_map_f = \"../../misc/04__jaspar_id_map/2018_03_09_gencode_jaspar_curated.txt\"\n",
    "fimo_f = \"../../misc/05__fimo/pool1_fimo_map.txt\"\n",
    "cov_map_f = \"../../misc/05__fimo/seqID_nFimoMotifs_Coverage.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "index_dir = \"../../data/00__index\"\n",
    "index_f = \"%s/tss_oligo_pool.index.txt\" % index_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ_f = \"../../data/02__activs/POOL1__pMPRA1__HepG2__activities_per_element.txt\"\n",
    "hela_activ_f = \"../../data/02__activs/POOL1__pMPRA1__HeLa__activities_per_element.txt\"\n",
    "k562_activ_f = \"../../data/02__activs/POOL1__pMPRA1__K562__activities_per_element.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>Mus musculus; human ortholog selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name          Gene ID                                comment\n",
       "0  MA0002.2      RUNX1  ENSG00000159216                                    NaN\n",
       "1  MA0003.3     TFAP2A  ENSG00000137203                                    NaN\n",
       "2  MA0007.3         Ar  ENSG00000169083  Mus musculus; human ortholog selected\n",
       "3  MA0009.2          T  ENSG00000164458                                    NaN\n",
       "4  MA0014.2       PAX5  ENSG00000196092                                    NaN"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_map = pd.read_table(tf_map_f)\n",
    "tf_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>31.920162</td>\n",
       "      <td>0.012392</td>\n",
       "      <td>39.989468</td>\n",
       "      <td>47.588514</td>\n",
       "      <td>34.921604</td>\n",
       "      <td>46.439177</td>\n",
       "      <td>10.383956</td>\n",
       "      <td>0.242817</td>\n",
       "      <td>24.534779</td>\n",
       "      <td>13.546740</td>\n",
       "      <td>15.400938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.729510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.052697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012272</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>34.321019</td>\n",
       "      <td>54.130236</td>\n",
       "      <td>22.474711</td>\n",
       "      <td>22.001707</td>\n",
       "      <td>80.122759</td>\n",
       "      <td>58.227585</td>\n",
       "      <td>42.266614</td>\n",
       "      <td>52.947538</td>\n",
       "      <td>119.129811</td>\n",
       "      <td>50.828462</td>\n",
       "      <td>64.445213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>4.715796</td>\n",
       "      <td>9.902367</td>\n",
       "      <td>2.222304</td>\n",
       "      <td>2.693052</td>\n",
       "      <td>8.641308</td>\n",
       "      <td>2.948341</td>\n",
       "      <td>3.283652</td>\n",
       "      <td>6.761766</td>\n",
       "      <td>7.120837</td>\n",
       "      <td>4.792727</td>\n",
       "      <td>6.164291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>11.382585</td>\n",
       "      <td>20.353400</td>\n",
       "      <td>12.625342</td>\n",
       "      <td>6.522418</td>\n",
       "      <td>34.807272</td>\n",
       "      <td>6.199940</td>\n",
       "      <td>3.842012</td>\n",
       "      <td>27.191509</td>\n",
       "      <td>15.206148</td>\n",
       "      <td>5.591343</td>\n",
       "      <td>3.786800</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id       A549    GM12878    H1-hESC      HUVEC    HeLa-S3  \\\n",
       "0  ENSG00000000003.10  31.920162   0.012392  39.989468  47.588514  34.921604   \n",
       "1   ENSG00000000005.5   0.000000   0.000000   0.729510   0.000000   0.000000   \n",
       "2   ENSG00000000419.8  34.321019  54.130236  22.474711  22.001707  80.122759   \n",
       "3   ENSG00000000457.9   4.715796   9.902367   2.222304   2.693052   8.641308   \n",
       "4  ENSG00000000460.12  11.382585  20.353400  12.625342   6.522418  34.807272   \n",
       "\n",
       "       HepG2      IMR90       K562       MCF-7       NHEK  SK-N-SH        \n",
       "0  46.439177  10.383956   0.242817   24.534779  13.546740      15.400938  \n",
       "1   0.052697   0.000000   0.000000    0.012272   0.000000       0.000000  \n",
       "2  58.227585  42.266614  52.947538  119.129811  50.828462      64.445213  \n",
       "3   2.948341   3.283652   6.761766    7.120837   4.792727       6.164291  \n",
       "4   6.199940   3.842012  27.191509   15.206148   5.591343       3.786800  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr = pd.read_table(expr_f)\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>FLIPPED__Enhancer.flip.antisense__chr19:185263...</td>\n",
       "      <td>32</td>\n",
       "      <td>52</td>\n",
       "      <td>+</td>\n",
       "      <td>28.7347</td>\n",
       "      <td>2.680000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAAGAGGAGGAGGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif                                          unique_id  start  end  \\\n",
       "0  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     60   80   \n",
       "1  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     60   80   \n",
       "2  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     57   77   \n",
       "3  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     57   77   \n",
       "4  ZNF263  FLIPPED__Enhancer.flip.antisense__chr19:185263...     32   52   \n",
       "\n",
       "  strand    score          pval          qval                    seq  \n",
       "0      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA  \n",
       "1      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA  \n",
       "2      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA  \n",
       "3      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA  \n",
       "4      +  28.7347  2.680000e-12  2.160000e-07  GGAGGAGGAAGAGGAGGAGGG  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = pd.read_table(fimo_f, sep=\"\\t\")\n",
    "fimo.columns = [\"motif\", \"unique_id\", \"start\", \"end\", \"strand\", \"score\", \"pval\", \"qval\", \"seq\"]\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqID</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>fractionBPcovered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>99</td>\n",
       "      <td>0.868421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>0.394737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>24</td>\n",
       "      <td>53</td>\n",
       "      <td>101</td>\n",
       "      <td>0.885965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>84</td>\n",
       "      <td>0.736842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.473684</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               seqID  maxCov  numMotifs  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       5         16   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       6         10   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...      24         53   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...      12         24   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       9         13   \n",
       "\n",
       "   numBPcovered  fractionBPcovered  \n",
       "0            99           0.868421  \n",
       "1            45           0.394737  \n",
       "2           101           0.885965  \n",
       "3            84           0.736842  \n",
       "4            54           0.473684  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_map = pd.read_table(cov_map_f)\n",
    "cov_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "index = pd.read_table(index_f, sep=\"\\t\")\n",
    "index_elem = index[[\"element\", \"oligo_type\", \"unique_id\", \"dupe_info\", \"SNP\", \"seq_name\"]]\n",
    "index_elem = index_elem.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ = pd.read_table(hepg2_activ_f, sep=\"\\t\")\n",
    "hela_activ = pd.read_table(hela_activ_f, sep=\"\\t\")\n",
    "k562_activ = pd.read_table(k562_activ_f, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_reps = [x for x in hepg2_activ.columns if \"rna\" in x]\n",
    "hela_reps = [x for x in hela_activ.columns if \"rna\" in x]\n",
    "k562_reps = [x for x in k562_activ.columns if \"rna\" in x]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. find expr of TFs in HeLa, HepG2, K562"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>31.920162</td>\n",
       "      <td>0.012392</td>\n",
       "      <td>39.989468</td>\n",
       "      <td>47.588514</td>\n",
       "      <td>34.921604</td>\n",
       "      <td>46.439177</td>\n",
       "      <td>10.383956</td>\n",
       "      <td>0.242817</td>\n",
       "      <td>24.534779</td>\n",
       "      <td>13.546740</td>\n",
       "      <td>15.400938</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.729510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.052697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012272</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>34.321019</td>\n",
       "      <td>54.130236</td>\n",
       "      <td>22.474711</td>\n",
       "      <td>22.001707</td>\n",
       "      <td>80.122759</td>\n",
       "      <td>58.227585</td>\n",
       "      <td>42.266614</td>\n",
       "      <td>52.947538</td>\n",
       "      <td>119.129811</td>\n",
       "      <td>50.828462</td>\n",
       "      <td>64.445213</td>\n",
       "      <td>ENSG00000000419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>4.715796</td>\n",
       "      <td>9.902367</td>\n",
       "      <td>2.222304</td>\n",
       "      <td>2.693052</td>\n",
       "      <td>8.641308</td>\n",
       "      <td>2.948341</td>\n",
       "      <td>3.283652</td>\n",
       "      <td>6.761766</td>\n",
       "      <td>7.120837</td>\n",
       "      <td>4.792727</td>\n",
       "      <td>6.164291</td>\n",
       "      <td>ENSG00000000457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>11.382585</td>\n",
       "      <td>20.353400</td>\n",
       "      <td>12.625342</td>\n",
       "      <td>6.522418</td>\n",
       "      <td>34.807272</td>\n",
       "      <td>6.199940</td>\n",
       "      <td>3.842012</td>\n",
       "      <td>27.191509</td>\n",
       "      <td>15.206148</td>\n",
       "      <td>5.591343</td>\n",
       "      <td>3.786800</td>\n",
       "      <td>ENSG00000000460</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id       A549    GM12878    H1-hESC      HUVEC    HeLa-S3  \\\n",
       "0  ENSG00000000003.10  31.920162   0.012392  39.989468  47.588514  34.921604   \n",
       "1   ENSG00000000005.5   0.000000   0.000000   0.729510   0.000000   0.000000   \n",
       "2   ENSG00000000419.8  34.321019  54.130236  22.474711  22.001707  80.122759   \n",
       "3   ENSG00000000457.9   4.715796   9.902367   2.222304   2.693052   8.641308   \n",
       "4  ENSG00000000460.12  11.382585  20.353400  12.625342   6.522418  34.807272   \n",
       "\n",
       "       HepG2      IMR90       K562       MCF-7       NHEK  SK-N-SH        \\\n",
       "0  46.439177  10.383956   0.242817   24.534779  13.546740      15.400938   \n",
       "1   0.052697   0.000000   0.000000    0.012272   0.000000       0.000000   \n",
       "2  58.227585  42.266614  52.947538  119.129811  50.828462      64.445213   \n",
       "3   2.948341   3.283652   6.761766    7.120837   4.792727       6.164291   \n",
       "4   6.199940   3.842012  27.191509   15.206148   5.591343       3.786800   \n",
       "\n",
       "        ensembl_id  \n",
       "0  ENSG00000000003  \n",
       "1  ENSG00000000005  \n",
       "2  ENSG00000000419  \n",
       "3  ENSG00000000457  \n",
       "4  ENSG00000000460  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr[\"ensembl_id\"] = expr[\"gene_id\"].str.split(\".\", expand=True)[0]\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>comment</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000159216.14</td>\n",
       "      <td>2.424524</td>\n",
       "      <td>0.534881</td>\n",
       "      <td>0.675734</td>\n",
       "      <td>6.063154</td>\n",
       "      <td>1.859569</td>\n",
       "      <td>4.693606</td>\n",
       "      <td>38.174505</td>\n",
       "      <td>9.631455</td>\n",
       "      <td>24.160086</td>\n",
       "      <td>31.768711</td>\n",
       "      <td>12.047190</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000137203.6</td>\n",
       "      <td>5.412460</td>\n",
       "      <td>0.248758</td>\n",
       "      <td>0.165780</td>\n",
       "      <td>0.175177</td>\n",
       "      <td>21.243269</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.269146</td>\n",
       "      <td>1.048992</td>\n",
       "      <td>2.160615</td>\n",
       "      <td>18.070842</td>\n",
       "      <td>0.073853</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>Mus musculus; human ortholog selected</td>\n",
       "      <td>ENSG00000169083.11</td>\n",
       "      <td>5.525005</td>\n",
       "      <td>0.012590</td>\n",
       "      <td>6.074826</td>\n",
       "      <td>7.937502</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.279713</td>\n",
       "      <td>0.001697</td>\n",
       "      <td>7.263081</td>\n",
       "      <td>0.357436</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000164458.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.433036</td>\n",
       "      <td>0.344499</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000196092.8</td>\n",
       "      <td>0.001572</td>\n",
       "      <td>8.895258</td>\n",
       "      <td>0.335530</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001473</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.426108</td>\n",
       "      <td>0.001250</td>\n",
       "      <td>0.148886</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name          Gene ID  \\\n",
       "0  MA0002.2      RUNX1  ENSG00000159216   \n",
       "1  MA0003.3     TFAP2A  ENSG00000137203   \n",
       "2  MA0007.3         Ar  ENSG00000169083   \n",
       "3  MA0009.2          T  ENSG00000164458   \n",
       "4  MA0014.2       PAX5  ENSG00000196092   \n",
       "\n",
       "                                 comment             gene_id      A549  \\\n",
       "0                                    NaN  ENSG00000159216.14  2.424524   \n",
       "1                                    NaN   ENSG00000137203.6  5.412460   \n",
       "2  Mus musculus; human ortholog selected  ENSG00000169083.11  5.525005   \n",
       "3                                    NaN   ENSG00000164458.5  0.000000   \n",
       "4                                    NaN   ENSG00000196092.8  0.001572   \n",
       "\n",
       "    GM12878   H1-hESC     HUVEC    HeLa-S3     HepG2      IMR90      K562  \\\n",
       "0  0.534881  0.675734  6.063154   1.859569  4.693606  38.174505  9.631455   \n",
       "1  0.248758  0.165780  0.175177  21.243269  0.000000   6.269146  1.048992   \n",
       "2  0.012590  6.074826  7.937502   0.000000  0.000000   1.279713  0.001697   \n",
       "3  1.433036  0.344499  0.000000   0.000000  0.000000   0.000000  0.000000   \n",
       "4  8.895258  0.335530  0.000000   0.001618  0.000000   0.001473  0.000000   \n",
       "\n",
       "       MCF-7       NHEK  SK-N-SH             ensembl_id  \n",
       "0  24.160086  31.768711      12.047190  ENSG00000159216  \n",
       "1   2.160615  18.070842       0.073853  ENSG00000137203  \n",
       "2   7.263081   0.357436       0.000000  ENSG00000169083  \n",
       "3   0.000000   0.000000       0.000000  ENSG00000164458  \n",
       "4   0.426108   0.001250       0.148886  ENSG00000196092  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_expr = tf_map.merge(expr, left_on=\"Gene ID\", right_on=\"ensembl_id\", how=\"left\")\n",
    "tf_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tf_expr = tf_expr[[\"motif_id\", \"motif_name\", \"ensembl_id\", \"HeLa-S3\", \"HepG2\", \"K562\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. calculate tissue-sp of TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>1.859569</td>\n",
       "      <td>4.693606</td>\n",
       "      <td>9.631455</td>\n",
       "      <td>0.439869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>21.243269</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.048992</td>\n",
       "      <td>0.650207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001697</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name       ensembl_id    HeLa-S3     HepG2      K562  \\\n",
       "0  MA0002.2      RUNX1  ENSG00000159216   1.859569  4.693606  9.631455   \n",
       "1  MA0003.3     TFAP2A  ENSG00000137203  21.243269  0.000000  1.048992   \n",
       "2  MA0007.3         Ar  ENSG00000169083   0.000000  0.000000  0.001697   \n",
       "3  MA0009.2          T  ENSG00000164458   0.000000  0.000000  0.000000   \n",
       "4  MA0014.2       PAX5  ENSG00000196092   0.001618  0.000000  0.000000   \n",
       "\n",
       "   tissue_sp  \n",
       "0   0.439869  \n",
       "1   0.650207  \n",
       "2   0.666667  \n",
       "3        NaN  \n",
       "4   0.666667  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specificities = calculate_tissue_specificity(tf_expr[[\"HepG2\", \"HeLa-S3\", \"K562\"]])\n",
    "tf_expr[\"tissue_sp\"] = specificities\n",
    "tf_expr.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. find avg specificity per tile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>seq</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>FLIPPED__Enhancer.flip.antisense__chr19:185263...</td>\n",
       "      <td>32</td>\n",
       "      <td>52</td>\n",
       "      <td>+</td>\n",
       "      <td>28.7347</td>\n",
       "      <td>2.680000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAAGAGGAGGAGGG</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif                                          unique_id  start  end  \\\n",
       "0  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     60   80   \n",
       "1  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     60   80   \n",
       "2  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     57   77   \n",
       "3  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     57   77   \n",
       "4  ZNF263  FLIPPED__Enhancer.flip.antisense__chr19:185263...     32   52   \n",
       "\n",
       "  strand    score          pval          qval                    seq  \\\n",
       "0      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA   \n",
       "1      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA   \n",
       "2      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA   \n",
       "3      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA   \n",
       "4      +  28.7347  2.680000e-12  2.160000e-07  GGAGGAGGAAGAGGAGGAGGG   \n",
       "\n",
       "   motif_id motif_name       ensembl_id   HeLa-S3     HepG2      K562  \\\n",
       "0  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "1  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "2  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "3  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "4  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "\n",
       "   tissue_sp  \n",
       "0   0.120346  \n",
       "1   0.120346  \n",
       "2   0.120346  \n",
       "3   0.120346  \n",
       "4   0.120346  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = fimo.merge(tf_expr, left_on=\"motif\", right_on=\"motif_name\", how=\"left\")\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "125873"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fimo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "105323"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_nonan = fimo[~pd.isnull(fimo[\"tissue_sp\"])]\n",
    "len(fimo_nonan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "75501"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_deduped = fimo_nonan.drop_duplicates(subset=[\"motif\", \"unique_id\"])\n",
    "len(fimo_deduped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.249383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.416909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.429598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.435871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.462181</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  avg_tf_tissue_sp\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.249383\n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.416909\n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.429598\n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.435871\n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.462181"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_sp = fimo_deduped.groupby([\"unique_id\"])[\"tissue_sp\"].agg(\"mean\").reset_index()\n",
    "avg_sp.columns = [\"unique_id\", \"avg_tf_tissue_sp\"]\n",
    "avg_sp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.212183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.412933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.414182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.419604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.481961</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  med_tf_tissue_sp\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.212183\n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.412933\n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.414182\n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.419604\n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.481961"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "med_sp = fimo_deduped.groupby([\"unique_id\"])[\"tissue_sp\"].agg(\"median\").reset_index()\n",
    "med_sp.columns = [\"unique_id\", \"med_tf_tissue_sp\"]\n",
    "med_sp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  avg_tf_tissue_sp  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.249383   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.416909   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.429598   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.435871   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.462181   \n",
       "\n",
       "   med_tf_tissue_sp  log_avg_tf_tissue_sp  log_med_tf_tissue_sp  \n",
       "0          0.212183              0.222650              0.192423  \n",
       "1          0.412933              0.348478              0.345668  \n",
       "2          0.414182              0.357394              0.346551  \n",
       "3          0.419604              0.361772              0.350378  \n",
       "4          0.481961              0.379929              0.393366  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tissue_sp = avg_sp.merge(med_sp, on=\"unique_id\")\n",
    "tissue_sp[\"log_avg_tf_tissue_sp\"] = np.log(tissue_sp[\"avg_tf_tissue_sp\"]+1)\n",
    "tissue_sp[\"log_med_tf_tissue_sp\"] = np.log(tissue_sp[\"med_tf_tissue_sp\"]+1)\n",
    "tissue_sp.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. find tissue specificity per tile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ[\"overall_mean\"] = hepg2_activ.mean(axis=1)\n",
    "hela_activ[\"overall_mean\"] = hela_activ.mean(axis=1)\n",
    "k562_activ[\"overall_mean\"] = k562_activ.mean(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  \n",
       "0  -3.002306  \n",
       "1  -0.964436  \n",
       "2  -1.394383  \n",
       "3  -1.650828  \n",
       "4  -1.343125  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = hepg2_activ[[\"unique_id\", \"overall_mean\"]].merge(hela_activ[[\"unique_id\", \"overall_mean\"]], on=\"unique_id\").merge(k562_activ[[\"unique_id\", \"overall_mean\"]], on=\"unique_id\")\n",
    "mean_activ.columns = [\"unique_id\", \"hepg2_mean\", \"hela_mean\", \"k562_mean\"]\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>102302</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>-1.692439</td>\n",
       "      <td>-1.727006</td>\n",
       "      <td>-2.688372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105340</th>\n",
       "      <td>WILDTYPE__Enhancer.noflip.NA__chr12:56752751.....</td>\n",
       "      <td>0.171876</td>\n",
       "      <td>-0.247542</td>\n",
       "      <td>-0.492796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105605</th>\n",
       "      <td>WILDTYPE__Lincrna.flip.sense__chrX:73072514..7...</td>\n",
       "      <td>3.459639</td>\n",
       "      <td>1.138591</td>\n",
       "      <td>2.952391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102534</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr6...</td>\n",
       "      <td>-0.903161</td>\n",
       "      <td>-1.035248</td>\n",
       "      <td>-1.436370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102776</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__chr4:40...</td>\n",
       "      <td>-0.227584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.279668</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                unique_id  hepg2_mean  \\\n",
       "102302  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...   -1.692439   \n",
       "105340  WILDTYPE__Enhancer.noflip.NA__chr12:56752751.....    0.171876   \n",
       "105605  WILDTYPE__Lincrna.flip.sense__chrX:73072514..7...    3.459639   \n",
       "102534  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr6...   -0.903161   \n",
       "102776  WILDTYPE_BUT_HAS_SNP__mrna.flip.sense__chr4:40...   -0.227584   \n",
       "\n",
       "        hela_mean  k562_mean  \n",
       "102302  -1.727006  -2.688372  \n",
       "105340  -0.247542  -0.492796  \n",
       "105605   1.138591   2.952391  \n",
       "102534  -1.035248  -1.436370  \n",
       "102776        NaN   0.279668  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ[~(mean_activ[\"unique_id\"].str.contains(\"SNP_INDIV\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"SNP_PLUS_HAPLO\")) & \n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"SCRAMBLED\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"FLIPPED\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"RANDOM\"))]\n",
    "mean_activ.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  \n",
       "0  -3.002306       -1.974712  \n",
       "1  -0.964436       -0.700792  \n",
       "2  -1.394383       -0.886722  \n",
       "3  -1.650828       -1.445209  \n",
       "4  -1.343125       -1.062973  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ[\"tile_mean_expr\"] = mean_activ[[\"hepg2_mean\", \"hela_mean\", \"k562_mean\"]].mean(axis=1)\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  \n",
       "0              0.222650              0.192423  \n",
       "1              0.348478              0.345668  \n",
       "2              0.357394              0.346551  \n",
       "3              0.361772              0.350378  \n",
       "4              0.379929              0.393366  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ.merge(tissue_sp, on=\"unique_id\", how=\"left\")\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# first scale ranges to be positive\n",
    "mean_activ[\"hepg2_scaled\"] = scale_range(mean_activ[\"hepg2_mean\"], 0, 100)\n",
    "mean_activ[\"hela_scaled\"] = scale_range(mean_activ[\"hela_mean\"], 0, 100)\n",
    "mean_activ[\"k562_scaled\"] = scale_range(mean_activ[\"k562_mean\"], 0, 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "      <th>hepg2_scaled</th>\n",
       "      <th>hela_scaled</th>\n",
       "      <th>k562_scaled</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "      <td>20.306424</td>\n",
       "      <td>35.475835</td>\n",
       "      <td>9.487524</td>\n",
       "      <td>0.386721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "      <td>31.843142</td>\n",
       "      <td>44.506588</td>\n",
       "      <td>35.136723</td>\n",
       "      <td>0.165019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "      <td>26.883511</td>\n",
       "      <td>46.540818</td>\n",
       "      <td>29.725285</td>\n",
       "      <td>0.261225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "      <td>19.474389</td>\n",
       "      <td>38.210172</td>\n",
       "      <td>26.497599</td>\n",
       "      <td>0.265622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "      <td>38.747601</td>\n",
       "      <td>33.473407</td>\n",
       "      <td>30.370430</td>\n",
       "      <td>0.117438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  hepg2_scaled  hela_scaled  \\\n",
       "0              0.222650              0.192423     20.306424    35.475835   \n",
       "1              0.348478              0.345668     31.843142    44.506588   \n",
       "2              0.357394              0.346551     26.883511    46.540818   \n",
       "3              0.361772              0.350378     19.474389    38.210172   \n",
       "4              0.379929              0.393366     38.747601    33.473407   \n",
       "\n",
       "   k562_scaled  tile_tissue_sp  \n",
       "0     9.487524        0.386721  \n",
       "1    35.136723        0.165019  \n",
       "2    29.725285        0.261225  \n",
       "3    26.497599        0.265622  \n",
       "4    30.370430        0.117438  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specificities = calculate_tissue_specificity(mean_activ[[\"hepg2_scaled\", \"hela_scaled\", \"k562_scaled\"]])\n",
    "mean_activ[\"tile_tissue_sp\"] = specificities\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "      <th>hepg2_scaled</th>\n",
       "      <th>hela_scaled</th>\n",
       "      <th>k562_scaled</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>seqID</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>fractionBPcovered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "      <td>20.306424</td>\n",
       "      <td>35.475835</td>\n",
       "      <td>9.487524</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>0.394737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "      <td>31.843142</td>\n",
       "      <td>44.506588</td>\n",
       "      <td>35.136723</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>24</td>\n",
       "      <td>53</td>\n",
       "      <td>101</td>\n",
       "      <td>0.885965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "      <td>26.883511</td>\n",
       "      <td>46.540818</td>\n",
       "      <td>29.725285</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>84</td>\n",
       "      <td>0.736842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "      <td>19.474389</td>\n",
       "      <td>38.210172</td>\n",
       "      <td>26.497599</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.473684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "      <td>38.747601</td>\n",
       "      <td>33.473407</td>\n",
       "      <td>30.370430</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>99</td>\n",
       "      <td>0.868421</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  hepg2_scaled  hela_scaled  \\\n",
       "0              0.222650              0.192423     20.306424    35.475835   \n",
       "1              0.348478              0.345668     31.843142    44.506588   \n",
       "2              0.357394              0.346551     26.883511    46.540818   \n",
       "3              0.361772              0.350378     19.474389    38.210172   \n",
       "4              0.379929              0.393366     38.747601    33.473407   \n",
       "\n",
       "   k562_scaled  tile_tissue_sp  \\\n",
       "0     9.487524        0.386721   \n",
       "1    35.136723        0.165019   \n",
       "2    29.725285        0.261225   \n",
       "3    26.497599        0.265622   \n",
       "4    30.370430        0.117438   \n",
       "\n",
       "                                               seqID  maxCov  numMotifs  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       6         10   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...      24         53   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...      12         24   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       9         13   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       5         16   \n",
       "\n",
       "   numBPcovered  fractionBPcovered  \n",
       "0            45           0.394737  \n",
       "1           101           0.885965  \n",
       "2            84           0.736842  \n",
       "3            54           0.473684  \n",
       "4            99           0.868421  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ.merge(cov_map, left_on=\"unique_id\", right_on=\"seqID\", how=\"left\")\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_activ[\"log_max_cov\"] = np.log(mean_activ[\"maxCov\"]+1)\n",
    "mean_activ[\"log_num_motifs\"] = np.log(mean_activ[\"numMotifs\"]+1)\n",
    "mean_activ[\"log_bp_cov\"] = np.log(mean_activ[\"numBPcovered\"]+1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. plot correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:5248266..5248380,-__1715.1.0.0</th>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:67993585..67993700,+__2318.1.0.0</th>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:33571488..33571603,+__2316.1.0.0</th>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:76375037..76375152,+__2319.1.0.0</th>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:153518247..153518361,-__1714.1.0.0</th>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    tile_mean_expr  \\\n",
       "unique_id                                                            \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...       -1.974712   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...       -0.700792   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...       -0.886722   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...       -1.445209   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...       -1.062973   \n",
       "\n",
       "                                                    tile_tissue_sp  \\\n",
       "unique_id                                                            \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...        0.386721   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...        0.165019   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...        0.261225   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...        0.265622   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...        0.117438   \n",
       "\n",
       "                                                    avg_tf_tissue_sp  \\\n",
       "unique_id                                                              \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...          0.249383   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...          0.416909   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.429598   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.435871   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...          0.462181   \n",
       "\n",
       "                                                    med_tf_tissue_sp  \\\n",
       "unique_id                                                              \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...          0.212183   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...          0.412933   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.414182   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.419604   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...          0.481961   \n",
       "\n",
       "                                                    log_avg_tf_tissue_sp  \\\n",
       "unique_id                                                                  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...              0.222650   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...              0.348478   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.357394   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.361772   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...              0.379929   \n",
       "\n",
       "                                                    log_med_tf_tissue_sp  \n",
       "unique_id                                                                 \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...              0.192423  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...              0.345668  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.346551  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.350378  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...              0.393366  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for_joint = mean_activ[[\"unique_id\", \"tile_mean_expr\", \"tile_tissue_sp\", \"avg_tf_tissue_sp\", \"med_tf_tissue_sp\", \n",
    "                        \"log_avg_tf_tissue_sp\", \"log_med_tf_tissue_sp\"]]\n",
    "for_joint.set_index(\"unique_id\", inplace=True)\n",
    "for_joint.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## tissue specificity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for_joint.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#cmap = sns.light_palette(\"#8da0cb\", as_cmap=True)\n",
    "cmap = sns.light_palette(\"darkslategray\", as_cmap=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.0909131237771, spearman p: 3.0548323287e-05\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHcAAABoCAYAAADLuW/EAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAD51JREFUeJztnXlQ1OUfx1/LubogyiUKHqAp5pliqYz+PNC0aaQYkWYa\nQ00da8guYUAlTCFQUBEcUrOJolTwyhzHbMhxdMQO80ByEYUOITAx8OAQWL6/P2y3BXf3uwh7uq8Z\nZpbdZ5/vZ7/v7+d5Ps8tEQRBwIZVYmdqA2wYDpu4VoxNXCvGJq4VYxPXirGJa8XYxLVibOJaMTZx\nrRgHU158yZIl+Pn5mdIEs6G8vJxdu3Z1aZ4mFdfPz4+1a9ea0gSzwRD3QVTc9PR0QkJCGDFiRJdf\nXBuCICCRSPRKe/r0afbv34+TkxPx8fH06NEDgAMHDnD27FmcnZ1Zu3Yt69at48GDB/j4+PDee+8Z\n0nyzQVTcuXPncvr0aXJzc3F1dWX69OkEBQUZxJjMzEyqqqpYsGABgYGBFBcXs3v3btXnixYtwt/f\nv813vv32W9LT0zl79izHjx8nPDwcgHPnzpGWlsa+ffs4ceIEEomEjRs3kpiYSFlZGQEBAQb5DeaE\nqLhKD2ppaaGpqYlr166xf/9+UlJSRDM/efIkP/30EzExMXob9OqrrxIYGAhAYGAg69ateyTNjRs3\nSE1NZfbs2SgUCiQSCb1796awsFCVRjnY5e3tzd27d+nfvz9JSUncuHGDO3fu6G2PJSMqbl5eHnPm\nzCEyMpKGhga6devGyJEjRTO+dOkS8N9NVpKbm0tubi4AdnaPBusymUz1+tdffyUnJ0f1/9KlSxk0\naBD9+vUjIyMDePgAtba2UllZSZ8+fVRpW1paAKiqqsLf3x+FQsGSJUuIjo5mwIABovZbAzrFTU9P\n5+LFi0ilUk6cOMEvv/xCTk6OXvXvqVOncHV15cqVK9TU1NCrVy8AIiIiiIiIAMSDiOHDh4uWEKGh\noURHRyOVSomPj2fLli0sW7aM4OBg4uLikMlkhIeHk5KSwoULFxg4cCDu7u6i9lsFgg7Ky8uFvLw8\noaKiQqioqBDq6up0JddIamqq1s8SEhI6nJ+1Yoh7odNzjxw5QllZGT///LMqgt24cWOHHp6VK1d2\n6uEzFZfLyjS+P9KCAjGd4i5fvpxDhw4xc+ZMXFxcjGWTydEmrPIzSxFYNKDy9PQkMzOT5uZmgoKC\neOGFF4xhl8nQJaylIdq3PG7cOMaOHUtrayunTp0yhk0mQ13Y2zU1Gv/apzNnRD03ISGBkJAQ4uLi\ncHZ2NoZNJqG9sNaATs/dvn07APn5+axZs6ZDnRHWiiV5r2hAVVJSwpAhQwAoLi42ilHGxhKEehx0\nihsdHU1xcTHDhg0DoKmpifT0dKMYZizaC6utSFa+79GrF7dravD4t1PGnNEpbmpqKoWFhXh6eqJQ\nKHBzczOWXSahvbBida+5N4tEo+VPP/0UeNhXGx8fb3CDjImuIErMgy0B0WjZ0dERb29vBEGgW7du\nxrDJRhchKu7ChQtZvXo1LS0tqg5/fcjPz6e4uJjKyko+/PBDHBxMOulDJ9q8trqmBk8LqFu1IVos\nOzk5MX78eCZMmEB+fr7eGbu7uxMVFYWTkxONjY2q93NzcwkLCyMsLIyioqLHs9oIVKsJbKmIirtt\n2zZKSkpwdnbGyclJ74x79+5NSkoKL730Upt+6YiICA4ePMjBgweNOnVHF2KRryaBLaHuFRXXxcWF\nKVOmIJPJUCgUemeckZGBi4sLP/74I/fv3++UkYZAW9v2dk2NRjEt0YNFK8L58+fj4eHBl19+Sb9+\n/fTOeMOGDZ0yzBzRVAebc3NIVNwxY8YAEBcXZ3BjTIWyiNVU1P5TWwuAe8+eRrWpK7CtOGiHtuJX\nKTJYRn0Leop79+5d9u7dy7Jlywxtj9HR5bWWjs5iOT8/n6NHj3Lv3j1aW1vZtm2bsewyCe29Vt1b\n1f+3lLavTs/NysrCy8uLVatW8dRTT9G9e3dj2WVUdHlt9b/RsyU2h3R67sGDB5HL5ezevZuCggIG\nDx6smtFvDaiLI+a1lohonTts2DDWrFnD4cOHcXV1NYZNRkVXhKxv29Zcx4N1eu7OnTspLS0lICAA\nuVxOY2Mjs2fPNpZtRkfMW6traiyqSaRT3N9//50NGzYQGRlJdna23ivvzB0xT6vWUFwrg6h/amvx\ntJABe53iCv+u8/H39++wsEeOHKG6uhp4uDrPXFAKqy0Y0jZgoN47pf7anEXWWedeuXKFrVu3UlhY\nyNatW9m6daveGV+/fp1Fixbx999/d9pIQyEmsKWj03OzsrKQSCTMmzevwxk3NDQAPOLxYqv8DIk+\ngY9SWEvudlSi8+4ePXqU1NRU5HI5vr6++Pr66p1xYGAgO3fufOQ75jDkJ9Y+VQ+s9G0SmWPErNNz\nr169ypYtW4iJiSEkJKRDGYeFhXXKMGPyuMWwOde3IOK5ysF5a+2ZUuef2lqrqWuV6BS3tLRUNXc5\nOjr6iVxx0L7Ora6peeRBMNdVCDqL5b179z6y7cGTxj+1tSqBLW3CnE5xR44cydixY9tsBJacnGxw\no8wNfSJnZf1rTjMzdIr7/fffc/z4cf7880/69u3L888/byy7zBJNQ37mHFTprHN9fHyIjIxkxYoV\n1NXVsXTpUmPZ1eU8Tn2oPtynqUuyPeY2BCg65FdUVISbmxshISGsWLHCWHYZjY5EyOoDB7oG7s2l\naNbpuYcPH+bevXuUl5eTnZ39REbL7dHUqWFuHqtEp+d+/vnnxrLDZHj26tUl7Vtl3WtOgZVt9qMG\nlEWvPs0ec+74EJ23fO/ePUpKSmhtbQVg/Pjxopm2traSnJyMk5MTo0aNMosoe2RAgCqoUvew2xra\nrsq6VTl2qz6mqxRenwEFU3uvqLjx8fFMnDhRNYKjj7j19fXMmzcPmUzG/v3724hrylEhMbQJphTa\nvWdPPHv1avMwKF+bY3NIVNzBgwfrvXQzIyOD69evAzBixAgkEglvvPFGmzQd2fvRkHR0+wOliOre\nq6nYNieRRcW9evUqK1asQCqVAujcHlDZVCorK2P9+vVMnDiRH374gf/9739dZG7nUC+a4T+B1V+r\nC6b0WHWU3quOhxbvNXVAJSrupk2buHnzJoIg6L2eNiAggM8++6zTxhkCTQLrQltQ1f575iYs6CHu\nypUrqaurY9CgQdy5c8cqtgdUv/EdEVodbWnNQVQlouJ6e3sTHByMm5sbBQUFxrDJqLQX43JZ2WPV\nm+YkqhJRcQMCAvD29mbHjh2qDbGtmfbFtrY0loCouJMnT0Yul/Pmm28ycOBAI5hkeixFPDFEG5rb\nt2/nwoUL2NnZkZSUZAybbHQRouI6Ozsjk8kYMmQI9fX1xrDJRhchKu6MGTO4cuUKy5cvt+p1QtaI\naJ3r6+vLzJkzaWlpeWLO47EWRMXNysoiPDy8Q0ew2TAPRItlDw8P6uvrUSgUqoOY9KGhoYEFCxZQ\nV1fXKQNtPD6inltRUcHt27eRSCR6e64gCOTk5DB69OhOG2jj8REVt2/fvjz99NNtjmHThnJUqKam\nhm7duiEIAmfOnGHWrFmqNOY85GdtSASRWeftd7CJiorSO/OsrCwiIyO1PhhdcThyUVGRSRaUdfV1\nDXE4ss7j3iyBl19++Ym6bkewlYtWjMWL25ENvq3huh1BtM61YblYvOfa0I75Hjyghfa75BjrLAVN\nu/M8zrHtxsTiPLf9LjnazlIw9HW1HdtuTlic57bfJUfbWQqGvq62Y9vNCfu1ppw8/BjcunWL06dP\n4+joyP379/nqq6/w8vKiqqqKoUOHduiQjc5cNyIigjFjxlBaWsqMGTMMcs3OYouWrRiLq3Nt6I9N\nXCvGJq4VYxPXiumSplBmZiZz585lwIABOtP99ttvlJSUdOl63aKiIrKzs7l48SKjR49m+PDhlJSU\n0NjYiEQiwdfXl5UrVwIP9/jw8fGhqqqK4OBgevfu3WV2PA4ff/wxM2bMICkpiddffx13d/c2w4hy\nuZybN29y+fLlR+7vpUuXaGhoYMKECVrz79J27o4dO7h16xYtLS0kJCTw0Ucf4ebmxjfffMN3331H\nXl4e77//PrGxsfj5+SGXy/H396e8vJyEhAQKCgo4d+4cTU1NxMTEkJOTQ21tLbW1taSlpbFgwQKe\ne+45SktLSUtLw97enhEjRpCWlkZsbCyJiYk4ODgQGxtLWlraI71VBQUF9OzZEzc3NxobG1m3bh0O\nDg54eXkxc+ZMMjMzcXR0ZOHChezatYu0tDT27dtH//79qa6ubmOb8qBo9Tw8PDxUzaXg4GCCgoLY\nuXMngiAwbdo0PD09yc7Oprm5mYSEBEpLS2lsbEQqlVJdXY2DgwP79u2jsLAQhUJBaGgot27doqCg\nABcXFy5cuEBGRgabNm1i3rx5HDlyRKe4XVYsP3jwgIqKCtasWUO/fv04f/48Xl5eREVF4e/vD0Bl\nZaXqhr/22msMGTKE0NBQpkyZQklJCXv27MHd3R0nJyeuXbtGcHAwzz77LKWlpSgUClxdXYmKisLX\n11d0H+fVq1cTGxvLyZMnVe9NmjSJ6dOnq/7/448/GDVqFFOmTOH27du0trby4osv4u3t/Uh+7W3T\nlAfArFmz2LhxIydOnGD37t24urri5eVFYWEhOTk5JCcnExMTo+pNmzRpUptNU8+cOcP69esJDw+n\nubm5jd0eHh7U1tby119/qbxY2bmiiS4VVzltpq6uDkEQcHR0BFD9EHt7e1X67t27Y2dnp+p0UCgU\ndO/enbfeeos5c+bQt29ftm/fjo+PD35+fgiCoNpg1N7eXvTQyKSkJFJSUpg6darGzwVB4J133sHT\n05PNmzfj6enJ22+/TVlZGQcOHFD1RNX8u363vW0Azc3NbfKA/zZDdXZ2pqWlhbCwMJYuXcrQoUNV\nEwybmppU21C0R/m7GhoaHpmzNnXqVL7++mvV7BWZTKayTxNdViz36NGDHj16kJiYiJ2dHc888wyH\nDh0iJSWFyspK4OEP1sXkyZOJi4tDJpMRFxeHVCrl1KlTNDc3c/ny5U7b2KdPH44dO4aXlxcSiYQ9\ne/YglUoJDAykoaGBTz75BEdHR0JDQ3F0dCQ9PR25XM7o0aPb2LZq1Srg4ang6nkAHDp0iPz8fKZN\nm0ZgYCCbN29GJpOxePFi5s+fT3x8PA0NDaxfv16jjePGjeODDz7A3t5etQjA19eXY8eOsXjxYhIS\nEsjIyAAeOpHObk9DTfGor68Xzp8/LwiCIERFRQmCIAgpKSnCgwcPDHVJk3PgwAHhzJkzBr3Gu+++\nq3qdnJysM63BmkJSqZS8vDwSExMZO3YsAK+88kqHTs+20ZYvvviCoKAgAAoLC1X1vDZsfctWjK0T\nw4qxiWvF2MS1YmziWjE2ca0Ym7hWzP8BFkDuJNxh0yEAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321fd3f7b8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_mean_expr\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(mean TF tissue specificity)\")\n",
    "\n",
    "r, p = stats.spearmanr(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_3.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.0851904541668, spearman p: 9.38739678006e-05\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHkAAABtCAYAAACFvc7EAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEBtJREFUeJztnXtMU+f/x1/lZgcFFFpEwDmyi03Q8RV0Tp0OZXNucerI\nnMZF42XqtqCb6BDwwlRQcOgM/jHnNGEz0aATFZOBhk2jEbPFKQpOouLCrAO5M8e9cH5/kPZXajkH\nXVta2ldi5LQPz3m3b57b57kcmSAIAk4GNC79LcCJ5XGa7AA4TXYAnCY7AE6THQCnyQ7AE5nc1NRk\nKR1OLIibVIKuri4uXrxIbm4uTU1N7Nu3zxq6nJgRUZNTU1O5d+8eEREReHh4sGvXLmvpcmJGRKtr\nNzc3fH198ff3x93d3VqanJgZmVRYU6vVcu7cOXJzc1EoFOzYscNa2pyYCUmTm5qa8PLyAqC6uhqV\nSmUVYU7Mh2ibnJiYiKenJxqNBrlcjre3NykpKdbS5sRMiJrs4uLCpk2bWLNmDV9//bW1NDkxM6Im\nd3V1AeDr62tRER999BEhISEWvYe9oNFoOHDggFnzFDW5rKyM+Ph4/vrrL7744gtkMhk7d+40qwCA\nkJAQvvzyS7Pna49Y4nsQNTk7O5vGxkYGDx5s9htLIQgCMplMNM327dtpbm4mODiYTz75BOiufRIT\nE3F1dWXMmDFERETw1Vdf4ePjw8yZM5kyZYo15NsUoiavX78ehUKBUqnk008/tYqgvXv3UllZycKF\nC1Gr1ZSWlnL48GH9+0uWLCE0NJSamhrkcjlJSUkkJibS2tqKXC6nuLiYMWPGMH/+fOLi4nj06BEr\nV64kPDyclStXOk02RiaTsXnzZpKSkqylB4APP/wQtVoNgFqtZuvWrY+l0Wg0DBs2DAA/Pz8aGxuR\ny+VoNBqCgoIAcHV15d1332Xfvn2cPXuWjo4O630IG0Kyd93V1YWbm5u+E+biYvmJK924HODmzZsc\nOnRIf718+XKef/55lEollZWVADQ0NDBkyBAAVCoV5eXlQPcfaXl5OYsWLcLf359NmzZZXLstIjlB\nsWHDhh7/i0W8Tp8+TU1NDdBdrV69epXi4mLu37/Pxo0bn0pgWFgYaWlpj70eEhJCe3s7mzdv1pf6\n9PR01q1bR25uLkVFRURFRREcHMzOnTsRBIFly5Y9lQZ7R9TkWbNmPVFmd+/eZc2aNaSnpwNw/vx5\nvL29CQ0N7XMeq1at6nPa9evXm7w2Dtg4+hhf1OStW7eiVCqJjo7uUw+7paUFQN8rbm5uJi4ujszM\nTOrq6vDz89Onzc7OJjs7G7BOE+DIiH67eXl5JCUl8ejRI0pKSvTtcm+o1Wr279+Pt7c3586dY/z4\n8WRlZdHR0aFvM3XMmzePnJwccnJyGDVq1H//JE56RbJNfuGFF6ioqKCwsJCioiJiYmJ6TSv2npP+\nQ3KCoqGhgddee40FCxY455TtFNHqOjg4mLCwMOrr6/npp584deqUtXQ5MSOiJTk2NtZaOpxYEGe3\n1gGQ7Hhdu3aNy5cv63vWztJtf0iafOzYMZYvX46rq6s19NgcRXfumHz9fy++aGUlT49kda1QKGhr\na0Mmk0lO/Q00ejNY6j1bQ7Ik+/j4UFBQoL92lOrankyUQtTkH3/8Ea1Wq792lJJsaHB1bS0AKn9/\n0XQ6bLEaFzU5KiqKtrY2a2npd4xN0xls/LMpww3zsDWjRU1WKpXW0tHviBlsTHVtrajRtoZkx+v+\n/fvW0NGvDKT21xR9GkJVVFQQEhJCdHS06IyR8aIB6J5T/u2334iPjzeTZMtjWIqr6uoIMJgiNUxj\nL6VZ0uS4uDgaGxs5efIkq1atIiIignHjxjF//vzH0hovGrh+/TrQvfLSGFuZTxYrxVV1dfr/A/z8\n9NemTLdlJE1evXo1KpWK119/nTNnzuDh4dFjzZUhxosGLly4gLe3N3/88Qf19fU95pTnzZvHvHnz\nAMusNX5adKVYZ6gOw2ud6fZSmkVNvn//PhEREURHRyMIAllZWaxYsYKFCxeaTG+8aEC3lKempuax\nRQP2Tm/VuC0iavKpU6e4fPky//77L4DkVpbeFg2sW7fuKeVZF7Eetdjv2HppFjV59OjRDB06VH9t\nGBhxVMSCI7aKqMlDhw7F09OT9vZ2ZDLZgFsZ0lunS9f+Pk3JtkVEu7VqtZpTp06hUCiQyWTk5+db\nS5cTMyI5dnFxcSE8PJwJEyYM6G0mxqV2oJRi6MMQavTo0cTFxaHVaomMjLSGJqtgahICHh86VRtd\nq+ykR22IqMk3b94kJCSEuXPnWkuPTaAz3dhg3WtiRtva5ARImFxfX68PUzoChqXYlMGm0tvDWFm0\nTT5w4AD5+flotVo6OjocYghlqi2uqq2lSqTNtvX2W9TkDRs2oFarGTFiBM899xwjRoywlq5+xbAU\nG5pbJVKN67DFGS3R6vrnn3/mwYMHHD16FOiOSY8bN84qwixJX40wLr261wLsKBACEiZ//PHH1NXV\n0draSmdnZ7+cHdJfmDK4N2w9+iU5Tt6yZQvQHdJ82o3k9oDx0MmY6tpam297e0NynOzu7k5AQACC\nIPDMM8+IpjVeNFBQUEBpaSkVFRVs2bIFNzfJ29kcfTHWcJLCFodQkiV58eLFbNy4kYSEBMmtqXfv\n3mXJkiVUVVUB3Qe2xMbG4uHhQWtra4+02dnZxMTEEBMTQ0lJyX/4CE+OoRHGveTqurr/72AZdLT0\n/+ywNEuaXFNTQ1BQENOnT+9xUoApjBcNDB06lLS0NObMmYNCoeiR1tY3oYsFRKpEImS2iKTJ586d\nw8XFhaioKA4ePCia1njRQGZmJgqFgl9//VU/Jz2QsJexsmQjKZPJaG9vp6KiQvIZFMbV+dSpU/+b\nOhtGF97UmWvLkS/JkrxixQq6urrIzs5m7dq11tBkNXpbzyX6O08Y+rQFJE2+cuUKGo2Guro6SktL\nraHJ5qk2Eea0ZSRNLiwsZM+ePaSnp3PhwgVraLIqA20ViCkk2+T29nbOnDmDu7s7//zzD5cvXwZg\nwoQJFhfnxDxImjx58mSam5uRyWRMmzaNyspKh9jdqPL3HzClW7K6rqurIywsjPz8fMrLy3nvvfeY\nM2eONbRZBXP0igP8/Gw6fi1pclVVlT6SNdDGur0ZYzjLpPLz63UliMrIXFsMaUIfTNZqteTk5ODl\n5aWPaA1EdGYZGqoyYbbOWHuabpRsk3VnRGu1WlJTUy0uyFr878UX9fPKus1sunZYZ3SVyO4Ivdk2\nXlVDH0zWJ7TDGaS+oDPW0Oi+YstRLkMknXv06BF37twhKCiIwMBAa2iyGrrSbGg00Cez9dW7DU8x\n6pA0OT09nUGDBjFjxgwOHjyoP8F+oGBo9JNgD9W0DkmTPTw8GDJkCOPGjePEiROiaY0XDZg6ecAW\n0ZVCXRvdF/PsoQTrkDR57NixHDlyhJKSEiZPniya1vikAeNrQ2zlpAFDDA3rS+m2B4OhDyYrlUr9\nZnKpSJfxogHja0Ns9aQBHcal214MNYWkybrH7zQ0NPD777+LLsk1XjSguw4ODjafYitjz+bqkDTZ\ncB9Uc3OzaFrn4wlsE0mTdQ/ilMlkhIWFWUOTEzMjafKyZcsICAggJyeHkSNHWkOTEzMj2a39/vvv\nKS0tZdCgQZw8edIampyYGcmS3NHRwfHjx1m9ejVFRUUWEaHRaP5zD7ukpKRflvaa+74ajcZseemQ\nCaaOyzOgvLyc2tpaVCoVzc3NNltlx8TEkJOT4zD3fRIkq+uCggIUCgWpqank5eVZQ5MTMzNgFg3o\nAiuOct8nYcAsGnCa3DuibXJVVdVjG9WeffZZi4tyYl5Ee9eXLl3iwYMHPV6zlQeN9Nc2WXs801u0\nug4MDKSxsZHIyEgiIyP1W1JtgSfZJmvJ+4qd6W0riP65V1ZW0tDQwMOHDxEEgXfeecdauiR5km2y\nlryv2JnetoLkONlWycnJoaamhs7OTtRqNfn5+QwfPhwPDw8WLFhgMaON76vbuZmRkWGzRz7brclO\n+o5tLMlwYlGcJjsATpMdAKfJDoBZIwZ79+5l1qxZkmdw/vnnn9y+fZu33nrLbPcuKSkhKyuLoqIi\nwsPDCQsL4/bt27S2tiKTyQgODtb3fnNycggMDKSyspJJkyb1eM5Gf/DNN98QHR1Namoqy5Ytw8/P\nr8f05a1bt3j48CHFxcWPfb/Xr1+npaWFV199tdf8LRIW+vbbb6murkar1ZKcnMz27dvx9fUlNzeX\ns2fPcvToUdauXUtCQgIhISHcunWL0NBQNBoNycnJFBYWcuXKFdrb24mPj+fQoUM0NDTQ0NBARkYG\nCxcuZPz48ZSVlZGRkYGrqyujRo0iIyODhIQEUlJScHNzIyEhgYyMjMeiX4WFhQwePBhfX19aW1vZ\nunUrbm5uqFQq3nzzTfbu3Yu7uzuLFy/mwIEDZGRkcOzYMZ599llqamp6aPP19QXokYe/vz8XL17E\n3d2dSZMmMXbsWPbv348gCEydOhWlUklWVhYdHR0kJydTVlZGa2srcrmcmpoa3NzcOHbsGDdu3KCz\ns5PZs2dTXV1NYWEhCoWCa9eukZmZya5du3j//fc5ffq0qMlmr67b2tp48OABGzduZPjw4Vy9ehWV\nSkVsbCyhoaEAVFRU6L/4RYsW8dJLLzF79mymTJnC7du3OXLkCH5+fnh4eHDnzh0mTZrEK6+8QllZ\nGZ2dnXh7exMbG0twcLBkFG7Dhg0kJCRw/vx5/WsTJ05k2rRp+uvy8nJefvllpkyZQm1tLV1dXcyc\nOZOAgIDH8jPWZioPgOnTp7Nz505++eUXDh8+jLe3NyqVihs3bnDo0CF27NhBfHy8Pjo3ceJE3njj\nDX1+ly5dYtu2bcydO1f/WAidbn9/fxoaGvj777/1pVps8sgiJusWyzc1NSEIgv4pNLoP5Orqqk/v\n6emJi4sLHh4eAHR2duLp6cmqVat4++23CQoKYt++fQQGBhISEoIgCHh6eurz6ezsFNWTmppKWloa\nUVFRJt8XBIHPP/8cpVLJ7t27USqVfPbZZ9y7d4/jx4/rI1v19fV6vYbaoHv1jGEegP7zDBo0CK1W\nS0xMDMuXL2fkyJH6c8Pb29vp6uoyqUv3uVpaWh5btx4VFcXJkyf1z+ny8vLS6zOF2atrHx8ffHx8\nSElJwcXFhTFjxnDixAnS0tKoqKgAuj+4GJMnTyYxMREvLy8SExORy+VcuHCBjo4OiouL/7PGYcOG\nkZeXh0qlQiaTceTIEeRyOWq1mpaWFr777jvc3d2ZPXs27u7u7Nmzh1u3bhEeHt5DW1JSEtB9/qhh\nHgAnTpygoKCAqVOnolar2b17N15eXixdupQPPviATZs20dLSwrZt20xqjIyMZPPmzbi6ujJjxgwA\ngoODycvLY+nSpSQnJ5OZmQl0FybRcKpgYZqbm4WrV68KgiAIsbGxgiAIQlpamtDW1mbpW/cbx48f\nFy5dumTRe6xZs0b/844dO0TTWnwIJZfLOXr0KCkpKURERAAwf/58CgoKLH3rAcsPP/zA2LFjAbhx\n44a+H9Abzti1A+AMhjgATpMdAKfJDoDTZAfAabID4DTZAfg/8gKZYDDoo+4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321fd3f748>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_tissue_sp\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(mean TF tissue specificity)\")\n",
    "\n",
    "r, p = stats.spearmanr(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_6.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## number of bp covered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_activ.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.28149830273, spearman p: 1.70689654122e-39\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAH4AAABoCAYAAAA3rEQOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEHRJREFUeJztnWtQVOUfxz+r3K+DCJJgCZICSibiTBSaKVopOWqaljmr\nkzaW4OgoFCiiBgppiWKkDk7ecgQdnWR8oYOOA4OleRdhQVFKNkwxRGS5Lez/hf/dWXDZswgsB3Y/\nr5azz/7Ol/N97ud5zpGoVCoVZkyOPt0twEz3YDbeRDEbb6KYjTdRzMabKGbjTRSz8SaK2XgTxWy8\niWLRnSdftGgRXl5e3SlBNJSVlZGenm6083Wr8V5eXqxbt647JYgGY18HQeNTUlIICwtjxIgRxtAD\ngEqlQiKR6E2zceNGFAoFnp6efPXVVwDU1tYSFxeHlZUVw4YNY+rUqSQkJCCRSPj4448JDQ01hvwe\ngaDx06ZNIzc3l4yMDBwdHZkwYQLBwcFdIiY1NZUHDx4wf/58/Pz8kMlkHDp0SPP9woUL8fb2pqKi\nAhsbG2JjY4mJiaGurg4bGxtKS0sJDw9n/PjxrFy5kgEDBjB58mRGjx5Nenq62XgtBI1XlzylUklD\nQwO3b9/m6NGjJCUlCQY/d+4cFy9eJDo62mBB8+bNw8/PDwA/Pz82bNjwQpqysjJeeeUVAPr160dV\nVRU2Njb4+/vj6+vLnj17GDNmDKNGjUIqleLg4MDnn39usAZTQND4zMxMPvzwQ6RSKbW1tdja2hIY\nGCgY+Pr168DzalubjIwMMjIyAOjT58VBhb29vebzrVu3OHDggObvxYsXM2TIEPr378+DBw8AePLk\nCS4uLgA8e/aMxMREpFIpfn5+7Nu3j+TkZEaOHMny5cuZPn26oG5TQa/xKSkpXLt2DRsbG86ePcvl\ny5c5cOCAQe19Tk4Ojo6OFBQUUFlZqTFnzpw5zJkzBxDu0AwfPlxnzeLl5UVDQwNr167V1A7JyckE\nBgYil8vZu3cvrq6uzJgxg+3bt2NlZcXo0aMFNZsUKj2UlZWpMjMzVXK5XCWXy1U1NTX6kutk8+bN\nbX4XHx/f7ni9FWNfC70lPisri7t37/Lnn39qetrff/99uzLWqlWrOpQxxUzxv4oWfw8dYNdNStqP\nXuOXLFnC8ePHmTRpEg4ODsbSJHpaG67vuFgzg2Dnrn///qSmptLY2EhwcDBTpkwxhi7R0pbp+tKL\n0XzBufrRo0cTFBREc3MzOTk5xtAkWgw1/XF1A4+rG7pYTccQLPHx8fGEhYURExODtbW1MTSJEl2m\nC5n7uLoBV0crUZZ6vcbv3LkTgOzsbLKzs1+qc9dbEXuJFkKwc1dcXMzQoUMBkMlkRhEldvSZXlpd\nA8BgR/s204gBvcZHRUUhk8nw9/cHoKGhgZSUFKMI60mozW59bLCjvaa6Fxt6jd+8eTM3btygf//+\nNDU14ezsbCxdPQZdpvcEBHv1e/bsAZ7fpImLi+tyQb2V9g4DuxpB4y0tLXF3d8fLywtbW1tjaOox\nCJV29fdi7AgKDucWLFjA6tWrUSqVmpsrhpCdnY1MJqO8vJz169djYdGti31eivaW0qJHz9MPc9M9\ndBPTsE6wxFtZWTFmzBjeeustsrOzDQ7cr18/IiIisLKyoq6uTnM8IyODmTNnMnPmTPLz819OtQhR\nm976s1hLvaDxO3bsoLi4GGtra6ysDO+dDhgwgKSkJKZPn95inn/OnDkcO3aMY8eOGXU5V1eibbSu\nY2LsAAoa7+DgwLhx47C3t6epqcngwNu3b8fBwYELFy7w7NmzDonsDtqq5ttTcnVlCLF08gQb3k8+\n+QRXV1cOHjzIoEGDDA6cnJzcIWFiR12KdZmrTdEjRZttfnciaPybb74JQExMTJeL6encedgyE/i6\ni89wNeadNJ1Ea9PbOiYWDDL+6dOnHD58mC+//LKr9YgCoXa4PZ21Ow8VFD1SUFpdI6qevd6qPjs7\nm5MnT1JdXU1zczM7duwwli5R0to4ofZdzOgt8Wlpabi5uREbG8vrr7+OnZ142ywxUfrgKaUPnna3\nDL3oLfHHjh2jsLCQQ4cOcf78eXx9fZk9e7axtIkG7ZKuqzev3ZaL3XA1gr16f39/1qxZQ2NjI2fO\nnDGGJtFgSJusNl2X4aUPnjLYw6nTdXUGeo3fvXs3JSUl+Pj4UFhYSF1dHR988IGxtImO1qVdl+ly\n+VM8PcVptjZ6jS8tLSU5ORmpVMrevXsFd7D2ZoRMl8tbmg+IOgPoNV71/31v3t7e7TY9KyuLiooK\n4Pku155MW8O31qY/Ln+s+c71FdeuF9YB9PbqCwoK2LZtGzdu3GDbtm1s27bN4MB37txh4cKFPHz4\nsMMijYkhc+nanTldpqv/lsvF27vXW+LT0tKQSCTMmjWr3YFra2sBXqgphHbLipkWd9y0DFWb3ii/\nozlm6elrPGEvgV7jT548SWFhIVOnTiUsLKxdgf38/Ni9ezeenp4tjrdnt2xPQtv0noBe44uKiti6\ndSvR0dHtNn7mzJkdEtYdaFfz6qFc6/ZdVzXfE9Fb16oXXpjCjJ1Q295WNS9EjxzHl5SUEBUVxf37\n94mKiuq1O2lam966tLcewkHPLu0gYPzhw4dfeJRJb8NQ09WUPnjawnR1x87S0/eFdr7HjuMDAwMJ\nCgpq8RDCTZs2dbkoY9Hekq6rJ68LS09f0Y/j9Rp/5swZTp06xd9//83AgQN5//33jaWry2nLdDX6\nZujaMl3sQzht9HbuPDw8kEqlLFu2jJqaGhYvXmwsXUZF1903eG669i3Wnt6uayN4WzY/Px9nZ2fC\nwsJYtmyZsXR1KbqGbaB/AaWuGTp1m96TSroavcb/9ttvuLu7U11drblJ09N79YYub+7MHvwwNzsG\nO9qLatesXuP37dtnLB1dzss80UIIQ0q6WFfa9qzJ8i5E3wLK9g7LXF9xxdPTSefkjVj2zgmuwKmu\nrqa4uJjm5mYAxowZIxi0ubmZTZs2YWVlxRtvvNEjRgODHe1bmO/rbqdzebR6mNa6Z689fFNnFF93\nO001LzYEjY+LiyMkJERzJ80Q4xUKBbNmzcLe3p6jR4+2ML4n3Z1TV9OlD1quqpHLn+ocp6vT6Crp\nYmrfwQDjfX19Dd4evX37du7ced7THTFiBBKJRPMseTXddXdu6AC7F9p5V0erFu28utSrtzwVPVK0\nq41WG67+TetOnViqeTDA+KKiIpYtW4aNjQ2A3l69erh39+5dvvvuO0JCQvjjjz949913O0lux2iP\n+dD2Pve2mgFtw9WxxGg6GGD8Dz/8wL///otKpTJ4P7uPjw+//PJLh8V1BW2Zr+ZxdYNBbfIwN7sW\n433tTKL+vVhNBwOMX7VqFTU1NQwZMoSqqqpe8UhTtRG6hnit22J9GUF9XP2EK11xxGg6GGC8u7s7\n77zzDs7Ozpw/f94YmoxGa1MMyQjaqJsIbdO104vVdDDAeB8fH9zd3dm1a5fmZQO9lbaMamu2T1+m\nELPpYIDxY8eOpbCwkK+//prBgwcbQZL4ELuJL4PgQHrnzp1cvXqVPn36kJiYaAxNZoyAoPHW1tbY\n29szdOhQFIqeuy3YTEsEjZ84cSIFBQUsWbLEpPfN9TYE23hPT08mTZqEUqmkqqrKGJrMGAFB49PS\n0pg9e7ZBr/0003MQrOpdXV1RKBQ0NTWhVCoNDlxbW8v8+fOpqRHfw/3MGFDi5XI5jx8/RiKRGFzi\nVSoVBw4cYOTIkR0WaKZrEDR+4MCBBAQEtHj1Z1uo785VVlZia2uLSqUiLy+PyZMna9L0pNuyvRmJ\nSmDHROsnXUVERBgcPC0tDalU2mamWbRoUYs1+x0lPz+/y5+P21XnKCsrIz09vdPjtolR32vZxcyY\nMaNXnMMYmOtaE6VXGd+eFymI+RzGQLCNN9M76VUl3ozh9LwXxbTB/v37qa+v57///uObb77pknOc\nO3eOixcvEh0d3SXxjUmvKfGfffYZnp6eXTZTeP36dYBe87yAXmN8Xl4eU6ZMwcnJSbP5ozPJycmh\ntLSUgoICKisrOz2+sek1Vf29e/coKirC2tq6S2YEIyMjAaioqOgVS9DMvXoTpddU9Wbah9l4E8Vs\nvIliNt5EMdj41NRU/vrrL8F09+7d49SpUy8taNWqVYJp9PVHV69ezdmzZ9sVrzPQp0lbw4ULFzhy\n5EinnVcd++eff6a4uBipVEpGRgZPnjzR+7t2D+d27drFo0ePUCqVxMfHs3HjRpydnTlx4gSnT58m\nMzOTlStXEhsbi4eHB/fu3WP58uWkpqayZcsWjhw5wquvvsr+/fvx9/dHJpPx2muv8c8//7BlyxZu\n3brF+vXrqampISkpibS0NJ48eYKlpSXz5s0jOjqajz76iE8//RSAo0ePkp+fT11dHV988QU3btwg\nPDxco1c7XmRkJFFRUQQEBGBlZcW3334LoPlfmpubCQkJwdnZmdOnT1NfX09UVBRbt24lKSmJ2NhY\nYmJi2Lp1K/D8tS19+/YlJyeHuXPncunSJY3WFStWEBcXh6OjI/fv329xDbOysrh58yYDBw6ktraW\nadOmAXDixAlWrFgBwNKlS3Ven/T0dM31nzVrFjKZDJlMRklJCXV1ddjY2DB58mQOHjyod+1Eu6r6\n+vp65HI5a9asYdCgQVy5cgU3NzciIiLw9vYGoLy8HAsLC5qbm1mwYAEzZszg6tWrOuMtXboUOzs7\nli5dio+PD+Xl5Xh7exMfH4+vry/Xr1/n6tWrODs78/DhQxQKBQEBARrTAX7//XfWrVvH+PHjuXv3\nLsOGDSMkJETzvXa86upq/P39Wbt2LQqFQrNqOC8vj+DgYDZt2oSHhwcnTpwgMTERqVTKyZMnUSqV\nKBQKLCwsyMrKQqlU4uLiQlFREQBTp07Fzc2thdbc3FzGjh3LmjVrcHR0bPF/T5gwgQ0bNiCTyXjv\nvffIzc0lNzeXiRMn6r0+JSUlLa4/PH9KuJ+fHwBvv/02YWFhuLi4UF5ertfLdhuvnhypqalBpVJh\naWkJoHlVeN++fTXp7ezsNJlAvV5PPetla2uLRCKhT58+moclNzU1aWbdmpubsbW1JSAggMjISMaO\nHYudnV2bq3lqamp0vu1aO17rN2KrPzc2NiKRSFCpVC1eea6OGRoayk8//cS4ceNQKpVMmjSJyMhI\nzUoce3t7GhsbW2hVXy94sRlQa7KwsGDkyJHIZDJu375NYGCgJo2u69P6+ut7u3djY2Ob30E7jXdy\ncsLJyYmEhASePXvGqFGjuHPnDklJSZocZm1trfO3w4cPJyUlhcuXL+s9R1lZGQkJCVRUVODn50dV\nVRVr166lurpaZ/rAwEDWrVvHlStXCA0N1RvP2tqaS5cukZCQgIuLC7a2tgCEhoZy/vx5oqKiaGho\nYMKECaxevZrjx48THh7OxIkTycrKYty4cYSHh3PkyBFiYmLw8PDQnKe11tDQUPLy8khMTHzBhLy8\nPOLj4wkKCkIikWBvb4+Tk5PgYtbW13/o0KE0NDQgk8leSKsukG3SkeU7CoVCdeXKFZVKpVJFRESo\nVCqVKikpSVVfX9+RsF3G/fv3VT/++GN3y3iBX3/9VXXx4sVOi1dZWalKS0vTm6ZDwzkbGxsyMzNJ\nSEggKCgIgLlz55Kdnd2RsCbFzZs3uXbtGsHBwZ0W8/Tp04Irhcxz9SaKeQLHRDEbb6KYjTdRzMab\nKGbjTRSz8SbK/wDfFpvJwKToswAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321fcd3c50>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_mean_expr\"], cmap=\"Blues\", \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(number of bp covered by motif)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_1.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.164330518614, spearman p: 3.66858885488e-14\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAIAAAABtCAYAAABp5GmXAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAERZJREFUeJztnXlMVNf7h59BdgYoCkJxC9VWlBJT0DQqWhXbGGutmDZa\njVFjNU0FU7SobG4BxdZ+MZraJZpYuxjEokJcMLQ2ECUad9GOVVwKCgVkKTCyjDO/P/jN7cw43oss\n1wszT0KYO3PmzHvv+ZztPe+9R2UwGAzYsVkcXrQBdl4sdgHYOHYB2Dh2Adg4dgHYOHYB2DjPJYDG\nxsbussPOC8JRKoFer6egoIDs7GwaGxv59ttv5bDLjkyICiA1NZU7d+4QFhaGs7MzX331lVx22ZEJ\n0S7A0dERb29v+vXrh5OTk1w22ZERlZQrWKfTcerUKbKzs1Gr1WzZskUu2+zIgKQAGhsb8fDwAKCy\nshI/Pz9ZDLMjD6JjgPj4eNzd3SktLcXV1RVPT09SUlLkss2ODIgKwMHBgeTkZGJjY0lPT5fLJjsy\nIioAvV4PgLe3d7ca8fHHHzNw4MBu/Y2eQmlpKbt375bt90QFUFxczOrVq/n777+Ji4tDpVLxxRdf\ndLkRAwcOZMOGDV2eb09E7usgKoCMjAzq6up46aWX5LJHwGAwoFKp2pW2oKCAgwcP4uzsTHJyMl5e\nXgBotVr27NlDeHg448aNY//+/dy4cQOtVsvmzZtxcXHpzlPoEYgKYM2aNajVanx9ffn0009lMWjn\nzp2Ul5ezYMECgoOD0Wg0/PLLL8LnixcvJigoyOw7J06cYPv27RQWFpKbm8uHH34IwJ9//sm9e/cI\nDw9Hp9Nx7NgxgoKCGDZsmL3w/x9RAahUKtatW0dCQoJc9gAwf/58goODAQgODmbTpk1PpSkpKeHL\nL79k2rRpPHnyBJVKhb+/P1evXhXShIeHc//+fQAePXrEkydP2LRpE1u2bKG4uJihQ4fKc0IKRnIW\noNfrcXR0FAaEDg7dv4Bo9DsAXL9+nR9//FE4Xrp0KUOHDmXQoEHs2LEDgD/++AO9Xk9ZWRkvv/yy\n1Ty9vb2FwWy/fv2E87F1JBeDEhMTzf6LeQJzcnKoqqoC2prqixcvcu3aNUpKSkhKSuqQgSEhIaSl\npYmmef/994mLi8PV1ZXk5GTS09NZtmyZmZBcXV0JCQkhPj4eLy8vXn311Q7Z09sQFcDMmTOfK7Pb\nt28TGxvL1q1bgbaa6enp+VSfLUZMTMxz/SbA+PHjGT9+vHAcGxsrvJ49e7bwOjo6+rnz7u2ICmDT\npk34+voSGRnZrpnA48ePAYTRu1arZeXKlezYsYPq6mr69u0rpM3IyCAjIwOQp1uxYx3RK3/8+HES\nEhKor6+nqKhIst8MDg7m+++/x9PTk1OnTvHmm2+yd+9eWltb8fHxMUs7Z84csrKyyMrK4vXXX+/8\nmdjpEJJjgGHDhlFWVsaZM2e4fPmyWZNqidhndpSJ5GJQbW0tERERzJs3zx4T0AsR7QIGDBhASEgI\nNTU1HDt2jCNHjshllx2ZEG0B7KPm3o99+G3jSA4CL126RGFhoTADsLcKvQtJAWRmZrJ06VL69Okj\nhz09guLKx5Jphvq5yWBJ55EUgFqtprm52cytaku0p7Cf9b2eIAJJAXh5eZGXlycc21IX0NHCN/2+\n0kUgKoCDBw+i0+mE4/YGaPQGpAq/uqFFeN1X7dzd5nQbogKYNGkSzc3NctmiGMQK37TgLd+zJgSl\ntwKiAvD19ZXLDsVjreDbi5JFIOkHKCkpkcMOxWBZ+6sbWtpd+Ma01r7T2fFEd9GuaWBZWRkDBw4k\nMjJSdOXOMiAE2mICzp07x+rVq7vIZGXwsOG/Ag1UW6/d1Q0tih8fSApg5cqV1NXVcfjwYWJiYggL\nC2PMmDHMnTv3qbSWASFXrlwB2iJ8LenJ8QCmhW88bo8IlNgVSApgxYoV+Pn58dZbb5Gbm4uzs7NZ\njJ4plgEh+fn5eHp6cuPGDWpqasxiAubMmcOcOXMA+WPhO4Nl4Yu9bxSFklsCUQGUlJQQFhZGZGQk\nBoOBvXv3smzZMhYsWGA1vWVAiDG8q6qq6qmAkJ7IswpfLL2lCJTWCogK4MiRIxQWFtLQ0AAgefvW\nswJCPv/88w6aJz9D/dw6NWC7X/ffd4d4u4l2D0pAVAChoaH4+/sLx6ZOIVuhr9r5mbMA08J+1udK\nF4GoAPz9/XF3d6elpQWVSmWPCOoARhEoFdHhd3BwMEeOHEGtVqNSqThx4oRcdr0wOtr836xseurP\nks44k7oLyfmXg4MDo0aNYuzYsbS2tsph0wuhuPJxpwr/ed5XEpLTwNDQUFauXIlOpyM8PFwOm2Tn\neX3/HUGp4wBRAVy/fp2BAwcKd9va+Q/jANBYy0sq22ZKg/zUVtMqdRwgKoCamhrBtWunfVgK4WZl\nE8P9XM3SKMkXICqA3bt34+rqytSpU5/rgQ29EVMn0LNqvyliLYKSEBVAYmIiR48eZciQIXLZ0yux\nHAwqyS0sKoDffvuNBw8ecODAAaDNxz9mzBhZDOupPKz4rzUI7K+mpLLBaiuglG5AVACffPIJ1dXV\nNDU18eTJkxfyrCClEKh2k1wLMC1843Fgf2V3AZJ+gI0bNwJtbuCOPuTBTls3cL/uMQ8bHivKISTp\nB3BycqJ///4YDAbc3MSbLMuAkLy8PDQaDWVlZWzcuBFHR8mfeyF0dgEI/qv9/5T/K7znH+AlvK/U\nwaBkC7Bo0SKSkpJYu3at5O3ft2/fZvHixVRUVADQt29foqOjcXZ2pqnJfCCUkZHB7NmzmT17NkVF\nRZ04ha7BWn9sWlNNm/9nzelNC9/asSVKCBOTrJJVVVUEBgYyfPhwsyd8WMMyIMTf35+0tDRmzZqF\nWm1eA3pqQAiYrwKWVDaY9f3V5dXC674BbdfL9HOlOYQkW4BTp07h4ODApEmT2LNnj2hay4CQHTt2\noFarOXv2rBBT0FN4Vu03Yjm1s1bbq8urJVuBF41kC6BSqWhpaaGsrExyzyDLLmLy5Mmds06BWNZ+\nU4y1X19+B4eAVyTzUsI0ULIFWLZsGXq9noyMDFatWiWHTYpCzAMIT0/99OV3zP4rHUkBnD9/ntLS\nUqqrq9FoNHLY1CMwrf3/lP9r1vc/C8s1ASUgKYAzZ86wfft2tm7dSn5+vhw2KRJr4V+Wtd+SntAK\nSI4BWlpayM3NxcnJiX///ZfCwkIAxo4d2+3GyU1H/AGmgzyxAjf1A/SYtQCACRMmoNVqUalUTJky\nhfLycpteFewsxqAQJQwAoR1dQHV1NSEhIZw4cYL79+8TFRXFrFmz5LBNUbRn/m458rc8VpoPANoh\ngIqKCsHD19Pm8h3BWDOtNdPGAhzu58ogPzWB/dX4B7RtTmF0+jgEvCL8GVHygpCkAHQ6HVlZWXh4\neAiePlvCMo7PWi02FYHln/EzJc4AoB1jgOTkZKBNCKmpqd1ukBIwDgaNN4WYisDUL2Ac2D2saBAK\n2pLA/mohXaDaTVEDQGiHAISECl3Jkxtrkb2D/NRWw8KMnw33czVrOZQyAIR2CKC+vp5bt24RGBhI\nQECAHDYpAtNWwBTLFsGIWBM/xNtNkbUf2iGArVu34uLiwrRp09izZ4+wc4gtYM0vYK1bkMI0rZJq\nP7RDAM7Ozvj4+DBmzBgOHTokmtYyIMTaE0N6GsYCMxXC89ZkJdZ8I5ICGD16NPv376eoqIgJEyaI\nprV8QojlsSk97QkhpjW3o2JQWu2HdgjA19dXeNCDlAfQMiDE8tiUnhwQosSC7CiSAjDuu1dbW8uF\nCxdEw8ItA0KMxwMGDOg6i+10KZICML0vUKvViqa1bxnT85AUgHHTaJVKRUhIiBw22ZERSQEsWbKE\n/v37k5WVxfDhw+WwyY6MSA6/f/jhBzQaDS4uLhw+fFgOm+zIiGQL0Nrayq+//sqKFSu4fPlytxhR\nWlrapTOBoqKibt2LsDvzLy0t7ZZ8n4XKYO0xnibcv3+fR48e4efnh1ar7RHdwOzZs8nKyuqx+cuJ\nZBeQl5eHWq0mNTWV48ePy2GTHRnplQEhRgdTT81fTnplQIhdAO1HdAxQUVHx1E2dgwcP7naj7MiH\n6Czg9OnTPHjwwOw9JW8atW/fPpqbm6murmbNmjXd9ju9aQ8E0S4gICCAuro6wsPDCQ8PF277Virz\n5s1jwIABkvcwdgaxPRB6IqICKC8vp7a2ln/++Yfy8nKmT58ul10d4vTp00yfPh0vLy9hp9OuJj8/\nn3v37gl7IPR0RLuAqKgooqKi5LKl09y9e5ebN2/i4uLSbTEGvW0PBElHkJ3ejfJDcex0K3YB2Dh2\nAdg4dgHYOM8tgJ07dwpxgmLcvXuX3NzcDhkF7dtoSmz8mpiYyO+///5c+XUFYjaZ2nD27FkyMzO7\n7HeNeX/zzTf89ddfLFy4kIyMDGpra0W/1+H7vb777jsqKyvR6XSsX7+ezZs34+3tTXZ2NidPnuTA\ngQOsWrWKhIQEAgICuHv3Lp999hk7d+5k27ZtZGZmMnjwYPbt28eIESPQaDQMGTKEhw8fsm3bNq5f\nv87GjRtpbGwkLS2NXbt2UVtbi5OTE/Pnz2f16tW89957fPTRR0DbTudFRUU0NTWxZMkSrl69yowZ\nMwR7TfOLiYkhLi6OkSNH4uzszNq1awGEc9Hr9YwdOxZvb29OnjxJc3MzcXFxpKenk5aWRkJCAvHx\n8aSnpwMwYsQI+vTpQ35+PnPnzuX8+fOCrbGxsSQnJ+Pp6fnUNrw5OTlcu3aNwMBAHj9+zMyZMwHI\nzs4mNjYWgOXLl1u9Prt37xau/wcffIBGo0Gj0VBcXExTUxOurq688847/PTTT6Le2w51Ac3NzTx4\n8ICkpCQGDRrExYsX8fPzIzo6mqCgIADKyspwdHREr9ezaNEioqKiuHTpktX8li9fjru7O8uXL+eV\nV16hrKyMoKAg1q9fz7Bhw7hy5QqXLl3C29ubiooKtFotI0eOFAofoLCwkA0bNjBp0iTu3LnD8OHD\nzZ5iYppffX09I0aMYN26dWi1Wurq6oA2R9Lo0aPZsmULAQEBZGdnk5qaysKFCzl69Cg6nQ6tVouj\noyM5OTnodDp8fHy4efMmAO+++y5+fn5mthYUFDBhwgSSkpLw9PQ0O+8pU6awadMmNBoNkydPpqCg\ngIKCAiIjI0WvT3Fxsdn1h7aI7ODgYADGjRvH1KlT8fHxoaysTLQsOywAo6OlsbERg8Eg7ChmXDzq\n06ePkN7d3V0Qg/EeAaMXzc3NDZVKhYODA87ObTdZPHnyRPDk6fV63NzcGDlyJDExMUyYMAF3d3c8\nPDys2tbY2CjkY4ppfi4uLmZpjK9bW1tRqVQYDAazRTBjnhEREXz99ddMnDgRnU7H22+/TUxMjBAd\n5OHhQWtrq5mtxusFT3cPRpscHR0ZNWoUGo2GW7duERoaKqSxdn0sr7+18zUitc9ThwTg5eWFl5cX\nKSkpNDQ08MYbb3D79m3S0tIExbm4uFj9bkhICNu3b+fChQuiv1FaWkpKSgpVVVUEBwdTV1fHunXr\nqK+vt5o+NDSUDRs2cPHiRSIiIkTzc3Fx4fz586SkpODj4yM8AzkiIoIzZ84QFxdHS0sLU6ZMITEx\nkUOHDjFjxgwiIyPJyclh4sSJzJgxg8zMTOLj481umrW0NSIigtOnT5OamvpUYZw+fZr169cTFhaG\nSqXCw8MDLy8vyRtwLK//a6+9RktLi9WnuElu9WfoArRareHixYsGg8FgiI6ONhgMBkNaWpqhubm5\nK7LvckpKSgz/+9//XrQZT/Hzzz8bzp0712X51dTUGHbt2iWapkumga6urhw4cICUlBTCwsIAmDt3\nLnl5eV2RvU1w7do1Ll++zOjRo7ssz5MnT0oGr9jXAmwcuyPIxrELwMaxC8DGsQvAxrELwMaxC8DG\n+T82PCUwO5tvngAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321fafa668>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_tissue_sp\"], cmap=\"Blues\", \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(number of bp covered by motif)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_4.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## max coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmap = sns.light_palette(\"firebrick\", as_cmap=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.329209707519, spearman p: 3.43536181424e-54\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHYAAABoCAYAAAAkewT6AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAD7hJREFUeJztnXtMU9cfwD/ghCK6Bin4gjjAEdx84WNCfGxx6IZmLyM6\nXRy6zAU3ZOoUBwpMRaVzCz4IUbdlbmwqzMyo08WlGOKiEUU3HwNUdHMp8QFEUR6lUO7vj639la69\nt4VSSuknWWLbe8/97n74nnvuOffc4yEIgoAbl8OzqwNw0zm4xboobrEuilusi+IW66K4xboobrEu\nilusi+IW66I80ZUHf/fddwkKCurKEJwGtVrNl19+abfyulRsUFAQn3zySVeG4DTY+zxIit22bRsx\nMTGMGDHCrgcWQxAEPDw8RLfZvHkzDQ0NDBkyhKVLlwJQV1dHeno6Op2OyZMnM2XKFDZv3oxOpyMu\nLo4XXnjBAdE7B5JiX331VX799Vfy8/Pp168f06ZNY/z48Z0SzM6dO7l79y4LFy4kIiKC8vJy9u3b\nZ/h98eLFhISEUF1djUwmIzU1lZSUFDQaDTKZDLVazYIFC4iMjCQ5OZlBgwaxYsUK/Pz8yM7Odos1\nRp85LS0taLVabty4wcGDB8nKypIsvKioiHPnzpGcnGx1QG+99RYREREAREREsGHDhv9so1arGTRo\nEAD9+/entrYWmUxGREQEDx8+JCEhgaioKCZPnszt27dJSEhg0aJFVsfgCkiKLSgoIDY2lvj4eBob\nG/Hx8WHkyJGSBV+6dAn4p1o1Jj8/n/z8fAA8Pf/bKPf19TX8+48//iAvL8/wecmSJYSFhaFQKLh7\n9y4ADx8+xM/PD4CSkhIGDBjAF198wcqVK1GpVEycOJHvv/+eNWvW8PLLL0vG7SqIit22bRu///47\nMpmMkydPcuHCBfLy8qy63p46dYp+/fpRWlrKgwcPDCd/3rx5zJs3D5BuMDz77LNma4agoCC0Wi3p\n6emG7FYqlcyfPx+lUolMJiMyMhKFQkFqaiq9evXipZdekozZpRBEUKvVQkFBgVBZWSlUVlYK9fX1\nYpubZevWrRZ/y8jIsLk8V8Xe50I0Y48ePcqtW7c4f/68oaX66aef2vSHs2rVqg794XU1tRcumP1e\nPm6cgyOxDVGxCQkJHDp0iOnTp9O3b19HxdTlWJJpbhtnFSzZeFIoFOzcuZPm5mbGjx/PzJkzHRFX\nlyAmtKmqCgDvgABHhdMhJPuKx40bx9ixY2ltbeXUqVOOiMnh1F64YFZqU1WV4T/T74z3dUYkMzYj\nI4OYmBhSUlLw9vZ2REwOw5IUY3GWaKqqcursFRW7a9cuAFQqFSqVql2NJ2fFWTPNXkg2nq5fv054\neDgA5eXlDgmqM5ESapqtmpqaNp9l/v52j6kzEBW7evVqysvLGT58OABarZZt27Y5JDB70t7sNJWq\n/647yBUVu3XrVi5fvoxCoUCn0yGXyx0Vl91w9SrXEpKt4q+++gr4ZxAgLS2t0wPqSoyrYXPZ2p2Q\nbBX37t2bwMBABEHAx8fHETF1Cda0hLsTkmIXLVrE2rVraWlpMXTeW4NKpaK8vJw7d+6wfv16nnjC\n8Q9rWFsNizWYmqqrDf/2VijsE5gDkKyKvby8mDBhAlFRUahUKqsL7t+/P4mJiXh5eaHRaAzf5+fn\nM3v2bGbPns3Vq1fbF7UdEctUY6liOON1XDKNcnJyGDhwICNGjMDLy8vqggcMGEBWVhavv/56m35m\nW4btOgOpKre7X1v1SGZs3759mTp1Kr6+vuh0OqsL3rFjB3379qW4uJi6uroOBekoLFXBlnDmrkXJ\njJ07dy7+/v589913BAcHW12wUqnsUGAdxdYTbY3UpupqvBWKbnEvKyl2zJgxAKSkpHR6MPaiI1Jt\n2Ufm79+mz7j2wgWnGcbr8TMBTKVKVcHWNqi6GqvEPnr0iAMHDvDee+91djwOxVaptm7XlYhWxSqV\nimPHjvH48WNaW1vJyclxVFydgqWeJVuy1Phe1pmvtaIZm5ubS0BAAKmpqTz99NP06dPHUXHZFdPB\n8fZK7U6IZuyPP/5IWVkZ+/bt48yZMwwbNoy4uDhHxdYpmJPaXeWJIdkqHj58OOvWraO5uZnCwkJH\nxNQhTFvEUpnaEamm1bCztIhBQuyePXu4efMmoaGhlJWVodFonPpp+vZK1VghV9aN+olBQuxff/2F\nUqkkPj6evXv3Ss6A60qs7vA3kmoqVOxJRE11tVm5zvrck6hY4d95NyEhITZLPXr0KNX/nrjFixe3\nMzzrsPSEoR59tlqSatp/LCZY3/vk7Ii2iktLS9m+fTuXL19m+/btbN++3eqCKyoqWLx4Mffv3+9w\nkGK0R6qlbcXK6W6IZmxubi4eHh7MmTPH5oIbGxsB/pPpUrPtbEHsmgqOGalx1sdQRcUeO3aMsrIy\nZs2aRUxMjE0FR0REsGfPHoYMGdLme3sN21nbUIL2NZYs0R2qYZAQe+3aNbKzs0lOTrZZ7OzZszsU\nmC1YI7U9GGeiuYaTac+TMw0CiIrVD6w7U4+Ts417OiuiF7mbN28ani1evXq1Ta8c6AysbSgZfjNu\n+ZrJXHs1jpyxkSWasQcOHPjPqwacifb0/+qvr20mWhnt621UtTprw8gaRMWOHDmSsWPHtnnJ1pYt\nWzo9KHPY2qtk7rOp1CYzreammpo2csUwN7LjLNdZUbGFhYWcOHGCv//+m8GDBzvNexyskWpOqPG+\n5qQatjGSay5rTVvGzpjVotfYgQMHEh8fT1JSEvX19SxZssRRcdmM8S2N8W2NcZZaI9VQnsk2prdI\n+mx1RqlgxbDd1atXkcvlxMTEkJSU5Ki4LGJtr5K5ayn8X5jG6HuZHeU4QzUMEmIPHz5MYGAgjx8/\nNgwCdIf5sVINpPZgzQCAs0gFCbHffPONo+LoMKa9StZKtSVbvRUKvBUKs40mZ5IKXfz2067EVKil\nlrA111BnkwpWiH38+DHXr1+ntbUVgAkTJkgW2traypYtW/Dy8mLUqFFd0pr2Dgj4//CbvoVrJnON\nhRrvo/8M/1TD+mw1/c1ZkRSblpZGdHS0YSTGGrENDQ3MmTMHX19fDh482EZsR0d3TE++4XuFgqZ/\nB8P11bH+5JsKNi3P0mf9ddVYqDM/DmOMpNhhw4ZZPX1yx44dVFRUADBixAg8PDwM7xLWY4/RHb1c\nmb8/mpoag1RjudYitq23iVjTWxxnlQpWiL127RpJSUnIZDIA0Vax/nbo1q1bbNy4kejoaM6ePcvz\nzz/f4UDl48a16X0Sk2uK/jdziA3DmctUZ6+C9UiK/fzzz7l37x6CIFg9nzU0NJSvv/66w8GZIiZX\nivaMo5qWayzVmbMVrBC7atUq6uvrCQsLo7a2tstfuac/oXrBxtdRez+Vbyk7nV0qWCE2MDCQSZMm\nIZfLOXPmjCNisgpLgh1xzO6ApNjQ0FACAwPZvXu34WXSzoSpYHuW2Z2RFDtlyhTKysp4//33eeqp\npxwQUvtwBRn2RPJGcteuXfz22294enqyadMmR8Tkxg5IivX29sbX15fw8HAaGhocEZMbOyAp9sUX\nX6S0tJSEhASnnrfjpi2S19ghQ4Ywffp0WlpaqK2tdURMbuyApNjc3Fzi4uKsWpbMjfMgWRX7+/vT\n0NCATqejpaXF6oIbGxtZuHAh9fX1HQrQTfuQzNjKykpqamrw8PCwOmMFQSAvL4/Ro0d3OEA37UNS\n7ODBg3nmmWfaLE1mCf3ozoMHD/Dx8UEQBE6fPs2MGTMM29hzUpYby3gIEk+Em74pJjEx0erCc3Nz\niY+Pt/hH4ciFga9everQpVJtPb69FwYWXQLNlXjjjTd61PHddaGL0mPE2vISbVc4vuQ11k33pMdk\nbE+jR4hVqVTk5OQY1jZwNEVFRQ6fQdEjxFpan8ARWFrSvLPpETMBLK1P4AgsLWne2fQIsTt27CA4\nOJji4mLCwsIcKnfZsmUAVFdXO/TRIner2EXpEdfYnohbrIviFuuiuMW6KHYTu3PnTm7fvi253Z9/\n/smJEyfadYyysjKKioratW97UavVZGdnG46dn59v18WR29N23bNnj+Q2dr/d2b17N1VVVbS0tJCR\nkcHmzZuRy+UcOXKEX375hYKCAj766CM+/vhjgoKCKCsrIyQkBLVaTUZGBsePH+fWrVvcu3cPpVLJ\n8uXLyczMZNOmTSxYsICqqiqL+27atInPPvuM7Oxs4uLiWLNmDdHR0ZSWlhIaGkptbS0bN240xPrK\nK68wffp0ysvLGTZsGK2trSxfvpwNGzbQq1cvhg4diq+vL8XFxYwZM4b79+9z5MgRpk2bZiijqKiI\nn3/+Ga1WS2ZmJkql0rDvvXv3mDt3Lq2trRw/fpzQ0FBKSkrQarUkJyfz4YcfEh4eTmxsLAUFBdTV\n1fHBBx9QX1/PTz/9hE6nIyAggGnTpvHDDz/Q3NzMvHnzCA0N5ezZs0RFRVn0YNequKmpicrKStat\nW0dwcDAXL14kICCAxMREQkJCALhz545hydG3336b8PBwXnvtNaZOncr169cZPXo0kyZNoqGhgZqa\nGhITE5k/fz7Lli1r82iOuX1NkcvlJCYm4unpyapVq2hpaTHMzAcIDg4mKSkJT09PVq5cSVVVFadP\nn2b8+PFkZGRw6dIloqOjmThxIn369MHT05OoqKg2Lww9fPgwSqWSpUuXUlhY2Gbf2NhYTp48SWFh\nITNnzmT//v30798fLy8vbty4gY+PD6mpqSgUCmJjYxkwYABXrlwhPz+ftLQ0w8Ia3377Lf7+/igU\nCi5fvkxkZKTkPCq7i9U/7lJfX48gCPTu3RvA0JXXq1cvw/b6k6V/GadOp2P37t3I5XJCQ0MRBIG6\nujpkMpnh/cdi++rFP3z40LANYJjbKwhCG7Gmv+t0Opqbmw1/eNb0K+u30Wg06HS6NvuOGjWKiooK\n1Go1YWFh9OnTh2XLlhEbG8vgwYMNT5bk5eWh1WoZOXKkoVwPDw+ampoM5/Wdd95hwYIFhIWFIZfL\nqZJ4f6Ndq+Inn3ySJ598kszMTDw9PYmMjOTQoUNkZWVx584d4J+ZBWL4+flx+vRpamtruXLlCkeP\nHmXfvn2sWbOG+Ph40X0HDRpETk4O165da/f/w+TJk0lLS6OkpIRJkybh5+dHSUmJYY0/U2bMmMHa\ntWvRaDSsX7+e9evXG/aFf2Yr6nu6pkyZQkpKCr6+vqSmphrKCAoKoqSkBE9PT7RaLbNmzSI9PR2N\nRsPQoUNZuHAhqamp+Pj4sGLFCh49eoRCar5vZz6e0dDQIFy8eFEQBEFITEwUBEEQsrKyhKamps48\nbLfn3LlzglarFc6fPy/s3bv3P7+rVCqhuLhYtIxOvd2RyWQUFBSQmZnJ2LFjAXjzzTdtWjm6J9LY\n2Eh6ejr79+83O9G8oqKC5557TrQMd1+xi+LuoHBR3GJdFLdYF8Ut1kVxi3VR3GJdlP8BLhhFpjKk\nvp8AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321faafba8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_max_cov\"], mean_activ[\"tile_mean_expr\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(maximum motif coverage)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_max_cov\"], mean_activ[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_2.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.246344610807, spearman p: 2.32313173595e-30\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHgAAABtCAYAAABqf6X6AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEMpJREFUeJztnXtQVPX7x1+7KiJXUS5eIAZwjBnyhpo6aGNaJjalMWMU\njZWVThaY4AVF0DAoKB0c7I90dLKc0UDDwEJo8DI4ShqRF1RKtIx1EFAEldty2e8f/vb8lnX3nBV3\nD7u4rxlHds9zPufZfe/zuT3n8zkKjUajwU6fRdnbDtixLHaB+zh2gfs4doH7OHaB+zh2gfs4jyRw\nU1OTpfywYyH6Sxl0dXVx4sQJ8vLyaGpq4ptvvpHDLztmQlTg1NRUrl27RmhoKA4ODmzZskUuv+yY\nCdEqun///ri7uzN06FAGDBggl092zIhCaqqyo6ODY8eOkZeXh4uLC1988YVcvtkxA5ICNzU14ezs\nDEBdXR1eXl6yOGbHPIi2wevWrcPJyQmVSoWjoyOurq6kpKTI5ZsdMyAqsFKpJCkpidjYWDIyMuTy\nyY4ZERW4q6sLAHd3d4s68cEHH+Dr62vRa9gKKpWKnTt3mq08UYGvXr3KmjVr+O+//1i9ejUKhYIv\nv/zSbBfX4uvry6effmr2cm0Rc38PogJnZWXR2NjI4MGDzXpRU9BoNCgUCpNsT5w4wYEDB3BwcCAp\nKQk3NzcAvv32W65evUpLSwuff/45GRkZNDQ04OHhQXx8vCXdtxpEBY6Pj8fFxQVPT08++ugjWRza\ntm0bN2/eZNGiRQQHB1NRUcHevXuF44sXLyYgIKDbOQUFBWzdupWSkhIKCwtZuHAhAK2traSkpLBn\nzx4qKiooLy/H39+foKAgWT6LNSAqsEKhYMOGDSQkJMjlDwBvvfUWwcHBAAQHB7Np06aHbKqqqvjq\nq6+YO3cunZ2dKBQKfHx8OH/+vGCzbNkySktLuXTpEpGRkcTFxREaGsrq1at56aWXcHV1le0z9RaS\nveiuri769+8vdLiUSssnoLTjboCLFy+yZ88e4fWSJUsICgrCz8+PzMxMAI4fP05XVxfV1dUMHz5c\nsN25cydubm6kpqbS0NDAvXv3AHBzc+NJuRVNMtmwfv36bv+LzWQdOnSIW7duAQ+q0rKyMi5cuEBV\nVRWJiYk9cjAkJIS0tDRRm/nz57N69WocHR1JSkoiIyODJUuWkJOTw9ixYykrK2Pp0qUUFBRQXFyM\np6en0E73dUQFfvXVVx+psMrKSmJjY0lPTwceRJarq+tDbaYYMTExj3RNgLCwMMLCwoTXsbGxAOTn\n53ezexKnWUUF3rRpE56ensyePduknnRLSwuA0Pttbm4mLi6OzMxM6uvrGTJkiGCblZVFVlYWIE+1\n/6Qi+s0ePnyYhIQE7t27R3l5udAOGyM4OJgdO3bg6urKsWPHmDJlCrt376a9vR0PD49utpGRkeTk\n5JCTk8Mzzzzz+J/EjkEk2+BRo0ZRXV3NqVOnOHv2LBEREUZtxY7Z6R0kkw0NDQ1Mnz6dqKgoe07Y\nBhGtokeOHElISAh37twhPz+f3NxcufyyYyZEIzg6OlouP+xYCHv3tY8j2cn6888/KSkpEXrQfTWq\nG86ceei9wc8+2wuemBdJgffv38+SJUvo16+fHP6YFUOi6TL42WdFbRrOnLF5kSUFdnFxoa2trdv8\nsLUiJWhP7G1dZEmB3dzcKCoqEl5bYxX9qML2pHxbFVlU4AMHDtDR0SG8NjUBby201tSIHnf08TG5\nLFsVWVTgmTNn0tbWJpcvZkFKVGO2jyK2LSEqsKenp1x+9BhzVc+miG2LUSw5Dq6qqpLDD7PwKNEr\nVY6xsizd3psbk4ZJ1dXV+Pr6Mnv2bNHMj37CHx7khM+cOcOaNWvM5LI0rbW1wt+O3t5Gj+mibwf/\n/4PRj2hbimTJCI6LiyMxMZHBgwcTExPDypUr+eGHHwzaVlZWsnjxYmr/70s8d+4cgMHbY7KysoiI\niCAiIoLy8vLH+QyiaAVtra01Kq6uncFjBqLZViJZUuDly5eTmZlJQEAAhYWFbNmyhfb2doO2+gn/\n4uJi/v33Xy5dusSdO3e62ZorH2xKJImJZ6qdoWrbFkQWraKrqqoIDQ1l9uzZaDQadu/ezdKlS1m0\naJFBe/2Ev/b2m1u3bj2U8DcXPf2SW3TEGqRTBbfW1hqsroXjNTU21eMWFTg3N5eSkhLu378PILm8\nxFjCf9WqVT10r2c4ensL0aj7tzFaamoeElm3LDGsvT0WFXjMmDH46Hxw3UkPa0dXGH2RWwy0qfoi\na9H9oQjv2VAUiwrs4+ODk5MTarUahUJhdXd0mLsNNCYySFfd1opoJys4OJjc3FxcXFxQKBQUFBTI\n5ZdZMbWTBQ9ENhTh+uXodrisubMl2YtWKpWMGzeOadOmGe0990WMCW3sx2KtIktOdIwZM4a4uDg6\nOjqYOHGiHD7Jhr5YhqpgQ71tbXVtC22xqMAXL17E19dXWK1ni+iLaKz6NQVDbbSuyNbYoxYV+M6d\nO8LUo7XxuFWioapWvyNlaLikFdlWOl2ibfDOnTspKCigo6OD9vZ2mxom9RTtlKb+D0D3PW0tIEyD\nWnGHSzSC169fzy+//IK/v79c/lg9thK5WkQFPnLkCDdu3CA7Oxt4MMc8efJkWRyzNMZmuFrr6oyf\no7NHmNiY2ZoQFfjDDz+kvr6e1tZWOjs7e2WvjsdBavyrL7KYuNrjjja2EZzkODg5ORl4ME3Z00Xc\nlqCnvVWjM1U64mozR8aS/o8ycdLbSI6DBwwYgLe3NxqNhkGDBona6if8i4qKqKiooLq6muTkZPr3\nl7zcY+Ho49Ojuzq04ho611jSXzhu5W2yZAS/++67JCYmsnbtWsnlofoJ/yFDhhAdHY2DgwOtra3d\nbOVK+IvxKJGoFVo30nXH1Oa6XcjcSAp869YtRowYwZw5c7qt0DeEfsLfx8eHtLQ0FixYgIuLSzfb\n3loArj/RYSh6jQ6V9GyMYU1DJUmBjx07hlKpZObMmezatUvUVj/hn5mZiYuLC6dPnxZyynKiX3W2\nmCCQ2PhXeE+vM2bNbbJko6hQKFCr1VRXV0s+s0G/Cn/++ecfzzsRHidKjEWm1D1bhuafrX24JBnB\nS5cupauri6ysLFauXCmHTxbBYGZIYlj0kL3OzFVrXZ1VR64WSYFLS0tRqVTU19dTUVEhh08m0VuT\n+qaKai3tsKTAp06dYuvWraSnp1NcXCyHT4+FVPrOFqLOnEgKrFarKSws5OjRo9y9e5eSkhJKSkrk\n8K3H9FaO1hpvq5UUeMaMGTQ3N3P37l1mzZrFzZs3qbHSMd/j4ujtbfZJi94WWVLg+vp6QkJCKCgo\n4Pr167z22mssWLBADt8sgiEB9SPenEL39g0AkgLX1tYKM1S9MZbtKVrRtELpDmW07+kmDhx9fIR/\nunaPI3RviwsmCNzR0UFOTg7Ozs7CTJWtYkxk3X/wsNhaoaUE1/1xWIO4YMJER1JSEvBA6NTUVIs7\n9ChIbaKiTT7opgXFJiVaa2vNkg60FnHBBIEFQwtngnrKo4isxRSxtXa6gmsnRoRI9/ZmkI+PVWeT\nJFW7d+8eV65cYcSIEQwbNkwOnx4ZQxGjK7p+GtHgWuDaWknB9c/TFVdbPVtT9IIJAqenpzNw4EDm\nzp3Lrl27hJ3frR3tF60V2mg+V5vvNbBQ3FTBrVVcMEFgBwcHPDw8mDx5MgcPHhS11U/4G1rxLzf6\nQutjSHj9Kt0Y1iysFkmBJ02axL59+ygvL2fGjBmitvpb+uu/1kXuHd+lRNCv0s1Vbm8jKbCnp6ew\nkFtqnyz9hL/+a10iIyOJjIwEzP+0r55gTCj9yLd2QfWRFPj69esANDQ08Mcff4jeNquf8Ne+Hjly\npPk8lhlbE1QfSYF11yU1NzeL2tq39Lc+JAXWPpRSoVAQEhIih092zIikwO+//z7e3t7k5OTw9NNP\ny+GTHTMi2X397rvvqKioYODAgfz0009y+GTHjEhGcHt7Oz/++CPLly/n7NmzFnFCpVLJ1pMuLy/v\ntec0mXJtlUpl1msqNBJPabx+/Tq3b9/Gy8uL5uZmm6+mIyIiyMnJeWKuLVlFFxUV4eLiQmpqKocP\nH5bDJztmpM8m/I2hnVx5Uq79RCX84ckTWLQNrq2tfWjR2FNPPWVxp+yYD9Fe9MmTJ7lx40a396zx\noRymIvdyVn16Y+9s0Sp62LBhNDY2MnHiRCZOnCgsC7VVxJazWhqxvbMtiehP+ObNmzQ0NFBTU4NG\no2HevHly+WURxJazWpri4mJcXV2FvbMttb2yPpLj4L5EfHw8fn5+ODg4EBUVJbvIAJs3b5Z1e+Un\nSuAnEfvTR/s4doH7OHaB+zh2gfs4Zhd427Ztwn1cYvzzzz8UFhb26BqXL1/m+PHjPTq3p6hUKjIy\nMoRrZ2VlsXXrVrOV35O+7o4dOyRtLDaVs337durq6ujo6GDjxo18/vnnuLu7k5eXx6+//kp2djYr\nV65k7dq1+Pr6cvnyZQICAlCpVGzcuJH8/HyuXbtGTU0N6enprFixgpSUFFJTU4mKiqKurs7ouamp\nqWzevJmMjAwWLlxIfHw806ZN49KlSwQGBtLY2Mhnn30m+PrKK6/w4osvUlFRwahRo+jq6mLFihVs\n2rSJfv364e/vj7OzM6dPn2b8+PHU1taSl5fHrFmzhDKOHz/O4cOHUavVpKSkkJ6eLpxbU1PD66+/\nTldXF/n5+QQGBlJaWoparWbNmjV88sknjB49mvDwcLKzs7l//z4ff/wxTU1N/Pzzz3R2duLl5cWs\nWbPYv38/7e3tREZGEhgYyG+//cbUqVON6mCRKrqtrY0bN26QmJiIn58fZWVleHl5ER0dTUBAAADV\n1dXCVOHbb7/N6NGjmT9/Ps899xx///0348aNIywsjObmZm7fvk10dDRvvvkmMTEx3W7DNXSuPu7u\n7kRHR6NUKlm1ahUdHR3CI+sB/Pz8WL58OUqlkri4OOrq6jh58iSTJk1i48aNnDt3jmnTpjFlyhSc\nnJxQKpVMnTqVF154QSgjNzeX9PR0li1bxpEjR7qdGx4eztGjRzly5Ajz5s1j3759DBkyBAcHB65c\nucKgQYNISEjA09OT8PBwfHx8uHDhAllZWSQlJQk3Pn7//fcMHToUT09Pzp8/z4QJEzh16pSoFhYT\nWHsze1NTExqNRnhii3aKUPeR8dovzcHBAYDOzk62b9+Ou7s7gYGBaDQa7t+/j6Oj40MZLUPnan8A\nDQ0Ngg2Ao6Mj8KA61BVY/3hnZyft7e3CD9CUfbK1NtqNW3XPHTt2LJWVlahUKoKCgnByciImJobw\n8HBGjBghPF19z549qNVqxowZI5SrUCiER/y2tbXx3nvvERUVRVBQEO7u7tRJ7BRkkSrazc0NNzc3\nUlJSUCqVTJgwgYMHD5KWlkZ1dTUAAwcOFC3Dw8ODkydP0tjYyIULFzh06BB79+4lPj6ed955R/Tc\n4cOH8/XXX/PXX3/1+DNMnz6dpKQkSktLCQsLw8PDg9LSUsaPH2/Qfs6cOaxfv57W1laSk5NJTk4W\nzgXw9vYWZs5mzJjBunXrcHZ2JiEhQSjD19eX0tJSlEolarWal19+mQ0bNtDa2oq/vz+LFi0iISGB\nQYMGERsby927d6UfAayRgebmZk1ZWZlGo9FooqOjNRqNRpOWlqZpa2uT4/I2y5kzZzRqtVrz+++/\na3bv3v3Q8aKiIs3p06dFy5BlmOTo6Eh2djYpKSmEhoYC8MYbb1BUVCTH5W2WlpYWNmzYwL59+wwm\neiorK3lWYuWFfS66j2Of6Ojj2AXu49gF7uPYBe7j2AXu49gF7uP8D/ELOVfgu7OGAAAAAElFTkSu\nQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b321eaf4828>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_max_cov\"], mean_activ[\"tile_tissue_sp\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(maximum motif coverage)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_max_cov\"], mean_activ[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)\n",
    "fig.savefig(\"Fig_3C_5.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. write files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>log_num_motifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>log_bp_cov</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>log_max_cov</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>10</td>\n",
       "      <td>2.397895</td>\n",
       "      <td>45</td>\n",
       "      <td>3.828641</td>\n",
       "      <td>6</td>\n",
       "      <td>1.945910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>53</td>\n",
       "      <td>3.988984</td>\n",
       "      <td>101</td>\n",
       "      <td>4.624973</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "      <td>84</td>\n",
       "      <td>4.442651</td>\n",
       "      <td>12</td>\n",
       "      <td>2.564949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>13</td>\n",
       "      <td>2.639057</td>\n",
       "      <td>54</td>\n",
       "      <td>4.007333</td>\n",
       "      <td>9</td>\n",
       "      <td>2.302585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>16</td>\n",
       "      <td>2.833213</td>\n",
       "      <td>99</td>\n",
       "      <td>4.605170</td>\n",
       "      <td>5</td>\n",
       "      <td>1.791759</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  tile_mean_expr  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       -1.974712   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...       -0.700792   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       -0.886722   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       -1.445209   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       -1.062973   \n",
       "\n",
       "   tile_tissue_sp  avg_tf_tissue_sp  log_avg_tf_tissue_sp  numMotifs  \\\n",
       "0        0.386721          0.249383              0.222650         10   \n",
       "1        0.165019          0.416909              0.348478         53   \n",
       "2        0.261225          0.429598              0.357394         24   \n",
       "3        0.265622          0.435871              0.361772         13   \n",
       "4        0.117438          0.462181              0.379929         16   \n",
       "\n",
       "   log_num_motifs  numBPcovered  log_bp_cov  maxCov  log_max_cov  \n",
       "0        2.397895            45    3.828641       6     1.945910  \n",
       "1        3.988984           101    4.624973      24     3.218876  \n",
       "2        3.218876            84    4.442651      12     2.564949  \n",
       "3        2.639057            54    4.007333       9     2.302585  \n",
       "4        2.833213            99    4.605170       5     1.791759  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final = mean_activ[[\"unique_id\", \"tile_mean_expr\", \"tile_tissue_sp\", \"avg_tf_tissue_sp\", \"log_avg_tf_tissue_sp\",\n",
    "                    \"numMotifs\", \"log_num_motifs\", \"numBPcovered\", \"log_bp_cov\", \"maxCov\", \"log_max_cov\"]]\n",
    "final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>MPRA_mean_activ</th>\n",
       "      <th>MPRA_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>num_motifs</th>\n",
       "      <th>log_num_motifs</th>\n",
       "      <th>num_bp_covered</th>\n",
       "      <th>log_num_bp_covered</th>\n",
       "      <th>max_coverage</th>\n",
       "      <th>log_max_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>10</td>\n",
       "      <td>2.397895</td>\n",
       "      <td>45</td>\n",
       "      <td>3.828641</td>\n",
       "      <td>6</td>\n",
       "      <td>1.945910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>53</td>\n",
       "      <td>3.988984</td>\n",
       "      <td>101</td>\n",
       "      <td>4.624973</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "      <td>84</td>\n",
       "      <td>4.442651</td>\n",
       "      <td>12</td>\n",
       "      <td>2.564949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>13</td>\n",
       "      <td>2.639057</td>\n",
       "      <td>54</td>\n",
       "      <td>4.007333</td>\n",
       "      <td>9</td>\n",
       "      <td>2.302585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>16</td>\n",
       "      <td>2.833213</td>\n",
       "      <td>99</td>\n",
       "      <td>4.605170</td>\n",
       "      <td>5</td>\n",
       "      <td>1.791759</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  MPRA_mean_activ  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...        -1.974712   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...        -0.700792   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...        -0.886722   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...        -1.445209   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...        -1.062973   \n",
       "\n",
       "   MPRA_tissue_sp  avg_tf_tissue_sp  log_avg_tf_tissue_sp  num_motifs  \\\n",
       "0        0.386721          0.249383              0.222650          10   \n",
       "1        0.165019          0.416909              0.348478          53   \n",
       "2        0.261225          0.429598              0.357394          24   \n",
       "3        0.265622          0.435871              0.361772          13   \n",
       "4        0.117438          0.462181              0.379929          16   \n",
       "\n",
       "   log_num_motifs  num_bp_covered  log_num_bp_covered  max_coverage  \\\n",
       "0        2.397895              45            3.828641             6   \n",
       "1        3.988984             101            4.624973            24   \n",
       "2        3.218876              84            4.442651            12   \n",
       "3        2.639057              54            4.007333             9   \n",
       "4        2.833213              99            4.605170             5   \n",
       "\n",
       "   log_max_coverage  \n",
       "0          1.945910  \n",
       "1          3.218876  \n",
       "2          2.564949  \n",
       "3          2.302585  \n",
       "4          1.791759  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final.columns = [\"unique_id\", \"MPRA_mean_activ\", \"MPRA_tissue_sp\", \"avg_tf_tissue_sp\", \"log_avg_tf_tissue_sp\",\n",
    "                 \"num_motifs\", \"log_num_motifs\", \"num_bp_covered\", \"log_num_bp_covered\", \"max_coverage\",\n",
    "                 \"log_max_coverage\"]\n",
    "final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_dir = \"../../data/04__coverage\"\n",
    "!mkdir -p $out_dir\n",
    "final.to_csv(\"%s/motif_coverage.txt\" % out_dir, sep=\"\\t\", index=False)\n",
    "tf_expr.to_csv(\"%s/tf_tissue_sp.txt\" % out_dir, sep=\"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
