{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 13__motif_chip_tf_ts_redo\n",
    "# testing whether gene tissue-sp correlates with chip/motif tissue-sp\n",
    "\n",
    "after talking to lucas: find tissue-sp for each TF, find all genes that have that motif/chip peak, calc. average tissue-sp. of those genes, correlate that with the tissue-sp. (and subset by class, if needed)\n",
    "\n",
    "run outside of notebook:\n",
    "    ````bedtools intersect -wo -a ../../data/00__index/0__all_tss/TSS.start.1perGencGene.geneIdOnly.3kb.bed -b ../../misc/07__chip/ALL_CELL_LINES__all_ChIP_peaks.bed````\n",
    "also for enhancers\n",
    "\n",
    "took fimo mappings from marta's directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from os import walk\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from plotting_utils import *\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "chip_f = \"chip_all.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fimo_f = \"fimo_all_biotypes.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_ts_f = \"hg19.cage_peak_phase1and2combined_counts.osc.tissue_specificity.txt\"\n",
    "enh_ts_f = \"human_permissive_enhancers_phase_1_and_2_expression_count_matrix.tissue_specificity.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "annot_f = \"../../misc/00__tss_properties/TSS_FantomCat_all.TSSperENSG.txt\"\n",
    "fimo_map_f = \"../../misc/04__jaspar_id_map/2018_03_09_gencode_jaspar_curated.txt\"\n",
    "chip_map_f = \"ensembl_92_gene_id_to_name.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tf_ts_f = \"gtex_tissue_specificity_tau.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>shuffled</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2533673</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>25</td>\n",
       "      <td>+</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>gtctgtggttt</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2533674</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr2:66610364..66610365,+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>23</td>\n",
       "      <td>-</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>GTCTGTGGTTT</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2533675</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr6:29424953..29424961,+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47</td>\n",
       "      <td>57</td>\n",
       "      <td>-</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>GTCTGTGGTTT</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2533676</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p1@ENST00000412759</td>\n",
       "      <td>chr1:120838328..120838358,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>169</td>\n",
       "      <td>179</td>\n",
       "      <td>-</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>GTCTGTGGTTT</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>div_lnc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2533677</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p1@TXNDC11</td>\n",
       "      <td>chr16:11836595..11836726,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>235</td>\n",
       "      <td>245</td>\n",
       "      <td>-</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>GTCTGTGGTTT</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>div_pc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        #pattern name   short_description                       tss_id  \\\n",
       "2533673         RUNX1            p1@SYVN1   chr11:64901978..64902009,-   \n",
       "2533674         RUNX1                 NaN    chr2:66610364..66610365,+   \n",
       "2533675         RUNX1                 NaN    chr6:29424953..29424961,+   \n",
       "2533676         RUNX1  p1@ENST00000412759  chr1:120838328..120838358,-   \n",
       "2533677         RUNX1          p1@TXNDC11   chr16:11836595..11836726,-   \n",
       "\n",
       "        shuffled  start  stop strand    score       p-value  q-value  \\\n",
       "2533673      NaN     15    25      +  15.6034  2.490000e-07    0.338   \n",
       "2533674      NaN     13    23      -  15.6034  2.490000e-07    0.338   \n",
       "2533675      NaN     47    57      -  15.6034  2.490000e-07    0.338   \n",
       "2533676      NaN    169   179      -  15.6034  2.490000e-07    0.338   \n",
       "2533677      NaN    235   245      -  15.6034  2.490000e-07    0.338   \n",
       "\n",
       "        matched sequence motif_id         biotype  \n",
       "2533673      gtctgtggttt    RUNX1  protein_coding  \n",
       "2533674      GTCTGTGGTTT    RUNX1      intergenic  \n",
       "2533675      GTCTGTGGTTT    RUNX1  protein_coding  \n",
       "2533676      GTCTGTGGTTT    RUNX1         div_lnc  \n",
       "2533677      GTCTGTGGTTT    RUNX1          div_pc  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = pd.read_table(fimo_f, sep=\"\\t\")\n",
    "fimo = fimo[fimo[\"shuffled\"] != \"shuffled\"]\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>chip_id</th>\n",
       "      <th>chip_chromosome</th>\n",
       "      <th>chip_start</th>\n",
       "      <th>chip_end</th>\n",
       "      <th>chip_score</th>\n",
       "      <th>biotype</th>\n",
       "      <th>pos</th>\n",
       "      <th>pos_rel</th>\n",
       "      <th>motif_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003524</td>\n",
       "      <td>10003525</td>\n",
       "      <td>24.543343</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003524.5</td>\n",
       "      <td>-139.5</td>\n",
       "      <td>ARID3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003413</td>\n",
       "      <td>10003414</td>\n",
       "      <td>42.461327</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003413.5</td>\n",
       "      <td>-28.5</td>\n",
       "      <td>ATF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003497</td>\n",
       "      <td>10003498</td>\n",
       "      <td>69.683125</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003497.5</td>\n",
       "      <td>-112.5</td>\n",
       "      <td>ATF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003457</td>\n",
       "      <td>10003458</td>\n",
       "      <td>27.073183</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003457.5</td>\n",
       "      <td>-72.5</td>\n",
       "      <td>ATF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003424</td>\n",
       "      <td>10003425</td>\n",
       "      <td>55.235219</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003424.5</td>\n",
       "      <td>-39.5</td>\n",
       "      <td>BCL3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end short_description                     tss_id  \\\n",
       "0       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "1       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "2       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "3       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "4       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "\n",
       "   score strand chip_id chip_chromosome  chip_start  chip_end  chip_score  \\\n",
       "0      0      -  ARID3A            chr1    10003524  10003525   24.543343   \n",
       "1      0      -    ATF2            chr1    10003413  10003414   42.461327   \n",
       "2      0      -    ATF2            chr1    10003497  10003498   69.683125   \n",
       "3      0      -    ATF3            chr1    10003457  10003458   27.073183   \n",
       "4      0      -    BCL3            chr1    10003424  10003425   55.235219   \n",
       "\n",
       "  biotype         pos  pos_rel motif_id  \n",
       "0  div_pc  10003524.5   -139.5   ARID3A  \n",
       "1  div_pc  10003413.5    -28.5     ATF2  \n",
       "2  div_pc  10003497.5   -112.5     ATF2  \n",
       "3  div_pc  10003457.5    -72.5     ATF3  \n",
       "4  div_pc  10003424.5    -39.5     BCL3  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip = pd.read_table(chip_f, sep=\"\\t\")\n",
    "chip.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tss_id</th>\n",
       "      <th>tissue_spec</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>01STAT:MAPPED</td>\n",
       "      <td>0.087543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100013403..100013414,-</td>\n",
       "      <td>0.980206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100027943..100027958,-</td>\n",
       "      <td>0.782627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100076685..100076699,+</td>\n",
       "      <td>0.978467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100150910..100150935,-</td>\n",
       "      <td>0.970738</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         tss_id  tissue_spec\n",
       "0                 01STAT:MAPPED     0.087543\n",
       "1  chr10:100013403..100013414,-     0.980206\n",
       "2  chr10:100027943..100027958,-     0.782627\n",
       "3  chr10:100076685..100076699,+     0.978467\n",
       "4  chr10:100150910..100150935,-     0.970738"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tss_ts = pd.read_table(tss_ts_f, sep=\"\\t\")\n",
    "tss_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tss_id</th>\n",
       "      <th>tissue_spec</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:839741-840250</td>\n",
       "      <td>0.974827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:840753-841210</td>\n",
       "      <td>0.967327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:845485-845678</td>\n",
       "      <td>0.966489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:855764-856157</td>\n",
       "      <td>0.982428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:856539-856757</td>\n",
       "      <td>0.976588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               tss_id  tissue_spec\n",
       "0  chr1:839741-840250     0.974827\n",
       "1  chr1:840753-841210     0.967327\n",
       "2  chr1:845485-845678     0.966489\n",
       "3  chr1:855764-856157     0.982428\n",
       "4  chr1:856539-856757     0.976588"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_ts = pd.read_table(enh_ts_f, sep=\"\\t\")\n",
    "enh_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "all_ts = tss_ts.append(enh_ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>PromType2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id       PromType2\n",
       "0   ENSG00000000003.10  protein_coding\n",
       "8    ENSG00000000005.5  protein_coding\n",
       "13   ENSG00000000419.8  protein_coding\n",
       "17   ENSG00000000457.9  protein_coding\n",
       "24  ENSG00000000460.12  protein_coding"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annot = pd.read_table(annot_f, sep=\"\\t\")\n",
    "promtype2 = annot[[\"gene_id\", \"PromType2\"]].drop_duplicates()\n",
    "promtype2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>Mus musculus; human ortholog selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name          Gene ID                                comment\n",
       "0  MA0002.2      RUNX1  ENSG00000159216                                    NaN\n",
       "1  MA0003.3     TFAP2A  ENSG00000137203                                    NaN\n",
       "2  MA0007.3         Ar  ENSG00000169083  Mus musculus; human ortholog selected\n",
       "3  MA0009.2          T  ENSG00000164458                                    NaN\n",
       "4  MA0014.2       PAX5  ENSG00000196092                                    NaN"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_map = pd.read_table(fimo_map_f, sep=\"\\t\")\n",
    "fimo_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000276626</td>\n",
       "      <td>RF00100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000201317</td>\n",
       "      <td>RNU4-59P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000200823</td>\n",
       "      <td>SNORD114-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000221598</td>\n",
       "      <td>MIR1249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000199595</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene stable ID   Gene name\n",
       "0  ENSG00000276626     RF00100\n",
       "1  ENSG00000201317    RNU4-59P\n",
       "2  ENSG00000200823  SNORD114-2\n",
       "3  ENSG00000221598     MIR1249\n",
       "4  ENSG00000199595     RF00019"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_map = pd.read_table(chip_map_f, sep=\"\\t\")\n",
    "chip_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>0.589677</td>\n",
       "      <td>0.986427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>0.203154</td>\n",
       "      <td>0.225353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              GeneID  tissue_spec  tissue_spec_tau\n",
       "0  ENSG00000223972.4     0.589677         0.986427\n",
       "1  ENSG00000227232.4     0.203154         0.225353\n",
       "2  ENSG00000243485.2          NaN              NaN\n",
       "3  ENSG00000237613.2          NaN              NaN\n",
       "4  ENSG00000268020.2          NaN              NaN"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_ts = pd.read_table(tf_ts_f, sep=\"\\t\")\n",
    "tf_ts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. grab gene_ids for motif_names in chip/fimo "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "435"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_tfs = list(fimo[\"motif_id\"].unique())\n",
    "len(fimo_tfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "160"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_tfs = list(chip[\"motif_id\"].unique())\n",
    "len(chip_tfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "manual_aliases = {\"SIN3AK20\": \"SIN3A\", \"KAP1\": \"TRIM28\", \"SREBP1\": \"SREBF1\", \"ZZZ3\": \"AC118549.1\", \n",
    "                  \"RPC155\": \"POLR3A\", \"RDBP\": \"NELFE\", \"FAM48A\": \"SUPT20H\", \"MZF1(VAR.2)\": \"MZF1\",\n",
    "                  \"RORA(VAR.2)\": \"RORA\", \"JUN(VAR.2)\": \"JUN\", \"JUND(VAR.2)\": \"JUND\", \"NKX2-5(VAR.2)\": \"NKX2-5\",\n",
    "                  \"JDP2(VAR.2)\": \"JDP2\", \"NR2F6(VAR.2)\": \"NR2F6\", \"RARA(VAR.2)\": \"RARA\", \"TFAP2A(VAR.2)\": \"TFAP2A\",\n",
    "                  \"TFAP2B(VAR.2)\": \"TFAP2B\", \"TFAP2B(VAR.3)\": \"TFAP2B\", \"TFAP2C(VAR.2)\": \"TFAP2C\", \n",
    "                  \"SREBF2(VAR.2)\": \"SREBF2\", \"SREBF1(VAR.2)\": \"SREBF1\", \"RARB(VAR.2)\": \"RARB\", \n",
    "                  \"RARG(VAR.2)\": \"RARG\", \"TFAP2A(VAR.3)\": \"TFAP2A\", \"TFAP2C(VAR.3)\": \"TFAP2C\", \"MIX-A\": \"MIXL1\"}\n",
    "manual_gene_ids = {\"HSF1\": \"ENSG00000185122\", \"HNF1B\": \"ENSG00000275410\", \"KLF13\": \"ENSG00000169926\",\n",
    "                   \"POU5F1\": \"ENSG00000204531\", \"SMARCB1\": \"ENSG00000099956\", \"RXRB\": \"ENSG00000204231\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_gene_id(motif_names, fimo_map, chip_map, manual_aliases, manual_gene_ids):\n",
    "    gene_id_map = {}\n",
    "    for motif in motif_names:\n",
    "        motif = motif.upper()\n",
    "        \n",
    "        if \"::\" in motif:\n",
    "            # fusion protein, continue\n",
    "            gene_id_map[motif] = np.nan\n",
    "            continue\n",
    "        \n",
    "        if motif in manual_gene_ids:\n",
    "            gene_id_map[motif] = manual_gene_ids[motif]\n",
    "            continue\n",
    "            \n",
    "        try:\n",
    "            fimo_id = fimo_map[fimo_map[\"motif_name\"] == motif][\"Gene ID\"].iloc[0]\n",
    "        except:\n",
    "            fimo_id = \"none\"\n",
    "        try:\n",
    "            chip_id = chip_map[chip_map[\"Gene name\"] == motif][\"Gene stable ID\"].iloc[0]\n",
    "        except:\n",
    "            chip_id = \"none\"\n",
    "        if fimo_id == \"none\" and chip_id == \"none\":\n",
    "            try:\n",
    "                alias = manual_aliases[motif]\n",
    "            except:\n",
    "                print(\"%s: no id, no alias\" % motif)\n",
    "                gene_id_map[motif] = np.nan\n",
    "                continue\n",
    "            try:\n",
    "                real_id = chip_map[chip_map[\"Gene name\"] == alias][\"Gene stable ID\"].iloc[0]\n",
    "            except:\n",
    "                print(\"%s: no id found for alias %s\" % (motif, alias))\n",
    "                gene_id_map[motif] = np.nan\n",
    "        elif fimo_id != \"none\" and chip_id == \"none\":\n",
    "            #print(\"found fimo id\")\n",
    "            gene_id_map[motif] = fimo_id\n",
    "        elif fimo_id == \"none\" and chip_id != \"none\":\n",
    "            #print(\"found chip id\")\n",
    "            gene_id_map[motif] = chip_id\n",
    "        elif fimo_id != \"none\" and chip_id != \"none\":\n",
    "            if fimo_id == chip_id:\n",
    "                #print(\"found fimo/chip id that agrees\")\n",
    "                gene_id_map[motif] = fimo_id\n",
    "            else:\n",
    "                #print(\"%s: found fimo/chip id that disagree: %s, %s\\n\" % (motif, fimo_id, chip_id))\n",
    "                real_id = manual_gene_ids[motif]\n",
    "                gene_id_map[motif] = real_id\n",
    "    return gene_id_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRP20: no id, no alias\n"
     ]
    }
   ],
   "source": [
    "chip_id_map = get_gene_id(chip_tfs, fimo_map, chip_map, manual_aliases, manual_gene_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RHOX11: no id, no alias\n"
     ]
    }
   ],
   "source": [
    "fimo_id_map = get_gene_id(fimo_tfs, fimo_map, chip_map, manual_aliases, manual_gene_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID3A</td>\n",
       "      <td>ENSG00000116017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATF2</td>\n",
       "      <td>ENSG00000115966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATF3</td>\n",
       "      <td>ENSG00000162772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BCL3</td>\n",
       "      <td>ENSG00000069399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BCLAF1</td>\n",
       "      <td>ENSG00000029363</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  motif_name          gene_id\n",
       "0     ARID3A  ENSG00000116017\n",
       "1       ATF2  ENSG00000115966\n",
       "2       ATF3  ENSG00000162772\n",
       "3       BCL3  ENSG00000069399\n",
       "4     BCLAF1  ENSG00000029363"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_id_map = pd.DataFrame.from_dict(chip_id_map, orient=\"index\").reset_index()\n",
    "chip_id_map.columns = [\"motif_name\", \"gene_id\"]\n",
    "chip_id_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AR</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  motif_name          gene_id\n",
       "0      RUNX1  ENSG00000159216\n",
       "1     TFAP2A  ENSG00000137203\n",
       "2         AR  ENSG00000169083\n",
       "3          T  ENSG00000164458\n",
       "4       PAX5  ENSG00000196092"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_id_map = pd.DataFrame.from_dict(fimo_id_map, orient=\"index\").reset_index()\n",
    "fimo_id_map.columns = [\"motif_name\", \"gene_id\"]\n",
    "fimo_id_map.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. merge fimo/chip with tss tissue-sp values\n",
    "why are some TSSs missing?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4455254"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fimo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4022690\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>shuffled</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>biotype</th>\n",
       "      <th>tissue_spec</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>790880</th>\n",
       "      <td>KLF5</td>\n",
       "      <td>p2@CDC45</td>\n",
       "      <td>chr22:19466980..19467003,+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2774</td>\n",
       "      <td>2783</td>\n",
       "      <td>-</td>\n",
       "      <td>16.2857</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.015</td>\n",
       "      <td>GCCCCGCCCT</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.963875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2972599</th>\n",
       "      <td>TFEC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr8:2418829-2418999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2984</td>\n",
       "      <td>2993</td>\n",
       "      <td>+</td>\n",
       "      <td>13.9146</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1.000</td>\n",
       "      <td>agcacgtgac</td>\n",
       "      <td>TFEC</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>0.974603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3648603</th>\n",
       "      <td>HOXA10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7:86609428-86609776</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1717</td>\n",
       "      <td>1727</td>\n",
       "      <td>+</td>\n",
       "      <td>13.5660</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.505</td>\n",
       "      <td>ggtaataaaac</td>\n",
       "      <td>HOXA10</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>0.993728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85864</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p7@PLD1</td>\n",
       "      <td>chr3:171529110..171529141,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>612</td>\n",
       "      <td>622</td>\n",
       "      <td>+</td>\n",
       "      <td>13.6724</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.401</td>\n",
       "      <td>TTCTGTGGTCT</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.975843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2099405</th>\n",
       "      <td>SREBF2(var.2)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr20:40321359-40321668</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1976</td>\n",
       "      <td>1985</td>\n",
       "      <td>-</td>\n",
       "      <td>16.8545</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.000</td>\n",
       "      <td>GTCACGTGAC</td>\n",
       "      <td>SREBF2(var.2)</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>0.923712</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         #pattern name short_description                       tss_id  \\\n",
       "790880            KLF5          p2@CDC45   chr22:19466980..19467003,+   \n",
       "2972599           TFEC               NaN         chr8:2418829-2418999   \n",
       "3648603         HOXA10               NaN       chr7:86609428-86609776   \n",
       "85864            RUNX1           p7@PLD1  chr3:171529110..171529141,-   \n",
       "2099405  SREBF2(var.2)               NaN      chr20:40321359-40321668   \n",
       "\n",
       "        shuffled  start  stop strand    score   p-value  q-value  \\\n",
       "790880       NaN   2774  2783      -  16.2857  0.000001    0.015   \n",
       "2972599      NaN   2984  2993      +  13.9146  0.000005    1.000   \n",
       "3648603      NaN   1717  1727      +  13.5660  0.000001    0.505   \n",
       "85864        NaN    612   622      +  13.6724  0.000007    0.401   \n",
       "2099405      NaN   1976  1985      -  16.8545  0.000003    1.000   \n",
       "\n",
       "        matched sequence       motif_id         biotype  tissue_spec  \n",
       "790880        GCCCCGCCCT           KLF5  protein_coding     0.963875  \n",
       "2972599       agcacgtgac           TFEC        enhancer     0.974603  \n",
       "3648603      ggtaataaaac         HOXA10        enhancer     0.993728  \n",
       "85864        TTCTGTGGTCT          RUNX1  protein_coding     0.975843  \n",
       "2099405       GTCACGTGAC  SREBF2(var.2)        enhancer     0.923712  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_ts = fimo.merge(all_ts, on=\"tss_id\")\n",
    "print(len(fimo_ts))\n",
    "fimo_ts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7619"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_tss_ids_fimo = fimo[~fimo[\"tss_id\"].isin(fimo_ts[\"tss_id\"])][\"tss_id\"].unique()\n",
    "len(missing_tss_ids_fimo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5940131"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(chip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5293069\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>chip_id</th>\n",
       "      <th>chip_chromosome</th>\n",
       "      <th>chip_start</th>\n",
       "      <th>chip_end</th>\n",
       "      <th>chip_score</th>\n",
       "      <th>biotype</th>\n",
       "      <th>pos</th>\n",
       "      <th>pos_rel</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>tissue_spec</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3099756</th>\n",
       "      <td>chr10</td>\n",
       "      <td>3850734</td>\n",
       "      <td>3853734</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr10:3852273-3852886</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>ZBTB7A</td>\n",
       "      <td>chr10</td>\n",
       "      <td>3851854</td>\n",
       "      <td>3851855</td>\n",
       "      <td>22.160684</td>\n",
       "      <td>enhancer</td>\n",
       "      <td>3851854.5</td>\n",
       "      <td>-879.5</td>\n",
       "      <td>ZBTB7A</td>\n",
       "      <td>0.955008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601120</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120970158</td>\n",
       "      <td>120973158</td>\n",
       "      <td>p1@RNF10</td>\n",
       "      <td>chr12:120972147..120972167,+</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>POLR2A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120972187</td>\n",
       "      <td>120972188</td>\n",
       "      <td>97.088955</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>120972187.5</td>\n",
       "      <td>29.5</td>\n",
       "      <td>POLR2A</td>\n",
       "      <td>0.234204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820742</th>\n",
       "      <td>chr14</td>\n",
       "      <td>64193773</td>\n",
       "      <td>64196773</td>\n",
       "      <td>p1@SGPP1</td>\n",
       "      <td>chr14:64194745..64194799,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>chr14</td>\n",
       "      <td>64194964</td>\n",
       "      <td>64194965</td>\n",
       "      <td>80.010361</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>64194964.5</td>\n",
       "      <td>-191.5</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>0.420277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800343</th>\n",
       "      <td>chr14</td>\n",
       "      <td>36788807</td>\n",
       "      <td>36791807</td>\n",
       "      <td>p1@MBIP</td>\n",
       "      <td>chr14:36789789..36789860,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>POLR2A</td>\n",
       "      <td>chr14</td>\n",
       "      <td>36789791</td>\n",
       "      <td>36789792</td>\n",
       "      <td>90.643676</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>36789791.5</td>\n",
       "      <td>15.5</td>\n",
       "      <td>POLR2A</td>\n",
       "      <td>0.483707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2313692</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31512627</td>\n",
       "      <td>31515627</td>\n",
       "      <td>p4@NFKBIL1</td>\n",
       "      <td>chr6:31514622..31514641,+</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>RELA</td>\n",
       "      <td>chr6</td>\n",
       "      <td>31515256</td>\n",
       "      <td>31515257</td>\n",
       "      <td>46.954125</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>31515256.5</td>\n",
       "      <td>629.5</td>\n",
       "      <td>RELA</td>\n",
       "      <td>0.789795</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chromosome      start        end short_description  \\\n",
       "3099756      chr10    3850734    3853734               NaN   \n",
       "601120       chr12  120970158  120973158          p1@RNF10   \n",
       "820742       chr14   64193773   64196773          p1@SGPP1   \n",
       "800343       chr14   36788807   36791807           p1@MBIP   \n",
       "2313692       chr6   31512627   31515627        p4@NFKBIL1   \n",
       "\n",
       "                               tss_id  score strand chip_id chip_chromosome  \\\n",
       "3099756         chr10:3852273-3852886      0      +  ZBTB7A           chr10   \n",
       "601120   chr12:120972147..120972167,+      0      +  POLR2A           chr12   \n",
       "820742     chr14:64194745..64194799,-      0      -  TCF7L2           chr14   \n",
       "800343     chr14:36789789..36789860,-      0      -  POLR2A           chr14   \n",
       "2313692     chr6:31514622..31514641,+      0      +    RELA            chr6   \n",
       "\n",
       "         chip_start   chip_end  chip_score         biotype          pos  \\\n",
       "3099756     3851854    3851855   22.160684        enhancer    3851854.5   \n",
       "601120    120972187  120972188   97.088955  protein_coding  120972187.5   \n",
       "820742     64194964   64194965   80.010361  protein_coding   64194964.5   \n",
       "800343     36789791   36789792   90.643676  protein_coding   36789791.5   \n",
       "2313692    31515256   31515257   46.954125  protein_coding   31515256.5   \n",
       "\n",
       "         pos_rel motif_id  tissue_spec  \n",
       "3099756   -879.5   ZBTB7A     0.955008  \n",
       "601120      29.5   POLR2A     0.234204  \n",
       "820742    -191.5   TCF7L2     0.420277  \n",
       "800343      15.5   POLR2A     0.483707  \n",
       "2313692    629.5     RELA     0.789795  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_ts = chip.merge(all_ts, on=\"tss_id\")\n",
    "print(len(chip_ts))\n",
    "chip_ts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6469"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_tss_ids_chip = chip[~chip[\"tss_id\"].isin(chip_ts[\"tss_id\"])][\"tss_id\"].unique()\n",
    "len(missing_tss_ids_chip)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. find tissue-sp per TF\n",
    "#### use gtex for now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>0.589677</td>\n",
       "      <td>0.986427</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>0.203154</td>\n",
       "      <td>0.225353</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              GeneID  tissue_spec  tissue_spec_tau          gene_id\n",
       "0  ENSG00000223972.4     0.589677         0.986427  ENSG00000223972\n",
       "1  ENSG00000227232.4     0.203154         0.225353  ENSG00000227232\n",
       "2  ENSG00000243485.2          NaN              NaN  ENSG00000243485\n",
       "3  ENSG00000237613.2          NaN              NaN  ENSG00000237613\n",
       "4  ENSG00000268020.2          NaN              NaN  ENSG00000268020"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_ts[\"gene_id\"] = tf_ts[\"GeneID\"].str.split(\".\", expand=True)[0]\n",
    "tf_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>BDP1</td>\n",
       "      <td>ENSG00000145734</td>\n",
       "      <td>ENSG00000145734.14</td>\n",
       "      <td>0.206479</td>\n",
       "      <td>0.308354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>ENSG00000106462</td>\n",
       "      <td>ENSG00000106462.6</td>\n",
       "      <td>0.246927</td>\n",
       "      <td>0.729700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>SP1</td>\n",
       "      <td>ENSG00000185591</td>\n",
       "      <td>ENSG00000185591.5</td>\n",
       "      <td>0.203233</td>\n",
       "      <td>0.227516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CREB1</td>\n",
       "      <td>ENSG00000118260</td>\n",
       "      <td>ENSG00000118260.10</td>\n",
       "      <td>0.206613</td>\n",
       "      <td>0.311374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>BRF2</td>\n",
       "      <td>ENSG00000104221</td>\n",
       "      <td>ENSG00000104221.8</td>\n",
       "      <td>0.206340</td>\n",
       "      <td>0.305182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif_name          gene_id              GeneID  tissue_spec  \\\n",
       "148       BDP1  ENSG00000145734  ENSG00000145734.14     0.206479   \n",
       "126       EZH2  ENSG00000106462   ENSG00000106462.6     0.246927   \n",
       "77         SP1  ENSG00000185591   ENSG00000185591.5     0.203233   \n",
       "12       CREB1  ENSG00000118260  ENSG00000118260.10     0.206613   \n",
       "150       BRF2  ENSG00000104221   ENSG00000104221.8     0.206340   \n",
       "\n",
       "     tissue_spec_tau  \n",
       "148         0.308354  \n",
       "126         0.729700  \n",
       "77          0.227516  \n",
       "12          0.311374  \n",
       "150         0.305182  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_id_map_ts = chip_id_map.merge(tf_ts, on=\"gene_id\", how=\"left\")\n",
    "chip_id_map_ts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>GRP20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif_name gene_id GeneID  tissue_spec  tissue_spec_tau\n",
       "127      GRP20     NaN    NaN          NaN              NaN"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_id_map_ts[pd.isnull(chip_id_map_ts[\"tissue_spec\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>ALX3</td>\n",
       "      <td>ENSG00000156150</td>\n",
       "      <td>ENSG00000156150.6</td>\n",
       "      <td>0.352883</td>\n",
       "      <td>0.923747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>ENSG00000212993</td>\n",
       "      <td>ENSG00000212993.3</td>\n",
       "      <td>0.240622</td>\n",
       "      <td>0.697340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>TCF21</td>\n",
       "      <td>ENSG00000118526</td>\n",
       "      <td>ENSG00000118526.6</td>\n",
       "      <td>0.252364</td>\n",
       "      <td>0.753331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ZNF143</td>\n",
       "      <td>ENSG00000166478</td>\n",
       "      <td>ENSG00000166478.5</td>\n",
       "      <td>0.203062</td>\n",
       "      <td>0.222797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>HOXD12</td>\n",
       "      <td>ENSG00000170178</td>\n",
       "      <td>ENSG00000170178.5</td>\n",
       "      <td>0.431913</td>\n",
       "      <td>0.959631</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif_name          gene_id             GeneID  tissue_spec  \\\n",
       "191       ALX3  ENSG00000156150  ENSG00000156150.6     0.352883   \n",
       "320    POU5F1B  ENSG00000212993  ENSG00000212993.3     0.240622   \n",
       "347      TCF21  ENSG00000118526  ENSG00000118526.6     0.252364   \n",
       "44      ZNF143  ENSG00000166478  ENSG00000166478.5     0.203062   \n",
       "380     HOXD12  ENSG00000170178  ENSG00000170178.5     0.431913   \n",
       "\n",
       "     tissue_spec_tau  \n",
       "191         0.923747  \n",
       "320         0.697340  \n",
       "347         0.753331  \n",
       "44          0.222797  \n",
       "380         0.959631  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_id_map_ts = fimo_id_map.merge(tf_ts, on=\"gene_id\", how=\"left\")\n",
    "fimo_id_map_ts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DDIT3::CEBPA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAX::MYC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PPARG::RXRA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>RXRA::VDR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>TAL1::TCF3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>HAND1::TCF3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>NR1H2::RXRA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>NFIC::TLX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GATA1::TAL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>POU5F1::SOX2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>EWSR1-FLI1</td>\n",
       "      <td>ENSG00000182944::ENSG00000151702</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>HNF1B</td>\n",
       "      <td>ENSG00000275410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>RARA::RXRA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>ARNT::HIF1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>BATF::JUN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>DUX4</td>\n",
       "      <td>ENSG00000260596</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>NR1H3::RXRA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>MAF::NFE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>SMAD2::SMAD3::SMAD4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>STAT1::STAT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>STAT5A::STAT5B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>BACH1::MAFK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>NEUROG1</td>\n",
       "      <td>ENSG00000181965</td>\n",
       "      <td>ENSG00000181965.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>RHOX11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>ESX1</td>\n",
       "      <td>ENSG00000123576</td>\n",
       "      <td>ENSG00000123576.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>KLF14</td>\n",
       "      <td>ENSG00000266265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>SCRT1</td>\n",
       "      <td>ENSG00000261678</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              motif_name                           gene_id             GeneID  \\\n",
       "6           DDIT3::CEBPA                               NaN                NaN   \n",
       "27              MAX::MYC                               NaN                NaN   \n",
       "30           PPARG::RXRA                               NaN                NaN   \n",
       "36             RXRA::VDR                               NaN                NaN   \n",
       "46            TAL1::TCF3                               NaN                NaN   \n",
       "47           HAND1::TCF3                               NaN                NaN   \n",
       "65           NR1H2::RXRA                               NaN                NaN   \n",
       "68            NFIC::TLX1                               NaN                NaN   \n",
       "77           GATA1::TAL1                               NaN                NaN   \n",
       "79          POU5F1::SOX2                               NaN                NaN   \n",
       "85            EWSR1-FLI1  ENSG00000182944::ENSG00000151702                NaN   \n",
       "87                 HNF1B                   ENSG00000275410                NaN   \n",
       "91            RARA::RXRA                               NaN                NaN   \n",
       "95           ARNT::HIF1A                               NaN                NaN   \n",
       "97             BATF::JUN                               NaN                NaN   \n",
       "103                 DUX4                   ENSG00000260596                NaN   \n",
       "126          NR1H3::RXRA                               NaN                NaN   \n",
       "132            MAF::NFE2                               NaN                NaN   \n",
       "143  SMAD2::SMAD3::SMAD4                               NaN                NaN   \n",
       "147         STAT1::STAT2                               NaN                NaN   \n",
       "149       STAT5A::STAT5B                               NaN                NaN   \n",
       "159          BACH1::MAFK                               NaN                NaN   \n",
       "181              NEUROG1                   ENSG00000181965  ENSG00000181965.4   \n",
       "187               RHOX11                               NaN                NaN   \n",
       "201                 ESX1                   ENSG00000123576  ENSG00000123576.5   \n",
       "271                KLF14                   ENSG00000266265                NaN   \n",
       "274                SCRT1                   ENSG00000261678                NaN   \n",
       "\n",
       "     tissue_spec  tissue_spec_tau  \n",
       "6            NaN              NaN  \n",
       "27           NaN              NaN  \n",
       "30           NaN              NaN  \n",
       "36           NaN              NaN  \n",
       "46           NaN              NaN  \n",
       "47           NaN              NaN  \n",
       "65           NaN              NaN  \n",
       "68           NaN              NaN  \n",
       "77           NaN              NaN  \n",
       "79           NaN              NaN  \n",
       "85           NaN              NaN  \n",
       "87           NaN              NaN  \n",
       "91           NaN              NaN  \n",
       "95           NaN              NaN  \n",
       "97           NaN              NaN  \n",
       "103          NaN              NaN  \n",
       "126          NaN              NaN  \n",
       "132          NaN              NaN  \n",
       "143          NaN              NaN  \n",
       "147          NaN              NaN  \n",
       "149          NaN              NaN  \n",
       "159          NaN              NaN  \n",
       "181          NaN              NaN  \n",
       "187          NaN              NaN  \n",
       "201          NaN              NaN  \n",
       "271          NaN              NaN  \n",
       "274          NaN              NaN  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_id_map_ts[pd.isnull(fimo_id_map_ts[\"tissue_spec\"])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. merge fimo/chip with tf spec values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>chip_id</th>\n",
       "      <th>chip_chromosome</th>\n",
       "      <th>chip_start</th>\n",
       "      <th>chip_end</th>\n",
       "      <th>chip_score</th>\n",
       "      <th>biotype</th>\n",
       "      <th>pos</th>\n",
       "      <th>pos_rel</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>motif_upper</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003524</td>\n",
       "      <td>10003525</td>\n",
       "      <td>24.543343</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003524.5</td>\n",
       "      <td>-139.5</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ARID3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003413</td>\n",
       "      <td>10003414</td>\n",
       "      <td>42.461327</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003413.5</td>\n",
       "      <td>-28.5</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003497</td>\n",
       "      <td>10003498</td>\n",
       "      <td>69.683125</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003497.5</td>\n",
       "      <td>-112.5</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003457</td>\n",
       "      <td>10003458</td>\n",
       "      <td>27.073183</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003457.5</td>\n",
       "      <td>-72.5</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003424</td>\n",
       "      <td>10003425</td>\n",
       "      <td>55.235219</td>\n",
       "      <td>div_pc</td>\n",
       "      <td>10003424.5</td>\n",
       "      <td>-39.5</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>BCL3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end short_description                     tss_id  \\\n",
       "0       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "1       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "2       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "3       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "4       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "\n",
       "   score strand chip_id chip_chromosome  chip_start  chip_end  chip_score  \\\n",
       "0      0      -  ARID3A            chr1    10003524  10003525   24.543343   \n",
       "1      0      -    ATF2            chr1    10003413  10003414   42.461327   \n",
       "2      0      -    ATF2            chr1    10003497  10003498   69.683125   \n",
       "3      0      -    ATF3            chr1    10003457  10003458   27.073183   \n",
       "4      0      -    BCL3            chr1    10003424  10003425   55.235219   \n",
       "\n",
       "  biotype         pos  pos_rel motif_id  tissue_spec motif_upper  \n",
       "0  div_pc  10003524.5   -139.5   ARID3A       0.3734      ARID3A  \n",
       "1  div_pc  10003413.5    -28.5     ATF2       0.3734        ATF2  \n",
       "2  div_pc  10003497.5   -112.5     ATF2       0.3734        ATF2  \n",
       "3  div_pc  10003457.5    -72.5     ATF3       0.3734        ATF3  \n",
       "4  div_pc  10003424.5    -39.5     BCL3       0.3734        BCL3  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_ts[\"motif_upper\"] = chip_ts[\"motif_id\"].str.upper()\n",
    "chip_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>chip_id</th>\n",
       "      <th>chip_chromosome</th>\n",
       "      <th>chip_start</th>\n",
       "      <th>...</th>\n",
       "      <th>pos</th>\n",
       "      <th>pos_rel</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>tissue_spec_x</th>\n",
       "      <th>motif_upper</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec_y</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003524</td>\n",
       "      <td>...</td>\n",
       "      <td>10003524.5</td>\n",
       "      <td>-139.5</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>ARID3A</td>\n",
       "      <td>ENSG00000116017</td>\n",
       "      <td>ENSG00000116017.6</td>\n",
       "      <td>0.243316</td>\n",
       "      <td>0.711899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003413</td>\n",
       "      <td>...</td>\n",
       "      <td>10003413.5</td>\n",
       "      <td>-28.5</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>ENSG00000115966</td>\n",
       "      <td>ENSG00000115966.12</td>\n",
       "      <td>0.204556</td>\n",
       "      <td>0.262339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003497</td>\n",
       "      <td>...</td>\n",
       "      <td>10003497.5</td>\n",
       "      <td>-112.5</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>ENSG00000115966</td>\n",
       "      <td>ENSG00000115966.12</td>\n",
       "      <td>0.204556</td>\n",
       "      <td>0.262339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003457</td>\n",
       "      <td>...</td>\n",
       "      <td>10003457.5</td>\n",
       "      <td>-72.5</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>ATF3</td>\n",
       "      <td>ENSG00000162772</td>\n",
       "      <td>ENSG00000162772.12</td>\n",
       "      <td>0.216412</td>\n",
       "      <td>0.483653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10002385</td>\n",
       "      <td>10005385</td>\n",
       "      <td>p1@LZIC</td>\n",
       "      <td>chr1:10003372..10003465,-</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10003424</td>\n",
       "      <td>...</td>\n",
       "      <td>10003424.5</td>\n",
       "      <td>-39.5</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>0.3734</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>ENSG00000069399</td>\n",
       "      <td>ENSG00000069399.8</td>\n",
       "      <td>0.211082</td>\n",
       "      <td>0.400526</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end short_description                     tss_id  \\\n",
       "0       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "1       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "2       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "3       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "4       chr1  10002385  10005385           p1@LZIC  chr1:10003372..10003465,-   \n",
       "\n",
       "   score strand chip_id chip_chromosome  chip_start       ...        \\\n",
       "0      0      -  ARID3A            chr1    10003524       ...         \n",
       "1      0      -    ATF2            chr1    10003413       ...         \n",
       "2      0      -    ATF2            chr1    10003497       ...         \n",
       "3      0      -    ATF3            chr1    10003457       ...         \n",
       "4      0      -    BCL3            chr1    10003424       ...         \n",
       "\n",
       "          pos  pos_rel motif_id  tissue_spec_x  motif_upper motif_name  \\\n",
       "0  10003524.5   -139.5   ARID3A         0.3734       ARID3A     ARID3A   \n",
       "1  10003413.5    -28.5     ATF2         0.3734         ATF2       ATF2   \n",
       "2  10003497.5   -112.5     ATF2         0.3734         ATF2       ATF2   \n",
       "3  10003457.5    -72.5     ATF3         0.3734         ATF3       ATF3   \n",
       "4  10003424.5    -39.5     BCL3         0.3734         BCL3       BCL3   \n",
       "\n",
       "           gene_id              GeneID tissue_spec_y tissue_spec_tau  \n",
       "0  ENSG00000116017   ENSG00000116017.6      0.243316        0.711899  \n",
       "1  ENSG00000115966  ENSG00000115966.12      0.204556        0.262339  \n",
       "2  ENSG00000115966  ENSG00000115966.12      0.204556        0.262339  \n",
       "3  ENSG00000162772  ENSG00000162772.12      0.216412        0.483653  \n",
       "4  ENSG00000069399   ENSG00000069399.8      0.211082        0.400526  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_ts = chip_ts.merge(chip_id_map_ts, left_on=\"motif_upper\", right_on=\"motif_name\", how=\"left\")\n",
    "chip_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>shuffled</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>biotype</th>\n",
       "      <th>tissue_spec</th>\n",
       "      <th>motif_upper</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>25</td>\n",
       "      <td>+</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>gtctgtggttt</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>RUNX1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1604</td>\n",
       "      <td>1614</td>\n",
       "      <td>+</td>\n",
       "      <td>13.8909</td>\n",
       "      <td>9.460000e-06</td>\n",
       "      <td>0.155</td>\n",
       "      <td>CCCCTCAGGCT</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>TFAP2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>638</td>\n",
       "      <td>656</td>\n",
       "      <td>+</td>\n",
       "      <td>14.5484</td>\n",
       "      <td>5.460000e-06</td>\n",
       "      <td>0.107</td>\n",
       "      <td>ctgggcgacagagcaagac</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>PAX5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>E2F1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>616</td>\n",
       "      <td>627</td>\n",
       "      <td>+</td>\n",
       "      <td>14.1475</td>\n",
       "      <td>8.620000e-06</td>\n",
       "      <td>1.000</td>\n",
       "      <td>gatggcgccaat</td>\n",
       "      <td>E2F1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>E2F1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ELK1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1947</td>\n",
       "      <td>1956</td>\n",
       "      <td>+</td>\n",
       "      <td>16.2128</td>\n",
       "      <td>3.920000e-06</td>\n",
       "      <td>0.605</td>\n",
       "      <td>gccggAAGTG</td>\n",
       "      <td>ELK1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>ELK1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name short_description                      tss_id shuffled  start  \\\n",
       "0         RUNX1          p1@SYVN1  chr11:64901978..64902009,-      NaN     15   \n",
       "1        TFAP2A          p1@SYVN1  chr11:64901978..64902009,-      NaN   1604   \n",
       "2          PAX5          p1@SYVN1  chr11:64901978..64902009,-      NaN    638   \n",
       "3          E2F1          p1@SYVN1  chr11:64901978..64902009,-      NaN    616   \n",
       "4          ELK1          p1@SYVN1  chr11:64901978..64902009,-      NaN   1947   \n",
       "\n",
       "   stop strand    score       p-value  q-value     matched sequence motif_id  \\\n",
       "0    25      +  15.6034  2.490000e-07    0.338          gtctgtggttt    RUNX1   \n",
       "1  1614      +  13.8909  9.460000e-06    0.155          CCCCTCAGGCT   TFAP2A   \n",
       "2   656      +  14.5484  5.460000e-06    0.107  ctgggcgacagagcaagac     PAX5   \n",
       "3   627      +  14.1475  8.620000e-06    1.000         gatggcgccaat     E2F1   \n",
       "4  1956      +  16.2128  3.920000e-06    0.605           gccggAAGTG     ELK1   \n",
       "\n",
       "          biotype  tissue_spec motif_upper  \n",
       "0  protein_coding     0.349684       RUNX1  \n",
       "1  protein_coding     0.349684      TFAP2A  \n",
       "2  protein_coding     0.349684        PAX5  \n",
       "3  protein_coding     0.349684        E2F1  \n",
       "4  protein_coding     0.349684        ELK1  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_ts[\"motif_upper\"] = fimo_ts[\"motif_id\"].str.upper()\n",
    "fimo_ts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>short_description</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>shuffled</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>biotype</th>\n",
       "      <th>tissue_spec_x</th>\n",
       "      <th>motif_upper</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>tissue_spec_y</th>\n",
       "      <th>tissue_spec_tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>25</td>\n",
       "      <td>+</td>\n",
       "      <td>15.6034</td>\n",
       "      <td>2.490000e-07</td>\n",
       "      <td>0.338</td>\n",
       "      <td>gtctgtggttt</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>ENSG00000159216.14</td>\n",
       "      <td>0.235779</td>\n",
       "      <td>0.667911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1604</td>\n",
       "      <td>1614</td>\n",
       "      <td>+</td>\n",
       "      <td>13.8909</td>\n",
       "      <td>9.460000e-06</td>\n",
       "      <td>0.155</td>\n",
       "      <td>CCCCTCAGGCT</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>ENSG00000137203.6</td>\n",
       "      <td>0.301559</td>\n",
       "      <td>0.872627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>638</td>\n",
       "      <td>656</td>\n",
       "      <td>+</td>\n",
       "      <td>14.5484</td>\n",
       "      <td>5.460000e-06</td>\n",
       "      <td>0.107</td>\n",
       "      <td>ctgggcgacagagcaagac</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>ENSG00000196092.8</td>\n",
       "      <td>0.505705</td>\n",
       "      <td>0.975997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>E2F1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>616</td>\n",
       "      <td>627</td>\n",
       "      <td>+</td>\n",
       "      <td>14.1475</td>\n",
       "      <td>8.620000e-06</td>\n",
       "      <td>1.000</td>\n",
       "      <td>gatggcgccaat</td>\n",
       "      <td>E2F1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>E2F1</td>\n",
       "      <td>E2F1</td>\n",
       "      <td>ENSG00000101412</td>\n",
       "      <td>ENSG00000101412.9</td>\n",
       "      <td>0.234943</td>\n",
       "      <td>0.662354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ELK1</td>\n",
       "      <td>p1@SYVN1</td>\n",
       "      <td>chr11:64901978..64902009,-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1947</td>\n",
       "      <td>1956</td>\n",
       "      <td>+</td>\n",
       "      <td>16.2128</td>\n",
       "      <td>3.920000e-06</td>\n",
       "      <td>0.605</td>\n",
       "      <td>gccggAAGTG</td>\n",
       "      <td>ELK1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.349684</td>\n",
       "      <td>ELK1</td>\n",
       "      <td>ELK1</td>\n",
       "      <td>ENSG00000126767</td>\n",
       "      <td>ENSG00000126767.13</td>\n",
       "      <td>0.206946</td>\n",
       "      <td>0.318789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name short_description                      tss_id shuffled  start  \\\n",
       "0         RUNX1          p1@SYVN1  chr11:64901978..64902009,-      NaN     15   \n",
       "1        TFAP2A          p1@SYVN1  chr11:64901978..64902009,-      NaN   1604   \n",
       "2          PAX5          p1@SYVN1  chr11:64901978..64902009,-      NaN    638   \n",
       "3          E2F1          p1@SYVN1  chr11:64901978..64902009,-      NaN    616   \n",
       "4          ELK1          p1@SYVN1  chr11:64901978..64902009,-      NaN   1947   \n",
       "\n",
       "   stop strand    score       p-value  q-value     matched sequence motif_id  \\\n",
       "0    25      +  15.6034  2.490000e-07    0.338          gtctgtggttt    RUNX1   \n",
       "1  1614      +  13.8909  9.460000e-06    0.155          CCCCTCAGGCT   TFAP2A   \n",
       "2   656      +  14.5484  5.460000e-06    0.107  ctgggcgacagagcaagac     PAX5   \n",
       "3   627      +  14.1475  8.620000e-06    1.000         gatggcgccaat     E2F1   \n",
       "4  1956      +  16.2128  3.920000e-06    0.605           gccggAAGTG     ELK1   \n",
       "\n",
       "          biotype  tissue_spec_x motif_upper motif_name          gene_id  \\\n",
       "0  protein_coding       0.349684       RUNX1      RUNX1  ENSG00000159216   \n",
       "1  protein_coding       0.349684      TFAP2A     TFAP2A  ENSG00000137203   \n",
       "2  protein_coding       0.349684        PAX5       PAX5  ENSG00000196092   \n",
       "3  protein_coding       0.349684        E2F1       E2F1  ENSG00000101412   \n",
       "4  protein_coding       0.349684        ELK1       ELK1  ENSG00000126767   \n",
       "\n",
       "               GeneID  tissue_spec_y  tissue_spec_tau  \n",
       "0  ENSG00000159216.14       0.235779         0.667911  \n",
       "1   ENSG00000137203.6       0.301559         0.872627  \n",
       "2   ENSG00000196092.8       0.505705         0.975997  \n",
       "3   ENSG00000101412.9       0.234943         0.662354  \n",
       "4  ENSG00000126767.13       0.206946         0.318789  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_ts = fimo_ts.merge(fimo_id_map_ts, left_on=\"motif_upper\", right_on=\"motif_name\", how=\"left\")\n",
    "fimo_ts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. find avg. tissue-spec for genes containing a given chip peak/motif"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>tf_ts</th>\n",
       "      <th>tss_ts</th>\n",
       "      <th>tss_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID3A</td>\n",
       "      <td>0.711899</td>\n",
       "      <td>0.822828</td>\n",
       "      <td>10119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATF1</td>\n",
       "      <td>0.287917</td>\n",
       "      <td>0.763838</td>\n",
       "      <td>8882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATF2</td>\n",
       "      <td>0.262339</td>\n",
       "      <td>0.811896</td>\n",
       "      <td>17871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATF3</td>\n",
       "      <td>0.483653</td>\n",
       "      <td>0.716232</td>\n",
       "      <td>23784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BACH1</td>\n",
       "      <td>0.345260</td>\n",
       "      <td>0.763086</td>\n",
       "      <td>8477</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  motif_id     tf_ts    tss_ts  tss_count\n",
       "0   ARID3A  0.711899  0.822828      10119\n",
       "1     ATF1  0.287917  0.763838       8882\n",
       "2     ATF2  0.262339  0.811896      17871\n",
       "3     ATF3  0.483653  0.716232      23784\n",
       "4    BACH1  0.345260  0.763086       8477"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_grp = chip_ts.groupby([\"motif_id\", \"tissue_spec_tau\"])[\"tissue_spec_x\"].agg([\"mean\", \"count\"]).reset_index()\n",
    "chip_grp.columns = [\"motif_id\", \"tf_ts\", \"tss_ts\", \"tss_count\"]\n",
    "chip_grp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "152"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(chip_grp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>tf_ts</th>\n",
       "      <th>tss_ts</th>\n",
       "      <th>tss_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALX3</td>\n",
       "      <td>0.923747</td>\n",
       "      <td>0.938096</td>\n",
       "      <td>3223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATF4</td>\n",
       "      <td>0.167687</td>\n",
       "      <td>0.931278</td>\n",
       "      <td>2904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATF7</td>\n",
       "      <td>0.242404</td>\n",
       "      <td>0.786313</td>\n",
       "      <td>2069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Alx1</td>\n",
       "      <td>0.909153</td>\n",
       "      <td>0.933308</td>\n",
       "      <td>2369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Alx4</td>\n",
       "      <td>0.910912</td>\n",
       "      <td>0.922418</td>\n",
       "      <td>1828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  motif_id     tf_ts    tss_ts  tss_count\n",
       "0     ALX3  0.923747  0.938096       3223\n",
       "1     ATF4  0.167687  0.931278       2904\n",
       "2     ATF7  0.242404  0.786313       2069\n",
       "3     Alx1  0.909153  0.933308       2369\n",
       "4     Alx4  0.910912  0.922418       1828"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_grp = fimo_ts.groupby([\"motif_id\", \"tissue_spec_tau\"])[\"tissue_spec_x\"].agg([\"mean\", \"count\"]).reset_index()\n",
    "fimo_grp.columns = [\"motif_id\", \"tf_ts\", \"tss_ts\", \"tss_count\"]\n",
    "fimo_grp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "389"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fimo_grp)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. plot correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAALMAAACzCAYAAADCFC3zAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJztnXlUVFe2/781j4CAYR4UVFBBBSGgOMYYjW0nwbSd0dWZ\nVlr7aeykNZ3E/Exe0hlM9CUro8+kW1tj1NaXxI5JKzFOCKgooFTEBJlBECiKkpqo6f7+cJ2bqqKo\nWxRVBZT3sxZLgVvnHKq+d9999tlnHw5FURRYWAIA7lAPgIXFW7BiZgkYWDGzBAysmFkCBlbMLAED\nK2aWgIEVM0vAwIqZJWBgxcwSMLBiZgkY+EM9gKHkwoULLn8/ffp0P42ExRuwlpklYGDFzBIw3NJu\nhi0qlQqNjY3o6ekBAAQFBWH06NFISEgAh8MZ4tGxuAPnVk4BvXDhAvR6PS5evIjOzk5IpVKEhYUB\nALq6uqDT6RAbG4ulS5ciKipqiEfLwsQtLebDhw+jrKwMXC4XkydPRnR0NG2FKYpCSEgICgoK0NXV\nhfnz5yMvL4+10sOYW9bNqK2tRWlpKcLCwpCZmQmhUGj3ew6Hg/Hjx2Ps2LE4fvw4fvzxR7S1teGe\ne+7pcy3L8OCWtMwNDQ3YvXs3QkJCkJ2dDR6Px/iaa9eu4eLFi5BKpcjOzsbs2bP9MFKWgXDLRTOa\nm5vx5ZdfIiYmxm0hA0BMTAzy8vJgNptRWFiIuro6H4+UZaDcUmJubW3FF198gYiICDz00ENuC5kQ\nHByM2bNnIzg4GLt27cLZs2dxCz7Yhi23jJtx/fp1/POf/0RoaChWrFgBsVjMuALYH1arFZ2dnTh3\n7hxSUlKwdOlSyOVyL4+YZaAEvJgvXLiAnp4elJSUQCQSYcaMGV6ZwE2fPh2XL1/GoUOHAACLFi3C\nlClT2GjHEBLwYj527BjOnDkDsViM3NxciEQir7bf29uLyspKtLW1ITg4GCkpKYiIiEBWVpZX+2Fh\nJqDF3NTUhJ07d0ImkyEnJ8enITWlUokrV65ApVJBJpMhLy8PaWlpCAoK8lmfLPYErJjLysrw/fff\nIzg4GLfffjsEAoHP+6QoCkqlEo2NjWhra4PVakV0dDSSk5MRExOD6OhohISEDMgVMZvNMBgM6O3t\nhcFgoL96e3thsVggEAggEAggFAoRHByM4OBgiMXiW9LdCTgx9/T04D//+Q+qqqowbdo0REdHDzhq\n4Q1SU1Nx9epV/PLLL6ivr4dOpwMAcLlcBAcHQyqVQiAQgM/ng8vlwmw2w2Qy0f8SwZrN5gH3zePx\nIJfLaXGPGjWKvqkDmYARc3d3N86dO4fz589DIBBg0aJFSE9PR1lZ2VAPDRRFwWAwQK1WQ6/XQ6/X\nw2QywWKxwGKxgKIo8Hg8cLlc8Hg88Hg88Pl82uo6+z+fzweHw4HVaoXFYqEtOGm/p6cHN27cgEaj\nAUVR4HA4iI6ORkxMDOLi4hAbG4vw8PCAsuAjTswURUGv10Oj0aCzsxPNzc1obGxES0sLhEIhsrOz\nMXPmTEilUgDMCfiBjsViwY0bN9Dd3Q0ej4fm5mZ0dXUBAMRiMWJiYhAbG4vY2FiEhIRALpdDKpWC\nyx15SxDDUsxNTU34/vvvYTabYbFYaOtjtVphMBhgtVrpa4OCghAXF4eUlBSkpqb2iVbc6mK2heyc\n0ev1aGlpob+am5uh1+vp6zgcTp+nAJfLBYfDAYfDQWZm5rB0WVyKub6+Hvv27YNGo4HJZAKfz0do\naCiWLVuGsWPHem0QO3bsQH19/YBfp1AokJaW5rVxeLO9W21sS5cuxWOPPea1Nj2CckFBQQFlMpno\n73t7eymLxUL98MMPrl7mN/Lz84dte+zY/I9Lx2jhwoV46623YLVacfToUaxZswZcLhd33nmnv+41\nFha34b366quvurogMjISq1atwm233YaXX3552M1+vfm49HZ77Nj8i0ufef369QCAjo4O8Hg8hIeH\n45133vH6IHbs2DH0/haLVxmKz9TlTpN3330XRqMR169fB0VRUCgUPhmEJ5M/luHNUHymjNum1q1b\nB61Wi+TkZKjVaixZssQf42JhGTCMYo6IiEBeXh5CQkJQXFzsjzGxsHgE4zJPUlISIiIisGPHDnR0\ndPhjTCwBQHd3t9/7ZLTMs2fPRlVVFf70pz9hzJgxfhgSSyBQW1vr9z4Zxbx161YEBwdjzJgxeOON\nN/D666/7Y1y3BDqdDp988gkA4IEHHkB8fDw6OzuxdetW8Pl8LFq0CBcvXoTBYIBKpcKLL77oUT9F\nRUX44YcfYBuFfe+99yAQCDBq1Cg8+uijfV5z/fp17Nq1CxaLBatWrcLJkyehVCoBwK0ohW3Kgb9g\nFLNIJIJMJsOECRPoNMZA47XXXsPYsWNRX1+PVatW4fPPP4dIJMKSJUtw+vRpCAQC8Hg8dHV1QSQS\nQSKRgKIoSCQSyGQyNDc3Q6VSYfbs2fjxxx8hFArx4IMP4s0338TDDz+MuXPnOu23tLQUd9xxB+Lj\n4/Htt9/iiSeeQHh4OP74xz/izTffhFgspjPhDAYDdDodCgsLsWjRIjQ3N+Pdd99FUlISpk2b1m8f\nAJCQkAA+/9ePmqIozJs3D0lJSfjwww/x/fff49q1a+ju7sa6desAAEePHsWKFSvQ2tqKkpISnDhx\nAhMnTsS4cePcek9dRHx9BqPPvGDBAly+fBkrV67E4sWL/TEmvyOTybBixQpERUWhtLQUfD4f0dHR\naGpqglgshlgsph+bDz30EKZMmYLe3l5wOBycP38eTz/9NCIjI9Ha2oq8vDzMmzcPdXV1SExMpEVW\nXV2Nbdu2Ydu2bTh8+DAAwGg0wmq1wmq10llq1dXVMJlMeP3113Hs2DG0t7fj2WefhUgkglQqxaJF\ni+hxz5gxA2vXrsWZM2cAAG1tbXQf+/fvp6+Lj4+325zA4XBw/fp17Ny5E8888wxOnDiBsLAw8Hg8\nlJaW4sMPP0R3dzc9Nh6Ph9DQUDz11FM4efKkW+/psBRzREQEYmNjMW7cOEyYMMEfY/I7PT092L17\nN5RKJTIzM6HT6dDa2oq4uDj88ssv4PP5uHHjBoCbyfWVlZUQi8VoaWlBWloa/v73v6OjowMxMTF2\n5b2YNgXMmDEDhw8fxs6dO3H33Xfj4MGDCA0NpQU5Z84cJCQk4LPPPsOoUaOg0+lw5MgR+vXFxcX4\n6KOPBlSQ5uTJk2hpacH27dshEolQWFiIvLw8XLt2DQCQlZWFNWvWID8/H9u3b8ePP/6IGTNmIDY2\nFjt37kRycrJb/QyFm+Ey0YiiKGr16tVUR0cH1d7eTq1du9YnCSKvvPKKT9p1l7fffntI+3cXo9FI\nXbhwgaIoimpqaqJ27do1xCPqn7vuusvvfTL6zPHx8eDxeBAKhYiIiKDvuJGYvN0ff/3rX4d6CG4h\nEAiQmZkJAIiLi3M6cRsueLLda7AwilmlUtnlY2zYsAEA8NZbb/luVCwjHrKVy3bi6WsYexqIaJua\nmrBp0yZs2rQJMpkM3377LTo7OwEAjz/+uOejZBmRaLVahISE+K0/l2K2nWW7g1wuR0REBP391atX\n8eyzz2LTpk19rt23bx/27dsHILBcllsZ288UANRq9fAR8yeffAKr1Yq0tDTMmjWLsYhKaGgoQkND\n6e/JvjJnOdAPPPAAHnjgAQAAQ0o1ywjB9jOdP38+rl+/joSEBL/171LMq1evBnBzj9f27dthMBiw\ndu1axkZ7enpw7tw5pKamYtu2bYiNjfXOaFlGDBKJBG1tbX7tk9Fn3rhxI0aPHo2cnBx6Ju2KNWvW\nAAB7BsgtzqhRo1BTU0PX7PAHjM7qa6+9hpkzZ2L79u1YunSpP8bEEgCEh4dDrVajvb3db30yWua5\nc+diwoQJePrpp5Gdne2PMbEEAKGhoRCJRLh06RIWLlzolz4ZLfPx48fx5JNP4u9//zu7y2QYodPp\n0NTUNGyTv7hcLqZNm4by8nIYjUa/9MlomdevX4/Y2FisWLGCrTk8TFAoFDhz5gwdOs3NzR0Wu6Md\nycnJQWlpKc6ePeuXA40YxbxlyxafD2IkodPpoFQqER4eTtezG8x1nvRPhAzcXAs4c+YMkpKSvNqP\nNwgNDUVmZiaKioowbdo0n9eqvmXPAfQEdy3iYCwn002gVCphNpvppWIulwur1QqlUjkkYqYoCr29\nvRCLxXY/v3btGi5cuICwsDBYrVbs2bMH06dPp+vd+QJGMRcXF2PMmDH4xz/+gWnTpt2yEQ13LeJg\nLKc7N0F7eztdxRO4mYstk8kQHh7urT/VLSwWC11DmqKoPmImCIVCTJ48GRUVFWhsbPSpmBkngIcP\nH0ZNTQ0yMzNx7tw5nw1kuKNUKvvk6BKL6Ml1jvR3E9hO8HQ6HcrLyyGTyeifabVaZGZm+s0qG41G\nukSuwWBwKwk/Li4O8fHxUCgUPl1IYRSzVqvF7t27kZKSgp6eHp8NZLgTHh7eJ4eEy+X2sYjuXueI\ns5vAbDbj8uXLtKDJNRKJBGFhYZDL5ZDL5T73Ra1WK/R6PVQqFXp6emAymQbcRlpaGuRyOV1V1hcw\ninnjxo3YsGEDgoOD8dJLL/lkEEPBQENbUqkUubm5tFC5XC4yMzOhVCrt2nB2XW5uLqPldLwJ9Ho9\nurq6cP78eXz55ZdQKBR21/T29kKj0UCj0eD48eM+qTZlNpuh0WigUqmg0+kGtXuEx+MhKysLZrMZ\nX3zxBQwGgxdHehOXPvPBgwcB3FxnP3ToENLS0vD000/3e71jyuf//d//oa2tDSKRCE899ZQXhz04\nPJ2gpaWlISkpCUqlEu3t7SgrK3Pahu117kYzyE1w5swZmM1maLVayGQyeoJH/O6MjAwUFxdDo9GA\nw+HQLoe3IhoURcFoNMJgMHg9wV4qleLRRx/F9u3bsWfPHjz66KNePTjJpWU+deoU7r33XlRUVOCD\nDz5gPC/66tWrePzxx+klzMrKSvzXf/0XVCrV0OwJc4I7vqnj9bYWXCqVIjw8HOXl5S7bkEqliI+P\n7yMuV0+EtLQ0PPzww8jKykJYWBgkEgn9O6vVitLSUpSXl9v1IRKJYDQaYTabGf1yV1itVuh0OqhU\nKmg0Gp/tFImMjMRDDz2Ea9eu4cCBA7BYLF5r26Vl5vP5sFgsdKII0x/omPJ5zz334LPPPkNbW1sf\nP3Ko8pldTdAchdefBW9paaHFKBQK3Q6PufNEkEqlmDRpEioqKuzGSVEUfv75ZwA3t09xOBxoNBpo\ntVr6/W5vb0d8fPyA3g+z2Qy9Xj/gVTqKoqBSqVBbW4va2lqsWrXK7jN11V5iYiKWL1+Offv24eDB\ng7jvvvu8ogGXYp42bRqee+45/PGPf8Snn36K1NRUl42RlM+goCAcP34cbW1tkMlkmDNnTp9rhyqf\nmfidtkJxNkHr7OzEqVOnwOVy7R71vb29OH36NNRqNf3aoKAgxvBYf+05C+0plUpkZGTQ1p/L5WL8\n+PG0mLlcLqRSKW7cuEGfNSKTyVBeXo6JEydCKpW6jFeT2LDBYBiQZaQoCu3t7aipqUFtbW2fEly2\nn2l/7qjtGTNTp05FeXk51Go10tPT+82uczec51LMDz30EO6//34IhUJQFMVYamDZsmVudTqU2Pqm\nthaSfOA6nQ6lpaWoqKhAT08POBwO5HI5JBIJzGYzSkpKoNVqaUFarVZoNBrMmjWrX6usUChw8uRJ\n+gaQyWSQSCR9rLmj5c7IyEBERAR9k1RXV9M3IZ/PB5/Ph1wu7/N0qK2tdfoEsFgstIjdCakBN59a\n165dQ01NDerq6uwiEVwuF7GxsW6XH3AkNjYWFosFly5dAp/Px8SJEweVLupSzO+88w7S09Nx7do1\nmM1m7N27NyDKc6WlpSE6Ohp1dXUYO3YsLRaFQoHi4mK0t7fTFovD4aCnpwcikYg+sw8AbREpikJQ\nUBC9XczRIhIf3baGhlarhUgkAp/Pp/t25suXl5fj4YcfpsVuexMKhULI5XK7xQpisb///nv6QCUA\nKCkpQUREhNuHe5rNZjQ1NaG2thZ1dXV2kQc+n4+EhAQkJycjMTGx38USd0lISKBDkFKpdFD1DF2K\nuaurC3fffTe2bNmCv/zlLwETmrO1gBUVFcjNzUVSUhLtRlitVlqoFEXBYrHAZDJh4sSJuHr1Kt0O\nOUpMJBIhPDzcqU8cEhJCfy+TyaDVagHcFGtGRgY9aXPHl09KSqKPhouNje1jgUkehFKppG86Mkls\nb29HdHR0v++J0WhEfX09amtr0dDQYBdLFolEGDNmDJKSkpCQkOD1o5uTkpKg0+nw008/ISgoyOPV\nTJdi7u3tRUtLCz14XwW7/YmtBbRarTAajSgqKgIAeiLk7BFsMBhQW1uL3t5e8Hg82nLL5XLk5eUB\ngNMoyeLFi2EymcDj8SCRSGgL7+gTZ2RkuPTl+5s82oYKS0tL0dnZCbPZTPvmOp0OQqEQo0aNcvpe\n1NXVoba2Fk1NTXZ9S6VSJCUlISkpCbGxsT4/snnSpElQq9W4dOkS5syZ41F/LsW8YsUK/POf/6Qn\ngMS5H8kQC6jX6+2s5KFDh/oVMgCYTCZotVro9XratZg0aRJ9GqyjGICb7sTBgwdhNpuhVqvpPIrs\n7Gy70J7RaMTp06cxbdo02i8mAlcqldDr9Yz5HocPH0Zvby8A0DcF8T/j4+PpMN+NGzfoCERra6vd\n3xscHIykpCQkJycjKirKr4cxcblcpKeno7CwEPX19R754S7FfPLkSdq1WLVqlWejHGaEh4eDoiha\nyMSNsHUtHCE/JxM/8iH/8ssvyMjIoGPPtpbVarVCq9UiLCwMAoEAIpEIVqsV+fn5dqtp5DxtACgs\nLMSUKVOQkZGB9vZ2WvDkcHjHuLNSqQSfz0djYyNMJpPdqiP524CbE0eVSgWtVtunYHx4eDgt4KE+\nSzs4OBgxMTFoaGhAUlLSgMfiUsw1NTVoamqy+9lA45jDDalUitTUVNq1AG5++K5m98R3Jm+u1Wql\nIx179+7F3LlzkZaWZjdBs1qt9Aoe6YM89onwjUZjn2N+FQoFMjMz7Sw3l8ulJ41krBwOBzweDz09\nPZDL5eBwOOByuZBIJH2WnsmyNyEqKooWsDfqWnjzBoiLi0NLSwt6enoQHBw8oNe6FHNjYyO9pE0g\n5QdGMtnZ2aiqqoLBYIBAIEB3dzfMZrNLQQsEAruYLPkAeTwe/ci39WGlUim+/vprpz6wVCpFRkYG\nTpw4QQvT1uJfvny5T86yTCaD2WwGj8cDh8NBamoqWltbAdxcVZsyZQrOnz8Pk8nkdLWVw+FAIpHg\n3nvvHXS6KAkLki9v+tPk5vK6mFNSUgJCvI5UVVVBp9NBr9fT0Qg+n0/n5jpj4sSJCAkJQWVlJV3e\nlsSeAdACJl8A+o1nKxQKlJeXQywW0zkWtitgIpGIzlmmKAoymQxisRh33nknDAYDlEolysrKoNVq\n7UrnOmaz2VpMMgEd6Eofj8frI1xvuyK2iyK9vb0oKCjAmDFjMGXKlAG141LMzz33nGejG6bodDoU\nFRWhtLQUAGghy+Vy/OY3v8HXX38NjUbT5wPncDhobW3FnXfeieTkZBw4cICeRBI3wdlSsrOEI9to\nikAggFQqpSeVHA4H6enpqK6uhlQqpf16krMcGhoKtVqNgoICaDQa+sYjllgoFGLs2LFISkqCRqNB\nUVGRnQ9vNpudRjVs/04iWIFAQD8V/Am5iW0rY7mLSzG7ikuONBQKBYqKitDZ2QmLxUL7sL29vZDJ\nZKipqek3R4FMAEtLS1FdXQ2KoqDRaOg2HJeSbbG11EDfeHJISAhkMhnGjRuHzMxMqNVqNDc3QygU\n2uXGqNVqfPvtt06jJhwOB1KpFHfddRddPUqv16O8vNwuX8bxqWNrdclRF0ONSqUCAISFhQ34tS7F\nfODAAQQHB9sd/D4SixwSa2j7GCbRCw6HQ69A9Zd0Tq6xTfQhH/yoUaPA5/PtdpO4Sv10lhsiEAjo\nODNZFCFLz0ajESaTif6QgV8nrOSzIGK2FUB3d7ddXJvH44HH48FgMCAyMhJ8Pn/YnYMO3LTM5Ik1\nUFyKuampCSdOnMDx48fpn7kqceuYz/zVV1+hvb0dDQ0NQ1rPmVhD8gESMZFHe2pqKq5cuQLAuQUj\n15BzTWyFYBtxaG9vx5EjRxiz4nJzc1FSUkJPKKdOnQoOhwOlUkk/IRx39YSHhyM5OZl2W86ePUtn\n7slkMkyePJlO/JFIJBg9ejR4PB64XC6dZcflchETE+P1FTxvolKpEBYW5tGN5lLMzz77LB555BF8\n9913EAqF+O1vf+uyMWclbNVq9ZAXKrHdoUGWlLlcLoKDgzFz5kwkJiaiqqqK/sBtoxYymQxTp05F\nRkYG6uvr+yxNE78yMzOTTtYHnC9skGy1+Ph4jBo1CiqVCmazGc3NzTh37lyfLDQi4JSUFLsQ2ujR\noxEREYG6ujrI5XL6yUImg7m5uUhPT0deXl6/CVXDFZVK5ZG/DLixO3vz5s34wx/+AI1Gg7feesul\nhXXMZ1YoFNi4cSO2bt2Krq4uu8egP/OZbTPlJBIJxGIxUlNTkZ2djdraWjqEZpuRJhaLERMTg56e\nHtTV1aGhoQGxsbFoaWmh9+FNnjwZiYmJiI2N7Te3orOzE1FRUTAajTAajbBYLG5loY0dO9Zu4yqB\nw+GgpqYGZWVl9JOFbPUnT5WzZ88iOTnZox0vg8HdfGZXqFQqTJw40aPXMoqZbBUHbroRrnDMZ54y\nZQq2bdsGtVrd527zdz5zf5GFoqIimEwmiEQiCAQC9Pb2Ys6cOUhMTMQ333xDv95qtaKlpQX5+flQ\nKBS4cuUKamtrUV9fj9zcXERHR9M5GMS6k1ixSqVCU1MTLWCy7AwwZ6ERN4FM1IxGIy5evEj7+0aj\nERqNhk4DJWN1Fir0Ne7kMzOh0Wggl8s9ei2jmBcuXEgfdHjHHXe4vHYk5DPbUlxcjI6ODrvVPS6X\nizNnzkCj0Ti1tF1dXaiurrbbfVNQUACpVAqTyYTu7m7w+XyYTCYIBAL861//gtlstnNdXGWhCQQC\nO/E6+o6tra124yJpnmazmS4Gb5ugRFJSyc3rDwvtKWT7l7Mnkju4VQXU1emfIwWSdUaEFR0djcuX\nLzvdacHhcGjrR4RmKxYiJp1OB41GA4vFAp1OB7FYDKFQSM8RbNt2loXG5XL7hMaYJj5EpEajkfbX\ng4KC6HE6Ls6cOXMGWq2W3iArk8mGXW06svuEuCaeTlBvifJcJDRHPlSKouioiyPEzyXWmnxxuVzI\n5XJ6kkay6IiwLRYLvchBIKE2kUiEO+64A7GxsbTl9TSuW1tbC4PBQPvaQUFBmD9/fr+LM2SnN/Dr\npoDhWpuO3MiebnJlFPNLL72EuXPnYs6cOXZZWyMJUp/NNlOuP2x/RxKMSOSDHMc7ZswYVFZWOt3g\nS/xWYi25XC54PB4SExMHnGvgCBEoeQKYzWYIBAJamM4WZxzHSHKdh6o2nSvIe+czMW/YsAElJSV4\n//330d3djZkzZ2LRokWD3i7jL5RKJZqamuxm18TaDmQfHFnxMxqNfdIobRcvwsLCEBcXh7a2NrtQ\n2WCFTP4W27g2cXucCZOEIx3P4SOuib9r07kDeR89LXPAKOby8nKcP38e3d3dGDt2LEJDQ/Hiiy/i\nvffe86hDf/Kf//yHrjNBQm9kUiWRSOjEIndqethGIDgcDuLi4iCXy9He3k67C6mpqcjJyWHcHe0p\n7u4sB+zDkSQmLpPJwOfzh2282VmcfyAwirmyshJPPfUURo8ejevXryMyMhJJSUkedeZPOjs77Qqm\nkLteLpfT/mp4eDjq6+vd3g4mEAiQmppKi0EoFMJisaC7u7uPaPsLiQ1G5Ew7yx37IBPOkRLNAG5+\nTj6xzOvXr8eVK1dQV1dHT5q2b9+OuLg4jzrzJ/X19fT/iTvB4XAwefJkREZGQiwW44cffqAttSuX\ngwj33nvvRVRUlF1MF4DbhQu9UfGeaSHEVR/D0bVwZCDunyMuxfzuu++ipKQEM2bM8KjxocJqtSI2\nNtbpm5KYmEhnwCmVSpePNJIwLxKJMGPGjD7b4J1ZWaVS2aeEAbnWWxXvXVl9Z31ER0ePCKs8WFyK\neevWrfRyL8H2UPjhAEVR9Eob+SJ1JSZOnIiqqir6uoiICHz33Xd9ciD6gyTtC4VCXLx4ERKJhLZy\nzixgU1OT3TJzZmYm7r77bgADKwvmiLuuibM+tFot9u7dS0dWhluM2Zu4FPPKlSvpfAaj0UinJw4l\nJNxENnn2519RFIW0tDRwOBzU1tZCo9HQ24wA9JlIOYMsc5Nri4qKIBKJEBYW1scCnj59ml5NJFy4\ncAG33347wsPDBzR5s2UgromrTbXk++EaYyaQCkqewJjhc+DAAUyaNAnp6enYsWOHR50MBnLcQE9P\nD1QqFV3wWq/X9xEyRVFoa2tDcXExvvjiC+zZsweXLl2iJ3jh4eGQyWQICwtzeWOSWTXZkU12lSiV\nShw5cgT79u3rs0DirH6x1Wqlc6A9qds80Iqljn04bqolPxtMtVBfYrFYYLFYGM9o7w/GaAaXy8XF\nixcBMO/Cdcxn3r9/P9ra2tDS0oIXXnjB5ZYdwN7qki+myQDJQiO1IGxFxuVyERcXh6SkJIwdOxZc\nLhfffvstTCaTXajNEduVP+DmMqvjgovtbmmg/yVY2/jyQLPYPHFN3N1UOxwhn4nPxPz//t//w3ff\nfQej0chYnssxn3n58uVobW3F8ePHXQq5t7d3QDWcSS00piy0MWPG2FlgHo+HuLi4PuViHSG7UGxL\nEJDrSdI82a1Ntk7l5eXh2LFjfWo0O4YxB5LF5qlr4s6m2uEIeYL6JNGITADJB6pQKFxOAB3zmQFg\n165d+POf/9znWtvcV3cWLnp7e+laaKToCYFkoSUnJyM+Pr6PlRQKhRCLxTCZTLh27RqCgoLQ09Pj\nsk+ylC2TyTBz5kwUFBTY/d5kMuHee+8FRVG0lRUKhTh9+jQMBgPEYrHLyqDu4G5c2RX+zGkebD7z\njRs3wOMz7OBYAAARVElEQVTxfCPmlJQUZGRkICcnx63GHPOZs7KyIBQKnT42bHNf//rXvzptj9RC\nq6mpQXNzs9NaaMnJyYiJiemTtEOSfMRiMe0KFBcX00UF3XkKkAlUbW2tXdFD4Kb10Ov1dsv6vhCO\nN9r0V07zYPOZVSoVQkJCPN6b6FLMP/74Iy5dumRXZNyVZXaWz+zMKrvCthaa46w2JCSEXtXqrxaa\nQCCAWCym970RdDodqqqqaIvrLlarFVeuXIFYLKaTxoVCIYxGI7030jbK4Avh+DPBfqiwWCzo6Ohg\nrAHuCpdiXrNmDXQ6nc+3oJMC37W1tX2SeEaPHk0LuL9aaBwOB2KxGCKRqN+xlpaWoquri64tNxAo\nioJer4fJZKKr1DtGCJhCXkyxYl8dTzxS6OjogNVqRWRkpMdtuBRzcXExWlpa7H7miwpHlZWVGD16\nNP19dHQ0LWBXtdB4PB4tYlePJp1OR+++9nSpFLi565mcIuDol7uKMjDFikfKwe6+pLGxEcHBwR5v\nmQIYxJyfn49//OMfSE9Ph06n8ziYzQSHw0FCQgIdQmOaAJCEd3d3JCiVShgMBrtNq8Cvy9UkKkH+\n398YxWIxXSODCM+2LWdRBqZl7IEcYxyollun06G9vR2/+c1vkJWV5XE7jKG5+vp6PPHEEwBuhul8\nQVZWFu655x6X17hrhZ0hkUjo8gK2kz/b0Btpk4TkeDye3YxcIpHQucF8Pr9PsfD+ogxMsWJ3YsmB\nbrlramogEomQnp4+qHYYxSyVSvHxxx/T8VRf4KpdElbzZF8YsWYGg8GuXgaZBBILS4QsFovtDjYn\nO2syMjLsioATMU2cOJHRWjLFipl+780EpeGIXq9HU1MT5syZM+h0CUYxr1+/HjU1Nejp6fHpifSO\niEQiSCQSjyefttYM+LWElVarBY/Ho7dD2VprnU4HDocDq9VKV/okR6JlZ2f3Ea5tlKE/N4ApVsz0\n+8EkKI0EqqurwePx3A7/uoJRzH/7299gMpkwd+5cfPTRRz4tcUv8UtvYsCc4WjMAdJ4FscIikYiO\n1BB/2VnSUm9vL4qKilwWkbS9cSiKogvMELExxYpd/d7TVcCRgFqtRmNjIyZNmuSVbXhu5WZERUVh\n4cKFKCwsHHSH/fVBzrPzRjE/Z9aMx+MhKCiI3hdnNBqh0+nodE1H35lAKn7u27ePFlVOTg79lLK9\nccg5KUVFRfj5558xc+ZM2rdlihX393tvrAIORyiKgkKhgFwuH9RxabYwijkpKQmHDh1CfX09XS7V\n25DohLdwZs3IFicSrSC7RYhbQY5VcIxocDgcGAwG8Hg8Ou/iyJEjAG4WySY3DlktJBiNRq/5tv4u\ns+UPWlpaoFKpkJOT47W5GKOY77//fsjlcsjlcqfHBg9HnFmz2NhYXL16lU4UIhahsbGRts6JiYlQ\nqVRQq9X060idZMe0yzNnzmDixIl255PYYlvm1hviC6RVQIPBgJ9++gnR0dG47bbbvNYuo5hfeOEF\n3HXXXTAYDHjppZfw7rvveq1zX+IsFZLEpslZISRxnXxvMBiQn5+PxsZGtLW1ob6+HkKhEGq12q4e\nMnDTbVEqlYiPj0dubq7dgT9khTBQfFtvMX36dFAUhT179kAoFOKRRx7xOKnIGYxijoiIwJIlSwAA\nly5dcnmtYz5zWVkZKisr0dTUhJdfftkLwx0YxJrZVpsn1pZYUttEKK1Wi2+++QYGg4GOepBDfGwz\nAuVyud1RweTGKS0ttTusPRB8W29TUVGB6upqPPDAA14VMuCGmKuqqrBy5UoIhUJ0dHRg/fr14HA4\nThOOHPOZT5w4gaCgIIwdO7bPtf4sadufD227tG21/nocGolqkCMYjEajXaEXDofTR6hSqRRz5851\nGsK7VWBKAVWr1Thy5AimTJlil7zmLdyqzyyTyVBYWIgJEya4PDnTMZ9Zp9PhueeewwcffNCnPrM/\nS9r2FxEAfj0imORHOy5nm81menIYGhpKb5aNjo5GU1OT2/UybgVcpYBSFIV///vfEAqFWLx4sU/6\nZxTze++9h/z8fFRVVeHYsWMufWbHfOacnBzs2LEDJpPJ42ro3qK/iAD5mVqtxsGDB50mIpEkfbKc\nrdfrsW/fPrrUVaAtL/uChoYG1NbW4pFHHvFZzUJGMXM4HBw+fBhPPPEEPv74Y5fXDvf6zM6sJvlZ\neHg4JBKJ3ZFkwK/n55HXjcQdz0ONVqtFVVUVMjIyMG7cOJ/1w+isLl++HNnZ2dDpdIxnmoxkiM9L\nzsHj8/mQy+WIiIjAvHnz7CpUjqQdz0MNWRwRCoVYtGiRT/titMyZmZn46KOPsGTJEqSkpPh0MEMN\nWdU7e/YsvVLomFQ00nY8DzXt7e3o6OjA9OnTfV53xa1i442NjT4dxHBi+vTpTrPhRuqO56HEarXi\n8uXLCAsLQ1RUlM/7YxRzc3MzFixYgCtXrmDMmDEjpi7zYGCKSATi8rIvIGm3Dz74oF+KbTL6zJ9+\n+ikqKirA5XLxxhtv+HxAIwWpVIr4+HhWyC44d+4c4uPj/VY1llHMIpEIMpkMEyZMGPLDKVlGDmaz\nGQ0NDZg2bZrf+mQU84IFC3D58mWsXLnSZ8FulsCDGD5/Bg0Yfeb29nYsXLgQAPoUC2Rh6Q+DwYCw\nsDCv51+4gtEyR0VFISoqCiKRCMXFxf4YE0sAYDAYfJb/3h+MlpnszbJYLHQ10FuNQN7m7yvMZrPf\nUxjcOjoNuJlsPmvWLJfXOqaAHjt2DNXV1Rg9ejTuv/9+LwzX/wT6Nn9fYTab3T7rxVswuhlLly7F\nmjVrIBKJ7CpvOuPq1at4/PHH0d7eDgAoKyujj1IYiQy02DeLPYOpTuQJjGI+cuQIampqkJmZiXPn\nzrm81jEF9MEHH8Rjjz2Gn376qc8G03379mHZsmVYtmwZFAqFp+P3Ka62+bP0xfYzBeD3E30ZTaZW\nq8Xu3buxfv16/PDDDy6vdUwBbWhoAAB6V7Qt/sxn9pRA3ubvC2w/03nz5nlcAd9TGMW8ceNG3Lhx\nAwKBgPaf+2O4p4AOlEDd5u8vPKlCNRgYxfz5559j8eLF2LJlC+Lj4/Hf//3f/hjXsIHNw/Acf1tm\nRp/ZZDLhypUrWLt2rd/vtOECm4fhGf7WC6OYpVIpiouL6V3NLCzuMqx85p9++gnZ2dnIzs6GRqPB\n7Nmz/TUulhEOKQ3sT1yKuauriw1DsXgEKXXmT1zeOm1tbSgoKKCP9j18+LC/xsUywvG3VQYYLHNi\nYiImTJiAhIQEAMCLL77ol0GxjHz8bZUBBjErFAosXrwY48ePp4t+V1ZW4vz583j88cf9MkCWkcmw\nE/OKFStw6NAhbN++HVarFRKJBFlZWVixYoW/xscyQhl2YhYIBFiwYAHi4+PpJd3bb7/dLwNjYRko\njF76xo0bUVNTg4aGhluq5ADL4Bh2lhkAxo0bRyePMOGYzwzcrAR67tw5PP/884MYJstIYyjEzGiZ\nf/75ZzzzzDN4/vnnGQXpmM9MdqYM5lRUlpHJsLTMW7ZswfXr1+maYa5wzGc+deoUgoKCcPnyZahU\nKrttNP6sz8ziH2w/U4PB4Pf+ORSD2XzmmWeg1WqRnJwMtVpNFxJ3xldffYXOzk5YLBakpqZi/vz5\nAG7WeF63bl2/r3v11VeHbU4zi2fk5+fj66+/9mufbh0DkZeXh5CQEMbd2f3lM7sSMktgMmnSJL/3\nyfh8T0pKQkREBHbs2IGOjg5/jIklABiKdGFGyzx79mxUVVXhT3/6k9cOH2Rh8QWMlnnr1q0oLy9n\nCyeyDHvYwoksAQOjm7FgwQLs3r0bK1eu9Fkhl+bmZo+iGQqFwqsFWbzZ3q02tqVLl+Kxxx6jfzYk\nLinFQF1dHfXVV19R//rXv6j9+/czXe5X8vPzh2177Nj8D6Nl/uSTT7B8+XL6fGkWluEK71WG53tZ\nWRkiIyPpYijx8fF+Gpp7eLvumzfbY8fmX9xaAZw3bx59zO59993nr7GxsAwIRjcjJiYGkyZN8mvR\naBYWT2AUs1wux9GjR+nvV69e7dMB9Ydjeqnt90uWLMGuXbtgsViwatUqBAcHD6ito0eP4sqVK2ht\nbcUrr7yCd955BxEREbj//vvdqivnqpTvrFmzBjW2/fv3o62tDS0tLVi3bh22bt06oLE1NTVh06ZN\n2LRpE2Qy2aDeN8e2Bvu+eRvGOPPq1avtvoYKx/RS2++PHj2KFStWYNGiRSgpKRlwW2FhYVi9ejWE\nQiEqKiqg1+thsVjcPibOVSnfwY5t+fLl+N3vfocpU6agvb19wGMjp8w6a3+gY3Nsa7Dvm7cZMbmX\njumltt+bTCZYrVZYrVZ64+1A2oqMjMTbb7+N++67D+PGjcPLL7+M2bNno6CgwKOx2Zby7e3tHdTY\nAGDXrl343e9+h6ioqAGPLTQ01C71djDvm2Nbg33fvA1jNGO40NHRgcLCQggEAmg0GsjlchQWFiI8\nPBx33nknduzYgYaGBixbtoyxLJRjW7t378Ztt92GtrY2iMVi7Nu3D1evXsXdd9/tVvV3x/bKy8tR\nUVEBALjnnnsGNbbRo0fj0qVLyMvLQ319PXbt2jWgsQE3z+MbN24cSktLB/W+ObY12PfN2zBGM1hY\nRgojxs1gYWGCFTNLwMCKmSVgYMXMEjCMzDPN3OSrr75CVFQU2trakJeXh8jIyCEdz7Zt2/Dkk09i\n9erVePrpp3HmzBmsWrXK7ppPP/0Uv/3tb7F//348++yzbrW7YcMGLFiwAHfccQcUCgV27NiBiooK\nTJ06FZMnT8Yvv/wCg8EADoeD2NjYgN2TGdBiLi4uxqhRoxASEgKDwYDXXnsNfD4ft912GxYuXIgP\nP/wQAoEAjz32GD7//HNs3rwZ+/fvR0JCAjo7O3H+/HkYjUY8//zzCAkJAQAcPnwYxcXF6O3txSuv\nvILf//73yM3NhVarxZtvvomPP/4Y3d3dEAgE+Mtf/oJXXnkFVqsVM2bMQG1tLRQKBZqbmyESiVBT\nU4Ompia8//77MJlMeOaZZ1BTU4OSkhKcPXsWb7/9NpYvXw4A+Pe//02L+8CBA1AoFDAYDHjyySdx\n6dIlLF26FMDNhJ/NmzfjhRdewN/+9jfw+Xy88MIL2Lx584g9j9FdAvqvmzlzJqKionDhwgUAQEND\nA/Lz8zF+/Hj6jL+lS5farWoR9uzZg5ycHHR1daG6uho7d+5ETk4OtFotwsLCMGvWLPB4PMTExODl\nl1/G//7v/6KiogLl5eWYNm0a6uvrcerUKWRlZSE/P58+Q3Hy5MlISUmhdy/v3bsXzz33HKRSKb3M\nPGPGDDQ3N2P+/PkoLCwEcHOTBKGkpARbtmzB4cOHUVtbi5SUFMyYMcPle7FhwwZwOBwsXrwY8+bN\nG/R7OxwJaDHbQlEU/vznP0Or1eJ//ud/8NJLL2Ht2rU4deoUqqqq6BU3lUqFhIQESKVSrFmzBufO\nnUNMTAw++OADADd3VUilUnz22WcQCoV2Cw0SiQSTJk3CmjVr8M0338BqtYLD4YCiqH6LohiNRnC5\nXJhMpj4n4E6dOhV79+4Fj8fDH/7whz6vJTeWO7zxxhsBb5kDegIYHR1NV/vncDjYs2cPCgoKkJqa\nCr1ejw8//BCXL19GWloaJk+ejPfff5+24rNnz8aLL76IgoICREVF0W1WV1djx44d6O3tRUxMDKqr\nq/Hmm2+ipaUFqampUKvV2LhxI3p6ejBr1iwUFxdj/fr1MBqNTsf4+9//Hps3b8brr79Or5qFhobi\n/Pnz0Ol0kMlkCA4OtlveTk9Px6uvvoqysjLG88xvJdgVwEGybt06bN682Wftf/nllxg/fjyys7N9\n1kegENCWeaRTWVmJiooKZGVlDfVQRgSsZWYJGFjLzBIwsGJmCRhYMbMEDKyYWQIGVswsAQMrZpaA\n4f8Dz01D59gitcEAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b937f3dca90>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "g = sns.jointplot(data=chip_grp, x=\"tf_ts\", y=\"tss_ts\", kind=\"reg\", size=2.5, color=\"gray\")\n",
    "g.set_axis_labels(\"tissue-specificity of TF\", \"mean(tissue-specificity of TSSs w/ peak)\")\n",
    "g.savefig(\"chip_corr.pdf\", dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAALcAAACzCAYAAADL/42JAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJztnXl0VFW2xr9b85SEJKYICQQSmiRAmAkQEiCAiNiinRa0\nW6EdupuWXqKC8J6iD7EdEIGnS1rl4cBg47BUoBtaAohhDAECARIIEBKGzCHzUHPd+/5In+OtSk0Z\nakhRv7VYmhruPVX13X332WefvRmO4zgECOCHCLw9gAAB3EVA3AH8loC4A/gtAXEH8FsC4g7gtwTE\nHcBvCYg7gN8SEHcAvyUg7gB+S0DcAfwWkbcHEKBnOXv2rEuvGzdunJtH4n0CljuA3xIQdwC/JSDu\nAH5LwOf2M1iWhV6vh9FohEwmg0Qi8faQvEZA3H6A2WxGYWEh8vPzUVxcDLPZTJ9TKBSIiYnBwIED\nIRaLvThKz8MENiv0Xtra2nD27Fnk5uaipaUF/fv3h1KpRHBwMMRiMbRaLWpqalBZWQmJRIKRI0dC\nrVYDuDuiJQFx90IqKipw+vRpFBQUgGEYjBgxAhMmTEBkZKTNUKBGo8HFixdRW1uLoUOHIi4uDuPH\nj/fCyD1LQNwuwrIsqqurUV1djZaWFrS2tqK6uhpCoRBisRhKpRIqlQoqlQoCwS/z9J6ykFqtFj/+\n+CNu376N5uZmyGQyDBo0CDExMS751RzH4cqVKyguLkZcXBwWLFgAhmF6ZGy+SsDndkBrayuuXLmC\nK1eu4NatWzCZTAAAmUwGlUoFvV4Ps9kMo9EIo9EIABCJRAgLC0N4eDjCw8PBsqyF2DtDQ0MDioqK\nUFRUhBs3boBlWajVagwZMgR9+/bt1HEZhsHQoUMhlUpx+fJlHDp0CDNnzvRrgQcstxUcx+HmzZs4\ncOAAqqqqAADh4eGIiIhAaGgoQkJCIBKJOrzHYDCgpaUF9fX1qKurQ0NDA1iWhVQqRUxMDAYMGAC1\nWo2IiAgolUpIJBIwDEOjG83NzWhsbER1dTUqKytRUVGB5uZmCAQCxMTEYMiQIeA4DjKZrNufsbi4\nGIWFhUhPT8e0adO6fTxfJSDu/8BxHK5fv46srCxUVlZCqVRi0KBBiIqKglQq7fTxzGYzGhsbIZVK\ncevWLZSXl8NgMNDnGYah4uYjkUgQFRWFfv36oX///oiLi6OCdnVp3RVaW1tx+PBh3HvvvUhNTe2x\n4/oSAbcEwM2bN5GVlYXbt2+jf//+WLBgARoaGrp1yxYKhQgPD6c+N8dxaG5uRm1tLTQaDXQ6HViW\nhVgshlgsRlVVFeRyOWQyGT2vVqvFpUuXeuQzWjN16lQYjUb89NNP4DgOaWlpbjmPN7nrLDff+jU2\nNuLKlSuora1FcHAwEhISoFar/doPJYwbNw4cxyErKwvHjh1Deno6pk6d6lef/a603M3Nzbh69Sqq\nq6uhVCoxduxY9OvXz69+WFdgGAYzZsyAUCjE4cOH0dLSgjlz5kAoFHp7aD3CXSNujuNQWlqKs2fP\norKyEnK5HKNGjUJ0dHSXoxn+wrRp06BUKrFv3z5UV1fjoYceQkREhLeH1W383i1pbGykS9OVlZVQ\nKBSIi4tDTEzMXS1qW/H327dvY/fu3WhqakJycjJSUlIQEhLihdH1DH4hbhKKa2trQ0NDA+rr61FR\nUYGysjLU1tZCKBRi8ODBGDduHJqbm+8698MW9haXjEYjjh8/jlOnTsFgMGDgwIEYPHgw+vbtC7Va\nDaVS2SEU6qu4JG69Xo+GhgaEhoZ2KSzWHViWxY4dO9Dc3AyO48CyrMV/TSYTdDod+B+DYRhERESg\nf//+GDhwIOLj490STuvNOFs51ev1yM/Px5UrV3D79m26SAW0L1TJ5XKIxWKEhIRgwYIFPnkXdCju\nU6dO4eDBgwAAlUqFtrY2iEQiTJ06FSkpKW4Z0NatW3Hz5s0uv7+goABJSUk9N6Ae4G4c06BBg/DU\nU0+57fguwTngwoULXFlZWYfHL1686OhtXiUjI8PbQ+hAYEzewaHz9O6776K1tRWJiYngOA4Mw+C9\n997DiBEjPHXtBQjQZRyK+6uvvsLJkydddkFKS0uxdu1arF27FkqlEnv27EFtbS0A4IEHHsCXX34J\ns9mMxYsXIzg4uPujt8Fjjz3mluN2h8CYvINDn3vTpk0oKSkBAAvLbY+GhgZs3LgRL730EpRKJd5/\n/30sXboUa9euRf/+/XHvvfeisrIS1dXVmD17ts1jbN261fu+WoAexVu/qUPL/ec//xlZWVmYMWMG\nOI7Djz/+6PBgoaGhCA0NpX9rtVoA7dELo9EIlmXBsmyHFbBvv/0W3377LQBAIBAExO0H+MJv6lDc\nr732GgoKCnDo0CEAQP/+/V06aEtLC06fPo3ExERs3rwZ0dHRmD59OrZs2QKxWIy//vWvFq9/7LHH\n6G1y9erVXfgYAXwNX/hNHYp7zZo1uHPnDvbu3QuJRIK5c+c6PeCSJUsAAJGRkR2eW7lyZReHGSBA\n53EaeV+3bh0mTJiAX/3qV1izZo0nxhTAz6ioqICDqZ3bcCpuiUSC4cOHY+LEiX6TLRbAs1y7dg11\ndXUeP6/TJIFZs2ZhxYoV4DgOM2bM8MSY7ko0Gg0+/vhjAO3+6oABA8CyLNasWUPLMlRWVkKn06Gh\noQGvvPJKp89hNBqxZs0ahISEIDExkUasTpw4gYMHD9r1jaurqy3CuNu2baOGznr+ZA9+LRVP4VTc\nAwcOxOTJk2EymaDRaDwxJq/yt7/9DbGxsbh58yYWL16Mzz77DFKpFA888ACOHz8OsVgMoVCI+vp6\nSKVSyOVycBwHuVwOpVKJsrIyNDQ0YMqUKTh06BAkEgl+97vf4Z133sHjjz9ud8/imTNnMGPGDAwY\nMAB79uzBM888A41Gg3nz5kGpVOL777+HTCaDVquFTqeDRqPBsWPHMHv2bJSVlWHdunWIi4vD6NGj\n7Z5DLBbjpZdewtq1ay0K9MTExNBkqB9//BEVFRVobGzE8uXLAQA//fQTFi5ciMrKSpw8eRItLS2Q\nyWSdKvLjDXE7dUs+/vhjmoA0cOBAT4zJqyiVSixcuBCRkZE4c+YMRCIR+vXrh9LSUshkMshkMhr7\n//3vf4+RI0dCr9eDYRjk5uZi0aJF6Nu3LyorK5Gamor09HTcuHEDAwcOpKIrKirC5s2bsXnzZmRm\nZgIADAYDDZWSJCSVSoXz589j3759WLx4MWpqarB06VJIpVIoFAqLtYKUlBS88MILyMnJAQBUVVXR\nc3z33XcA2i1waWkp3njjDYsEsgEDBlChHj58GGFhYRAKhThz5gw2btyIxsZGizCuyWTCsmXL0NLS\n4vL3SioHeBKn4g4PD4dGo4HZbPbKAD1NS0sLduzYgbq6OowdOxYajQaVlZXo378/rl27BpFIhObm\nZgDt8dv8/HzIZDKUl5cjKSkJn3/+Oe7cuYOoqCiaWstxnNP5SkpKCjIzM7F9+3bMmTMH//znP1FS\nUoLMzExwHIecnBzExMTg008/RZ8+faDRaLB//376/uzsbPz973/HlClT7J6jT58++Prrr/HRRx8h\nOTkZR44cQUNDg8VrUlNTUVFRAQAYP348lixZgoyMDGzZsgWHDh1CSkoKQkNDsXnz5k4ZO51O5/Jr\newxnySdLlizhfvjhB27nzp3crl273Jfl8h9ef/11t5/DEe+++65Xz+8qBoOBO3v2LMdxHFdaWsp9\n+eWXXh6RfdLT07lTp055/LxOfe6oqCgMGzYMSqXSE9ea1/nv//5vbw/BJcRiMcaOHQsAdMe+r6JS\nqXDr1i1MmDDBo+d1Km6VSoWffvqJ/v3cc8+5dUAB/I/w8HAUFRVBq9VCLpd77LxOxR0Qs2M0Gg3q\n6uoQHh4OhULh7eH4JP369QMAHDx4EA899JDHzutQ3N2pc3c3UFBQgJycHJhMJrAsi4kTJ/p0aWBv\nXYhSqRSzZ8/G3r170adPH0yZMsUj+1gdivvjjz8Gy7JISkpCWlqa02qi/Pztp59+Gj/88AOqqqog\nlUrxhz/8Ae+99x7UajUeeeQRhIeH99yn6CZd+dE1Gg1ycnLQ1taGtrY2AKDRC18QOPlMCoUCGo0G\nZWVlyM3NhdFohEKhQFpamke3vpHN2VlZWaitrcXcuXPdXgzfobiJS1JQUIAtW7ZAp9PhhRdesPv6\n69ev0/xtAMjPz8fq1auxbt06XLt2DVqtFmazuUeKOfYUxPqSu9SkSZPs/uj8i6Curg4mk4kKm5CT\nk4OhQ4d22jI6usDq6upw48YNxMbGWhgFe+8hn4lceCKRiKYfA+3hzr1796Jfv34eMTIVFRU4e/Ys\ngoODMWbMGFy4cAFVVVV4+OGHER0d7bbzOvW5V61ahXvuuQcTJ06ks3N78PO3AeChhx7Cp59+iqqq\nKkRFReG1115DcXExDhw4gIyMDPo+69xfT0GsLylGybIscnJyEBcX10Fg1hfBmDFjLHbiE0wmE8rL\nyzFkyBCXx2Hr2GSnUmFhIQoKCuhrY2NjER8fj8uXL9P6gkKhENHR0Rg7diy0Wi2ys7PBsixaW1tp\niWVr9Ho9Nm/ejLlz5/aYBWdZFgaDAQzDYPfu3fQ35RcAjY6ORlBQEK5du4bPP/8cEyZMwIwZM9zS\nu8el0g65ubn47LPPcOPGDYuFA2t27tyJ2tpamM1mJCYmoqqqChzHQalUIj4+Hnv27AHDMHjiiScQ\nFRVl8xirV6/2WP5vaWkp9u3b1+Hx6dOnQyaTQaFQoL6+Hk1NTcjOzgbQnkgmEAggEAgQFhaG/Pz8\nDu/v06cP0tLSoFar7bo6xOrK5XL88MMP0Ov14DgOer0eWq2WLvqYzWYIBAIwDOPSErZQKKS7plx5\nfd++ffGHP/yhyz44EbTBYKAXkUwmswgdL1q0CH/5y18s3jd69GicPHkSR44cgUqlwkMPPYTY2Ngu\njcEeTi33tGnTEB8fj0WLFiE5Odnha3/72986fH7o0KGdG10Pw7+NA6D1TviTG61Wi8zMTOpyWFtm\ngUAAuVwOhUKB8vJyCIXCDiJqbm7G/v37ERYWBpFIRF0djUaD8vJyXL9+HdevX4dQKERra6uFy0Dg\nH9O6zLEjOpvDodVqqW/uKizLwmg00q5pXUEoFCItLQ1Dhw7Fv/71L2zfvh3jx4/H7Nmze6zoj9Oj\nZGVl4fTp0/j888/x+uuvO91q5qvwb/06nQ4Mw0AsFkOj0UAoFEIul0Or1aKlpQUMw9hNNWBZFm1t\nbdBoNFAoFDaFR/IwmpubERwcjBMnTqCyshKXLl2iroKvwLKsS343sdB6vb7Dd6PX61FRUYHy8nKU\nl5ejsbHR6cYUfm7LiBEjoFKpcPbsWZSUlDhMr+7MZN2puFesWIHo6GgsXLiw1zYJ4vvWxBdlWZYW\ngAfai8jcuHHDZkF4W3Ac12EyaQ3J4BMIBKipqemUBfYUjkJyZrOZuhx8QVuL+c6dO90eQ1xcHEJC\nQnDq1Clcvny5R8qHOBX3hg0bun0Sb1NXV0eFZTKZqKtBfFmWZXHx4kUIBIIeTw7jOM6nLLU1DMNY\nuCVmsxl6vR4Gg4GOW6fTobKyEmVlZaioqLAp5uDgYERFRSE6OhqDBw/u0ljCw8MxePBgFBcXY/jw\n4d0OLvSOiobdJDw8HAKBACzLWvhzDMNYCJ3/2N1CW1sbpFIpNBoNFbROp7OwzGTtgk9wcDCio6Pp\nv6CgIPpcd0K9IpGox75/p+LOzs7GoEGD8MUXX2D06NF48MEHe+TEnkShUGDSpEk011mlUlHfmj+h\nlMlkaG1t9eZQPY5IJEJZWRnMZrNDMYeEhFDLbC1mgkAggFAo7NKEcNy4cbRPz9ChQ50GL1zB6Sgy\nMzMxa9YsjB07Fjk5Ob1S3ACQlJSEuLg4Gi0pLCxEdnY23dUiFAptRi38HZPJhD179nR4PCQkxMIy\nq1Qq+hzDMBAKhVTI5P+740a0trZi27ZtAGC3YFNncSrutrY27NixAytWrKAVX3srCoWC+pZSqRRi\nsRjNzc10Zu5swmcr7GcPV9wbgUBA4+akAZSnIZ/HnpgZhoFIJKIiJv/tSfR6PbZv3w6dTocnn3yy\nx0rtubRCSTrW+kvdERI94U8eGYaBUqm06ZaQUCEAl90WR8ImFxbDMNDpdDCbzdQ9sn6fRCKh4+xp\n8cfExCAhIYGKWSAQdBCxuyse6PV6Gsl68skncc899/TYsR2K+5///CcAQC6XY+/evUhKSsKiRYt6\n7OTeQKPR4PLlyzCZTDCZTB0WS/i3VvJ3WFgYBAJBj2yQJjF0g8GA4OBghIWF0b/5exKJ0Mn8QCQS\nWUQwuotAIEBraytkMhmioqIgEok83nHCbDbjzJkz0Ov1+NOf/tTjfXgcOklHjx7Fww8/jPPnz+PD\nDz/EjRs3evTknqagoABfffUVcnNzUVdXh5aWFipmjuOoeDmOA8dxEAgEUCgU0Gq1tH9kZ/1KiURi\n4Y8SAfEvFolEAolEAoVCYfFagUBAs/h6OkQpEAggFotx7tw5GI1Gjwt73LhxKCsrQ0tLCxYuXOiW\nBlMOLbdIJILZbKYfvDdvEOYv5Oj1erqgA8Ai3s0wDA0bkvfxX8fP3SAXgPUSPR+ZTAa5XA6dToe2\ntjYoFApIpVI0NjYCaP9OietBREYSsjiOo1EdsVgMiUTidNLraiiTfA8CgaDTy+89wbVr15Cfn4+H\nH37Y5RqUncWhGRo9ejSWLVuGX//61/jkk0+QmJjolkF4ArKQQ5bPiaDJf/n/gF+sN1+0JKeC4zhI\npVKaHEQsvC1EIhEEAgHtDCyXyyESieh7SVxXKBTSY1iLk/tPQyu+sG1ZWoZhEBQU5FIOD8uy1A3y\ndG49x3HYv38/Bg0ahFGjRrntPA4t9+9//3s88sgjkEgk4DgO8fHxDg/maLPC3LlzPVJ83h5kIYek\nX/KFzbfYfMtpD4ZhqPUn77OVeC+VSmmDLJFIhDFjxqC8vBwsy9LYe1hYGOrr63HixAla5tkRxGVJ\nT09Hbm4uTSUAfqmhXlpa6tJ3wrKszfRed1NTU4P6+nr85je/cas75FDcpEVIRUUFTCYTvvnmG7z5\n5pt2X+9os4J11aKeimW6ChETSV0F2sVH8o+VSiXNpQB+EZEtsRGrzsdelCUlJQVisRhqtRoSiQSt\nra0oKytDW1sbSktLkZOTQ10UZ/BdjlOnTkEsFiMiIgJ6vZ66LwKBwGnOC0EgEHil0FJpaSn69u3r\nNneE4FDc9fX1mDNnDjZs2ICXXnrJaSjQ0WaFiIgIl4vPuwuykHPmzBkUFhaivr4eDMNApVLRrECS\nNG82m10K+9nKHSF+uEajwcGDByEQCGiRHlKqzBryHqlUajcyY72a2tzcDKlU2uUFlKCgILfthOH/\npvzNCiaTCTU1NUhPT3f7JNahuPV6PcrLy+kt19mPTYrNBwUFISsrC1VVVVAqlZg6dSqSk5N9ovi8\nQqHAtGnTEBUVhf3791OfmPjiwcHBkEgkFj5pZ+Fbe3KMpqYmi9eEh4ejb9++9PslE1WWZfHwww/j\nxx9/pFWadDqdhb8tl8shkUigVCot8mWIe+FKPJzkU7vLJeH/pvzw8Z07d8CyrEdy+x2Ke+HChdi2\nbRv+8pe/4JNPPnHaJMjZZgVfWgSKjo6GTCazyBYEQIVCNir0JFKpFGlpaYiNjUVISAhu376Nqqoq\nC8tLRD558mQa3VEoFIiPj8e5c+foiqbBYIBMJsO8efOg0WiQl5eHgoICiyiQPQQCAUJCQqBWq3v0\n87lCVVUVVCpVjy7W2MOhuI8cOUIFuXjxYrcPxpPwk6lYlqWWEADtStwT8KMxQUFBiIqKQlRUFC5d\nuoQTJ05Qy65UKiGXyyEQCBAeHo4BAwYgLi4O5eXlANovxrCwMPz888/0DhoUFITKykrExcWhpqYG\nffr0QUNDA0QikcOwLbkAPB0lYVkWdXV1HtsX4FDcxcXFHWbeAwYMcOuA3A3xkc1mMwYPHgy1Wo26\nujoEBwfj9u3bOH/+vMVeQFJ91VWIiBMTE3Hu3Dn6uFKphFQqRWRkJLRaLXJycuhEluxSl8vlmDRp\nEnUVSkpKOmwclslkdJlcIBAgJycHUqmUCrYzmy08TWNjI3Q6Xac2T3cHh+K+ffs2XYIn9KYKVCzL\nWiyzk+I5fIRCIb09x8fHIyIiAoWFhbh69Srd2GAP6wUToVBI5xijRo2CWq1GTk4OzdUgwi0tLaXj\nkMvlkEqlMJlMmD59Ov3ha2trcfToUboZmWVZnDp1iiZb8T8jACp4Mi5+vN4W3li8uXPnDqRSqduj\nJASH4k5ISOgVYuZbY76QHQlTq9WisbERffr0QUtLC4qLi1FWVkYnPPaQSCR49NFHUVZWhgsXLlC3\nIzY2FnFxcYiJiaGCGTduHIYOHdqhtgh/8wQAizi5RqNBSUkJjhw5QiehxGUB2if5JKLCsizMZjPC\nwsKoi0XuBHK5nMbi+d8DET3DMB53S2praxEbG+ux8h0Oxb1s2TKPDKIzEBFbi9lVOI7DhQsXcOHC\nBbo/sDO3aIPBgJqaGowaNQpJSUloaWmBWq22awH5abb8x/j+PtmwfOjQIfp5yOIP2fNJQotyuRz1\n9fUQi8UwGo1QKpXYtWsXJk2ahIyMDNy4cQORkZHgOA4KhQKVlZXYv38/tFqtxcU0efJkj1ptlmXR\n1NTUI5sQXMWhuEkBQ29ArBJfwCQ1tDNwHIfm5ma6y4QsoFhj7WKQuLOt85nNZvTp0weA7aVrV8qz\nkZh7eXk5srKyoNVqUV9fT31nEskhlrelpQXBwcHUjWloaEBoaChEIhFYlkVWVhYkEgldyBkzZgzU\najXi4uIwe/ZsZGdn0w4QkydPdlpgqachK6me1JRDcX///fcIDg7GvffeSx9z9y3FZDKhoaGhy7nL\n1mIuLy+3GZ8n+cr8uht8Ids6PxGOoxhtZ8qzKRQKKmJywZGLTKvV0jGSY/GtOX/iSGLyJEbf1tbW\noW7KggULvFqNlrhYtvqTuguH4i4tLcXhw4eRlZVFH3N3L0pX4rR8OI5DU1OTxYZWazEzDAO1Wo2o\nqChERETg3LlzVBxNTU02swHJ+/giEgqFGDt2rF1ftTPl2Qjh4eEdzknGQbIOVSoVtFotzSAkE0fy\nXxL2EwgE0Ov19POTethkDN6MdJHccU/WiXQo7qVLl+KJJ57Av//9b5c7CLsbIma+ZbZ2M4iYyZap\nfv36dahFd/78eZrNRywlWYo3Go3Q6XQQCoU0XMcwDGbNmtXBClsXx7S+MEls15FPPnHiRIsydUFB\nQdBoNAgKCoJYLKaptuQzWCdhkXxw0piJzEE0Gg20Wi2USqVX0lr5KBQKt+RsO8LpNrP169fjySef\nRGtrK9asWePxLsIcx6GxsRHl5eXUOluLWSAQICIiwqGY+cTHxyMmJgatra1Qq9Uwm83Izs5GRUUF\ndbs0Gg3dO0jCcXFxcRbHsVXA0tr6k0UZR5AqSvywYUJCAhWvQCDAjBkzLDY4k9LERLQ7duywyOHg\nQ/LIvQmpvuVJnIqbdBAGYHOXtDtoaGiwsMzWSUQCgYBa5qioKKdi5kOsXFhYGBiGsRAowzBISEhA\ncnJyhwUU/uIKYNsFycvLw5gxY5CXl2f3ffz384VqK2xoa2LKPxaJxJA2gmQzA4mukEQrmUyGs2fP\nYty4cV6ri24rYc7duNRBmDTbdNZB2Dqfe+fOnaipqcGtW7fwxhtvuFR8Pjc3l06cCHwxE8vcmcLl\nZBcLycsgWAuUYRgUFRUhOTm5QykIWzWzyfzAZDJR/1etVuPxxx93OHmzN+m0DhvaCiPaEjyJmwOg\n2Y1Au3tDxH7mzBnk5uZizJgxmDNnjsvfXU/hjWJHLlV5tdeR1hrrfG6gfZas0Whw/fp1l4rPGwwG\nKub+/ftTy9zZKvxEzCQ8ZgtnPrItcRFIf87W1laL3BEiOnvvc3XSaUvEji4K66JDpLaIVqu1uKDz\n8vIwYcIEj1vw1tZW33NLOoN1PndBQQFWrVqFTZs2ITIy0qXi8yEhIfjzn//cpZYSrgiaD7F4ZIme\n+NjOfniNRoPdu3dbpLGSuLij99TV1dmsT2I96bQl4ri4OItinjqdDocPH6bdEazvNACwe/du3Llz\nh67gkrnDjRs33C5u63zukSNH+p7lXrlyJaZNm4apU6c6bbNmnc89cuRIbN68GU1NTaiqqsLevXtp\n8Xk+/NzflStXdkrYnRU0H4VCgejoaOTl5dHHxowZY9ffBdqFd/z4cVRXV1sci+RV24pK8MVK9kPy\nXS/+pJNfU4VccPzkKFJmmVwg27Ztw3333dfBrdFoNGhubrb4TkgUxROxZut8bp90S1599VWcPHkS\nH3zwARobGzF58mTMnj3bpmvhLJ972LBhXR/pfyA7wfmFbboKKQZPaoeIRCKUl5fj7NmzHSaFpHh8\nTk6OzXRYhmFgNBo7WERbfr31Sih/0llXV2fRRAqARZcC/p5JALRNiLVbU1dXRwvla7Vaek6RSITM\nzEyHi0vuwufEnZeXh9zcXDQ2NiI2NhahoaF45ZVX8P7773tifADaBUGsc3cFzYf43PxMO4PBgOPH\nj1skJxGfmLye3CWsf6yJEyfanXjykclktDWJ9Z1BLpd3CHW2tbUhLCwMCQkJqKmpoY+TpC2DwdDh\njkFcrpCQECgUCjQ0NIBhGPTp08elxSV/wKm48/Pz8ac//Qn33HMPqqur0bdv3w7xXnfgLkHzCQ8P\nB8dxMBqNEIlEdKMt0L49TC6XU6tJXBSSWsq3iEKhEKNGjcLIkSNRWlrqMAMQaBdldHS0TWFptVoo\nFAqL5XilUgmNRoPk5GRcvnwZ9fX19Dmg3TWzvmPwJ5lkpVOpVFpsfPbkwg4ZgydxKO4VK1bgypUr\nuHHjBjiOQ21tLbZs2eLWfFyhUEhX5ty9gbSkpMSijJnRaLQQYmtrK/R6PVQqFRUsEUxISAjdWZOa\nmorKykp89dVXDiMZzmLfQHvZA35cX6FQQKlU0vNPmTKlw26cyZMnA0CHC4ufnPXzzz9bfJ+uLC71\nJD4n7nUubgKJAAASoUlEQVTr1uHkyZNISUnx1HgslpndCfGFie/e1NRks24IWdwh8KMSZOIJwGF4\nz1nMnIynvLwcZ86cgVQqhV6vp49PnTrVpmCB9u1nJSUlNi8soP3iGDJkiEXBSWcXmDvwOXFv2rQJ\nJSUl2LVrF33svffec/ugPIG1L2xv6ZoU1uHfwhUKBUpKSrB//366iGM0Gmk0iWXbmyPx+1E6in2T\naEpTU5NFNSzS8s46J4MIFnA9bu7KBeZOyEXlSRyK+9lnn0VOTg7kcnmH8FVvh+8LO9onSURPFm1I\n70i+oEghHGJxib+clZUFvV7vMCrBD/2RKAxZqtbr9XRhyB7WF6mtC4vg6AJzNwaDwePndnopff/9\n9xg2bBhGjBiBrVu3emBInoH4wvzqq/bq7yUkJNBb/759+/Dtt99aRDTIZM1oNNLHyUQ0JyfHwofW\naDQoLS2ljxFxmkwmmu4K/BI2S0hIcCgK/tI72fBAeqzzOw97G2+I22m0RCAQ4MKFCwAct3XrbWg0\nGoSEhCAjIwP19fX497//DY1GQzcvkLCZVCrF8OHDsXv3bou87qamJnonM5lMkMvlmDx5Mg4fPkwz\nCQHLqIS9lUf+5l7ikgQFBUEqlTrdlkUu0hMnTti8sHwh3EfScPnxek/g1HL/z//8D65fv47Lly/7\nVFGd7kDqdO/btw+7du2CXq/Hvffei/DwcKhUKohEIhofTk1Ntdh/CPxiqTUaDbWUOp0Oer2ell4g\n7gHwi0tjyzcGgEmTJllUflWpVPTczoRJLtLU1FRazJ7v+9fV1fX499dZyMqoT1luMqEkt8iCgoJe\nP6G0J7LHH3+8QxSEn3pqHauWy+Uwm810Z4xAIKApr8ePH6eFKYOCglBSUoKQkBC7OSW2IjCuTPqs\nl/WtIxKeDvfZg+wUstUBzZ04Le0wZswYTJw40VPjcTuOMgEHDBhgsfhCsBWrHjJkCK5evWqRo8yy\nLOrr6zvsh8zJyUFGRobDjQz8yZ4rguzssr43IeL2qazAQ4cO4eLFixZF53u75ba3YuhMULay7oqK\niiyOw3Ecrl69So8J/NLEtL6+nl4Q5Pmuio/f14dvqR0t63sTk8lk0TTLUzgU95IlS2iPRldwtFnh\nxRdf9GrxeUJnVwyt38t/nS1rXlhY2OF9Go2GbrLmOA6JiYlITk7ukviIK2IymVBfX29RsMfRsr43\nMZlMCAoK8nhAwqG4s7Oz6UoYwVEFKkebFbxdfJ5PTy1o2LPmpOoTgW8c+Lt9OgvfFSGTWnJn4Jdr\n8zXMZrNFk1ZP4VDcGRkZ+OKLLzBixAhoNBpUVFQ4PJijzQok4mBrL52nis/z6akFDXvWnORfx8fH\no6SkxOI9XU1asp4vkAI948ePx7Bhw3xK2PzfVKPReGUB0Gmc++bNm3jmmWcAtIcFHeFos8KCBQt8\novi8K7hSMcoetqz5zZs3O+3j28LWfEEkEvmcsAHL3/TXv/51l/rBdxenZ1QoFPjoo4/oFiVH9Kbi\n8wRrIXemYpQ9nPnmjtwHRxdWd+YL3oTjON8U94oVK1BcXIyWlhZaX8NfsFV3hOzAAVyrGOUKrvr4\nrlxY3k6A6greErdTB/ett97C9u3bUV9fj7///e+eGJNHsLWYc+rUqQ4dCbqzysfPI1EoFBZxdFfG\nY52XQnB2LF+D7C/1NC7llkRGRmLWrFk4duyYJ8bkEWwt5ggEArpLnP9YV/zjzro3XSnF1lvwWcsd\nFxeHEydOYPny5QgNDfXEmDwCP5uOQMJp5HFnFaP42X3Wz1mXYThx4oTN1zoaj68sn3cXn/W5H3nk\nEahUKqhUKkydOtUTY/II9iZnSUlJNrsh8HFmlYkV1mq1FvHuM2fO2C1w1Fsni65A9pl6Gqfifvnl\nl3HfffdBp9Nh5cqVWLdunSfG5RHsTc66WzGKbDy23sV+9epVhyuTvXGy6Ao+K261Wo0HHngAAHDx\n4kW3D8jTdHYxxxXfWKFQIDExESdOnKCv4e+id3Q+b+6WcRc+K+7CwkI8++yzkEgkuHPnDlasWAGG\nYXp9AlVXcTXxKjk5GVevXoXBYLAog+wPPnRn8Vlxr1+/HkqlEseOHUN8fDwGDx7siXH5LK76xgqF\nwqILsD/50J3FZ8X9/vvvIyMjA4WFhfj555/9yufuKq76xo5e150l/t6IT4qbYRhkZmbimWeewUcf\nfeSJMfUKXPWNbb2uJ5b4exv2qgu4E6finj9/PqqqqqDRaJz2xLHO5/7uu+9QVVWF8vJyLF++HJs2\nbXJafN7fsVfF1Rc28roLgUBg0TXOY+d19oKxY8eipKQECQkJSE1Ndfja69ev4+mnn6bFGufPn495\n8+Zh5MiRqKmpcan4vL9DqriSjcVkW5ovbOR1F6QCrqdxadno9u3bLh3MOp8bAL788ku8+OKLaG1t\ndan4vKerEnkae1Vc/c1q839TlmV903KXlZVh5syZuHLlis261Hys87lbWlpopdbq6mps3LgR+/bt\n67Dh+LHHHsPOnTuxc+fOXuF78pfeHS3D24K0zuNDykT4E/zflLTy9jROLfcnn3yC4OBgxMbG4u23\n38abb75p97W28rlffPFFAMDQoUMddt7tLfAng6Rnu1QqdXlvZHh4OJRKJaRSaadalfRmGIbpkG3p\nCZxabqlUCqVSifj4eL+zLp3FOiGqtbUVLS0tNKx34sQJ/OMf/3BYxozEyUUiEa154q74d2fvKu7C\nWxNKp5Z75syZ2LFjB5599lk88sgjnhiTz8JfeieWiOSQkHmGXq/H0aNH0a9fP8jlcpuxbE/kkPhS\nuJFhGFp1ypM4FXdNTQ1mzZoFAB0mQv6Io8UV/tI7SeEkjUyB9okTabK0bds22rvHlri6mkPiyuJP\nV3rQuxNvuSVOxU06XzU0NODw4cP4zW9+4/ZBeQtn1o6/9A601/Szzv4jQtdqtdDpdLQzW0+Iy1Vr\n7GsbH3xW3CSyYTababVXf6SrRdyB9jzt/Px8NDc3A4BFZwSTyQSJRNJtcdXV1eHIkSMQCoUdGlE5\nusMQvJm05bNuyauvvtr+QpEIaWlpbh+Qt+iMtbN2KaZNm4akpCR88803NIeCiJtfmrir4iooKMDR\no0dpU1dSZcrR+Hxp44PP7sR58MEHERsbiy+++MIrM15P0R1rR+LdkyZNorvnSYtqoN16d6cuYE5O\njsXiFr/KlL3x+dLGB5/dILx//37MmjULY8eORU5ODh588EFPjMvjdNXa2SoPoVarER4ejsLCQuTk\n5EAoFOLcuXOQSCSdjljwe2Xyy7SZzWakpaX1io0PHMd1qd15d3Eq7ra2NuzYsQMrVqzAwYMHPTEm\nr2HLn7Zuf8fHlp+el5eHxx9/HEB7g1ryo3Y1YsG/o5DyaSzL4rHHHus1Cz8+65asWrUKzc3NEIvF\n1P/2Z4i1cyUy4chPJ/9v67nOiNv6jkIWfXqLsIH2z+2Tlvuzzz7D/fffjw0bNmDAgAF44403PDEu\nr2LLImdnZ0MqlVqUCLbnpysUCtTX11vEwMlzXRGlL/nPXcFn3RKj0YgrV67ghRdewJ49ezwxJq9B\nFkh0Op2FYEmJhv3790MmkznsDhwdHY1du3bRnjgcx9E+Od2JWPiK/9wVfHZCqVAokJ2djfj4eKed\nfR1tVli6dKlPFJ+3h3V/GdJ3k2VZOokTiUQdfGe+VZXL5RZdz0jZ3unTp/tkUXhP4ZOW+9KlS0hO\nTkZycjJaW1sxZcoUhwezLj4/f/58VFZWIisry2HxeW/nczvqL0NW1pRKpc32e8AvVrW0tNTmdiqZ\nTHbXCds6n9sb4naopPr6elRXV1v8c4S9zQrz5s2jfdVtFZ/3dj63rYkh6S8ze/ZshIeHW/Rzsec7\n+3NJtM7C/00B+J64q6qqcODAAdrbPDMz0+HBHG1WuO+++7BlyxYcOnQIKSkpPfohuos9UUZHR2PI\nkCFITU11qX4g8cFdee3dhjd8bobj93ez4vTp08jOzsbkyZMBAH379sXAgQPdOqDVq1d7pbuCs9Bf\nZ0ox3G1lG5yRnp6OTZs2WXTF8wQOL6eCggLcf//9GDJkCHUl8vPzkZubi6efftojA/QUzsJtnYlW\n9ObIhrvwuQnlwoULsXfvXmzZsoWukI0fPx4LFy701Pg8SkCU7sPnivKIxWLMnDkTAwYMoBOuCRMm\neGRgAfwLb0TBnJ5x1apVKC4uxq1bt1wu8RAggDXeELfTKeyvfvUr2nItQICu4pPivnr1Kp5//nla\nJepuLV0coHv4pLg3bNiA6upqcBznsGRBgACO8ElxL1++HG1tbRg8eDCamppol4UAAVyFYRiv1Id0\nqW1IamoqQkJCkJ2d7YkxBfAzJk2a5JVEOZda9anVamzduhV37tzxxJgC+BkkO9LTOLXcU6ZMQWFh\nIf76179i0KBBHhhSgAA9g1PLvWnTJuTl5UEgEODtt992+No9e/Zgy5Yt2LJlCwDg3Llz2LZtG956\n6y0YDAa89dZb2Lx5s1/Xog7gO/RoIUzr4vOHDx+GwWBAbGwsrl+/Hig+H8Cj9GghTOt8bo1Gg2XL\nluHDDz9EZGSkS8XnGxoaupUVWFBQ4HM1vu/GMTU0NKC0tBRAezapN3CY8goAN2/eRF5eHkwmExiG\nwbx58+y+dufOnaitrYXZbEZiYiJMJhPKy8tRV1eHOXPmYO/evWAYBk888QSioqJ6/MMA7TXCSYK8\nrxAYk3dwark//vhjzJ8/v8NOblvYKj7PZ9iwYZ0bXYAA3cCpuMPDw6HRaJxuDvYVfDEPJjAm7+DU\nLXn++eeRnp4OhmHAMIxflzAO4F84tdxRUVEYNmxYhyZFAQL4Ok7FrVKp8NNPP9G/n3vuObcOqLNY\n10qx/tsXxvTzzz+jqKgI99xzj9dar5SWlmLt2rVYu3YtlEqlT3xP7sZpnPu5556z+OdrWMfWrf/2\nhTGdO3eOlhz2FiqVCmq1mv7tC9+Tu+n1HU2tY+u2aqd4e0y/+93v8NRTT+HSpUte6YEOAKGhoQgN\nDbU7Rn/Ee6akh7CulUL+jo6O9pkx3bp1CwAQFBTk9Q7JLS0tOH36tE98T+7GabQkQIDeSq93SwIE\nsEdA3AH8loC4A/gtAXEH8Ft6fbTEVXbu3InIyEhUVVUhNTXVa2mYhM2bN+OPf/wjnnvuOSxatAg5\nOTlYvHixxWs++eQTzJ07F9999x2WLl3q0nFfffVVzJw5EzNmzEBBQQG2bt2K8+fPY9SoURg+fDiu\nXbsGnU4HhmEQHR2N5cuXu+Pj+QR3jbizs7PRp08fhISEQKfT4W9/+xtEIhEiIiIwa9YsbNy4EWKx\nGE899RQ+++wzrF+/Ht999x1iYmJQW1uL3NxcGAwG/Nd//RdCQkIAAJmZmcjOzoZer8frr7+ORx99\nFJMmTUJbWxveeecdfPTRR2hsbIRYLMZLL72E119/HSzLIiUlBSUlJSgoKEBZWRmkUimKi4tRWlqK\nDz74AEajEc8//zyKi4tx8uRJnDp1Cu+++y7mz58PAPjXv/5Fxf7999+joKAAOp0Of/zjH3Hx4kXa\nTjEpKQnr16/Hyy+/jLfeegsikQgvv/wy1q9f79UFJU/h/5/wP0yePBmRkZE4e/YsAODWrVvIyMjA\nkCFDaPH5Bx980GIVj/D1119j4sSJqK+vR1FREbZv346JEyeira0NYWFhSEtLg1AoRFRUFF577TX8\n3//9H86fP4+8vDyMHj0aN2/exNGjRzF+/HhkZGTg9OnTAIDhw4cjISGBpgJ/8803WLZsGRQKBV0a\nT0lJQVlZGaZPn45jx44BaN9AQjh58iQ2bNiAzMxMlJSUICEhwWn981dffRUMw+D+++9Henp6t79b\nX+WuETcfjuPw4osvoq2tDf/7v/+LlStX4oUXXsDRo0dRWFhIV+0aGhoQExMDhUKBJUuW4PTp04iK\nisKHH34IoH03i0KhwKeffkqL7BPkcjmGDRuGJUuW0D45pB2JTqezOS6DwQCBQACj0dihW/OoUaNo\n++0nn3yyw3vJheYKb7/99l1hue+aCWW/fv1oZwiGYfD111/jwIEDSExMhFarxcaNG3H58mUkJSVh\n+PDh+OCDD6iVnzJlCl555RUcOHAAkZGR9JhFRUXYunUr9Ho9oqKiUFRUhHfeeQfl5eVITExEU1MT\nVq1ahZaWFqSlpSE7OxsrVqyAwWCwOcZHH30U69evx5tvvomgoCAA7cvmubm50Gg0UCqVCA4Otlgy\nHzFiBFavXo1z584hLS3NXV9frySwQtmDLF++HOvXr3fb8b/66isMGTIEycnJbjuHP3HXWO7eTn5+\nPs6fP4/x48d7eyi9hoDlDuC3BCx3AL8lIO4AfktA3AH8loC4A/gtAXEH8FsC4g7gt/w/Wj/hH0XU\nEwgAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b937ed8d9e8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "g = sns.jointplot(data=fimo_grp, x=\"tf_ts\", y=\"tss_ts\", kind=\"reg\", size=2.5, color=\"gray\")\n",
    "g.set_axis_labels(\"tissue-specificity of TF\", \"mean(tissue-specificity of TSSs w/ motif)\")\n",
    "g.savefig(\"motif_corr.pdf\", dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
