{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 08__deletions\n",
    "# analyzing single-nucleotide deletions MPRA and preparing input for MIND\n",
    "\n",
    "in this notebook, i do two things: (1) i calculate the effect size of deletions (log2 foldchange between the deletion activity and the reference activity) from the deletion MPRA and (2) i compare those effect sizes to the number of TF motifs that are computationally predicted by FIMO to be gained or lost by the deletion. note that these FIMO predictions are done separately. \n",
    "\n",
    "------\n",
    "\n",
    "figures in this notebook:\n",
    "- **Fig 3B, S9A**: scatter plots showing the comparison between deletion effect sizes and motif disruptions\n",
    "- **Fig 3C, S9B**: specific examples of deletion profiles across a sequence (HOTAIR and DLEU1) and how they compare to motif disruption profiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from itertools import chain\n",
    "from decimal import Decimal\n",
    "from matplotlib import gridspec\n",
    "from random import shuffle\n",
    "from scipy import stats\n",
    "from scipy import signal\n",
    "from scipy.spatial import distance\n",
    "from scipy.cluster import hierarchy\n",
    "from scipy.stats import spearmanr\n",
    "from scipy.stats import pearsonr\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from plotting_utils import *\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "from del_utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "activ_dir = \"../../data/02__activs\"\n",
    "pval_dir = \"../../data/03__pvals\"\n",
    "index_dir = \"../../data/00__index\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pool2_hepg2_barc_activ_f = \"%s/POOL2__pMPRA1__HepG2__activities_per_barcode.txt\" % (activ_dir)\n",
    "pool2_k562_barc_activ_f = \"%s/POOL2__pMPRA1__K562__activities_per_barcode.txt\" % (activ_dir)\n",
    "\n",
    "pool2_hepg2_elem_activ_f = \"%s/POOL2__pMPRA1__HepG2__activities_per_element.txt\" % (activ_dir)\n",
    "pool2_k562_elem_activ_f = \"%s/POOL2__pMPRA1__K562__activities_per_element.txt\" % (activ_dir)\n",
    "\n",
    "pool2_hepg2_pvals_f = \"%s/POOL2__pMPRA1__HepG2__pvals.txt\" % (pval_dir)\n",
    "pool2_k562_pvals_f = \"%s/POOL2__pMPRA1__K562__pvals.txt\" % (pval_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pool2_index_f = \"%s/dels_oligo_pool.index.txt\" % (index_dir)\n",
    "pool2_index_del_f = \"%s/dels_oligo_pool.index.with_deletion_info.txt\" % (index_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "annot_f = \"%s/tssid_all_biotypes.txt\" % (index_dir)\n",
    "tfbs_f = \"../../data/04__coverage/pool2_n_motifs_map.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>element</th>\n",
       "      <th>rna_3</th>\n",
       "      <th>rna_4</th>\n",
       "      <th>rna_5</th>\n",
       "      <th>rna_6</th>\n",
       "      <th>rna_7</th>\n",
       "      <th>rna_8</th>\n",
       "      <th>rna_9</th>\n",
       "      <th>rna_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...</td>\n",
       "      <td>-1.790824</td>\n",
       "      <td>-1.506121</td>\n",
       "      <td>-1.253360</td>\n",
       "      <td>-1.425010</td>\n",
       "      <td>-1.616346</td>\n",
       "      <td>-1.290745</td>\n",
       "      <td>-1.357850</td>\n",
       "      <td>-1.364172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>TGGCCAATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGGGCTGG...</td>\n",
       "      <td>-0.654425</td>\n",
       "      <td>-0.593675</td>\n",
       "      <td>-0.604614</td>\n",
       "      <td>-0.724685</td>\n",
       "      <td>-0.759409</td>\n",
       "      <td>-0.652865</td>\n",
       "      <td>-0.591606</td>\n",
       "      <td>-0.501004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>ACGGGTGCCCACCTGGGAACAGGAGGCTTGGTTCCACGGCTGGGCT...</td>\n",
       "      <td>-1.210300</td>\n",
       "      <td>-1.264010</td>\n",
       "      <td>-1.127320</td>\n",
       "      <td>-1.101221</td>\n",
       "      <td>-1.304477</td>\n",
       "      <td>-1.178312</td>\n",
       "      <td>-1.065390</td>\n",
       "      <td>-1.429695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>AGTATCCGCTCTCCCCTGTCCCCTGCTATGGGCAGGGCCTGGCTGG...</td>\n",
       "      <td>0.469084</td>\n",
       "      <td>0.792547</td>\n",
       "      <td>0.496125</td>\n",
       "      <td>0.538051</td>\n",
       "      <td>0.669531</td>\n",
       "      <td>0.552507</td>\n",
       "      <td>0.692894</td>\n",
       "      <td>0.565142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...</td>\n",
       "      <td>TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...</td>\n",
       "      <td>-1.703193</td>\n",
       "      <td>-1.657239</td>\n",
       "      <td>-1.342610</td>\n",
       "      <td>-1.407414</td>\n",
       "      <td>-1.560008</td>\n",
       "      <td>-1.441266</td>\n",
       "      <td>-1.275676</td>\n",
       "      <td>-1.331183</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...   \n",
       "1  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...   \n",
       "2  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...   \n",
       "3  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...   \n",
       "4  CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...   \n",
       "\n",
       "                                             element     rna_3     rna_4  \\\n",
       "0  TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT... -1.790824 -1.506121   \n",
       "1  TGGCCAATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGGGCTGG... -0.654425 -0.593675   \n",
       "2  ACGGGTGCCCACCTGGGAACAGGAGGCTTGGTTCCACGGCTGGGCT... -1.210300 -1.264010   \n",
       "3  AGTATCCGCTCTCCCCTGTCCCCTGCTATGGGCAGGGCCTGGCTGG...  0.469084  0.792547   \n",
       "4  TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT... -1.703193 -1.657239   \n",
       "\n",
       "      rna_5     rna_6     rna_7     rna_8     rna_9    rna_10  \n",
       "0 -1.253360 -1.425010 -1.616346 -1.290745 -1.357850 -1.364172  \n",
       "1 -0.604614 -0.724685 -0.759409 -0.652865 -0.591606 -0.501004  \n",
       "2 -1.127320 -1.101221 -1.304477 -1.178312 -1.065390 -1.429695  \n",
       "3  0.496125  0.538051  0.669531  0.552507  0.692894  0.565142  \n",
       "4 -1.342610 -1.407414 -1.560008 -1.441266 -1.275676 -1.331183  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_elem_res = pd.read_table(pool2_hepg2_elem_activ_f, sep=\"\\t\")\n",
    "k562_elem_res = pd.read_table(pool2_k562_elem_activ_f, sep=\"\\t\")\n",
    "hepg2_elem_res.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>rna_3</th>\n",
       "      <th>rna_4</th>\n",
       "      <th>rna_5</th>\n",
       "      <th>rna_6</th>\n",
       "      <th>rna_7</th>\n",
       "      <th>rna_8</th>\n",
       "      <th>rna_9</th>\n",
       "      <th>rna_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAACAAAGTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.559995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAACAACGTG</td>\n",
       "      <td>-1.780485</td>\n",
       "      <td>-0.598085</td>\n",
       "      <td>-1.170739</td>\n",
       "      <td>-0.878212</td>\n",
       "      <td>-1.162959</td>\n",
       "      <td>-1.45008</td>\n",
       "      <td>-0.133118</td>\n",
       "      <td>-1.089508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAACAACTGG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.425661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAACAAGATG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAACAAGATT</td>\n",
       "      <td>-1.028853</td>\n",
       "      <td>-1.208253</td>\n",
       "      <td>-3.278023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.543835</td>\n",
       "      <td>-4.28303</td>\n",
       "      <td>-1.792000</td>\n",
       "      <td>-0.798199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       barcode     rna_3     rna_4     rna_5     rna_6     rna_7    rna_8  \\\n",
       "0  AAAACAAAGTC       NaN       NaN  0.559995       NaN       NaN      NaN   \n",
       "1  AAAACAACGTG -1.780485 -0.598085 -1.170739 -0.878212 -1.162959 -1.45008   \n",
       "2  AAAACAACTGG       NaN       NaN -2.425661       NaN       NaN      NaN   \n",
       "3  AAAACAAGATG       NaN       NaN       NaN       NaN       NaN      NaN   \n",
       "4  AAAACAAGATT -1.028853 -1.208253 -3.278023       NaN -2.543835 -4.28303   \n",
       "\n",
       "      rna_9    rna_10  \n",
       "0       NaN       NaN  \n",
       "1 -0.133118 -1.089508  \n",
       "2       NaN       NaN  \n",
       "3       NaN       NaN  \n",
       "4 -1.792000 -0.798199  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_barc_res = pd.read_table(pool2_hepg2_barc_activ_f, sep=\"\\t\")\n",
    "k562_barc_res = pd.read_table(pool2_k562_barc_activ_f, sep=\"\\t\")\n",
    "hepg2_barc_res.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>element</th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>rna_3_log2fc</th>\n",
       "      <th>rna_4_log2fc</th>\n",
       "      <th>rna_5_log2fc</th>\n",
       "      <th>rna_6_log2fc</th>\n",
       "      <th>rna_7_log2fc</th>\n",
       "      <th>rna_8_log2fc</th>\n",
       "      <th>rna_9_log2fc</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_6_padj</th>\n",
       "      <th>rna_7_padj</th>\n",
       "      <th>rna_8_padj</th>\n",
       "      <th>rna_9_padj</th>\n",
       "      <th>rna_10_padj</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "      <th>combined_class</th>\n",
       "      <th>downsamp_combined_sig</th>\n",
       "      <th>downsamp_combined_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP</td>\n",
       "      <td>-0.740530</td>\n",
       "      <td>-0.483852</td>\n",
       "      <td>-0.266400</td>\n",
       "      <td>-0.420682</td>\n",
       "      <td>-0.565798</td>\n",
       "      <td>-0.276644</td>\n",
       "      <td>-0.338944</td>\n",
       "      <td>...</td>\n",
       "      <td>5.301211e-02</td>\n",
       "      <td>9.182491e-06</td>\n",
       "      <td>9.165985e-01</td>\n",
       "      <td>2.859277e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.653376e-23</td>\n",
       "      <td>sig</td>\n",
       "      <td>not sig</td>\n",
       "      <td>sig</td>\n",
       "      <td>sig repressive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>TGGCCAATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGGGCTGG...</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP</td>\n",
       "      <td>0.395869</td>\n",
       "      <td>0.428595</td>\n",
       "      <td>0.382347</td>\n",
       "      <td>0.279643</td>\n",
       "      <td>0.291139</td>\n",
       "      <td>0.361236</td>\n",
       "      <td>0.427300</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.552552e-01</td>\n",
       "      <td>2.618122e-02</td>\n",
       "      <td>1.545699e-03</td>\n",
       "      <td>2.506378e-15</td>\n",
       "      <td>sig</td>\n",
       "      <td>not sig</td>\n",
       "      <td>sig</td>\n",
       "      <td>sig active</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>ACGGGTGCCCACCTGGGAACAGGAGGCTTGGTTCCACGGCTGGGCT...</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP</td>\n",
       "      <td>-0.160006</td>\n",
       "      <td>-0.241741</td>\n",
       "      <td>-0.140360</td>\n",
       "      <td>-0.096893</td>\n",
       "      <td>-0.253929</td>\n",
       "      <td>-0.164211</td>\n",
       "      <td>-0.046484</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>5.887921e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>not sig</td>\n",
       "      <td>not sig</td>\n",
       "      <td>not sig</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>AGTATCCGCTCTCCCCTGTCCCCTGCTATGGGCAGGGCCTGGCTGG...</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP</td>\n",
       "      <td>1.519378</td>\n",
       "      <td>1.814816</td>\n",
       "      <td>1.483085</td>\n",
       "      <td>1.542379</td>\n",
       "      <td>1.720079</td>\n",
       "      <td>1.566608</td>\n",
       "      <td>1.711800</td>\n",
       "      <td>...</td>\n",
       "      <td>9.008623e-25</td>\n",
       "      <td>3.404805e-34</td>\n",
       "      <td>3.490157e-28</td>\n",
       "      <td>2.103572e-30</td>\n",
       "      <td>2.605472e-31</td>\n",
       "      <td>2.232976e-230</td>\n",
       "      <td>sig</td>\n",
       "      <td>sig active</td>\n",
       "      <td>sig</td>\n",
       "      <td>sig active</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...</td>\n",
       "      <td>TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...</td>\n",
       "      <td>CONTROL_SNP_INDIV</td>\n",
       "      <td>-0.652900</td>\n",
       "      <td>-0.634970</td>\n",
       "      <td>-0.355650</td>\n",
       "      <td>-0.403086</td>\n",
       "      <td>-0.509460</td>\n",
       "      <td>-0.427165</td>\n",
       "      <td>-0.256770</td>\n",
       "      <td>...</td>\n",
       "      <td>1.095963e-01</td>\n",
       "      <td>8.470045e-04</td>\n",
       "      <td>3.288456e-01</td>\n",
       "      <td>5.823235e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>7.097918e-23</td>\n",
       "      <td>sig</td>\n",
       "      <td>not sig</td>\n",
       "      <td>sig</td>\n",
       "      <td>sig repressive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...   \n",
       "1  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...   \n",
       "2  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...   \n",
       "3  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...   \n",
       "4  CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...   \n",
       "\n",
       "                                             element           oligo_type  \\\n",
       "0  TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...  CONTROL_BUT_HAS_SNP   \n",
       "1  TGGCCAATCTACTCCCAGGAGCAGGGAGGGCAGGAGCCAGGGCTGG...  CONTROL_BUT_HAS_SNP   \n",
       "2  ACGGGTGCCCACCTGGGAACAGGAGGCTTGGTTCCACGGCTGGGCT...  CONTROL_BUT_HAS_SNP   \n",
       "3  AGTATCCGCTCTCCCCTGTCCCCTGCTATGGGCAGGGCCTGGCTGG...  CONTROL_BUT_HAS_SNP   \n",
       "4  TTGAAGTCCAACTCCTAAGCCAGTGCCAGAAGAGCCAAGGACAGGT...    CONTROL_SNP_INDIV   \n",
       "\n",
       "   rna_3_log2fc  rna_4_log2fc  rna_5_log2fc  rna_6_log2fc  rna_7_log2fc  \\\n",
       "0     -0.740530     -0.483852     -0.266400     -0.420682     -0.565798   \n",
       "1      0.395869      0.428595      0.382347      0.279643      0.291139   \n",
       "2     -0.160006     -0.241741     -0.140360     -0.096893     -0.253929   \n",
       "3      1.519378      1.814816      1.483085      1.542379      1.720079   \n",
       "4     -0.652900     -0.634970     -0.355650     -0.403086     -0.509460   \n",
       "\n",
       "   rna_8_log2fc  rna_9_log2fc           ...               rna_6_padj  \\\n",
       "0     -0.276644     -0.338944           ...             5.301211e-02   \n",
       "1      0.361236      0.427300           ...             1.000000e+00   \n",
       "2     -0.164211     -0.046484           ...             1.000000e+00   \n",
       "3      1.566608      1.711800           ...             9.008623e-25   \n",
       "4     -0.427165     -0.256770           ...             1.095963e-01   \n",
       "\n",
       "     rna_7_padj    rna_8_padj    rna_9_padj   rna_10_padj  combined_padj  \\\n",
       "0  9.182491e-06  9.165985e-01  2.859277e-01  1.000000e+00   1.653376e-23   \n",
       "1  1.000000e+00  1.552552e-01  2.618122e-02  1.545699e-03   2.506378e-15   \n",
       "2  5.887921e-01  1.000000e+00  1.000000e+00  1.000000e+00   1.000000e+00   \n",
       "3  3.404805e-34  3.490157e-28  2.103572e-30  2.605472e-31  2.232976e-230   \n",
       "4  8.470045e-04  3.288456e-01  5.823235e-01  1.000000e+00   7.097918e-23   \n",
       "\n",
       "   combined_sig  combined_class  downsamp_combined_sig  \\\n",
       "0           sig         not sig                    sig   \n",
       "1           sig         not sig                    sig   \n",
       "2       not sig         not sig                not sig   \n",
       "3           sig      sig active                    sig   \n",
       "4           sig         not sig                    sig   \n",
       "\n",
       "   downsamp_combined_class  \n",
       "0           sig repressive  \n",
       "1               sig active  \n",
       "2                  not sig  \n",
       "3               sig active  \n",
       "4           sig repressive  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_wt_pvals = pd.read_table(pool2_hepg2_pvals_f, sep=\"\\t\")\n",
    "k562_wt_pvals = pd.read_table(pool2_k562_pvals_f, sep=\"\\t\")\n",
    "hepg2_wt_pvals.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>full_oligo</th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>tile_number</th>\n",
       "      <th>local_start</th>\n",
       "      <th>local_end</th>\n",
       "      <th>RE_count_1</th>\n",
       "      <th>RE_count_2</th>\n",
       "      <th>RE_count_3</th>\n",
       "      <th>SNP</th>\n",
       "      <th>barcode</th>\n",
       "      <th>element</th>\n",
       "      <th>unique_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>GATTCGTCGTG</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.2</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CGAGATAAGTT</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.3</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CTCGATTAAGG</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.4</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGAATAAGTCA</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.5</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>none</td>\n",
       "      <td>CCTAATCGACA</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          full_oligo            oligo_type  \\\n",
       "0  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "1  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "2  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "3  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "4  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "\n",
       "        oligo_id dupe_info                                           seq_name  \\\n",
       "0  1.2.0.0.0.0.1      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  1.2.0.0.0.0.2      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  1.2.0.0.0.0.3      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  1.2.0.0.0.0.4      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  1.2.0.0.0.0.5      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand                        ...                          \\\n",
       "0  65187718.0      -                        ...                           \n",
       "1  65187718.0      -                        ...                           \n",
       "2  65187718.0      -                        ...                           \n",
       "3  65187718.0      -                        ...                           \n",
       "4  65187718.0      -                        ...                           \n",
       "\n",
       "   tile_number  local_start  local_end  RE_count_1  RE_count_2  RE_count_3  \\\n",
       "0          2.0         94.0      208.0           1           1           0   \n",
       "1          2.0         94.0      208.0           1           1           0   \n",
       "2          2.0         94.0      208.0           1           1           0   \n",
       "3          2.0         94.0      208.0           1           1           0   \n",
       "4          2.0         94.0      208.0           1           1           0   \n",
       "\n",
       "    SNP      barcode                                            element  \\\n",
       "0  none  GATTCGTCGTG  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "1  none  CGAGATAAGTT  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "2  none  CTCGATTAAGG  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "3  none  TGAATAAGTCA  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "4  none  CCTAATCGACA  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "\n",
       "                                           unique_id  \n",
       "0  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "1  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "2  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "3  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "4  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index = pd.read_table(pool2_index_f, sep=\"\\t\")\n",
    "index.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>n_tfs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  n_tfs\n",
       "0  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...     16\n",
       "1  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@HBB...     10\n",
       "2  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...     14\n",
       "3  CONTROL_BUT_HAS_SNP__control.noflip.NA__p1@S10...     16\n",
       "4  CONTROL_SNP_INDIV__control.noflip.NA__p1@HBB__...     16"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tfbs = pd.read_table(tfbs_f, sep=\"\\t\", header=None)\n",
    "tfbs.columns = [\"unique_id\", \"n_tfs\"]\n",
    "tfbs.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. add required deletion information to index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mapping 3964 unique deletion sequences\n"
     ]
    }
   ],
   "source": [
    "# first create a dict of del seq : wt seq map\n",
    "index_dels = index[index[\"oligo_type\"].str.contains(\"DELETION\")][[\"oligo_type\", \"tile_name\", \"element\"]].drop_duplicates()\n",
    "print(\"mapping %s unique deletion sequences\" % (len(index_dels)))\n",
    "index_wt = index[index[\"oligo_type\"] == \"WILDTYPE\"][[\"oligo_type\", \"tile_name\", \"element\"]].drop_duplicates()\n",
    "index_wt_w_snp = index[index[\"oligo_type\"] == \"WILDTYPE_BUT_HAS_SNP\"][[\"oligo_type\", \"tile_name\", \"element\"]].drop_duplicates()\n",
    "index_fl = index[index[\"oligo_type\"] == \"FLIPPED\"][[\"oligo_type\", \"tile_name\", \"element\"]].drop_duplicates()\n",
    "dels_dict = {}\n",
    "for i, row in index_dels.iterrows():\n",
    "    if \"WILDTYPE_BUT_HAS_SNP\" in row.oligo_type:\n",
    "        seq = index_wt_w_snp[index_wt_w_snp[\"tile_name\"] == row.tile_name][\"element\"].iloc[0]\n",
    "    elif \"WILDTYPE\" in row.oligo_type:\n",
    "        seq = index_wt[index_wt[\"tile_name\"] == row.tile_name][\"element\"].iloc[0]\n",
    "    elif \"FLIPPED\" in row.oligo_type:\n",
    "        seq = index_fl[index_fl[\"tile_name\"] == row.tile_name][\"element\"].iloc[0]\n",
    "    dels_dict[row.element] = seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finding deletion numbers...\n"
     ]
    }
   ],
   "source": [
    "# add deletion number to index\n",
    "print(\"finding deletion numbers...\")\n",
    "index[\"del_num\"] = index.apply(get_del_num, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finding deletion bases...\n"
     ]
    }
   ],
   "source": [
    "print(\"finding deletion bases...\")\n",
    "index[\"del_base\"] = index.apply(get_del_base, seq_map=dels_dict, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>tile_number</th>\n",
       "      <th>local_start</th>\n",
       "      <th>local_end</th>\n",
       "      <th>SNP</th>\n",
       "      <th>element</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>del_num</th>\n",
       "      <th>del_base</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1,1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGTTCCAGAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>11</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGTTCCAGAAAAAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>13</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGTTCCAGAAAACGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>14</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.15.1.1</td>\n",
       "      <td>1.2.0.0.15.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>none</td>\n",
       "      <td>TGTTCCAGAAAACACAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>15</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        oligo_type        oligo_id                  dupe_info  \\\n",
       "0             WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1                1.2.0.0.0.0   \n",
       "80   WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1,1.2.0.0.12.1   \n",
       "106  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1               1.2.0.0.13.1   \n",
       "132  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1               1.2.0.0.14.1   \n",
       "158  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.15.1.1               1.2.0.0.15.1   \n",
       "\n",
       "                                              seq_name  \\\n",
       "0    Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "80   Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "106  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "132  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "158  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                             tile_name    chr  locus_start  \\\n",
       "0    Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "80   Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "106  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "132  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "158  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "      locus_end strand  tile_start    tile_end  tile_number  local_start  \\\n",
       "0    65187718.0      -  65187510.0  65187624.0          2.0         94.0   \n",
       "80   65187718.0      -  65187510.0  65187624.0          2.0         94.0   \n",
       "106  65187718.0      -  65187510.0  65187624.0          2.0         94.0   \n",
       "132  65187718.0      -  65187510.0  65187624.0          2.0         94.0   \n",
       "158  65187718.0      -  65187510.0  65187624.0          2.0         94.0   \n",
       "\n",
       "     local_end   SNP                                            element  \\\n",
       "0        208.0  none  TGTTCCAGAAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTG...   \n",
       "80       208.0  none  TGTTCCAGAAACAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...   \n",
       "106      208.0  none  TGTTCCAGAAAAAGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...   \n",
       "132      208.0  none  TGTTCCAGAAAACGCAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...   \n",
       "158      208.0  none  TGTTCCAGAAAACACAGCAAACAGAGGCCCTTGGATTCCGGGGTGG...   \n",
       "\n",
       "                                             unique_id  del_num del_base  \n",
       "0    WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...        0        X  \n",
       "80   WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...       11        A  \n",
       "106  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...       13        C  \n",
       "132  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...       14        A  \n",
       "158  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...       15        G  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_elem = index.drop([\"RE_count_1\", \"RE_count_2\", \"RE_count_3\", \"barcode\", \"full_oligo\"], axis=1)\n",
    "index_elem = index_elem.drop_duplicates(subset=[\"oligo_type\", \"seq_name\", \"dupe_info\", \"element\"])\n",
    "index_elem[\"dupe_info\"] = index_elem.apply(fix_dupe_info, axis=1)\n",
    "index_elem.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. merge with index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>full_oligo</th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>...</th>\n",
       "      <th>del_num</th>\n",
       "      <th>del_base</th>\n",
       "      <th>rna_3</th>\n",
       "      <th>rna_4</th>\n",
       "      <th>rna_5</th>\n",
       "      <th>rna_6</th>\n",
       "      <th>rna_7</th>\n",
       "      <th>rna_8</th>\n",
       "      <th>rna_9</th>\n",
       "      <th>rna_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>-0.410643</td>\n",
       "      <td>0.272279</td>\n",
       "      <td>0.094076</td>\n",
       "      <td>0.030652</td>\n",
       "      <td>0.060766</td>\n",
       "      <td>-0.446546</td>\n",
       "      <td>0.225989</td>\n",
       "      <td>-0.505224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.2</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>-0.188333</td>\n",
       "      <td>0.517627</td>\n",
       "      <td>-0.291708</td>\n",
       "      <td>0.394602</td>\n",
       "      <td>-0.126533</td>\n",
       "      <td>-0.024516</td>\n",
       "      <td>-0.566155</td>\n",
       "      <td>-0.180420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.3</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>0.618850</td>\n",
       "      <td>0.138554</td>\n",
       "      <td>0.149168</td>\n",
       "      <td>-0.166496</td>\n",
       "      <td>0.478421</td>\n",
       "      <td>-0.172385</td>\n",
       "      <td>0.246760</td>\n",
       "      <td>0.187198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.4</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>-0.743942</td>\n",
       "      <td>-0.886037</td>\n",
       "      <td>-0.789264</td>\n",
       "      <td>-0.849808</td>\n",
       "      <td>-0.179054</td>\n",
       "      <td>-0.370554</td>\n",
       "      <td>-1.570538</td>\n",
       "      <td>0.126095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.5</td>\n",
       "      <td>none</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>0.330004</td>\n",
       "      <td>-0.114620</td>\n",
       "      <td>1.211775</td>\n",
       "      <td>-0.214544</td>\n",
       "      <td>0.263450</td>\n",
       "      <td>0.284756</td>\n",
       "      <td>0.356498</td>\n",
       "      <td>-0.247038</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          full_oligo            oligo_type  \\\n",
       "0  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "1  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "2  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "3  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "4  ACTGGCCGCTTCACTGTGTTCCAGAAAACAGCAGCAAACAGAGGCC...  WILDTYPE_BUT_HAS_SNP   \n",
       "\n",
       "        oligo_id dupe_info                                           seq_name  \\\n",
       "0  1.2.0.0.0.0.1      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  1.2.0.0.0.0.2      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  1.2.0.0.0.0.3      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  1.2.0.0.0.0.4      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  1.2.0.0.0.0.5      none  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand    ...     del_num  del_base     rna_3     rna_4  \\\n",
       "0  65187718.0      -    ...           0         X -0.410643  0.272279   \n",
       "1  65187718.0      -    ...           0         X -0.188333  0.517627   \n",
       "2  65187718.0      -    ...           0         X  0.618850  0.138554   \n",
       "3  65187718.0      -    ...           0         X -0.743942 -0.886037   \n",
       "4  65187718.0      -    ...           0         X  0.330004 -0.114620   \n",
       "\n",
       "      rna_5     rna_6     rna_7     rna_8     rna_9    rna_10  \n",
       "0  0.094076  0.030652  0.060766 -0.446546  0.225989 -0.505224  \n",
       "1 -0.291708  0.394602 -0.126533 -0.024516 -0.566155 -0.180420  \n",
       "2  0.149168 -0.166496  0.478421 -0.172385  0.246760  0.187198  \n",
       "3 -0.789264 -0.849808 -0.179054 -0.370554 -1.570538  0.126095  \n",
       "4  1.211775 -0.214544  0.263450  0.284756  0.356498 -0.247038  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_barc = index.merge(hepg2_barc_res, left_on=\"barcode\", right_on=\"barcode\", how=\"left\")\n",
    "k562_barc = index.merge(k562_barc_res, left_on=\"barcode\", right_on=\"barcode\", how=\"left\")\n",
    "hepg2_barc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>...</th>\n",
       "      <th>del_num</th>\n",
       "      <th>del_base</th>\n",
       "      <th>rna_3</th>\n",
       "      <th>rna_4</th>\n",
       "      <th>rna_5</th>\n",
       "      <th>rna_6</th>\n",
       "      <th>rna_7</th>\n",
       "      <th>rna_8</th>\n",
       "      <th>rna_9</th>\n",
       "      <th>rna_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>X</td>\n",
       "      <td>-0.203456</td>\n",
       "      <td>0.071141</td>\n",
       "      <td>0.089584</td>\n",
       "      <td>0.075924</td>\n",
       "      <td>0.054592</td>\n",
       "      <td>-0.059412</td>\n",
       "      <td>0.098149</td>\n",
       "      <td>0.068125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1,1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.214458</td>\n",
       "      <td>-0.023648</td>\n",
       "      <td>-0.495514</td>\n",
       "      <td>-0.228879</td>\n",
       "      <td>0.123623</td>\n",
       "      <td>-0.102686</td>\n",
       "      <td>0.028465</td>\n",
       "      <td>-0.181398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13</td>\n",
       "      <td>C</td>\n",
       "      <td>0.048487</td>\n",
       "      <td>-0.045041</td>\n",
       "      <td>0.100893</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>-0.103560</td>\n",
       "      <td>-0.040055</td>\n",
       "      <td>-0.167797</td>\n",
       "      <td>0.002691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.390815</td>\n",
       "      <td>0.126024</td>\n",
       "      <td>-0.288252</td>\n",
       "      <td>-0.050163</td>\n",
       "      <td>-0.047548</td>\n",
       "      <td>-0.107317</td>\n",
       "      <td>0.277935</td>\n",
       "      <td>-0.137774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.15.1.1</td>\n",
       "      <td>1.2.0.0.15.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.108030</td>\n",
       "      <td>0.202503</td>\n",
       "      <td>0.055752</td>\n",
       "      <td>-0.256612</td>\n",
       "      <td>0.159802</td>\n",
       "      <td>0.228066</td>\n",
       "      <td>0.140733</td>\n",
       "      <td>0.028319</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      oligo_type        oligo_id                  dupe_info  \\\n",
       "0           WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1                1.2.0.0.0.0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1,1.2.0.0.12.1   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1               1.2.0.0.13.1   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1               1.2.0.0.14.1   \n",
       "4  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.15.1.1               1.2.0.0.15.1   \n",
       "\n",
       "                                            seq_name  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand  tile_start    ...     del_num  del_base     rna_3  \\\n",
       "0  65187718.0      -  65187510.0    ...           0         X -0.203456   \n",
       "1  65187718.0      -  65187510.0    ...          11         A -0.214458   \n",
       "2  65187718.0      -  65187510.0    ...          13         C  0.048487   \n",
       "3  65187718.0      -  65187510.0    ...          14         A -0.390815   \n",
       "4  65187718.0      -  65187510.0    ...          15         G -0.108030   \n",
       "\n",
       "      rna_4     rna_5     rna_6     rna_7     rna_8     rna_9    rna_10  \n",
       "0  0.071141  0.089584  0.075924  0.054592 -0.059412  0.098149  0.068125  \n",
       "1 -0.023648 -0.495514 -0.228879  0.123623 -0.102686  0.028465 -0.181398  \n",
       "2 -0.045041  0.100893  0.000007 -0.103560 -0.040055 -0.167797  0.002691  \n",
       "3  0.126024 -0.288252 -0.050163 -0.047548 -0.107317  0.277935 -0.137774  \n",
       "4  0.202503  0.055752 -0.256612  0.159802  0.228066  0.140733  0.028319  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2 = index_elem.merge(hepg2_elem_res, on=[\"unique_id\", \"element\"], how=\"left\")\n",
    "k562 = index_elem.merge(k562_elem_res, on=[\"unique_id\", \"element\"], how=\"left\")\n",
    "hepg2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# add overall mean column and standard dev column\n",
    "hepg2_reps = [col for col in hepg2.columns if \"rna\" in col]\n",
    "hepg2[\"overall_mean\"] = hepg2[hepg2_reps].mean(axis=1)\n",
    "hepg2[\"lfcSE\"] = hepg2[hepg2_reps].std(axis=1)/np.sqrt(len(hepg2_reps))\n",
    "\n",
    "hepg2_barc[\"overall_mean\"] = hepg2_barc[hepg2_reps].mean(axis=1)\n",
    "hepg2_barc[\"lfcSE\"] = hepg2_barc[hepg2_reps].std(axis=1)/np.sqrt(len(hepg2_reps))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# add overall mean column and standard dev column\n",
    "k562_reps = [col for col in k562.columns if \"rna\" in col]\n",
    "k562[\"overall_mean\"] = k562[k562_reps].mean(axis=1)\n",
    "k562[\"lfcSE\"] = k562[k562_reps].std(axis=1)/np.sqrt(len(k562_reps))\n",
    "\n",
    "k562_barc[\"overall_mean\"] = k562_barc[k562_reps].mean(axis=1)\n",
    "k562_barc[\"lfcSE\"] = k562_barc[k562_reps].std(axis=1)/np.sqrt(len(k562_reps))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. calculate p-value for deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mapping barcode values to 3964 deletion sequences\n",
      "..row 0..\n",
      "..row 1000..\n",
      "..row 2000..\n",
      "..row 3000..\n"
     ]
    }
   ],
   "source": [
    "hepg2_barcode_value_dict = get_barcode_value_map(hepg2, hepg2_barc, hepg2_reps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mapping barcode values to 3964 deletion sequences\n",
      "..row 0..\n",
      "..row 1000..\n",
      "..row 2000..\n",
      "..row 3000..\n"
     ]
    }
   ],
   "source": [
    "k562_barcode_value_dict = get_barcode_value_map(k562, k562_barc, k562_reps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "calculating pvalues b/w deletion and wt for 3964 deletion tiles\n"
     ]
    }
   ],
   "source": [
    "hepg2_pvals, hepg2_l2fcs = calculate_p_value(hepg2_barcode_value_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "calculating pvalues b/w deletion and wt for 3964 deletion tiles\n"
     ]
    }
   ],
   "source": [
    "k562_pvals, k562_l2fcs = calculate_p_value(k562_barcode_value_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>rna_3_pval</th>\n",
       "      <th>rna_4_pval</th>\n",
       "      <th>rna_5_pval</th>\n",
       "      <th>rna_6_pval</th>\n",
       "      <th>rna_7_pval</th>\n",
       "      <th>rna_8_pval</th>\n",
       "      <th>rna_9_pval</th>\n",
       "      <th>rna_10_pval</th>\n",
       "      <th>combined_pval</th>\n",
       "      <th>rna_3_l2fc</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>rna_5_l2fc</th>\n",
       "      <th>rna_6_l2fc</th>\n",
       "      <th>rna_7_l2fc</th>\n",
       "      <th>rna_8_l2fc</th>\n",
       "      <th>rna_9_l2fc</th>\n",
       "      <th>rna_10_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3862</th>\n",
       "      <td>TTTCCCGAGAGATGGCAAAGGCCCTGGCAGGAAGTTTACAGGGTTT...</td>\n",
       "      <td>0.105570</td>\n",
       "      <td>0.027459</td>\n",
       "      <td>0.666765</td>\n",
       "      <td>0.353407</td>\n",
       "      <td>0.086355</td>\n",
       "      <td>0.458310</td>\n",
       "      <td>0.666765</td>\n",
       "      <td>0.255059</td>\n",
       "      <td>0.044477</td>\n",
       "      <td>0.485808</td>\n",
       "      <td>0.510173</td>\n",
       "      <td>0.109897</td>\n",
       "      <td>0.170552</td>\n",
       "      <td>0.265033</td>\n",
       "      <td>0.008015</td>\n",
       "      <td>-0.002408</td>\n",
       "      <td>0.178464</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346</th>\n",
       "      <td>CGCCAGCGAGCGGGAGAGAGAGCCGAGGAGGAAGAGGGAGAGGGCG...</td>\n",
       "      <td>0.062069</td>\n",
       "      <td>0.090143</td>\n",
       "      <td>0.074602</td>\n",
       "      <td>0.292568</td>\n",
       "      <td>0.940032</td>\n",
       "      <td>0.084289</td>\n",
       "      <td>0.435327</td>\n",
       "      <td>0.851483</td>\n",
       "      <td>0.089126</td>\n",
       "      <td>0.492443</td>\n",
       "      <td>0.453929</td>\n",
       "      <td>0.486026</td>\n",
       "      <td>0.301142</td>\n",
       "      <td>-0.081406</td>\n",
       "      <td>0.182938</td>\n",
       "      <td>0.015364</td>\n",
       "      <td>0.032526</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>CCCTTCTTTGCTGCAATCTGGGTGCGGCTAGAGCAATTTTCATAGA...</td>\n",
       "      <td>0.775938</td>\n",
       "      <td>0.648923</td>\n",
       "      <td>0.657123</td>\n",
       "      <td>0.237524</td>\n",
       "      <td>0.528314</td>\n",
       "      <td>0.780222</td>\n",
       "      <td>0.821430</td>\n",
       "      <td>0.203969</td>\n",
       "      <td>0.734108</td>\n",
       "      <td>0.211697</td>\n",
       "      <td>0.311141</td>\n",
       "      <td>0.028294</td>\n",
       "      <td>-0.371390</td>\n",
       "      <td>0.206838</td>\n",
       "      <td>-0.163643</td>\n",
       "      <td>0.115319</td>\n",
       "      <td>-0.197453</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3069</th>\n",
       "      <td>GGGGCTAGACGGGCCTGGACAACTACTAGAGCGCCTCGGGCTGTGC...</td>\n",
       "      <td>0.271449</td>\n",
       "      <td>0.089308</td>\n",
       "      <td>0.197391</td>\n",
       "      <td>0.165353</td>\n",
       "      <td>0.101273</td>\n",
       "      <td>0.177944</td>\n",
       "      <td>0.333992</td>\n",
       "      <td>0.975493</td>\n",
       "      <td>0.058434</td>\n",
       "      <td>-0.304279</td>\n",
       "      <td>-0.279800</td>\n",
       "      <td>-0.372580</td>\n",
       "      <td>0.508479</td>\n",
       "      <td>-0.538950</td>\n",
       "      <td>-0.221167</td>\n",
       "      <td>-0.099235</td>\n",
       "      <td>-0.049811</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2246</th>\n",
       "      <td>GCAGTTCTTCCCGGGGCGGGGAGAGTGGGTAGGCGGTCCCTCGCGG...</td>\n",
       "      <td>0.090765</td>\n",
       "      <td>0.106469</td>\n",
       "      <td>0.397689</td>\n",
       "      <td>0.146460</td>\n",
       "      <td>0.617075</td>\n",
       "      <td>0.578046</td>\n",
       "      <td>0.563990</td>\n",
       "      <td>0.160234</td>\n",
       "      <td>0.067372</td>\n",
       "      <td>-0.285757</td>\n",
       "      <td>-0.385582</td>\n",
       "      <td>0.070902</td>\n",
       "      <td>-0.494802</td>\n",
       "      <td>0.207201</td>\n",
       "      <td>-0.070299</td>\n",
       "      <td>-0.061336</td>\n",
       "      <td>-0.344962</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  index  rna_3_pval  \\\n",
       "3862  TTTCCCGAGAGATGGCAAAGGCCCTGGCAGGAAGTTTACAGGGTTT...    0.105570   \n",
       "1346  CGCCAGCGAGCGGGAGAGAGAGCCGAGGAGGAAGAGGGAGAGGGCG...    0.062069   \n",
       "968   CCCTTCTTTGCTGCAATCTGGGTGCGGCTAGAGCAATTTTCATAGA...    0.775938   \n",
       "3069  GGGGCTAGACGGGCCTGGACAACTACTAGAGCGCCTCGGGCTGTGC...    0.271449   \n",
       "2246  GCAGTTCTTCCCGGGGCGGGGAGAGTGGGTAGGCGGTCCCTCGCGG...    0.090765   \n",
       "\n",
       "      rna_4_pval  rna_5_pval  rna_6_pval  rna_7_pval  rna_8_pval  rna_9_pval  \\\n",
       "3862    0.027459    0.666765    0.353407    0.086355    0.458310    0.666765   \n",
       "1346    0.090143    0.074602    0.292568    0.940032    0.084289    0.435327   \n",
       "968     0.648923    0.657123    0.237524    0.528314    0.780222    0.821430   \n",
       "3069    0.089308    0.197391    0.165353    0.101273    0.177944    0.333992   \n",
       "2246    0.106469    0.397689    0.146460    0.617075    0.578046    0.563990   \n",
       "\n",
       "      rna_10_pval  combined_pval  rna_3_l2fc  rna_4_l2fc  rna_5_l2fc  \\\n",
       "3862     0.255059       0.044477    0.485808    0.510173    0.109897   \n",
       "1346     0.851483       0.089126    0.492443    0.453929    0.486026   \n",
       "968      0.203969       0.734108    0.211697    0.311141    0.028294   \n",
       "3069     0.975493       0.058434   -0.304279   -0.279800   -0.372580   \n",
       "2246     0.160234       0.067372   -0.285757   -0.385582    0.070902   \n",
       "\n",
       "      rna_6_l2fc  rna_7_l2fc  rna_8_l2fc  rna_9_l2fc  rna_10_l2fc  \\\n",
       "3862    0.170552    0.265033    0.008015   -0.002408     0.178464   \n",
       "1346    0.301142   -0.081406    0.182938    0.015364     0.032526   \n",
       "968    -0.371390    0.206838   -0.163643    0.115319    -0.197453   \n",
       "3069    0.508479   -0.538950   -0.221167   -0.099235    -0.049811   \n",
       "2246   -0.494802    0.207201   -0.070299   -0.061336    -0.344962   \n",
       "\n",
       "      combined_padj combined_sig  \n",
       "3862            1.0      not sig  \n",
       "1346            1.0      not sig  \n",
       "968             1.0      not sig  \n",
       "3069            1.0      not sig  \n",
       "2246            1.0      not sig  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine and adjust pvals\n",
    "hepg2_all_pvals = combine_and_adjust_pvals(hepg2_pvals, hepg2_l2fcs, 0.05, hepg2_reps)\n",
    "hepg2_all_pvals.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>rna_1_pval</th>\n",
       "      <th>rna_2_pval</th>\n",
       "      <th>rna_3_pval</th>\n",
       "      <th>rna_4_pval</th>\n",
       "      <th>combined_pval</th>\n",
       "      <th>rna_1_l2fc</th>\n",
       "      <th>rna_2_l2fc</th>\n",
       "      <th>rna_3_l2fc</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3118</th>\n",
       "      <td>GGGGGAGCGCGCAGCCGGCCAGTGGCGGCCTCGCCGGCGGCCTCTT...</td>\n",
       "      <td>0.601508</td>\n",
       "      <td>0.723674</td>\n",
       "      <td>0.563703</td>\n",
       "      <td>0.179712</td>\n",
       "      <td>0.518932</td>\n",
       "      <td>0.941407</td>\n",
       "      <td>-1.283407</td>\n",
       "      <td>1.158817</td>\n",
       "      <td>3.108161</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3858</th>\n",
       "      <td>TTTCCCGAGAGATGGCAAAGGCCCTGGCAGGAAGTTTACAGGGTTT...</td>\n",
       "      <td>0.786880</td>\n",
       "      <td>0.518194</td>\n",
       "      <td>0.598161</td>\n",
       "      <td>0.895173</td>\n",
       "      <td>0.879439</td>\n",
       "      <td>1.277676</td>\n",
       "      <td>0.662970</td>\n",
       "      <td>-0.685965</td>\n",
       "      <td>0.346213</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3931</th>\n",
       "      <td>TTTCCTTAGCCCCAGCCCACACCATGTGTATGAGTTGGGTTGAGAA...</td>\n",
       "      <td>0.780939</td>\n",
       "      <td>0.453537</td>\n",
       "      <td>0.109246</td>\n",
       "      <td>0.084278</td>\n",
       "      <td>0.164933</td>\n",
       "      <td>0.619287</td>\n",
       "      <td>-0.130405</td>\n",
       "      <td>0.492348</td>\n",
       "      <td>-0.719083</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3914</th>\n",
       "      <td>TTTCCTTAGCCCCAGCCCACACCATGTGTATGAGTTGGGTTAGAAG...</td>\n",
       "      <td>0.384189</td>\n",
       "      <td>0.920014</td>\n",
       "      <td>0.554658</td>\n",
       "      <td>0.790566</td>\n",
       "      <td>0.848083</td>\n",
       "      <td>-0.426039</td>\n",
       "      <td>0.306888</td>\n",
       "      <td>0.616516</td>\n",
       "      <td>-0.370664</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>AAGGGAAGTGAGGGGTAGAGGAAGGCGGGTCCCTGGCTCGGCTCTA...</td>\n",
       "      <td>0.471938</td>\n",
       "      <td>0.138945</td>\n",
       "      <td>0.055996</td>\n",
       "      <td>0.427586</td>\n",
       "      <td>0.071645</td>\n",
       "      <td>-1.054777</td>\n",
       "      <td>-0.436403</td>\n",
       "      <td>1.977409</td>\n",
       "      <td>-0.008234</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  index  rna_1_pval  \\\n",
       "3118  GGGGGAGCGCGCAGCCGGCCAGTGGCGGCCTCGCCGGCGGCCTCTT...    0.601508   \n",
       "3858  TTTCCCGAGAGATGGCAAAGGCCCTGGCAGGAAGTTTACAGGGTTT...    0.786880   \n",
       "3931  TTTCCTTAGCCCCAGCCCACACCATGTGTATGAGTTGGGTTGAGAA...    0.780939   \n",
       "3914  TTTCCTTAGCCCCAGCCCACACCATGTGTATGAGTTGGGTTAGAAG...    0.384189   \n",
       "142   AAGGGAAGTGAGGGGTAGAGGAAGGCGGGTCCCTGGCTCGGCTCTA...    0.471938   \n",
       "\n",
       "      rna_2_pval  rna_3_pval  rna_4_pval  combined_pval  rna_1_l2fc  \\\n",
       "3118    0.723674    0.563703    0.179712       0.518932    0.941407   \n",
       "3858    0.518194    0.598161    0.895173       0.879439    1.277676   \n",
       "3931    0.453537    0.109246    0.084278       0.164933    0.619287   \n",
       "3914    0.920014    0.554658    0.790566       0.848083   -0.426039   \n",
       "142     0.138945    0.055996    0.427586       0.071645   -1.054777   \n",
       "\n",
       "      rna_2_l2fc  rna_3_l2fc  rna_4_l2fc  combined_padj combined_sig  \n",
       "3118   -1.283407    1.158817    3.108161            1.0      not sig  \n",
       "3858    0.662970   -0.685965    0.346213            1.0      not sig  \n",
       "3931   -0.130405    0.492348   -0.719083            1.0      not sig  \n",
       "3914    0.306888    0.616516   -0.370664            1.0      not sig  \n",
       "142    -0.436403    1.977409   -0.008234            1.0      not sig  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine and adjust pvals\n",
    "k562_all_pvals = combine_and_adjust_pvals(k562_pvals, k562_l2fcs, 0.05, k562_reps)\n",
    "k562_all_pvals.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_3_l2fc</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>rna_5_l2fc</th>\n",
       "      <th>rna_6_l2fc</th>\n",
       "      <th>rna_7_l2fc</th>\n",
       "      <th>rna_8_l2fc</th>\n",
       "      <th>rna_9_l2fc</th>\n",
       "      <th>rna_10_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1,1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011002</td>\n",
       "      <td>-0.094789</td>\n",
       "      <td>-0.585099</td>\n",
       "      <td>-0.304803</td>\n",
       "      <td>0.069031</td>\n",
       "      <td>-0.043274</td>\n",
       "      <td>-0.069684</td>\n",
       "      <td>-0.249523</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.251943</td>\n",
       "      <td>-0.116182</td>\n",
       "      <td>0.011308</td>\n",
       "      <td>-0.075917</td>\n",
       "      <td>-0.158152</td>\n",
       "      <td>0.019357</td>\n",
       "      <td>-0.265946</td>\n",
       "      <td>-0.065434</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.187358</td>\n",
       "      <td>0.054883</td>\n",
       "      <td>-0.377836</td>\n",
       "      <td>-0.126087</td>\n",
       "      <td>-0.102139</td>\n",
       "      <td>-0.047905</td>\n",
       "      <td>0.179786</td>\n",
       "      <td>-0.205900</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.15.1.1</td>\n",
       "      <td>1.2.0.0.15.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095426</td>\n",
       "      <td>0.131363</td>\n",
       "      <td>-0.033832</td>\n",
       "      <td>-0.332535</td>\n",
       "      <td>0.105210</td>\n",
       "      <td>0.287478</td>\n",
       "      <td>0.042585</td>\n",
       "      <td>-0.039807</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      oligo_type        oligo_id                  dupe_info  \\\n",
       "0           WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1                1.2.0.0.0.0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1,1.2.0.0.12.1   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1               1.2.0.0.13.1   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1               1.2.0.0.14.1   \n",
       "4  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.15.1.1               1.2.0.0.15.1   \n",
       "\n",
       "                                            seq_name  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand  tile_start      ...       rna_3_l2fc  rna_4_l2fc  \\\n",
       "0  65187718.0      -  65187510.0      ...              NaN         NaN   \n",
       "1  65187718.0      -  65187510.0      ...        -0.011002   -0.094789   \n",
       "2  65187718.0      -  65187510.0      ...         0.251943   -0.116182   \n",
       "3  65187718.0      -  65187510.0      ...        -0.187358    0.054883   \n",
       "4  65187718.0      -  65187510.0      ...         0.095426    0.131363   \n",
       "\n",
       "   rna_5_l2fc  rna_6_l2fc rna_7_l2fc rna_8_l2fc rna_9_l2fc  rna_10_l2fc  \\\n",
       "0         NaN         NaN        NaN        NaN        NaN          NaN   \n",
       "1   -0.585099   -0.304803   0.069031  -0.043274  -0.069684    -0.249523   \n",
       "2    0.011308   -0.075917  -0.158152   0.019357  -0.265946    -0.065434   \n",
       "3   -0.377836   -0.126087  -0.102139  -0.047905   0.179786    -0.205900   \n",
       "4   -0.033832   -0.332535   0.105210   0.287478   0.042585    -0.039807   \n",
       "\n",
       "  combined_padj  combined_sig  \n",
       "0           NaN           NaN  \n",
       "1           1.0       not sig  \n",
       "2           1.0       not sig  \n",
       "3           1.0       not sig  \n",
       "4           1.0       not sig  \n",
       "\n",
       "[5 rows x 49 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add info back to original dfs\n",
    "hepg2 = hepg2.merge(hepg2_all_pvals, left_on=\"element\", right_on=\"index\", how=\"left\")\n",
    "hepg2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_2_pval</th>\n",
       "      <th>rna_3_pval</th>\n",
       "      <th>rna_4_pval</th>\n",
       "      <th>combined_pval</th>\n",
       "      <th>rna_1_l2fc</th>\n",
       "      <th>rna_2_l2fc</th>\n",
       "      <th>rna_3_l2fc</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1,1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.556914</td>\n",
       "      <td>0.849997</td>\n",
       "      <td>0.962435</td>\n",
       "      <td>0.950389</td>\n",
       "      <td>-0.246157</td>\n",
       "      <td>0.224402</td>\n",
       "      <td>-0.511201</td>\n",
       "      <td>0.231922</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.404710</td>\n",
       "      <td>0.200471</td>\n",
       "      <td>0.619516</td>\n",
       "      <td>0.390514</td>\n",
       "      <td>0.122225</td>\n",
       "      <td>-0.007327</td>\n",
       "      <td>0.556734</td>\n",
       "      <td>-0.352480</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.547251</td>\n",
       "      <td>0.527546</td>\n",
       "      <td>0.648285</td>\n",
       "      <td>0.826626</td>\n",
       "      <td>0.234730</td>\n",
       "      <td>-0.311519</td>\n",
       "      <td>0.429838</td>\n",
       "      <td>-0.099402</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.15.1.1</td>\n",
       "      <td>1.2.0.0.15.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.684543</td>\n",
       "      <td>0.606288</td>\n",
       "      <td>0.976772</td>\n",
       "      <td>0.961971</td>\n",
       "      <td>0.425954</td>\n",
       "      <td>-0.066967</td>\n",
       "      <td>-0.538121</td>\n",
       "      <td>-0.008081</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      oligo_type        oligo_id                  dupe_info  \\\n",
       "0           WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1                1.2.0.0.0.0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1,1.2.0.0.12.1   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1               1.2.0.0.13.1   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1               1.2.0.0.14.1   \n",
       "4  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.15.1.1               1.2.0.0.15.1   \n",
       "\n",
       "                                            seq_name  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "4  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand  tile_start      ...       rna_2_pval  rna_3_pval  \\\n",
       "0  65187718.0      -  65187510.0      ...              NaN         NaN   \n",
       "1  65187718.0      -  65187510.0      ...         0.556914    0.849997   \n",
       "2  65187718.0      -  65187510.0      ...         0.404710    0.200471   \n",
       "3  65187718.0      -  65187510.0      ...         0.547251    0.527546   \n",
       "4  65187718.0      -  65187510.0      ...         0.684543    0.606288   \n",
       "\n",
       "   rna_4_pval  combined_pval rna_1_l2fc rna_2_l2fc rna_3_l2fc  rna_4_l2fc  \\\n",
       "0         NaN            NaN        NaN        NaN        NaN         NaN   \n",
       "1    0.962435       0.950389  -0.246157   0.224402  -0.511201    0.231922   \n",
       "2    0.619516       0.390514   0.122225  -0.007327   0.556734   -0.352480   \n",
       "3    0.648285       0.826626   0.234730  -0.311519   0.429838   -0.099402   \n",
       "4    0.976772       0.961971   0.425954  -0.066967  -0.538121   -0.008081   \n",
       "\n",
       "  combined_padj  combined_sig  \n",
       "0           NaN           NaN  \n",
       "1           1.0       not sig  \n",
       "2           1.0       not sig  \n",
       "3           1.0       not sig  \n",
       "4           1.0       not sig  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add info back to original dfs\n",
    "k562 = k562.merge(k562_all_pvals, left_on=\"element\", right_on=\"index\", how=\"left\")\n",
    "k562.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. wrangle deletion data into format we need"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Enhancer.noflip.NA__chr11:65187297-65188198__chr11:65187510..65187718,-,1.2',\n",
       " 'Enhancer.noflip.NA__chr11:65264635-65265953__chr11:65265021..65265229,-,2.1',\n",
       " 'Enhancer.noflip.NA__chr11:65264635-65265953__chr11:65265021..65265229,-,2.2',\n",
       " 'Enhancer.noflip.NA__chr20:47893075-47893400__chr20:47893097..47893305,-,3.1',\n",
       " 'Enhancer.noflip.NA__chr20:47893075-47893400__chr20:47893097..47893305,-,3.2',\n",
       " 'Enhancer.noflip.NA__chr3:169482089-169482431__chr3:169482079..169482287,-,4.1',\n",
       " 'Enhancer.noflip.NA__chr3:169482089-169482431__chr3:169482079..169482287,-,4.2',\n",
       " 'Enhancer.noflip.NA__chr7:130795999-130796263__chr7:130796089..130796297,-,5.1',\n",
       " 'Enhancer.noflip.NA__chr7:130795999-130796263__chr7:130796089..130796297,-,5.2']"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_names = index_elem[index_elem.oligo_type.str.contains(\"DELETION\")][\"tile_name\"].unique()\n",
    "[x for x in unique_names if \"Enhancer\" in x]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_3_l2fc</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>rna_5_l2fc</th>\n",
       "      <th>rna_6_l2fc</th>\n",
       "      <th>rna_7_l2fc</th>\n",
       "      <th>rna_8_l2fc</th>\n",
       "      <th>rna_9_l2fc</th>\n",
       "      <th>rna_10_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011002</td>\n",
       "      <td>-0.094789</td>\n",
       "      <td>-0.585099</td>\n",
       "      <td>-0.304803</td>\n",
       "      <td>0.069031</td>\n",
       "      <td>-0.043274</td>\n",
       "      <td>-0.069684</td>\n",
       "      <td>-0.249523</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011002</td>\n",
       "      <td>-0.094789</td>\n",
       "      <td>-0.585099</td>\n",
       "      <td>-0.304803</td>\n",
       "      <td>0.069031</td>\n",
       "      <td>-0.043274</td>\n",
       "      <td>-0.069684</td>\n",
       "      <td>-0.249523</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.251943</td>\n",
       "      <td>-0.116182</td>\n",
       "      <td>0.011308</td>\n",
       "      <td>-0.075917</td>\n",
       "      <td>-0.158152</td>\n",
       "      <td>0.019357</td>\n",
       "      <td>-0.265946</td>\n",
       "      <td>-0.065434</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.187358</td>\n",
       "      <td>0.054883</td>\n",
       "      <td>-0.377836</td>\n",
       "      <td>-0.126087</td>\n",
       "      <td>-0.102139</td>\n",
       "      <td>-0.047905</td>\n",
       "      <td>0.179786</td>\n",
       "      <td>-0.205900</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      oligo_type        oligo_id     dupe_info  \\\n",
       "0           WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1   1.2.0.0.0.0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.12.1   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1  1.2.0.0.13.1   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1  1.2.0.0.14.1   \n",
       "\n",
       "                                            seq_name  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand  tile_start      ...       rna_3_l2fc  rna_4_l2fc  \\\n",
       "0  65187718.0      -  65187510.0      ...              NaN         NaN   \n",
       "1  65187718.0      -  65187510.0      ...        -0.011002   -0.094789   \n",
       "1  65187718.0      -  65187510.0      ...        -0.011002   -0.094789   \n",
       "2  65187718.0      -  65187510.0      ...         0.251943   -0.116182   \n",
       "3  65187718.0      -  65187510.0      ...        -0.187358    0.054883   \n",
       "\n",
       "   rna_5_l2fc  rna_6_l2fc rna_7_l2fc rna_8_l2fc rna_9_l2fc  rna_10_l2fc  \\\n",
       "0         NaN         NaN        NaN        NaN        NaN          NaN   \n",
       "1   -0.585099   -0.304803   0.069031  -0.043274  -0.069684    -0.249523   \n",
       "1   -0.585099   -0.304803   0.069031  -0.043274  -0.069684    -0.249523   \n",
       "2    0.011308   -0.075917  -0.158152   0.019357  -0.265946    -0.065434   \n",
       "3   -0.377836   -0.126087  -0.102139  -0.047905   0.179786    -0.205900   \n",
       "\n",
       "  combined_padj  combined_sig  \n",
       "0           NaN           NaN  \n",
       "1           1.0       not sig  \n",
       "1           1.0       not sig  \n",
       "2           1.0       not sig  \n",
       "3           1.0       not sig  \n",
       "\n",
       "[5 rows x 49 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# first flatten the dataframe so dupes at seq level are dealt with\n",
    "hepg2 = tidy_split(hepg2, \"dupe_info\", sep=\",\", keep=False)\n",
    "k562 = tidy_split(k562, \"dupe_info\", sep=\",\", keep=False)\n",
    "hepg2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>oligo_type</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>tile_name</th>\n",
       "      <th>chr</th>\n",
       "      <th>locus_start</th>\n",
       "      <th>locus_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_4_l2fc</th>\n",
       "      <th>rna_5_l2fc</th>\n",
       "      <th>rna_6_l2fc</th>\n",
       "      <th>rna_7_l2fc</th>\n",
       "      <th>rna_8_l2fc</th>\n",
       "      <th>rna_9_l2fc</th>\n",
       "      <th>rna_10_l2fc</th>\n",
       "      <th>combined_padj</th>\n",
       "      <th>combined_sig</th>\n",
       "      <th>del_num_fixed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP</td>\n",
       "      <td>1.2.0.0.0.0.1</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.11.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.094789</td>\n",
       "      <td>-0.585099</td>\n",
       "      <td>-0.304803</td>\n",
       "      <td>0.069031</td>\n",
       "      <td>-0.043274</td>\n",
       "      <td>-0.069684</td>\n",
       "      <td>-0.249523</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.11.1.1</td>\n",
       "      <td>1.2.0.0.12.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.094789</td>\n",
       "      <td>-0.585099</td>\n",
       "      <td>-0.304803</td>\n",
       "      <td>0.069031</td>\n",
       "      <td>-0.043274</td>\n",
       "      <td>-0.069684</td>\n",
       "      <td>-0.249523</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.13.1.1</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.116182</td>\n",
       "      <td>0.011308</td>\n",
       "      <td>-0.075917</td>\n",
       "      <td>-0.158152</td>\n",
       "      <td>0.019357</td>\n",
       "      <td>-0.265946</td>\n",
       "      <td>-0.065434</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION</td>\n",
       "      <td>1.2.0.0.14.1.1</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>Enhancer.noflip.NA__chr11:65187297-65188198__c...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>65187718.0</td>\n",
       "      <td>-</td>\n",
       "      <td>65187510.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054883</td>\n",
       "      <td>-0.377836</td>\n",
       "      <td>-0.126087</td>\n",
       "      <td>-0.102139</td>\n",
       "      <td>-0.047905</td>\n",
       "      <td>0.179786</td>\n",
       "      <td>-0.205900</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      oligo_type        oligo_id     dupe_info  \\\n",
       "0           WILDTYPE_BUT_HAS_SNP   1.2.0.0.0.0.1   1.2.0.0.0.0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.11.1   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.11.1.1  1.2.0.0.12.1   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.13.1.1  1.2.0.0.13.1   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION  1.2.0.0.14.1.1  1.2.0.0.14.1   \n",
       "\n",
       "                                            seq_name  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...   \n",
       "\n",
       "                                           tile_name    chr  locus_start  \\\n",
       "0  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "1  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "2  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "3  Enhancer.noflip.NA__chr11:65187297-65188198__c...  chr11   65187510.0   \n",
       "\n",
       "    locus_end strand  tile_start      ...        rna_4_l2fc  rna_5_l2fc  \\\n",
       "0  65187718.0      -  65187510.0      ...               NaN         NaN   \n",
       "1  65187718.0      -  65187510.0      ...         -0.094789   -0.585099   \n",
       "1  65187718.0      -  65187510.0      ...         -0.094789   -0.585099   \n",
       "2  65187718.0      -  65187510.0      ...         -0.116182    0.011308   \n",
       "3  65187718.0      -  65187510.0      ...          0.054883   -0.377836   \n",
       "\n",
       "   rna_6_l2fc  rna_7_l2fc rna_8_l2fc rna_9_l2fc rna_10_l2fc  combined_padj  \\\n",
       "0         NaN         NaN        NaN        NaN         NaN            NaN   \n",
       "1   -0.304803    0.069031  -0.043274  -0.069684   -0.249523            1.0   \n",
       "1   -0.304803    0.069031  -0.043274  -0.069684   -0.249523            1.0   \n",
       "2   -0.075917   -0.158152   0.019357  -0.265946   -0.065434            1.0   \n",
       "3   -0.126087   -0.102139  -0.047905   0.179786   -0.205900            1.0   \n",
       "\n",
       "  combined_sig  del_num_fixed  \n",
       "0          NaN              0  \n",
       "1      not sig             11  \n",
       "1      not sig             12  \n",
       "2      not sig             13  \n",
       "3      not sig             14  \n",
       "\n",
       "[5 rows x 50 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# then fix deletion numbers of those that have new rows due to a dupe\n",
    "hepg2[\"del_num_fixed\"] = hepg2.apply(fix_del_num, axis=1)\n",
    "k562[\"del_num_fixed\"] = k562.apply(fix_del_num, axis=1)\n",
    "hepg2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enhancer.noflip.NA__chr7:130795999-130796263__chr7:130796089..130796297,-,5.1 is missing 1 bases: [79]\n",
      "mrna.noflip.NA__p1@ZBTB37__chr1:173837344..173837552,+,9.2 is missing 8 bases: [77, 78, 79, 80, 81, 82, 83, 84]\n",
      "Antisense.noflip.NA__p1@MEG3__chr14:101292281..101292489,+,17.1 is missing 1 bases: [43]\n",
      "Lincrna.flip.sense__p1@CRNDE__chr16:54962763..54962971,-,18.1 is missing 6 bases: [21, 22, 23, 24, 25, 26]\n",
      "Lincrna.flip.sense__p1@CRNDE__chr16:54962763..54962971,-,18.2 is missing 1 bases: [56]\n",
      "Antisense.noflip.NA__p1@ZNFX1-AS1__chr20:47895005..47895213,+,22.1 is missing 1 bases: [26]\n",
      "Lincrna.flip.antisense__p1@CRNDE__chr16:54962763..54962971,+,31.1 is missing 1 bases: [59]\n",
      "Lincrna.flip.antisense__p1@CRNDE__chr16:54962763..54962971,+,31.2 is missing 6 bases: [89, 90, 91, 92, 93, 94]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delpos</th>\n",
       "      <th>mean.log2FC</th>\n",
       "      <th>sd</th>\n",
       "      <th>se</th>\n",
       "      <th>seq</th>\n",
       "      <th>padj</th>\n",
       "      <th>sig</th>\n",
       "      <th>wt_activ</th>\n",
       "      <th>wt_l2fc</th>\n",
       "      <th>wt_class</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>11</td>\n",
       "      <td>-0.178531</td>\n",
       "      <td>0.383383</td>\n",
       "      <td>0.135546</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-0.116371</td>\n",
       "      <td>0.911189</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>12</td>\n",
       "      <td>-0.451884</td>\n",
       "      <td>0.330879</td>\n",
       "      <td>0.116983</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-0.116371</td>\n",
       "      <td>0.911189</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>13</td>\n",
       "      <td>0.157971</td>\n",
       "      <td>0.294439</td>\n",
       "      <td>0.104100</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-0.116371</td>\n",
       "      <td>0.911189</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>14</td>\n",
       "      <td>-0.480071</td>\n",
       "      <td>0.454278</td>\n",
       "      <td>0.160612</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-0.116371</td>\n",
       "      <td>0.911189</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>15</td>\n",
       "      <td>0.272765</td>\n",
       "      <td>0.549605</td>\n",
       "      <td>0.194315</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-0.116371</td>\n",
       "      <td>0.911189</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    delpos  mean.log2FC        sd        se seq  padj      sig  wt_activ  \\\n",
       "67      11    -0.178531  0.383383  0.135546   T   1.0  not sig -0.116371   \n",
       "68      12    -0.451884  0.330879  0.116983   G   1.0  not sig -0.116371   \n",
       "69      13     0.157971  0.294439  0.104100   C   1.0  not sig -0.116371   \n",
       "70      14    -0.480071  0.454278  0.160612   A   1.0  not sig -0.116371   \n",
       "71      15     0.272765  0.549605  0.194315   G   1.0  not sig -0.116371   \n",
       "\n",
       "     wt_l2fc    wt_class tile_chr  tile_start    tile_end  \n",
       "67  0.911189  sig active    chr11  65265115.0  65265229.0  \n",
       "68  0.911189  sig active    chr11  65265115.0  65265229.0  \n",
       "69  0.911189  sig active    chr11  65265115.0  65265229.0  \n",
       "70  0.911189  sig active    chr11  65265115.0  65265229.0  \n",
       "71  0.911189  sig active    chr11  65265115.0  65265229.0  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_dels = wrangle_deletion_data(hepg2, unique_names, hepg2_wt_pvals)\n",
    "hepg2_dels[\"Enhancer.noflip.NA__chr11:65264635-65265953__chr11:65265021..65265229,-,2.1\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enhancer.noflip.NA__chr7:130795999-130796263__chr7:130796089..130796297,-,5.1 is missing 1 bases: [79]\n",
      "mrna.noflip.NA__p1@ZBTB37__chr1:173837344..173837552,+,9.2 is missing 8 bases: [77, 78, 79, 80, 81, 82, 83, 84]\n",
      "Antisense.noflip.NA__p1@MEG3__chr14:101292281..101292489,+,17.1 is missing 1 bases: [43]\n",
      "Lincrna.flip.sense__p1@CRNDE__chr16:54962763..54962971,-,18.1 is missing 6 bases: [21, 22, 23, 24, 25, 26]\n",
      "Lincrna.flip.sense__p1@CRNDE__chr16:54962763..54962971,-,18.2 is missing 1 bases: [56]\n",
      "Antisense.noflip.NA__p1@ZNFX1-AS1__chr20:47895005..47895213,+,22.1 is missing 1 bases: [26]\n",
      "Lincrna.flip.antisense__p1@CRNDE__chr16:54962763..54962971,+,31.1 is missing 1 bases: [59]\n",
      "Lincrna.flip.antisense__p1@CRNDE__chr16:54962763..54962971,+,31.2 is missing 6 bases: [89, 90, 91, 92, 93, 94]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delpos</th>\n",
       "      <th>mean.log2FC</th>\n",
       "      <th>sd</th>\n",
       "      <th>se</th>\n",
       "      <th>seq</th>\n",
       "      <th>padj</th>\n",
       "      <th>sig</th>\n",
       "      <th>wt_activ</th>\n",
       "      <th>wt_l2fc</th>\n",
       "      <th>wt_class</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>11</td>\n",
       "      <td>0.139440</td>\n",
       "      <td>1.632326</td>\n",
       "      <td>0.816163</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.164245</td>\n",
       "      <td>2.322216</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>12</td>\n",
       "      <td>-1.069173</td>\n",
       "      <td>1.283795</td>\n",
       "      <td>0.641897</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.164245</td>\n",
       "      <td>2.322216</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>13</td>\n",
       "      <td>-1.045590</td>\n",
       "      <td>0.632650</td>\n",
       "      <td>0.316325</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.164245</td>\n",
       "      <td>2.322216</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>14</td>\n",
       "      <td>0.123362</td>\n",
       "      <td>0.582553</td>\n",
       "      <td>0.291277</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.164245</td>\n",
       "      <td>2.322216</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>15</td>\n",
       "      <td>-0.486999</td>\n",
       "      <td>0.700334</td>\n",
       "      <td>0.350167</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.164245</td>\n",
       "      <td>2.322216</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr11</td>\n",
       "      <td>65265115.0</td>\n",
       "      <td>65265229.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    delpos  mean.log2FC        sd        se seq  padj      sig  wt_activ  \\\n",
       "67      11     0.139440  1.632326  0.816163   T   1.0  not sig  1.164245   \n",
       "68      12    -1.069173  1.283795  0.641897   G   1.0  not sig  1.164245   \n",
       "69      13    -1.045590  0.632650  0.316325   C   1.0  not sig  1.164245   \n",
       "70      14     0.123362  0.582553  0.291277   A   1.0  not sig  1.164245   \n",
       "71      15    -0.486999  0.700334  0.350167   G   1.0  not sig  1.164245   \n",
       "\n",
       "     wt_l2fc    wt_class tile_chr  tile_start    tile_end  \n",
       "67  2.322216  sig active    chr11  65265115.0  65265229.0  \n",
       "68  2.322216  sig active    chr11  65265115.0  65265229.0  \n",
       "69  2.322216  sig active    chr11  65265115.0  65265229.0  \n",
       "70  2.322216  sig active    chr11  65265115.0  65265229.0  \n",
       "71  2.322216  sig active    chr11  65265115.0  65265229.0  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k562_dels = wrangle_deletion_data(k562, unique_names, k562_wt_pvals)\n",
    "k562_dels[\"Enhancer.noflip.NA__chr11:65264635-65265953__chr11:65265021..65265229,-,2.1\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delpos</th>\n",
       "      <th>mean.log2FC</th>\n",
       "      <th>sd</th>\n",
       "      <th>se</th>\n",
       "      <th>seq</th>\n",
       "      <th>padj</th>\n",
       "      <th>sig</th>\n",
       "      <th>wt_activ</th>\n",
       "      <th>wt_l2fc</th>\n",
       "      <th>wt_class</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>0.418004</td>\n",
       "      <td>1.052097</td>\n",
       "      <td>0.526048</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-1.444146</td>\n",
       "      <td>-0.286175</td>\n",
       "      <td>not sig</td>\n",
       "      <td>chr14</td>\n",
       "      <td>101292281.0</td>\n",
       "      <td>101292395.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "      <td>0.148403</td>\n",
       "      <td>0.752713</td>\n",
       "      <td>0.376357</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-1.444146</td>\n",
       "      <td>-0.286175</td>\n",
       "      <td>not sig</td>\n",
       "      <td>chr14</td>\n",
       "      <td>101292281.0</td>\n",
       "      <td>101292395.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13</td>\n",
       "      <td>-0.904815</td>\n",
       "      <td>1.411657</td>\n",
       "      <td>0.705829</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-1.444146</td>\n",
       "      <td>-0.286175</td>\n",
       "      <td>not sig</td>\n",
       "      <td>chr14</td>\n",
       "      <td>101292281.0</td>\n",
       "      <td>101292395.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>0.179602</td>\n",
       "      <td>0.694621</td>\n",
       "      <td>0.347311</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-1.444146</td>\n",
       "      <td>-0.286175</td>\n",
       "      <td>not sig</td>\n",
       "      <td>chr14</td>\n",
       "      <td>101292281.0</td>\n",
       "      <td>101292395.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15</td>\n",
       "      <td>0.038227</td>\n",
       "      <td>0.743493</td>\n",
       "      <td>0.371747</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>-1.444146</td>\n",
       "      <td>-0.286175</td>\n",
       "      <td>not sig</td>\n",
       "      <td>chr14</td>\n",
       "      <td>101292281.0</td>\n",
       "      <td>101292395.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   delpos  mean.log2FC        sd        se seq  padj      sig  wt_activ  \\\n",
       "0      11     0.418004  1.052097  0.526048   C   1.0  not sig -1.444146   \n",
       "1      12     0.148403  0.752713  0.376357   T   1.0  not sig -1.444146   \n",
       "2      13    -0.904815  1.411657  0.705829   G   1.0  not sig -1.444146   \n",
       "3      14     0.179602  0.694621  0.347311   C   1.0  not sig -1.444146   \n",
       "4      15     0.038227  0.743493  0.371747   A   1.0  not sig -1.444146   \n",
       "\n",
       "    wt_l2fc wt_class tile_chr   tile_start     tile_end  \n",
       "0 -0.286175  not sig    chr14  101292281.0  101292395.0  \n",
       "1 -0.286175  not sig    chr14  101292281.0  101292395.0  \n",
       "2 -0.286175  not sig    chr14  101292281.0  101292395.0  \n",
       "3 -0.286175  not sig    chr14  101292281.0  101292395.0  \n",
       "4 -0.286175  not sig    chr14  101292281.0  101292395.0  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k562_dels[\"Antisense.noflip.NA__p1@MEG3__chr14:101292281..101292489,+,17.1\"].head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. write files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# hepg2\n",
    "filenames = []\n",
    "clean_names = []\n",
    "hepg2_dels_clean = {}\n",
    "path = \"../../data/05__deletions/HepG2\"\n",
    "for key in hepg2_dels:\n",
    "    df = hepg2_dels[key]\n",
    "    filename, clean_name = fix_names(key, \"HepG2\", hepg2, NAME_DICT, LOC_DICT)\n",
    "    filenames.append(filename)\n",
    "    clean_names.append(clean_name)\n",
    "    filename = \"%s/%s\" % (path, filename)\n",
    "    df.to_csv(filename, sep=\"\\t\", index=False)\n",
    "    hepg2_dels_clean[clean_name] = df\n",
    "name_map = pd.DataFrame({\"filename\": filenames, \"seq_name\": clean_names})\n",
    "name_map.to_csv(\"../../data/05__deletions/HepG2_filename_map.txt\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# k562\n",
    "filenames = []\n",
    "clean_names = []\n",
    "k562_dels_clean = {}\n",
    "path = \"../../data/05__deletions/K562\"\n",
    "for key in k562_dels:\n",
    "    df = k562_dels[key]\n",
    "    filename, clean_name = fix_names(key, \"K562\", k562, NAME_DICT, LOC_DICT)\n",
    "    filenames.append(filename)\n",
    "    clean_names.append(clean_name)\n",
    "    filename = \"%s/%s\" % (path, filename)\n",
    "    df.to_csv(filename, sep=\"\\t\", index=False)\n",
    "    k562_dels_clean[clean_name] = df\n",
    "name_map = pd.DataFrame({\"filename\": filenames, \"seq_name\": clean_names})\n",
    "name_map.to_csv(\"../../data/05__deletions/K562_filename_map.txt\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## 7. compare deletion profiles to predicted motif profiles (seq by seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def fix_names(row, name_dict, loc_dict):\n",
    "    old_name = row[\"unique_id\"]\n",
    "    chrom = old_name.split(\"__\")[3].split(\":\")[0]\n",
    "    start = int(old_name.split(\"__\")[3].split(\":\")[1].split(\"..\")[0])\n",
    "    end = int(old_name.split(\"__\")[3].split(\":\")[1].split(\"..\")[1].split(\",\")[0])\n",
    "    strand = old_name.split(\"__\")[3].split(\",\")[1]\n",
    "    locs = \"%s:%s-%s\" % (chrom, start, end)\n",
    "    if strand == \"+\":\n",
    "        text_strand = \"plus\"\n",
    "    else:\n",
    "        text_strand = \"minus\"\n",
    "    tile_num = int(old_name.split(\"__\")[4].split(\".\")[1])\n",
    "    \n",
    "    name = old_name.split(\"__\")[2]\n",
    "    coords = old_name.split(\"__\")[3].split(\",\")[0]\n",
    "    try:\n",
    "        gene = name.split(\",\")[0].split(\"@\")[1]\n",
    "        prom = name.split(\",\")[0].split(\"@\")[0]\n",
    "    except:\n",
    "        gene = \"X\"\n",
    "        prom = \"pX\"\n",
    "    \n",
    "    if gene not in name_dict.keys() and coords not in loc_dict.keys():\n",
    "        name = \"%s__%s__tile%s\" % (gene, prom, tile_num)\n",
    "    elif gene in name_dict.keys():\n",
    "        name = \"%s__%s__tile%s\" % (name_dict[gene], prom, tile_num)\n",
    "    elif coords in loc_dict.keys():\n",
    "        name = \"%s__%s__tile%s\" % (loc_dict[coords], prom, tile_num)\n",
    "    \n",
    "    clean_name = \"%s__%s\" % (name, text_strand)\n",
    "    return clean_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>del_id</th>\n",
       "      <th>delpos</th>\n",
       "      <th>del_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2068</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....</td>\n",
       "      <td>20</td>\n",
       "      <td>DLEU1__p1__tile2__plus</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1524</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>13</td>\n",
       "      <td>enhancer_ZFAS1__pX__tile2__minus</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3494</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__mrna.noflip.NA_...</td>\n",
       "      <td>58</td>\n",
       "      <td>ZBTB37__p1__tile1__plus</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>FLIPPED_DELETION__Lincrna.flip.antisense__p1@L...</td>\n",
       "      <td>36</td>\n",
       "      <td>FENDRR__p1__tile1__plus</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2666</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....</td>\n",
       "      <td>26</td>\n",
       "      <td>LINC00467__p1__tile2__plus</td>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              unique_id  n_tfs  \\\n",
       "2068  WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....     20   \n",
       "1524  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...     13   \n",
       "3494  WILDTYPE_BUT_HAS_SNP_DELETION__mrna.noflip.NA_...     58   \n",
       "298   FLIPPED_DELETION__Lincrna.flip.antisense__p1@L...     36   \n",
       "2666  WILDTYPE_BUT_HAS_SNP_DELETION__Lincrna.noflip....     26   \n",
       "\n",
       "                                del_id  delpos  del_num  \n",
       "2068            DLEU1__p1__tile2__plus      85       85  \n",
       "1524  enhancer_ZFAS1__pX__tile2__minus      29       29  \n",
       "3494           ZBTB37__p1__tile1__plus      68       68  \n",
       "298            FENDRR__p1__tile1__plus      34       34  \n",
       "2666        LINC00467__p1__tile2__plus      67       67  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tfbs[\"del_id\"] = tfbs.apply(fix_names, name_dict=NAME_DICT, loc_dict=LOC_DICT, axis=1)\n",
    "tfbs[\"delpos\"] = tfbs.apply(get_del_num, axis=1)\n",
    "tfbs[\"del_num\"] = tfbs.apply(get_del_num, axis=1)\n",
    "tfbs.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>del_id</th>\n",
       "      <th>delpos</th>\n",
       "      <th>del_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3697</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@C...</td>\n",
       "      <td>10</td>\n",
       "      <td>CRNDE__p1__tile1__minus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4055</th>\n",
       "      <td>WILDTYPE__Lincrna.flip.sense__p1@CRNDE__chr16:...</td>\n",
       "      <td>34</td>\n",
       "      <td>CRNDE__p1__tile2__minus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3723</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...</td>\n",
       "      <td>53</td>\n",
       "      <td>ZBTB37__p1__tile1__plus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3696</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr7...</td>\n",
       "      <td>37</td>\n",
       "      <td>enhancer_LINC_PINT__pX__tile1__minus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3716</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...</td>\n",
       "      <td>23</td>\n",
       "      <td>TERC__p1__tile2__minus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              unique_id  n_tfs  \\\n",
       "3697  WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__p1@C...     10   \n",
       "4055  WILDTYPE__Lincrna.flip.sense__p1@CRNDE__chr16:...     34   \n",
       "3723  WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__p1@ZBTB3...     53   \n",
       "3696  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr7...     37   \n",
       "3716  WILDTYPE_BUT_HAS_SNP__Lincrna.noflip.NA__p1@TE...     23   \n",
       "\n",
       "                                    del_id  delpos  del_num  \n",
       "3697               CRNDE__p1__tile1__minus       0        0  \n",
       "4055               CRNDE__p1__tile2__minus       0        0  \n",
       "3723               ZBTB37__p1__tile1__plus       0        0  \n",
       "3696  enhancer_LINC_PINT__pX__tile1__minus       0        0  \n",
       "3716                TERC__p1__tile2__minus       0        0  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tfbs_wt = tfbs[(~tfbs[\"unique_id\"].str.contains(\"DELETION\")) & \n",
    "               (~tfbs[\"unique_id\"].str.contains(\"SNP_INDIV\")) & \n",
    "               (~tfbs[\"unique_id\"].str.contains(\"HAPLO\"))]\n",
    "tfbs_wt.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>del_num</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>del_id</th>\n",
       "      <th>delpos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.11.1,1.2.0.0.12.1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.15.1</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  \\\n",
       "0  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...   \n",
       "4  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...   \n",
       "\n",
       "                   dupe_info  del_num  n_tfs  \\\n",
       "0                1.2.0.0.0.0        0     17   \n",
       "1  1.2.0.0.11.1,1.2.0.0.12.1       11     16   \n",
       "2               1.2.0.0.13.1       13     16   \n",
       "3               1.2.0.0.14.1       14     15   \n",
       "4               1.2.0.0.15.1       15     17   \n",
       "\n",
       "                             del_id  delpos  \n",
       "0  enhancer_NEAT1__pX__tile2__minus       0  \n",
       "1  enhancer_NEAT1__pX__tile2__minus      11  \n",
       "2  enhancer_NEAT1__pX__tile2__minus      13  \n",
       "3  enhancer_NEAT1__pX__tile2__minus      14  \n",
       "4  enhancer_NEAT1__pX__tile2__minus      15  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_del_info = index_elem[[\"unique_id\", \"dupe_info\", \"del_num\"]]\n",
    "index_tfbs = index_del_info.merge(tfbs, on=[\"unique_id\", \"del_num\"])\n",
    "index_tfbs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>del_num</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>del_id</th>\n",
       "      <th>delpos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.11.1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.12.1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id     dupe_info  del_num  \\\n",
       "0  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...   1.2.0.0.0.0        0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.11.1       11   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.12.1       11   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.13.1       13   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.14.1       14   \n",
       "\n",
       "   n_tfs                            del_id  delpos  \n",
       "0     17  enhancer_NEAT1__pX__tile2__minus       0  \n",
       "1     16  enhancer_NEAT1__pX__tile2__minus      11  \n",
       "1     16  enhancer_NEAT1__pX__tile2__minus      11  \n",
       "2     16  enhancer_NEAT1__pX__tile2__minus      13  \n",
       "3     15  enhancer_NEAT1__pX__tile2__minus      14  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_tfbs = tidy_split(index_tfbs, \"dupe_info\", sep=\",\", keep=False)\n",
    "index_tfbs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>del_num</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>del_id</th>\n",
       "      <th>delpos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...</td>\n",
       "      <td>1.2.0.0.0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.11.1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.12.1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.13.1</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.14.1</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id     dupe_info  del_num  \\\n",
       "0  WILDTYPE_BUT_HAS_SNP__Enhancer.noflip.NA__chr1...   1.2.0.0.0.0        0   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.11.1       11   \n",
       "1  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.12.1       11   \n",
       "2  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.13.1       13   \n",
       "3  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.14.1       14   \n",
       "\n",
       "   n_tfs                            del_id  delpos  \n",
       "0     17  enhancer_NEAT1__pX__tile2__minus       0  \n",
       "1     16  enhancer_NEAT1__pX__tile2__minus      11  \n",
       "1     16  enhancer_NEAT1__pX__tile2__minus      12  \n",
       "2     16  enhancer_NEAT1__pX__tile2__minus      13  \n",
       "3     15  enhancer_NEAT1__pX__tile2__minus      14  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_tfbs[\"delpos\"] = index_tfbs.apply(fix_del_num, axis=1)\n",
    "index_tfbs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delpos</th>\n",
       "      <th>mean.log2FC</th>\n",
       "      <th>sd</th>\n",
       "      <th>se</th>\n",
       "      <th>seq</th>\n",
       "      <th>padj</th>\n",
       "      <th>sig</th>\n",
       "      <th>wt_activ</th>\n",
       "      <th>wt_l2fc</th>\n",
       "      <th>wt_class</th>\n",
       "      <th>tile_chr</th>\n",
       "      <th>tile_start</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>del_id</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>del_num</th>\n",
       "      <th>n_tfs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>0.063207</td>\n",
       "      <td>0.066139</td>\n",
       "      <td>0.023384</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>4.731591</td>\n",
       "      <td>5.759151</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr20</td>\n",
       "      <td>47895099.0</td>\n",
       "      <td>47895213.0</td>\n",
       "      <td>ZFAS1__p1__tile2__plus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...</td>\n",
       "      <td>22.2.0.0.11.1</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "      <td>0.063207</td>\n",
       "      <td>0.066139</td>\n",
       "      <td>0.023384</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>4.731591</td>\n",
       "      <td>5.759151</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr20</td>\n",
       "      <td>47895099.0</td>\n",
       "      <td>47895213.0</td>\n",
       "      <td>ZFAS1__p1__tile2__plus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...</td>\n",
       "      <td>22.2.0.0.12.1</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13</td>\n",
       "      <td>-0.110958</td>\n",
       "      <td>0.071982</td>\n",
       "      <td>0.025450</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>4.731591</td>\n",
       "      <td>5.759151</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr20</td>\n",
       "      <td>47895099.0</td>\n",
       "      <td>47895213.0</td>\n",
       "      <td>ZFAS1__p1__tile2__plus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...</td>\n",
       "      <td>22.2.0.0.13.1</td>\n",
       "      <td>13</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>-0.002866</td>\n",
       "      <td>0.035329</td>\n",
       "      <td>0.012491</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>4.731591</td>\n",
       "      <td>5.759151</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr20</td>\n",
       "      <td>47895099.0</td>\n",
       "      <td>47895213.0</td>\n",
       "      <td>ZFAS1__p1__tile2__plus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...</td>\n",
       "      <td>22.2.0.0.14.1</td>\n",
       "      <td>14</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15</td>\n",
       "      <td>-0.002866</td>\n",
       "      <td>0.035329</td>\n",
       "      <td>0.012491</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>not sig</td>\n",
       "      <td>4.731591</td>\n",
       "      <td>5.759151</td>\n",
       "      <td>sig active</td>\n",
       "      <td>chr20</td>\n",
       "      <td>47895099.0</td>\n",
       "      <td>47895213.0</td>\n",
       "      <td>ZFAS1__p1__tile2__plus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...</td>\n",
       "      <td>22.2.0.0.15.1</td>\n",
       "      <td>14</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   delpos  mean.log2FC        sd        se seq  padj      sig  wt_activ  \\\n",
       "0      11     0.063207  0.066139  0.023384   T   1.0  not sig  4.731591   \n",
       "1      12     0.063207  0.066139  0.023384   T   1.0  not sig  4.731591   \n",
       "2      13    -0.110958  0.071982  0.025450   C   1.0  not sig  4.731591   \n",
       "3      14    -0.002866  0.035329  0.012491   G   1.0  not sig  4.731591   \n",
       "4      15    -0.002866  0.035329  0.012491   G   1.0  not sig  4.731591   \n",
       "\n",
       "    wt_l2fc    wt_class tile_chr  tile_start    tile_end  \\\n",
       "0  5.759151  sig active    chr20  47895099.0  47895213.0   \n",
       "1  5.759151  sig active    chr20  47895099.0  47895213.0   \n",
       "2  5.759151  sig active    chr20  47895099.0  47895213.0   \n",
       "3  5.759151  sig active    chr20  47895099.0  47895213.0   \n",
       "4  5.759151  sig active    chr20  47895099.0  47895213.0   \n",
       "\n",
       "                   del_id                                          unique_id  \\\n",
       "0  ZFAS1__p1__tile2__plus  WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...   \n",
       "1  ZFAS1__p1__tile2__plus  WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...   \n",
       "2  ZFAS1__p1__tile2__plus  WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...   \n",
       "3  ZFAS1__p1__tile2__plus  WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...   \n",
       "4  ZFAS1__p1__tile2__plus  WILDTYPE_BUT_HAS_SNP_DELETION__Antisense.nofli...   \n",
       "\n",
       "       dupe_info  del_num  n_tfs  \n",
       "0  22.2.0.0.11.1       11     24  \n",
       "1  22.2.0.0.12.1       11     24  \n",
       "2  22.2.0.0.13.1       13     24  \n",
       "3  22.2.0.0.14.1       14     26  \n",
       "4  22.2.0.0.15.1       14     26  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hepg2_dels_fixed = {}\n",
    "k562_dels_fixed = {}\n",
    "for dels_dict, fixed_dict in zip([hepg2_dels_clean, k562_dels_clean], [hepg2_dels_fixed, k562_dels_fixed]):\n",
    "    for name in dels_dict:\n",
    "        df = dels_dict[name]\n",
    "        df[\"del_id\"] = name\n",
    "        df = df.merge(index_tfbs, on=[\"del_id\", \"delpos\"], how=\"left\")\n",
    "        fixed_dict[name] = df\n",
    "hepg2_dels_fixed[\"ZFAS1__p1__tile2__plus\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HepG2\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAWYAAACNCAYAAACE0tDjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XlYVPX+wPH3wLCM7DhoKiplXXDJJTW39ME0TTMxTS2r\n69XQm134pUmuuZuC1+VqZk8uZI+W1zIycesqYnpzSbzhluCKgJosgqKMrPP7wztzBxhgGGdgkM/r\neXieYc75nu/nbJ/zPd+zjEKr1WoRQghhM+xqOgAhhBAlSWIWQggbI4lZCCFsjCRmIYSwMZKYhRDC\nxkhiFkIIGyOJWQghbIz93Llz59Z0EHXFoUOHWLNmDTk5OXh7ezN37lxatWrFhg0b6Natm9EyWq0W\nhUJhdFhYWBj9+vUzO56Kpm2uR4lp2rRpBAYGYmdXeXshKiqKrKwsmjZtSmFhITNmzKBXr14sWrSI\nXbt2ER0dTV5eHunp6Xz66adERERw5swZcnNzCQgIIDw8nIMHD9KrVy9SU1PZsGEDdnZ2hIWFERcX\nx969e7l37x4tW7as0jzk5+cTFRXF77//TuvWrSsdX7e8Pv/8c1JSUvTzZGjp0qXs3buX2NhYevfu\nXaV4dA4dOkROTg4NGjQwq/yPP/6Iu7s78+bN06/fFStW0LRpU9zd3Sss+69//YtVq1axc+dOTp48\nSc+ePdm1axcbNmwgOjqaLl26sGnTJjp27GhWbI8jZU0HUJfs2LEDd3d3rl+/zoMHDwA4evQox48f\nJycnhxUrVgDQsmVL7O3tOXToEG+88QZxcXFkZ2fj4ODApEmTmDVrFm5ubqSkpOinnZ2dzYIFC6hX\nrx79+/fHw8ODjRs3UlBQwJw5c1i/fj0FBQW4uLgwfPhwpkyZwquvvoq7uztxcXHk5+czZcoUVq5c\niVKpxMfHh3HjxgFw9+5d5s+fj4ODA507d6ZJkyZERkbi7++Pt7c3QUFBLFiwAG9vb3JycvQxRUVF\ncfjwYRwcHOjRowedOnVi7dq1aLVaevfujaenJ99++y337t3jb3/7GwA5OTl88sknTJs2jSVLluDu\n7k6rVq3o3bs3W7Zs4f333y93+W7bto3u3bvTp08fACZPnsyMGTPo2bMnYWFhLF26FICCggJu3brF\nnTt3KCwsLDGNYcOGMXz4cIqKipg5cyZDhw6tcJ1OmzYNLy8vkpOT+eCDD9BqtZw/f96khH727FkS\nEhJISEjg8uXLNGzYEIBdu3bp10lYWBju7u6EhYURHh5ORkYGarW63GkeP36cjRs30qBBA/Lz83Fz\nc6N169YUFRWhVCoJCQnB39+fixcvsmDBAhYsWMDSpUtZsWIFw4cPJzIyssz618UaFBRktM7Nmzdz\n5coVioqKGDt2LFOnTqVVq1Y4Ojry/vvvExsby8qVK4GH23tGRgbJyclERERw4MABjh07hrOzM8nJ\nyTRr1qzS5VYXSFdGNerevTt9+/Yt8V23bt3o0qUL0dHRFBYW4uXlRWJiIgCvvPIKPj4+/Pbbb3h4\neJCWlsbhw4fp2bMnH3/8MW5ubvrp5OTkkJOTQ9++fXnyySfZtGkTixcvZsqUKVy+fBl3d3c+/vhj\ncnJyyMrKolWrVrz55pts2bIFb29vHB0duXjxIteuXaNt27b06tVLP22FQsGQIUNo3749R48e1cf9\n4YcfcubMGXbv3s0777zD9OnTy7TA+/Xrx5IlSzhw4ADffPMNbm5u+Pj4cPr0adRqNQMGDKBhw4ac\nOXMGgAkTJjBhwgRUKhVpaWl07dqVDh064OXlVSYpb9iwgWnTpjFz5kwAEhMT6dKli354u3btuH79\nepn1EBsby/PPP89zzz3Hv//97xLDduzYwezZsxkxYgTdu3c3ab2OGzeOOXPm8O233+Lv70///v1N\nKtemTRsCAgIICAgo8b3hOrl8+TLjx48nJiaGwsLCCpOyTpcuXQgODsbT05MZM2bo15lOSEgI3bp1\n4+LFi2XKGlv/V69excXFRT982rRpTJs2jdjYWHJzc9m5cydeXl4UFBQADxsWs2fPJjc3l6SkJFq1\nagU8bGFv27aNc+fO8f7773PhwgX27t1L9+7dS2xbQhKzzSgsLOSll14iNDSUNm3aAODi4kJBQQGt\nWrUiNDSUnj17ApCXlwc87IrQcXR0ZOrUqdy7d4/IyEh9SzA/Px93d3eUyocnRxqNBgcHB/2OVq9e\nPUJDQxkwYACNGjVi4sSJqNVqli9frp/2sWPH+O233+jUqZO+Tl15hUJBUVEReXl5aLVaiouLS8yX\no6MjAE5OThQWFjJ06FDGjRuHv78/mzZtIj8/n2effVY//l/+8he+++47iouLmTZtGg4ODqxevdro\nMnv33XcJDw/nk08+AeCZZ57h5MmT+mVz6tQp/Pz8ypTbsWMH586dIyUlhR07dpQYNnjwYObPn89X\nX33FsWPHjNZrSKFQlJhHSzBcJ40bN2bJkiUUFRXx8ccfm1TexcWlRFyGZwXOzs4oFAqUSiVFRUX6\nA2l2djaFhYVG1//t27f1XSDNmzcnPDyc8PBwevfuTXFxMU2aNCE0NJTAwEDy8vL09QKo1WrOnz8P\nwKRJk2jTpg1ZWVn88MMPHDx4kIULF+Lp6YmHhwfp6emPtuAeI9KVUcO8vLyIi4tj6dKlLF68mN27\ndxMUFMSNGzcACAgI4JtvvmH27Nk888wzDBgwgOnTp5OQkKBvocDDBPHpp5/i5uZG586dadCgAbNm\nzUKj0bBgwQJSUlJYuHAhvr6+uLq66sv17NmT6dOn4+LiwowZM/jss89wdnYu0Yrz8fHhp59+Ys+e\nPdy9e7fM6f8rr7zCvHnziI2NLZOcfvjhB/bv30/v3r0JCAhg+fLluLi4MHbsWNLS0oiLi8POzo78\n/HwA+vbty5EjR7h+/Trr1q3Dzc2NZ599lqysrEq7MoYPH87ChQvZvn07xcXFDB48uEz/Z3p6Oq6u\nrixcuBCA8ePHc+/evTLTcnFxITMzs8J1Bw8PACtXruTWrVtMmjSp0vFLy8/PJyEhocR3hutkwoQJ\n/Pzzz9y+fZsDBw4wffp0PDw8qlxPeRo1asTq1atJTExEqVSyZcsWo+v/xIkTRsu7urri6+vLrFmz\n8PHxoVWrVsTFxbFw4UK8vLxo3Lgx7dq144MPPkClUuHh4YGHhwdbt27Fz8+P2bNnM3LkSJRKpUln\nA3WFQl5iJKwlKiqKJ554wuQuAVt06NAhMjIy9P/b2dmVOCs4fvw4n3zyif6MxFB6ejqHDx/W/9+s\nWTOSk5P1/wcEBOhP8011/vx5fQvU2DRfeOEFsy/wVWThwoUmtdhTU1P57rvvqnyQ+vrrr+nVq1eZ\nC591lbSYhaiAYV+rMb/++mu5w3x8fMpcPOzUqdMjxdOyZcsyFxYfdZqmaN++Pampqfj6+lp82lqt\nlvv370tSNiAtZiGEsDFy8U8IIWyMJGYhhLAxNtXHHBwcbJU+LCGEsAWpqamsX7++0vFsKjH7+voi\nT4gLIR5XpuY36coQQggbI4lZWEV8fHxNhyBErWVTXRmi9ouJiUGj0ZCYmEhmZqb+hUJCCNNJYhYW\npdFo9O+n0Gg0NRuMELWUdGUIqzD28iAhhGkkMQshhI2ptq6M/fv3k5CQwM2bN5k3b57Rl74IIYSo\nxhazt7c3ISEhODo66n+9A2Dr1q0MHTqUoUOHcvbs2eoKRwghbFa1JeaGDRsSHh7OkCFDSrwPeOTI\nkURFRREVFaV/QbwQQtRl1ZaYV61ahaurK8ePHzf6YnIhhBAPVVtHb0RERHVVJYQQtZrclSGEEDZG\nErMQQtgYScxCCGFjJDELIYSNkcQshBA2xuzEnJOTw+XLl0s8LCKEEOLRmXW73O7du9FoNPj4+HD6\n9GkcHBwYNGiQpWMTQog6yazEXFhYyLBhwywdixBCCMxMzLt27SIzMxOtVotCoWDMmDGWjksIIeos\nsxJzx44dJRkLIYSVmHXxLysri2+++YbExERLxyOEEHWeWYl56tSptG3blnXr1jFu3DhLxySEEHWa\nWYn5nXfeYdu2bYSFhbFu3TqTy6WkpBASEsL9+/fNqVYIIeoEs/qYN23axLlz51i2bBm3b99mw4YN\nJpVzdXWlQYMGJb7bunUrW7duBcDOTp53EUIIsxLzjBkzaN26NWPGjCEgIMDkcl5eXnh5eZX4buTI\nkYwcORKAuXPnmhOOEEI8Vsxqog4ZMoR69epx4cIFfv31V0vHJIQQdZpZLWbD3+b7448/6Nq1q8ll\nQ0NDzalSCCHqDLMS89ixY0lJSeHGjRs0btzY0jEJIUSdZvbVtujoaK5du8YPP/xgyXiEEKLOMzsx\nK5VKjhw5gr29vSXjEUKIOs+sxBwZGcn9+/d5+umnKSwstHRMQghRp5mVmMeOHYu3tzchISG4u7tb\nOiYhhKjTzO7KcHd354svvihzX7IQQohHY1Zi3rlzJ/Xq1aNTp044OjoSHR1t6bhELRAfH2/0sxDi\n0Zh1u9wLL7zApUuXcHBwoGPHjjRs2NDScQkbFxMTQ0pKCpmZmcDD96CkpqaSnZ1dw5EJUfuZlZhX\nrFhBu3btcHR05Pjx4/KGOSuLiYlBo9GQkZFB06ZN6dOnT02HhEajoVOnTiQlJQHQqVMnAP7zn//U\nYFSiLtPtJwAqlcom9hNzmdWV4enpSdu2bcnPz+fEiROWjumxV/q0v7IuAY1Gg5+fH2q1Wr/hCSFK\n0u0nfn5+tX4/MSsxT5o0iYsXL+Li4sKIESO4deuWpeOymor6QqujnzQmJoaYmBh27tzJpk2b2Llz\np/47wz9j/Pz8rB5fXWTOereFPnVbiMGQNeN51GnHxMSwcePGcvctW2NWVwZAQkICQUFB/Pjjj1y8\neJEJEyZUOH50dDQZGRkAlf4sVXx8PO3btzc3tHLFxMQQHx9Pamqq/lRHV5duWGZmZplTIEvGo9Fo\n8Pf3x8/Pj9u3b+uTre4I7+/vX+uP9qUZLj9rrVtz6E59ExMTja53S5WxtEeJwdjyt0RXmbF9yxzl\nxae7hmHutDUaTYkzTnO2ScNcYbi84OE1Fkt2M5qdmJ2dnTlw4ABKpZL69etXOv6lS5eYNGkSERER\nFY5XUYKER9vJDZOirm9UNw3dsNJJsaJkXpV4DIeVbvkaxmP4uSK2lOAqYuwiYVWSia68Wq0mKysL\nLy+vR04guullZ2frl6GpB0Pd6XJ5ZapjvVQWQ3kxlbdvVXV65cVUet+qiooONrrrGQBxcXHs3LnT\naFKsjC42w+UAmHRAMTw4GNtuLN3NaHZibt68Oampqfj7+9OvX79Kx9cFrVAoSnxf+kX5PXv21Cd6\nw878vLw86tevT3x8PJ6engAcPHhQ/1m3MTg7O/PgwQPg4Yq4fPkyGo0GlUqFSqVCo9GQlJREXl4e\nBw8eJDs7m6SkJP0wlUqlr1elUgHo48nOzubgwYPlxpCQkEB2djZ37tzRP6quUqlo0aIFCQkJeHp6\nolKp9LEqFAr9Z11dulhKJyQdXXldvYbzZ7icdUnMcNqWHE8Xh2Hc5c2TWq3Wx6/7XFHCLR1DeZKS\nkjh27Bhdu3bVLwdT5klHq9WWiLX0ei9vORiuC8MyLVq00G9TpbcBSy7/imIwLGNsn9Fty0lJSfp5\nLyoqKrMcDJNd6RiMJUXDfSsrK4udO3eaNH+lt4HS+77htgZU+AoIw/EMYzC2vRo2JnWfb9y4UaJM\nixYt9NsX/G/bLb3dGM6TYc4qPe9VaUiYnZjT0tIYP348GzduNGn8gIAA1q5dS5MmTUp8X/pF+YbB\nx8TEkJGRgVqtxtPTk8DAQDw9PfVH98DAQP3n8vpfz549qz9SDho0qMxw3fQMy+/cudOkMuXFoCsP\nD3cAw+mb2k9s2BfWuHHjMitVV6/h/BlO37CbxFhr/FHHM7ZcqqKivj5dC86UerKzs0ssB0DfZWaY\nQExpZet2SsOWj+Fn3fQNYzIs4+fnR2BgIO3bty+zDejKm7L8jdVrOF7pGHT7ieG8GttnjJ2pGRMT\nE0NmZiZNmzYt0ZrWLVdjSu+3uoSrUCjKXQ4qlYrMzEzUanWZbdxwuZqzrZnTPWNsv9VtX6but4bT\nKG/dmsKsxBwZGYlGo+Gzzz7D2dnZpDJDhw6tcj2luw2Acj+by5xpWDoGYyrbkGpDN0ZFTEmQpjC2\nHNRqNYMGDapyt5dKpaqwm0o3Tukyhq3LR10vxmKA/52JGCZgnYq618yJqU+fPtSvX19/gNElKd1y\nNaW8LgbdujRMTLrPhvWUVnq5VpVu+3rUrqWqllWpVMTFxaFWqx/pp/LMfh9zdbFEAjKnz8ucMsK2\nVDU5lU4U5X0uXaYixhKS7rNhK6y8ZFW63vISjbUaLLoDhe6zOeWTkpJITEw0euAoL1ZLXUSr7gZM\nZQclU5ndlVFbmHPkLd1qEbbPsKVSlQRSmqXOznQJqbyLVCkpKYDxlnBF9dZEoinvoGRqeQBfX99K\nD3KPi4oOSqZ67BOzOUfeR90Yje2UtVV5LTtbY6zbqybV5qfOSrN0l6GtrKPSDC8ePsrBXcfYQclU\nj31iNtejbDyW6t+qaYansbqWnTlH/+piq3GJ2sFaB1NztktJzFZUXYnClL5Lc/Tp04eUlJQSF3wy\nMjIeq9agELbI/MuGwiaoVCoyMjL092pmZGSQlJRUou/SUn2uxv4XQlietJhrOVNbr1W5Bc2QJGYh\nqp+0mOuQ8m7dskW2Hp8Q1iSJuY5QqVQkJibquzwSExNJSkp6pJvgrcUwViHqIunKqCPKe3Choiez\nasqj3q4oRG1ne80lYTXG7iU1vAnesDVd061VScqiLpMWcx1XF5/MEsLWVVuLeffu3axYsaK6qhNV\nVBuezBKirqi2FnNAQABnzpwp833p9zELIURdZ9XEvGrVKi5dugRAcHCw0XEM38ccHBzM3LlzrRmS\nEELUmNTUVJPGU2i1Wq2VY9GLiIhg6tSp1VWdEELUStWamIUQQlROOnWFEMLGSGIWQggbI4lZCCFs\njCRmIYSwMZKYhRDCxlj1Pub9+/eTkJDA9evX8fT0BGDChAm4u7sbHT84OBhfX19rhiSqwbVr12je\nvHlNhyGEzUlNTWX9+vWVjmfVxOzt7U1ISAj+/v7ExsaSlpbG0aNH6d+/v36c0k/+mRK0sF1FRUXU\nr1+ftLQ0HB0dazocIWyKqQ/QWbUro2HDhoSHhzN58mQUCgXFxcXY29uXGGfkyJFERUURFRVFmzZt\nrBmOqAZXrlzhzp07XLlypaZDEaLWsmpiXrVqFa6uruTn5/Pll18SExNDt27drFmlqGGnTp3C19eX\nxMTEmg5FiFrLql0ZERER1py8sEGnT59m+PDhJCQkEBQUVNPhCFEryV0ZwqJOnTrFiBEjpMUsxCOQ\nxCwsKjMzk06dOnH16tWaDkWIWksSs7CYO3fu4OHhgVKppLCwEHk/lhDmkcQsLObMmTM8++yzAPj4\n+JCRkVHDEQlRO0liFhZz6tQp2rVrB4C/v7/0MwthJknMwmJOnz5N27ZtgYc/JZaQkFDDEQlRO0li\nFhaTmJiIv78/8DAxS4tZCPNIYhYWUVxcTHFxMUrlw1vj/f39pcUshJkkMQuLuHz5Mi1atND/7+np\nyZ07d2owIiFqL0nMwiIM+5d1HBwcyM/Pr5b67927R1RUVLXUJYS1mZSY7969S1paGpcvX7Z2PKIc\ne/furekQKmR4R4bO008/XW3bzMqVKwkJCeHmzZvVUp8Q1mRSYv7yyy/Zvn07e/bssXY8woiTJ08y\ncOBAm36azliLubr6mbOysti9ezeRkZEsWrTI6vU9jn755ReKiopqOgzxX5Um5hs3buDk5IS/vz+u\nrq7VEVO1OXr0KJmZmRWOo9VqOXHiRDVFZNy8efOYNWsWO3bsqNE4KpKRkYFarS7xXXXdmbF8+XIm\nTpxI//79SUhIIDk5udxx79+/z7lz56weU22SkJDA0KFD+fzzz2s6FPFflSbmXbt2ceHCBa5cufJY\nPWIbERHBkiVLeOutt3j55ZdZs2YNOTk5JcbRarVMmTKF9957j3nz5tXI/B85coT69eszceJEdu/e\nXe31lyctLY19+/axb98+du3aZfRXaaqjxZyens7BgwcZNmwYCoWCWbNmsXDhQqPj3r17lyFDhjBm\nzBj27dtn1bhqC61Wy8SJE9m3bx9btmzhxo0bJYYXFhaSlJRULbFcuXJFWu3/VWlibt++PW3atLG5\npHz//v1KW7vGaLVa5s2bR3JyMt9//z179+5l8+bNKJVKXnrpJX13TXFxMf/3f/9HvXr1iIuL486d\nO8ycObPal8P8+fOZPXs2Xl5eKBQKbt++Xa31G1NcXMyoUaM4cuQIv/76K/Hx8cycObPMeH5+flbf\nqZcsWcKUKVOws3u4Kffq1YvU1FQuXbpUYrzs7GyCgoIICwtj//79REREWPxAp9VquXXrlkWnaW1f\nf/01HTt2pG3btkRERDB58mT9sIKCAt5++23eeustPvroI3Jzc60Wx759+wgKCmL06NEUFhZarZ7a\notL3MXfu3Bk/Pz98fHxIT0+vjpgqtXv3bubPn4+3tzdFRUW8+uqr9O3bFycnp0rLrl27lry8PFav\nXo1CoQBArVYzfvx4hg8fTlhYGFu2bMHR0ZGnnnqKGTNmALBs2TJmzpzJ5MmTCQ0N1U/Px8fHal08\nBw4cwM/PjyeffBKAgQMHsnv3bt5++22r1GeqL7/8kl69ejF79uwKx7O3t6eoqAitVqtf1qbIy8sr\n03IzJjs7mxMnTrBkyZIS38+ZM4dZs2bp+5sfPHjAe++9x5w5c3jxxRcB2L59O8OGDePOnTt07drV\n6PTVajVubm4mxZyUlERISAgZGRkEBgYyZ84cVCqVSWVryu3bt1m9ejUHDhwA4IUXXmDjxo389NNP\nBAYG8uabbzJ48GBGjx7Nli1b6NOnD3PmzNE/RATg6+uLg4PDI8Wxa9cu/vGPf/DLL7/wz3/+k1Gj\nRrF58+Y6/dNkCm0lTcDIyEhOnz5Nu3bt0Gq1jB071qyKoqOj9S+1GTNmjNFx5s6dy9y5c8u9//Xu\n3btMnz4dlUrF3//+dzw9Pbl9+zbR0dEcPnzYpNOgtm3bMnHixAoTxf79+7l+/TqjR48u8b1Wq+Wz\nzz7j5MmT+u+uXbuGu7s7r732GoMHD8bLy6vM9IqLiwH0rTpTaLVa+vbty1dffaX/gdpr167x0Ucf\n8e2335o8HUtLT08nKCiI2NhYkw6Er7/+OhEREWX6n0srLCzk559/5vvvvyc5OZkWLVqYlMwnTJjA\n888/X+b7xYsXc+HCBf3/wcHB9OjRo8Q4ubm5zJo1q9yzkJSUFOrVq8eQIUN4+eWXcXFxKTOOVqtl\n06ZNfPfdd6xcuZL27duzefNm1qxZw6JFi3juuecqnYeaEhYWxmuvvcbAgQP132VmZjJo0CB8fHx4\n4403GDVqlH5Yeno64eHh+uVVXFzMxYsX+dOf/sSwYcPo0aNHmZ+Oq8y+fftYv34927Zt0zdwvvrq\nK7Zv38769ev1DyzVVkqlssR2o8txlak0McPDi2RnzpyhQ4cOdO7c2awAV6xYwaRJk4iIiGDq1Kn6\n70v/GOt7773H8uXLyc/Pp6ioCBcXF+7fv4+DgwNKpZKhQ4cyYsQIbt68SVFREc2bN+fq1at4e3sD\nD1sBTz75JNeuXcPe3p5GjRqRnJyMWq2msLCQ7OxsWrRoweXLl3FycsLHx4fU1FQaNGiARqMhJydH\nP1ylUuHl5cWNGzd44oknyMnJ4f79+/rhLi4uuLm5cfr0aeLi4oiOjkahUDBixAhatmzJzZs32bdv\nH/Hx8Tg5OfHUU0/x4osv8sYbb5CcnIynpydKpZKMjAyaNWvGjRs3OHv2LMePH2f//v0MHjyYv/71\nryXm6c9//jN79uzh1q1bVp2nP/74g8aNG5OVlYVGo9EPnzdvHiNHjiQgIABfX1/S09PJy8vTDy89\nTxs3bmT79u3Uq1eP+/fv61tB+fn5uLi4kJubi0KhQKVS8fTTTzNixAiaNWtWrfPk5uaGSqUiLS2t\nzDwdP36co0ePcuTIEe7du0e9evV48OABWq1WP089evRg3Lhx5OTk6NdTVlYWkZGRXLlyBScnJ4qL\niykoKNBvz/b29jg6OqLRaHBycqKoqIjCwsISwx0cHHjw4AHOzs4UFhaWGK5UKlEqlfrhBQUFJfYX\npVKJvb09eXl5qFQqo/tT69ateffdd2nWrFmJ/WnDhg24urry8ssvm7Q/5efnExkZSUJCQpXnycvL\ni88//5ybN2+WWE8nTpzg66+/rvI82dnZkZeXZ3Q9lbftOTs7k5uba5X15OPjw5o1a/Tb3qpVqyyX\nmCMjIxk7diwbNmzg3XffrXJSBli0aBEzZszQ9wkaY+rRxJalp6fz448/smvXLlq2bMmwYcP0rab4\n+Hi+//57fvnlF6N91VqtVl8mMDDQ6CninDlz6Nq1KwMGDLD6vJT2888/s27dOjZv3lztdQvxODA1\nx5l0nuDs7MwXX3yBs7Oz2QEFBASwdu1amjRpYvY0agMfHx+Cg4MJDg4uM6xDhw506NDhkaYfFBTE\n2rVrqyUxHzlyhNDQUH0/q1arrdFuFCHqCpMSc1paGh4eHvq+UnMMHTrU7LLifzp06EB8fDzFxcVV\n6rOuqrS0ND788EP27t2Lj4+P1eoRQpRl0p794MEDPD098fT0tHY8ohIKhYLu3buzbNkyq93zWVRU\nxOjRo1mxYoUkZSFqgEmJ2dvbm969e9O7d29rxyNMEBERQXFxMf369eP8+fMWn/7s2bMZOHAg3bp1\ns/i0hRCVM6krIzs7m23btmFvb1/urW6i+jg4ODB16lSGDBlCaGgo3t7eFrvnMz8/Hzs7u3KfnhNC\nWJ9JiVl3F8Xvv/9u1WBE1fj7+7N3716uX79u0ScSGzduXKUHQoQQllVpYr569SqxsbHk5uaSlJTE\n0qVLqyMuYSI7OzuaNm1a02EIISyo0j7mZcuW4eTkxFtvvVXmtY5CCCEsr9LEvHr1avr378+ePXv4\n7bffqiNKgb2fAAABMElEQVQmIYSo00zqY1ar1YwaNarEc/NCCCGsQ37zTwghbIwkZiGEsDGSmIUQ\nwsZIYhZCCBsjiVkIIWyMJGYhhLAxkpiFEMLGSGIWQggbI4lZCCFsjFV+gra4uJjFixfj6OhI27Zt\nycjIoKCgAFdXV15//XVrVCmEEI8NqyTm3NxcXn/9dVxcXNi2bRtarVb/C9mllf6VbCGEqOssmphX\nrVrFpUuXAGjTpg0KhYIJEyawbNkyAKPv+B05ciQjR44EIDg4uNb/SrYQQpQnNTXVpPEUWku+Yf2/\nrly5woIFC+jWrRv+/v5cuXKFgoICGjVqxKuvvmrp6oQQ4rFilcQshBDCfNKpK4QQNkYSsxBC2BhJ\nzEIIYWMkMQshhI2RxCyEEDZGErMQQtgYScxCCGFj/h9DYY5UqWiYmwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ac1bf2f4978>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAWYAAACNCAYAAACE0tDjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XtUE2f6B/BvDOEWgwTBC5eCogWpeG9BXVHES9nSy5Ei\nStdtS7UrtmqpHEEPrVirVdfLCnW3olKPF5AWaauIrRWtukVZ6eLRbosX7jdFFCRCgITM74/+khNC\nQsKQhEGezzkeJ+GdmWduz7zzzpsZHsMwDAghhHDGgN4OgBBCSEeUmAkhhGMoMRNCCMdQYiaEEI6h\nxEwIIRxDiZkQQjiGEjMhhHAMJeZedunSJcTGxiIzMxP3799HTEwMKisrsXv3bp3jdNX1PCYmpkfx\nmKJbe09iiouLg1wuN6hsXl4eFi1ahNjYWKxYsQK3b99GXl4evv766w5lXn/9dURHRyM6OhpXr17t\nMI/MzEzk5uaiubkZSUlJyM3NZRV3cXExoqOjDSqblJSEsrIyJCcno729Xev6ksvluHTpEv7xj3+w\niqc726ChoQHHjx9XxQUAZWVlSEpK0jvu/fv3sWbNGnz44YdYvnw5iouL8eDBA3zwwQeIiYnBmTNn\nkJ6ejoaGBlbL0V9Y9HYA/d3JkydhZ2eHqqoqtLS0AACuXLmCvLw8SCQSVYIeM2YM+Hw+Ll26hEWL\nFiE/Px8NDQ0QCASIjo7GRx99BJFIhIqKCtW0GxoasGnTJtja2mL+/PkYNGgQDh06BJlMhg0bNuDA\ngQOQyWQQCoUICwvD2rVr8fLLL8POzg75+floa2vD2rVrsWfPHlhYWMDJyQnLli0DADQ2NuKTTz6B\nQCDA888/DxcXF6SkpMDLywsODg549dVXsWnTJjg4OEAikahiyszMxOXLlyEQCDB9+nRMmTIFycnJ\nYBgGgYGBsLe3x1dffYUnT57gvffeAwBIJBJs3rwZcXFx2L59O+zs7ODj44PAwECkpaVhxYoVqumH\nhoYiLCwMEokECQkJWLhwYad1Hh4ejrCwMNXnb7/9tlOZ33//HaWlpZg8ebLebZiXl4f9+/fD2dkZ\nbm5uiIyMxKVLlyCVSvWOCwC5ubkYOHAgiouLVSfGqqqqDuvl2WefRUFBgcEnzpCQEPj7+6OpqQlb\ntmwB8McJ4JVXXgHwx343evRo5ObmorW1FRs2bICtrS0A4NixY4iIiMDRo0c7Tfd///sfvv76a8hk\nMoSHhyM5ORnOzs6oqqrChg0bsHfvXsTFxcHJyQmPHz/G9evXAQDLli3Dc889hzVr1iA+Ph5Hjx7F\n+++/b9Cy9EdUY+5l06ZNw5w5czp8N3XqVPj5+eHUqVOQy+UQi8W4desWAOCll16Ck5MTCgoKMGjQ\nINTW1uLy5cuYMWMG4uPjIRKJVNORSCSQSCSYM2cORowYgSNHjuCzzz7D2rVrUVRUBDs7O8THx0Mi\nkaC+vh4+Pj5YvHgx0tLS4ODgAEtLS9y5cwdlZWUYN24cAgICVNPm8Xh47bXXMGHCBFy5ckUV94cf\nfoibN28iOzsbS5Yswbp168Dj8Tos37x587B9+3acP38eqampEIlEcHJywo0bN+Do6Ijg4GAMHToU\nN2/eBABERUUhKioKNjY2qK2thb+/PyZOnAixWNwhKasTiUQYMED77n3ixAnExcUhLi4OtbW1neLj\n8XiYPHkypk+fbsgmBAD4+/vjk08+wc2bN8Hn8/HWW29h4MCBBo07bdo0zJ49u8N3muvFxcUFoaGh\nBsfj7OyM+Ph4eHh44Pfff9dapry8HA4ODggLCwOfz1d9f+fOHYjFYgDAjh07EBcXhx07dgAADh8+\njMGDB8PR0RE3btyAXC5HbGwsli5diuzsbLS0tMDJyQkXLlzAzp07cfHiRcycORMjRozArl27EBwc\nDLFYjJqaGoOXpT+ixMxhcrkcc+fOxcqVKzF27FgAgFAohEwmg4+PD1auXIkZM2YAAFpbWwF0bIqw\ntLREbGwsnjx5gpSUFNXleltbG+zs7GBh8ccFk1QqhUAggFAoBADY2tpi5cqVCA4OxvDhw/HBBx/A\n0dERu3btUk376tWrKCgowJQpU1TzVI7P4/HQ3t6O1tZWMAwDhULRYbksLS0BAFZWVpDL5ViwYAGW\nLVsGLy8vHDlyBG1tbfD19VWVf+utt/D1119DoVAgLi4OAoEAn3/+eZfr7smTJ50SrlJoaCi2bt2K\nrVu3YsiQIRCLxaisrATwR03VwcGhy2lro1wma2vrbo+rjeZ6YRsP8Md6BgA+nw+FQqFqRpg2bRpe\neeUVnDhxQnXiV5ZTiomJwdatW1VNIa2trYiMjERERAQ8PT0hEAhUJ0DlPBsbGxEYGIjVq1ejpKQE\n1dXV2LRpE8LCwlSVEJlM1u1l6k+oKYODxGIx8vPzsWPHDnz22WfIzs7Gq6++iurqagCAt7c3UlNT\n8fHHH2P06NEIDg7GunXrUFhY2GGH5/F4SEpKgkgkwvPPP48hQ4bgo48+glQqxaZNm1BRUYFPP/0U\nrq6uHWp3M2bMwLp16yAUCrF+/Xrs3bsX1tbW8Pb2VpVxcnLCDz/8gDNnzqCxsbFTO/BLL72EjRs3\n4sKFC6rEoPTNN9/g3LlzCAwMhLe3N3bt2gWhUIjIyEjU1tYiPz8fAwYMQFtbGwBgzpw5yM3NRVVV\nFfbv3w+RSARfX1/U19d3aso4ceIE8vPz8fjxY8TExODhw4fIyMhAXl4ebG1t8dJLL6k+A8Crr76K\nN954A1u3boWVlRXs7e3x7LPPdnub5eTkoKioSHUC7Q4XFxecOXOmw3eLFi3qsF66686dO9iyZQta\nWlrg6ekJ4I9afUpKCgQCAcRiMe7cuYOCggK0trbC2dlZNa7m9lK3ZMkSrF+/HjY2NoiOjoZcLsfm\nzZtRW1uLjRs3wt/fH/Hx8bCxsYGVlRU8PT2RlZWF+vp67N27F6NGjcLSpUshEAi6vUz9CY8eYkTM\nKTMzE8OGDcO0adN6OxSDlZWV4ZdfflF99vb2RmFhoeqzXC4Hj8fr0G6t7vvvv0dzc7PqM8Mwqtq8\njY0NgoODuxVPU1MTfvjhB9VnS0tL1UkMAJ555hkcP35c1fzQXf/6178QFhYGR0dHvWVjYmK6PZ+G\nhgakpaUhKiqKVXz9AdWYCdHD3d0d7u7uHb7z8fFRDefl5aG8vFzn+C+++KJR4xEKhViwYEGXZY4f\nP856+hEREcjOzsbixYtZT6MrZ8+eRXh4uEmm/bSgGjMhhHAM3fwjhBCOocRMCCEcw6k25qVLl8LV\n1bW3wyCEEJOorKzEgQMH9JbjVGJ2dXVFQkJCb4dBCCEmYWh+o6YMQgjhGErMRqR8LgAhhPQEp5oy\n+rKcnBxcv34dlZWVsLGxQVBQUG+HRAjpo6jGbCRSqRReXl7w8PAw+KlihBCiDSVmI/Lw8DCoHDV5\nEEK6Qk0ZZkZNHoQQfajG3ENd1X61/U0qlcLa2pqaPAghOnEyMasnNK5f9muLr66uDllZWcjJyUFO\nTk6nv2s+EIeYnqH7FNf3N9I/cK4pQ3mp//DhQwDgxGX/9evXMWHChE7f5+Tk4OHDh52S74ABA1Tt\nzdpqxYa2RRPjUd+Guranvr8RYi6cqzErezdIpdIe9XQwZq1b1/jqsWrDhQRs6LI/zTVF9ROoclh5\nRaOrHFdobhdd+3Vfusok+nEuMQMdE1p3k1tOTk6HZgT1f92h3Lm7OpDZxGfofDWHjTE9Y5TjGkPi\nNvRkr+9Ea0gMXSVSNvpzYn5aloMNTiZmNpQbUSqVwsPDo9OByPZg03Ygm3KHYXOAaRtHeYLSVwM0\ntBwXaZ50u1pfhp7s2ZxoTZWY1SsFR44cUW0n9WH1qwB9VwR9DSXmPkRXElImGOWLJnUdiIZsbM1L\nWuX4+m7qGRq3rr+xOcC0XRlkZWWhpqYGHh4esLa27vKkpDyR6SvHReonXfVlN2dC0rXNNJMnG+o9\neBiGUW0n9WFl5UNzuK/2+tF2LPRWDF0Nd6ccG30mMWsmRfV/6rVazTcyazL00ldbolLe1GOTxAzp\nvcHmANN1ma5cD4b2ANFVzpw7IxvqN1l745eXuraZZvJkS3O7qH82ZLivUb9S7a3KAhcSM+d6Zeii\nraeD+oYz5BJUvUajr5dHVzt3d3f87vTeMPQAU+894OHhgdLSUtWwOvW/dUVbOfUfw9TX10MsFuPW\nrVsdesyYo/eMZk8JXT0nNJfd0B4WPe2J0dU2M3RfMXSZPDw88OjRI9Ww+nbXtQ9wga5eMcphZcVE\n/Tgx9wlGWwyaw8o3fovFYr3lenIs9Jkas5L6TtfdDdedyzxjtkN2p/eGoW2h5qihqtdClTVAbW33\npqytqjdRKJsHDG1K0reOetI0pa6rbaZrGxr7JiHX6atdamtSY/OIg56sx67uT/XkWGATE+vELJFI\nUFRUhJaWFraT6DE2NYPeuswzZi3GlO1vOTk5OHToUKf2dSU2J5Ge6OqA0EWZcPWtI+UVi67pmaO3\nRV+++WoIzeXT1SavvDfE5mpU2z0WbTeEDU3ghu7jhpZjs6+wasrIzs6GVCqFk5MTbty4AYFAgJCQ\nkC7HOXXqFOrq6gAAb7/9NpvZGgUXL/O6i21PE0On7ejoyKkbR91tnlEm3LKyMoOWo6tmHM2mGm1N\nOoZSv2xXT0jKy3ourXNjUdZC1beFeq1YeQX73//+F4DubVFRUQFHR0fV+q+rq4Obm1unY0F5Es/P\nz0dWVlaXTW8PHz406Y/WNCsI3ZkXq8Qsl8sRGhrarXHu3r2L6OhobNu2rcP36enpSE9PB/DHATVj\nxgyUlpbCxsYGAFQbicfjqYbV/8a2XH19PbKyslQbGAAqKirg5uYGGxsbs8TQ3ViV5WxsbCCVSrs1\nbeXOrb6syh1dSTkNzWn3JFbN6WkOK8cRi8Vax1FuCzbrVSgU9mj9Dx48WLVu1IfVPxu6r5SWluKn\nn36Cvb19h+kwDKNzO+ladmNuC33rv6fllPuX+rbQHC4tLdW7TLpoHgtSqRSlpaXg8/kAdG9D5bBm\n0lefrq71auj61xaDoVgl5tOnT+Phw4dgGAY8Hs+gGrByo/F4vA7fh4eHIzw8HMAf78My18+us7Ky\nOp3Jm5qaIJVK9db+zU0ZK/DHRmcTn7KWpkzMDx8+hKOjI5ydnY26zjVjBfTfpHr06JHOcbi2LXpi\n1qxZ8PDwgIeHBwYPHtzlz/zd3Nx6vF0M2RZA1+u/p+UAcHobKte3KY6FnmCVmCdPntzt5ghvb28k\nJyfDxcWFzSyNTlkbaWlpUZ3p1IefNkFBQaioqFDteBUVFZw+YJ5G6olYVy+QoKAgnUmbGB9X1zer\nxFxfX4/U1FRMnjwZXl5eBo2zYMECNrMyGWWCcnV1VW0ULm4gYzIkMZDeR9vGvLi4vln1yoiNjcW4\nceOwf/9+LFu2zNgxmVVfSlbKS0a2+tKyEtKfsaoxL1myBJ6enoiJicGwYcOMHRPRoGx2uXXrVp9J\nqOonka6G1dsidZUjpL9hVWM+cuQIwsLCsHPnTrzzzjvGjoloCAoKQkhICAYPHsyZmxP6lJWVdWtY\n398I6U9Y1ZjXr1+P5557Dm+//Ta8vb2NHRPRoS/UlvXdVNUcVu8q1V9uxJobXX30PawS82uvvYaq\nqircvn0bjY2N8Pf3N3ZcRIu+kJj13VTVNdzVOIQdzSawp/EHLE8rVon5119/VQ3fu3ePEjPpRNeN\nxq5uQNLNSePSPEkqf2xCuI9VYo6MjERFRQWqq6vh7Oxs7JgIIUZEJ7m+h/VDjE6dOoWysjJ88803\nxoyHEEL6PdaJ2cLCArm5uarfgxNCCDEOVok5JSUFTU1NGDVqFORyubFjIoSQfo1VYo6MjISDgwPe\nf/992NnZGTsmQgjp11g3ZdjZ2WHfvn2qx/sRQggxDlaJOSsrC7a2tpgyZQosLS1x6tQpY8dFCCH9\nFqvucn/6059w9+5dCAQCTJ48GUOHDjV2XIQQ0m+xqjHv3r0b5eXlKC8vx8mTJ40dEyHEROjn2X0D\nq8Rsb2+PcePGoa2tDdeuXTN2TIQQE+nq4VCUtLmDVWKOjo7GnTt3IBQKsXDhQty/f9/YcRFCzIye\n6McdrHtlFBYWYvTo0bh58yYyMzONGRMhxARsbGzoqX19BOvEbG1tjfPnz8PCwsKgt8BmZ2dj9+7d\nbGdHCOkh5fvtzPFMb2oW6RnWidnd3R0KhQJeXl5YuHCh3vLe3t5oa2tjOztCiBGoP9BImTwVCkWn\n52F3xZCkS80iPcOquxwA1NbW4t1338WhQ4d0lklMTMTdu3cBAEuXLtVaJj09Henp6QCAAQNYnycI\nIQZQT8xlZWXw8PCAo6MjQkJCUFdXh6CgIL2PBy0rK4O7u7vOvysUCtVLEKjZhB1WiTklJQVSqRR7\n9+6FtbW1znKrVq3q8PnMmTOdyoSHhyM8PBwAkJCQwCYcQogJab6bUR9HR0cAQEhIiIkievqxfh4z\nG7GxsazGI4SYnq7nNitr1ppNHvRGFNOhtgNC+iFlDw315gZtiVm9nJubm8EvBaaH8/cM6zZmQkjf\nFRQUhIqKCr3NDYaWAzo2eVBi7hlKzISQTrrqeaGrZn39+nXVMOkZSsyE9FNd1WrVe1509dJcJWUf\naaopGwe1MRPST+lKopq/EDQ02VJSNh5KzISQDsz5C0GiHSVmQkgnhtZ+6afXpkGJmRDSiaGJmX56\nbRqUmAkhrNDT6kyHEjMhhBVqizYdSsyEENaoJ4ZpUGImhLBGidk0KDETQgjHcOqXf5WVlfToT0LI\nU6uystKgcjyGYRgTx0IIIaQbqCmDEEI4hhIzIYRwDCVmQgjhGErMhBDCMZSYCSGEYygxE0IIx3Cq\nH/PSpUvh6ura22EQQohJVFZW4sCBA3rLcSoxu7q60g9MCCFPLUPzGzVlEEIIx5i0xnzu3DkUFhai\nqqoK9vb2AICoqCjY2dmZcraEENKnmTQxOzg44P3334eXlxcuXLiA2tpaXLlyBfPnz1eVSU9PR3p6\nOgBgwACqwBNCiEkz4dChQ7F161asWbMGPB4PCoUCfD6/Q5nw8HBkZmYiMzMTY8eONWU4hBDSJ5g0\nMScmJmLgwIFoa2vDl19+iZycHEydOtWUsySEkD7PpE0Z27ZtM+XkCSHkqUSNuoQQwjGUmAkhhGMo\nMRNCCMdQYiaEEI6hxEwIIRxDiZkQQjiGEjMhhHCMQYm5sbERtbW1KCoqMnU8hBBiVr/++mtvh9CJ\nQYn5yy+/xLfffoszZ86YOh5CSC84f/485HJ5b4dhdg8ePMCkSZPw22+/9XYoHehNzNXV1bCysoKX\nlxcGDhxojpgIIWZUXl6O0NBQHD16tLdDMbvs7GxERkbi448/7u1QOtCbmE+fPo3bt2+juLgYDMOY\nIyZCiBlt2rQJqamp+OKLL9Da2trb4XRLSkoKFi1apPpnyNtB1J08eRLx8fGwtLTEtWvXjBpbe3s7\ntm/fzmpcvc/KmDBhAvh8PhiGAY/HYzUTLqqqqsLKlSvx6NEj1XdJSUnw9fXtxah6V319ver5JrGx\nsRCLxb0cETG127dv4969ewgODkZRURGSk5OxcuXK3g5Lr7a2NqxevRpWVlbYvXu3KjeFhoYiIiIC\ntra2eqfR0tKCuro6uLq6YuPGjVi9ejWys7ONEh/DMFi+fDkmTpzIegJ61dbWdvjfVDZs2GDS6Std\nuXKF8fPzYwoKClTfXb16lYmKijLL/LlGKpUy27dvZ/z9/ZnMzEzmxIkTzNSpU5kdO3YwUqm0t8Mz\nmevXrzOzZ89mUlJSGLlc3tvh9IpFixYx+fn5DMMwTEtLC+Pn58dIJJIeTTM/P5+ZPXs2c/jwYaa9\nvd0YYXZQW1vLzJ07l0lJSen0tz179jAHDx40aDqnT59mNm7cqPr87rvvMufPnzdKjHFxcUxCQkKn\n7w3NcXprzCkpKbhx4wbGjx8PhmEQGRnJ6gRw6tQp1NXVAQDefvvtLss+fvxY6/dyuRx5eXn48ccf\nce3aNTAMg0mTJmHevHnw9/eHpaWl3jgyMzORlpaGkydPYsiQIarvX3jhBXzwwQdobW2FlZVVN5bM\nPJqbmyGTyfSWk0gkuHjxIs6ePYvffvsNvr6++OSTT7S+5La9vR3Hjh3D3r17ERkZicuXL8PC4o9d\n4uWXX8bBgwcxf/58g66U2traMHr0aMybNw+zZs0y2f0IPp+vc9pyuVwVvz4ZGRlISkpCcnIyMjIy\nMHPmTKxbtw5//vOfTXJlqFAoAHR8GURbWxv27duHgwcPIiQkBGvXru3wdh+GYdDY2Nij+fJ4PJ1v\nDCooKIBCocDkyZMBAFZWVoiKikJiYiLWr1/Pan5paWnYv38/9u3bh9TUVMycORPx8fHw9/fXO67m\n8c3n87VuCx6Phy1btmh9hPBf//pXvPbaawblqe+++w5RUVGqzx9//DH+8pe/YOLEiT3aB/bv348n\nT55gy5YtrKfBYxj9DcdXrlzBzZs3MXHiRDz//POsZrR7925ER0dj27ZtiI2NVX2v+QaT5cuXY9eu\nXWhra0N7ezuEQiGampogEAgwYMAAeHp6YsGCBRg2bBgUCgXq6+uRkZGBO3fuQKFQoK2tDUKhEM3N\nzeDxeLC2tkZzczOsrKygUCjg5uaGXbt2oaKiAlZWVnByckJlZSWGDBmCv//973BxccHf/vY3FBUV\nwcbGBmKxGNXV1Rg2bBgkEgmamprg6emJoqIiCIVCiEQi3Lt3D87Ozqivr4dUKlX9XSQSwcbGBrW1\ntXB1dcWDBw/Q2tqq+ru9vT0sLCxQV1eHZ555BmVlZSgtLYWzszOqqqrQ2tqKy5cvIzc3V3VwaVsm\nmUymWk82NjaYNWsWxowZgxdeeAE//fQTduzYgVmzZmHatGmwtbWFnZ0dfvnlFxw9ehRTp07F4sWL\n4evr26NlGjlyJHJycnD9+nVcu3YNEokEtra2aGlpAcMwsLW1RVNTk+rkqW87qS8Tn8+HpaUlpFIp\n5HI57O3tERUVhcDAQBQVFaG6uhpJSUkoLy+HWCzGlClTMHHiRPj4+KCiogI2NjYQCoWoq6uDk5MT\nUlNTUVJSgsOHD6O6uhoikQhNTU3YunUrfv/9dzzzzDPw8/NDUFAQqqqqIBKJwOfz0dDQgOHDh6Ou\nrg7Ozs4YNWoUSkpK4ODgAAB49OgRRowYgbKyMvD5fAwfPhxlZWW4du0a/vnPf0Iul2P06NHw9fWF\nUChEeno6goKCEBUVhe+++w4HDhzAm2++CYVCgby8PJSWlsLW1ha2traQy+WQy+WqdWJhYQELCwu0\ntLTA2toaMpmsw/FiYWEBPp+P5uZmtLS0wNPTE35+fggICEBlZSVEIhE2b96MVatWYdasWaipqUF7\neztcXV0REBCAbdu2wdraGo8fP4aLiwuqq6vB5/Ph5OSEmpoa2Nvbo729HRKJBG5ubqioqEBWVhae\nPHmCNWvWwM3NDVKpFCUlJcjMzMTNmzfB5/MhEAhUMWsuk0AgwKRJk+Dr64u5c+dCIpGwOp5WrFiB\n1atXY9iwYWhvb4e7u3un7eTu7o4ZM2bg+PHjcHZ2Rnl5ORwdHZGSkoIff/xR675nZWWF9vb2DjFr\nWyY3NzckJSWhpKSk0/GUmJho0AtZDUrMKSkpiIyMxMGDB/HOO+90LyP/vy1btmD9+vXYvn071q5d\nq7VMQkJCr74lu7y8HKtXr8Y333zTK/P/6KOPUFhYCEdHRwCAWCzGnDlzMG3aNFhbW7OeLsMwyMjI\nwPnz51XfDR48GCtXrsTQoUN7HLe5/fLLL4iPj4enpydkMhmqq6vx6aefYvz48WhsbMRPP/2Eixcv\norm5Wev448aNw/Lly7XWihiGQWFhIc6ePYvCwkKt48vlcty6dQv29vaYM2cOfHx8tE6rsbERn3/+\nOSZNmoR169bBwcEBZWVl+PHHH/HgwQOsWLECgwYNUpVvbW3FgQMHIJPJMG/ePIwZM8YotXeFQoEb\nN27g7NmzKCkpUX0/fvx4LF++vFP5n3/+mVUPjYkTJ2LZsmW9fi8qPz8fX3zxRZc3Aq9du4bDhw8j\nKSnJjJEZnuMMSsypqamQSCSwtrbGm2++ySqgzMxM1NXVQSgU4o033tBaprcTMwDMnz8fx44dUyVH\nc5o7dy6+++47g25c9HcMw+DcuXPg8/mYPXt2r8Rw//59nDt3DsXFxVr/bmFhgYiICLi7u5s5MjJr\n1ix8++23qpdAa4qPj8fMmTMxd+5cs8ZlaI4zqEGutrYWgwYNUrWTsbFgwQLW45rT4sWLkZaWZvY7\n0zKZDDKZjJKygXg8ntkPKk1Dhw7VWckgvWvJkiU4cuSIzuP44sWLnOu7rM6gxNzS0tJvzvqhoaEI\nCQkxe2IuKCjAhAkTzDpPQp5WixYtwvTp07U+RkImk8HV1dWgzgK9xaDE7ODggMDAQFPHwgkikQhu\nbm747bff4OPjY7b5/vzzz5g+fbrZ5kfI00woFCIrK0vVE0zTiBEjzBxR9xiUmBsaGpCRkQE+n6+3\nq9vTYMmSJTh27Bg2b95stnn+/PPPSExMNNv8CHnaubq6au0m2hcYlJiVvSi49qAPUwkMDDRrUmYY\nBjU1NXB2djbbPAkh3KU3MZeUlODChQtobm5GaWkpduzYYY64epWlpSUGDx6MmpoaDB8+3OTzKykp\n4fylFSHEfPQ+xGjnzp2wsrLCG2+8gXHjxpkjJk6YP38+fvjhB7PMi9qXCSHq9Cbmzz//HPPnz8eZ\nM2dQUFBgjpg44cUXXzTb86cpMRNC1BnUxuzo6IiIiAhERESYOh7O8PDwQGVlZbeev8DWr7/+iuee\ne86k8yCE9B30zr8uvPDCC/jPf/5j0nk0NDSonsVACCEAJeYuBQcH4/vvvzfpPK5evar1KVmEkP6L\nEnMXAgICcOnSJaNO8/Hjx1i1ahWOHDmCe/fuUfsyIaQTSsxdsLa2hkgkwoMHD4w2zX379sHFxQWN\njY2IiopCWloa/Pz8jDZ9QkjfZ9q7Wk+BefPm4ezZs0Z5WE1raysyMzNx+fJlCAQCvPfee0aIkBDy\ntDFJYlZKECaDAAACBklEQVQoFPjss89gaWmJcePGoa6uDjKZDAMHDsTrr79uilmaTHBwMBISEoyS\nmI8dO4aFCxdCIBAYITJCyNPKJIm5ubkZr7/+OoRCITIyMsAwjOrtJZo032DCNaNGjYKFhQVmzJiB\nqVOnYt68eQgMDOx2LwqFQoGDBw+a/GYiIaTvM2piTkxMxN27dwEAY8eOBY/HQ1RUFHbu3AkAWt9s\nEB4ejvDwcADo9Yfk63Lo0CG0tbXh6tWrOH36NBISErB+/XoEBwerlqmhoQFlZWUYP3681mmcPn0a\nAQEBEIlE5gydENIHGTUxr1q1CgBQXFyMTZs2YerUqbh69SqGDx+O5ORkjBkzxpizMytLS0sEBAQg\nICAADx48wKeffoo9e/bAz88P//73v2FpaQknJyc8evQImzdv7vRs5cTERBw+fLiXoieE9CUmacoY\nOXIkvvzyS9XnmTNnmmI2vcbJyQl79uxBcXEx7t69i7i4ONWbR27fvo34+HgIBAKMHTsWwB9d5Dw8\nPMzyQCRCSN9HvTJ6YOTIkRg5cmSH75599ll89dVXuHHjBmpqalTfU5c4QoihOJWYKysrOdvO3FNX\nrlzp7RAIIb2ssrLSoHIGvSWbEEKI+XCvfxohhPRzlJgJIYRjKDETQgjHUGImhBCOocRMCCEcQ4mZ\nEEI4hhIzIYRwDCVmQgjhmP8DJTn+zLHBxX4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ac1bf2f49b0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K562\n"
     ]
    }
   ],
   "source": [
    "hepg2_all_dels_tfs = pd.DataFrame()\n",
    "k562_all_dels_tfs = pd.DataFrame()\n",
    "for dels_fixed, dels_df, cell in zip([hepg2_dels_fixed, k562_dels_fixed], [hepg2_all_dels_tfs, k562_all_dels_tfs],\n",
    "                                    [\"HepG2\", \"K562\"]):\n",
    "    print(cell)\n",
    "    for name in dels_fixed:\n",
    "        df = dels_fixed[name].sort_values(by=\"delpos\")\n",
    "        seq = list(df[\"seq\"])\n",
    "        scores = list(df[\"mean.log2FC\"])\n",
    "        yerrs = list(df[\"se\"])\n",
    "\n",
    "        # get wt motif # for reference\n",
    "        del_id = df.del_id.iloc[0]\n",
    "        wt_tfs = tfbs_wt[tfbs_wt[\"del_id\"] == del_id][\"n_tfs\"].iloc[0]\n",
    "        df[\"delta_tfs\"] = df[\"n_tfs\"] - wt_tfs\n",
    "        motif_vals = list(df[\"delta_tfs\"])\n",
    "        dels_df = dels_df.append(df)\n",
    "\n",
    "        # plot\n",
    "        if name == \"HOTAIR__p1__tile2__minus\" and cell == \"HepG2\":\n",
    "            plot_peaks_and_tfbs((5.6, 2.5), 94, name, cell, scores, yerrs, \n",
    "                                motif_vals, seq, \"Fig_3C\", True)\n",
    "        if name == \"DLEU1__p1__tile1__plus\" and cell == \"HepG2\":\n",
    "            plot_peaks_and_tfbs((5.6, 2.5), 94, name, cell, scores, yerrs, \n",
    "                                motif_vals, seq, \"Fig_S13B\", True)\n",
    "#         if name == \"ZFAS1__p1__tile2__plus\" and cell == \"HepG2\":\n",
    "#             plot_peaks_and_tfbs((5.6, 2.5), 94, name, cell, scores, yerrs, \n",
    "#                                 motif_vals, seq, \"ZFAS_tile2.for_talk.pdf\", True)\n",
    "            \n",
    "#         if name == \"ZFAS1__p1__tile2__plus\" and cell == \"HepG2\":\n",
    "#             plot_peaks_and_tfbs((5.6, 1.5), 94, name, cell, scores, yerrs, \n",
    "#                                 motif_vals, seq, \"ZFAS_tile2.for_poster.pdf\", True)\n",
    "        \n",
    "    if cell == \"HepG2\":\n",
    "        hepg2_all_dels_tfs = dels_df.copy()\n",
    "    else:\n",
    "        k562_all_dels_tfs = dels_df.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. compare effect sizes w/ number of motifs gained or lost (for all seqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "922"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# hepg2\n",
    "hepg2_all_dels_tfs[\"delta_tfs_abs\"] = np.abs(hepg2_all_dels_tfs[\"delta_tfs\"])\n",
    "hepg2_all_dels_tfs[\"del_abs\"] = np.abs(hepg2_all_dels_tfs[\"mean.log2FC\"])\n",
    "hepg2_all_dels_tfs[\"delta_tfs_log\"] = np.log(hepg2_all_dels_tfs[\"delta_tfs\"]+1)\n",
    "no_nans = hepg2_all_dels_tfs[~pd.isnull(hepg2_all_dels_tfs[\"mean.log2FC\"])]\n",
    "sig_only = hepg2_all_dels_tfs[hepg2_all_dels_tfs[\"sig\"] == \"sig\"]\n",
    "len(sig_only)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delpos</th>\n",
       "      <th>mean.log2FC</th>\n",
       "      <th>sd</th>\n",
       "      <th>se</th>\n",
       "      <th>seq</th>\n",
       "      <th>padj</th>\n",
       "      <th>sig</th>\n",
       "      <th>wt_activ</th>\n",
       "      <th>wt_l2fc</th>\n",
       "      <th>wt_class</th>\n",
       "      <th>...</th>\n",
       "      <th>tile_end</th>\n",
       "      <th>del_id</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>del_num</th>\n",
       "      <th>n_tfs</th>\n",
       "      <th>delta_tfs</th>\n",
       "      <th>delta_tfs_abs</th>\n",
       "      <th>del_abs</th>\n",
       "      <th>delta_tfs_log</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>24</td>\n",
       "      <td>-0.359714</td>\n",
       "      <td>0.187131</td>\n",
       "      <td>0.066161</td>\n",
       "      <td>A</td>\n",
       "      <td>3.814204e-03</td>\n",
       "      <td>sig</td>\n",
       "      <td>0.024331</td>\n",
       "      <td>1.051891</td>\n",
       "      <td>sig active</td>\n",
       "      <td>...</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.24.1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.359714</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>36</td>\n",
       "      <td>0.804015</td>\n",
       "      <td>0.122571</td>\n",
       "      <td>0.043335</td>\n",
       "      <td>A</td>\n",
       "      <td>1.349856e-32</td>\n",
       "      <td>sig</td>\n",
       "      <td>0.024331</td>\n",
       "      <td>1.051891</td>\n",
       "      <td>sig active</td>\n",
       "      <td>...</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.36.1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.804015</td>\n",
       "      <td>-inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>39</td>\n",
       "      <td>-0.451598</td>\n",
       "      <td>0.178589</td>\n",
       "      <td>0.063141</td>\n",
       "      <td>C</td>\n",
       "      <td>7.015322e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>0.024331</td>\n",
       "      <td>1.051891</td>\n",
       "      <td>sig active</td>\n",
       "      <td>...</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.39.1</td>\n",
       "      <td>39.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.451598</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>40</td>\n",
       "      <td>-0.451598</td>\n",
       "      <td>0.178589</td>\n",
       "      <td>0.063141</td>\n",
       "      <td>C</td>\n",
       "      <td>7.015322e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>0.024331</td>\n",
       "      <td>1.051891</td>\n",
       "      <td>sig active</td>\n",
       "      <td>...</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.40.1</td>\n",
       "      <td>39.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.451598</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>41</td>\n",
       "      <td>-0.635980</td>\n",
       "      <td>0.167341</td>\n",
       "      <td>0.059164</td>\n",
       "      <td>G</td>\n",
       "      <td>1.241464e-18</td>\n",
       "      <td>sig</td>\n",
       "      <td>0.024331</td>\n",
       "      <td>1.051891</td>\n",
       "      <td>sig active</td>\n",
       "      <td>...</td>\n",
       "      <td>65187624.0</td>\n",
       "      <td>enhancer_NEAT1__pX__tile2__minus</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...</td>\n",
       "      <td>1.2.0.0.41.1</td>\n",
       "      <td>41.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.635980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    delpos  mean.log2FC        sd        se seq          padj  sig  wt_activ  \\\n",
       "13      24    -0.359714  0.187131  0.066161   A  3.814204e-03  sig  0.024331   \n",
       "25      36     0.804015  0.122571  0.043335   A  1.349856e-32  sig  0.024331   \n",
       "28      39    -0.451598  0.178589  0.063141   C  7.015322e-09  sig  0.024331   \n",
       "29      40    -0.451598  0.178589  0.063141   C  7.015322e-09  sig  0.024331   \n",
       "30      41    -0.635980  0.167341  0.059164   G  1.241464e-18  sig  0.024331   \n",
       "\n",
       "     wt_l2fc    wt_class      ...          tile_end  \\\n",
       "13  1.051891  sig active      ...        65187624.0   \n",
       "25  1.051891  sig active      ...        65187624.0   \n",
       "28  1.051891  sig active      ...        65187624.0   \n",
       "29  1.051891  sig active      ...        65187624.0   \n",
       "30  1.051891  sig active      ...        65187624.0   \n",
       "\n",
       "                              del_id  \\\n",
       "13  enhancer_NEAT1__pX__tile2__minus   \n",
       "25  enhancer_NEAT1__pX__tile2__minus   \n",
       "28  enhancer_NEAT1__pX__tile2__minus   \n",
       "29  enhancer_NEAT1__pX__tile2__minus   \n",
       "30  enhancer_NEAT1__pX__tile2__minus   \n",
       "\n",
       "                                            unique_id     dupe_info del_num  \\\n",
       "13  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.24.1    24.0   \n",
       "25  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.36.1    36.0   \n",
       "28  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.39.1    39.0   \n",
       "29  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.40.1    39.0   \n",
       "30  WILDTYPE_BUT_HAS_SNP_DELETION__Enhancer.noflip...  1.2.0.0.41.1    41.0   \n",
       "\n",
       "   n_tfs  delta_tfs  delta_tfs_abs   del_abs  delta_tfs_log  \n",
       "13  17.0        0.0            0.0  0.359714       0.000000  \n",
       "25  16.0       -1.0            1.0  0.804015           -inf  \n",
       "28  17.0        0.0            0.0  0.451598       0.000000  \n",
       "29  17.0        0.0            0.0  0.451598       0.000000  \n",
       "30  13.0       -4.0            4.0  0.635980            NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_only.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'g' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-a7c2815f078f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0max_joint\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'g' is not defined"
     ]
    }
   ],
   "source": [
    "g.ax_joint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.jointplot(data=sig_only, x=\"delta_tfs\", y=\"mean.log2FC\", kind=\"reg\", space=0, size=2.5, \n",
    "                  stat_func=spearmanr, \n",
    "                  marginal_kws={\"hist\": False}, color=\"darkgrey\", scatter_kws={\"s\": 25})\n",
    "print(\"HepG2\")\n",
    "g.set_axis_labels(r\"$\\Delta$ motifs (del-ref)\", \"deletion effect size\")\n",
    "\n",
    "# add n-value\n",
    "g.ax_joint.annotate(\"n = %s\" % len(sig_only), ha=\"right\", xy=(.95, .05), xycoords=g.ax_joint.transAxes, \n",
    "                    fontsize=fontsize)\n",
    "\n",
    "g.savefig(\"Fig_3B.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# k562\n",
    "k562_all_dels_tfs[\"delta_tfs_abs\"] = np.abs(k562_all_dels_tfs[\"delta_tfs\"])\n",
    "k562_all_dels_tfs[\"del_abs\"] = np.abs(k562_all_dels_tfs[\"mean.log2FC\"])\n",
    "k562_all_dels_tfs[\"delta_tfs_log\"] = np.log(k562_all_dels_tfs[\"delta_tfs\"]+1)\n",
    "no_nans = k562_all_dels_tfs[~pd.isnull(k562_all_dels_tfs[\"mean.log2FC\"])]\n",
    "sig_only = k562_all_dels_tfs[k562_all_dels_tfs[\"sig\"] == \"sig\"]\n",
    "len(sig_only)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.jointplot(data=sig_only, x=\"delta_tfs\", y=\"mean.log2FC\", kind=\"reg\", space=0, size=2.5, \n",
    "                  stat_func=spearmanr, \n",
    "                  marginal_kws={\"hist\": False}, color=\"darkgrey\", scatter_kws={\"s\": 25})\n",
    "print(\"K562\")\n",
    "g.set_axis_labels(r\"$\\Delta$ motifs (del-ref)\", \"deletion effect size\")\n",
    "\n",
    "# add n-value\n",
    "g.ax_joint.annotate(\"n = %s\" % len(sig_only), ha=\"right\", xy=(.95, .05), xycoords=g.ax_joint.transAxes, \n",
    "                    fontsize=fontsize)\n",
    "\n",
    "g.savefig(\"Fig_S13A.pdf\", bbox_inches=\"tight\", dpi=\"figure\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
