{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 06__tf_tissue_sp\n",
    "# calculating tissue specificity of TFs (across HepG2, HeLa, and K562)\n",
    "\n",
    "in this notebook, i calculate the tissue specificity of TFs across the 3 cell types in our MPRAs using ENCODE RNA-seq data. then, i correlate motif coverage (# bp covered and maximum coverage) as well as average TF specificity with MPRA activities and specificities.\n",
    "\n",
    "note: the FIMO mappings and coverage calculations were done separately (see methods)\n",
    "\n",
    "------\n",
    "\n",
    "figures in this notebook:\n",
    "- **Fig 3C**: KDE plot of correlations of MPRA activity & specificity with each of the 3 metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from itertools import chain\n",
    "from decimal import Decimal\n",
    "from scipy import stats\n",
    "from scipy import signal\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from plotting_utils import *\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "expr_f = \"../../misc/03__rna_seq_expr/Expression.all.cells.txt\"\n",
    "tf_map_f = \"../../misc/04__jaspar_id_map/2018_03_09_gencode_jaspar_curated.txt\"\n",
    "fimo_f = \"../../misc/05__fimo/pool1_fimo_map.txt\"\n",
    "cov_map_f = \"../../misc/05__fimo/seqID_nFimoMotifs_Coverage.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "index_dir = \"../../data/00__index\"\n",
    "index_f = \"%s/tss_oligo_pool.index.txt\" % index_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ_f = \"../../data/02__activs/POOL1__pMPRA1__HepG2__activities_per_element.txt\"\n",
    "hela_activ_f = \"../../data/02__activs/POOL1__pMPRA1__HeLa__activities_per_element.txt\"\n",
    "k562_activ_f = \"../../data/02__activs/POOL1__pMPRA1__K562__activities_per_element.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>Mus musculus; human ortholog selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name          Gene ID                                comment\n",
       "0  MA0002.2      RUNX1  ENSG00000159216                                    NaN\n",
       "1  MA0003.3     TFAP2A  ENSG00000137203                                    NaN\n",
       "2  MA0007.3         Ar  ENSG00000169083  Mus musculus; human ortholog selected\n",
       "3  MA0009.2          T  ENSG00000164458                                    NaN\n",
       "4  MA0014.2       PAX5  ENSG00000196092                                    NaN"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_map = pd.read_table(tf_map_f)\n",
    "tf_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>31.920162</td>\n",
       "      <td>0.012392</td>\n",
       "      <td>39.989468</td>\n",
       "      <td>47.588514</td>\n",
       "      <td>34.921604</td>\n",
       "      <td>46.439177</td>\n",
       "      <td>10.383956</td>\n",
       "      <td>0.242817</td>\n",
       "      <td>24.534779</td>\n",
       "      <td>13.546740</td>\n",
       "      <td>15.400938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.729510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.052697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012272</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>34.321019</td>\n",
       "      <td>54.130236</td>\n",
       "      <td>22.474711</td>\n",
       "      <td>22.001707</td>\n",
       "      <td>80.122759</td>\n",
       "      <td>58.227585</td>\n",
       "      <td>42.266614</td>\n",
       "      <td>52.947538</td>\n",
       "      <td>119.129811</td>\n",
       "      <td>50.828462</td>\n",
       "      <td>64.445213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>4.715796</td>\n",
       "      <td>9.902367</td>\n",
       "      <td>2.222304</td>\n",
       "      <td>2.693052</td>\n",
       "      <td>8.641308</td>\n",
       "      <td>2.948341</td>\n",
       "      <td>3.283652</td>\n",
       "      <td>6.761766</td>\n",
       "      <td>7.120837</td>\n",
       "      <td>4.792727</td>\n",
       "      <td>6.164291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>11.382585</td>\n",
       "      <td>20.353400</td>\n",
       "      <td>12.625342</td>\n",
       "      <td>6.522418</td>\n",
       "      <td>34.807272</td>\n",
       "      <td>6.199940</td>\n",
       "      <td>3.842012</td>\n",
       "      <td>27.191509</td>\n",
       "      <td>15.206148</td>\n",
       "      <td>5.591343</td>\n",
       "      <td>3.786800</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id       A549    GM12878    H1-hESC      HUVEC    HeLa-S3  \\\n",
       "0  ENSG00000000003.10  31.920162   0.012392  39.989468  47.588514  34.921604   \n",
       "1   ENSG00000000005.5   0.000000   0.000000   0.729510   0.000000   0.000000   \n",
       "2   ENSG00000000419.8  34.321019  54.130236  22.474711  22.001707  80.122759   \n",
       "3   ENSG00000000457.9   4.715796   9.902367   2.222304   2.693052   8.641308   \n",
       "4  ENSG00000000460.12  11.382585  20.353400  12.625342   6.522418  34.807272   \n",
       "\n",
       "       HepG2      IMR90       K562       MCF-7       NHEK  SK-N-SH        \n",
       "0  46.439177  10.383956   0.242817   24.534779  13.546740      15.400938  \n",
       "1   0.052697   0.000000   0.000000    0.012272   0.000000       0.000000  \n",
       "2  58.227585  42.266614  52.947538  119.129811  50.828462      64.445213  \n",
       "3   2.948341   3.283652   6.761766    7.120837   4.792727       6.164291  \n",
       "4   6.199940   3.842012  27.191509   15.206148   5.591343       3.786800  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr = pd.read_table(expr_f)\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>FLIPPED__Enhancer.flip.antisense__chr19:185263...</td>\n",
       "      <td>32</td>\n",
       "      <td>52</td>\n",
       "      <td>+</td>\n",
       "      <td>28.7347</td>\n",
       "      <td>2.680000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAAGAGGAGGAGGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif                                          unique_id  start  end  \\\n",
       "0  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     60   80   \n",
       "1  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     60   80   \n",
       "2  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     57   77   \n",
       "3  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     57   77   \n",
       "4  ZNF263  FLIPPED__Enhancer.flip.antisense__chr19:185263...     32   52   \n",
       "\n",
       "  strand    score          pval          qval                    seq  \n",
       "0      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA  \n",
       "1      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA  \n",
       "2      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA  \n",
       "3      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA  \n",
       "4      +  28.7347  2.680000e-12  2.160000e-07  GGAGGAGGAAGAGGAGGAGGG  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = pd.read_table(fimo_f, sep=\"\\t\")\n",
    "fimo.columns = [\"motif\", \"unique_id\", \"start\", \"end\", \"strand\", \"score\", \"pval\", \"qval\", \"seq\"]\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqID</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>fractionBPcovered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>99</td>\n",
       "      <td>0.868421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>0.394737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>24</td>\n",
       "      <td>53</td>\n",
       "      <td>101</td>\n",
       "      <td>0.885965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>84</td>\n",
       "      <td>0.736842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.473684</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               seqID  maxCov  numMotifs  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       5         16   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       6         10   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...      24         53   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...      12         24   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       9         13   \n",
       "\n",
       "   numBPcovered  fractionBPcovered  \n",
       "0            99           0.868421  \n",
       "1            45           0.394737  \n",
       "2           101           0.885965  \n",
       "3            84           0.736842  \n",
       "4            54           0.473684  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_map = pd.read_table(cov_map_f)\n",
    "cov_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "index = pd.read_table(index_f, sep=\"\\t\")\n",
    "index_elem = index[[\"element\", \"oligo_type\", \"unique_id\", \"dupe_info\", \"SNP\", \"seq_name\"]]\n",
    "index_elem = index_elem.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ = pd.read_table(hepg2_activ_f, sep=\"\\t\")\n",
    "hela_activ = pd.read_table(hela_activ_f, sep=\"\\t\")\n",
    "k562_activ = pd.read_table(k562_activ_f, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_reps = [x for x in hepg2_activ.columns if \"rna\" in x]\n",
    "hela_reps = [x for x in hela_activ.columns if \"rna\" in x]\n",
    "k562_reps = [x for x in k562_activ.columns if \"rna\" in x]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. find expr of TFs in HeLa, HepG2, K562"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>31.920162</td>\n",
       "      <td>0.012392</td>\n",
       "      <td>39.989468</td>\n",
       "      <td>47.588514</td>\n",
       "      <td>34.921604</td>\n",
       "      <td>46.439177</td>\n",
       "      <td>10.383956</td>\n",
       "      <td>0.242817</td>\n",
       "      <td>24.534779</td>\n",
       "      <td>13.546740</td>\n",
       "      <td>15.400938</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.729510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.052697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012272</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>34.321019</td>\n",
       "      <td>54.130236</td>\n",
       "      <td>22.474711</td>\n",
       "      <td>22.001707</td>\n",
       "      <td>80.122759</td>\n",
       "      <td>58.227585</td>\n",
       "      <td>42.266614</td>\n",
       "      <td>52.947538</td>\n",
       "      <td>119.129811</td>\n",
       "      <td>50.828462</td>\n",
       "      <td>64.445213</td>\n",
       "      <td>ENSG00000000419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>4.715796</td>\n",
       "      <td>9.902367</td>\n",
       "      <td>2.222304</td>\n",
       "      <td>2.693052</td>\n",
       "      <td>8.641308</td>\n",
       "      <td>2.948341</td>\n",
       "      <td>3.283652</td>\n",
       "      <td>6.761766</td>\n",
       "      <td>7.120837</td>\n",
       "      <td>4.792727</td>\n",
       "      <td>6.164291</td>\n",
       "      <td>ENSG00000000457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>11.382585</td>\n",
       "      <td>20.353400</td>\n",
       "      <td>12.625342</td>\n",
       "      <td>6.522418</td>\n",
       "      <td>34.807272</td>\n",
       "      <td>6.199940</td>\n",
       "      <td>3.842012</td>\n",
       "      <td>27.191509</td>\n",
       "      <td>15.206148</td>\n",
       "      <td>5.591343</td>\n",
       "      <td>3.786800</td>\n",
       "      <td>ENSG00000000460</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id       A549    GM12878    H1-hESC      HUVEC    HeLa-S3  \\\n",
       "0  ENSG00000000003.10  31.920162   0.012392  39.989468  47.588514  34.921604   \n",
       "1   ENSG00000000005.5   0.000000   0.000000   0.729510   0.000000   0.000000   \n",
       "2   ENSG00000000419.8  34.321019  54.130236  22.474711  22.001707  80.122759   \n",
       "3   ENSG00000000457.9   4.715796   9.902367   2.222304   2.693052   8.641308   \n",
       "4  ENSG00000000460.12  11.382585  20.353400  12.625342   6.522418  34.807272   \n",
       "\n",
       "       HepG2      IMR90       K562       MCF-7       NHEK  SK-N-SH        \\\n",
       "0  46.439177  10.383956   0.242817   24.534779  13.546740      15.400938   \n",
       "1   0.052697   0.000000   0.000000    0.012272   0.000000       0.000000   \n",
       "2  58.227585  42.266614  52.947538  119.129811  50.828462      64.445213   \n",
       "3   2.948341   3.283652   6.761766    7.120837   4.792727       6.164291   \n",
       "4   6.199940   3.842012  27.191509   15.206148   5.591343       3.786800   \n",
       "\n",
       "        ensembl_id  \n",
       "0  ENSG00000000003  \n",
       "1  ENSG00000000005  \n",
       "2  ENSG00000000419  \n",
       "3  ENSG00000000457  \n",
       "4  ENSG00000000460  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr[\"ensembl_id\"] = expr[\"gene_id\"].str.split(\".\", expand=True)[0]\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>comment</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>A549</th>\n",
       "      <th>GM12878</th>\n",
       "      <th>H1-hESC</th>\n",
       "      <th>HUVEC</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>IMR90</th>\n",
       "      <th>K562</th>\n",
       "      <th>MCF-7</th>\n",
       "      <th>NHEK</th>\n",
       "      <th>SK-N-SH</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000159216.14</td>\n",
       "      <td>2.424524</td>\n",
       "      <td>0.534881</td>\n",
       "      <td>0.675734</td>\n",
       "      <td>6.063154</td>\n",
       "      <td>1.859569</td>\n",
       "      <td>4.693606</td>\n",
       "      <td>38.174505</td>\n",
       "      <td>9.631455</td>\n",
       "      <td>24.160086</td>\n",
       "      <td>31.768711</td>\n",
       "      <td>12.047190</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000137203.6</td>\n",
       "      <td>5.412460</td>\n",
       "      <td>0.248758</td>\n",
       "      <td>0.165780</td>\n",
       "      <td>0.175177</td>\n",
       "      <td>21.243269</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.269146</td>\n",
       "      <td>1.048992</td>\n",
       "      <td>2.160615</td>\n",
       "      <td>18.070842</td>\n",
       "      <td>0.073853</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>Mus musculus; human ortholog selected</td>\n",
       "      <td>ENSG00000169083.11</td>\n",
       "      <td>5.525005</td>\n",
       "      <td>0.012590</td>\n",
       "      <td>6.074826</td>\n",
       "      <td>7.937502</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.279713</td>\n",
       "      <td>0.001697</td>\n",
       "      <td>7.263081</td>\n",
       "      <td>0.357436</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000164458.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.433036</td>\n",
       "      <td>0.344499</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000196092.8</td>\n",
       "      <td>0.001572</td>\n",
       "      <td>8.895258</td>\n",
       "      <td>0.335530</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001473</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.426108</td>\n",
       "      <td>0.001250</td>\n",
       "      <td>0.148886</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name          Gene ID  \\\n",
       "0  MA0002.2      RUNX1  ENSG00000159216   \n",
       "1  MA0003.3     TFAP2A  ENSG00000137203   \n",
       "2  MA0007.3         Ar  ENSG00000169083   \n",
       "3  MA0009.2          T  ENSG00000164458   \n",
       "4  MA0014.2       PAX5  ENSG00000196092   \n",
       "\n",
       "                                 comment             gene_id      A549  \\\n",
       "0                                    NaN  ENSG00000159216.14  2.424524   \n",
       "1                                    NaN   ENSG00000137203.6  5.412460   \n",
       "2  Mus musculus; human ortholog selected  ENSG00000169083.11  5.525005   \n",
       "3                                    NaN   ENSG00000164458.5  0.000000   \n",
       "4                                    NaN   ENSG00000196092.8  0.001572   \n",
       "\n",
       "    GM12878   H1-hESC     HUVEC    HeLa-S3     HepG2      IMR90      K562  \\\n",
       "0  0.534881  0.675734  6.063154   1.859569  4.693606  38.174505  9.631455   \n",
       "1  0.248758  0.165780  0.175177  21.243269  0.000000   6.269146  1.048992   \n",
       "2  0.012590  6.074826  7.937502   0.000000  0.000000   1.279713  0.001697   \n",
       "3  1.433036  0.344499  0.000000   0.000000  0.000000   0.000000  0.000000   \n",
       "4  8.895258  0.335530  0.000000   0.001618  0.000000   0.001473  0.000000   \n",
       "\n",
       "       MCF-7       NHEK  SK-N-SH             ensembl_id  \n",
       "0  24.160086  31.768711      12.047190  ENSG00000159216  \n",
       "1   2.160615  18.070842       0.073853  ENSG00000137203  \n",
       "2   7.263081   0.357436       0.000000  ENSG00000169083  \n",
       "3   0.000000   0.000000       0.000000  ENSG00000164458  \n",
       "4   0.426108   0.001250       0.148886  ENSG00000196092  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_expr = tf_map.merge(expr, left_on=\"Gene ID\", right_on=\"ensembl_id\", how=\"left\")\n",
    "tf_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tf_expr = tf_expr[[\"motif_id\", \"motif_name\", \"ensembl_id\", \"HeLa-S3\", \"HepG2\", \"K562\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. calculate tissue-sp of TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MA0002.2</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>ENSG00000159216</td>\n",
       "      <td>1.859569</td>\n",
       "      <td>4.693606</td>\n",
       "      <td>9.631455</td>\n",
       "      <td>0.439869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MA0003.3</td>\n",
       "      <td>TFAP2A</td>\n",
       "      <td>ENSG00000137203</td>\n",
       "      <td>21.243269</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.048992</td>\n",
       "      <td>0.650207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MA0007.3</td>\n",
       "      <td>Ar</td>\n",
       "      <td>ENSG00000169083</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001697</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MA0009.2</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000164458</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MA0014.2</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>0.001618</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   motif_id motif_name       ensembl_id    HeLa-S3     HepG2      K562  \\\n",
       "0  MA0002.2      RUNX1  ENSG00000159216   1.859569  4.693606  9.631455   \n",
       "1  MA0003.3     TFAP2A  ENSG00000137203  21.243269  0.000000  1.048992   \n",
       "2  MA0007.3         Ar  ENSG00000169083   0.000000  0.000000  0.001697   \n",
       "3  MA0009.2          T  ENSG00000164458   0.000000  0.000000  0.000000   \n",
       "4  MA0014.2       PAX5  ENSG00000196092   0.001618  0.000000  0.000000   \n",
       "\n",
       "   tissue_sp  \n",
       "0   0.439869  \n",
       "1   0.650207  \n",
       "2   0.666667  \n",
       "3        NaN  \n",
       "4   0.666667  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specificities = calculate_tissue_specificity(tf_expr[[\"HepG2\", \"HeLa-S3\", \"K562\"]])\n",
    "tf_expr[\"tissue_sp\"] = specificities\n",
    "tf_expr.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. find avg specificity per tile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>unique_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>seq</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>motif_name</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>HeLa-S3</th>\n",
       "      <th>HepG2</th>\n",
       "      <th>K562</th>\n",
       "      <th>tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>60</td>\n",
       "      <td>80</td>\n",
       "      <td>-</td>\n",
       "      <td>29.2449</td>\n",
       "      <td>9.790000e-13</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGAGGGGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...</td>\n",
       "      <td>57</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>29.0000</td>\n",
       "      <td>1.740000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAGGAGGGGGAGGA</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF263</td>\n",
       "      <td>FLIPPED__Enhancer.flip.antisense__chr19:185263...</td>\n",
       "      <td>32</td>\n",
       "      <td>52</td>\n",
       "      <td>+</td>\n",
       "      <td>28.7347</td>\n",
       "      <td>2.680000e-12</td>\n",
       "      <td>2.160000e-07</td>\n",
       "      <td>GGAGGAGGAAGAGGAGGAGGG</td>\n",
       "      <td>MA0528.1</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>ENSG00000006194</td>\n",
       "      <td>7.772455</td>\n",
       "      <td>8.319385</td>\n",
       "      <td>5.862709</td>\n",
       "      <td>0.120346</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    motif                                          unique_id  start  end  \\\n",
       "0  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     60   80   \n",
       "1  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     60   80   \n",
       "2  ZNF263  WILDTYPE_BUT_HAS_SNP__Antisense.noflip.NA__chr...     57   77   \n",
       "3  ZNF263  WILDTYPE_SNP_INDIV__Antisense.noflip.NA__chr9:...     57   77   \n",
       "4  ZNF263  FLIPPED__Enhancer.flip.antisense__chr19:185263...     32   52   \n",
       "\n",
       "  strand    score          pval          qval                    seq  \\\n",
       "0      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA   \n",
       "1      -  29.2449  9.790000e-13  2.160000e-07  GGAGGAGGAGGAGGAGGGGGA   \n",
       "2      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA   \n",
       "3      -  29.0000  1.740000e-12  2.160000e-07  GGAGGAGGAGGAGGGGGAGGA   \n",
       "4      +  28.7347  2.680000e-12  2.160000e-07  GGAGGAGGAAGAGGAGGAGGG   \n",
       "\n",
       "   motif_id motif_name       ensembl_id   HeLa-S3     HepG2      K562  \\\n",
       "0  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "1  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "2  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "3  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "4  MA0528.1     ZNF263  ENSG00000006194  7.772455  8.319385  5.862709   \n",
       "\n",
       "   tissue_sp  \n",
       "0   0.120346  \n",
       "1   0.120346  \n",
       "2   0.120346  \n",
       "3   0.120346  \n",
       "4   0.120346  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo = fimo.merge(tf_expr, left_on=\"motif\", right_on=\"motif_name\", how=\"left\")\n",
    "fimo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "125873"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fimo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "105323"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_nonan = fimo[~pd.isnull(fimo[\"tissue_sp\"])]\n",
    "len(fimo_nonan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "75501"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fimo_deduped = fimo_nonan.drop_duplicates(subset=[\"motif\", \"unique_id\"])\n",
    "len(fimo_deduped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.249383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.416909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.429598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.435871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.462181</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  avg_tf_tissue_sp\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.249383\n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.416909\n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.429598\n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.435871\n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.462181"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_sp = fimo_deduped.groupby([\"unique_id\"])[\"tissue_sp\"].agg(\"mean\").reset_index()\n",
    "avg_sp.columns = [\"unique_id\", \"avg_tf_tissue_sp\"]\n",
    "avg_sp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.212183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.412933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.414182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.419604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.481961</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  med_tf_tissue_sp\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.212183\n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.412933\n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.414182\n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.419604\n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.481961"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "med_sp = fimo_deduped.groupby([\"unique_id\"])[\"tissue_sp\"].agg(\"median\").reset_index()\n",
    "med_sp.columns = [\"unique_id\", \"med_tf_tissue_sp\"]\n",
    "med_sp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  avg_tf_tissue_sp  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...          0.249383   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...          0.416909   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.429598   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...          0.435871   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...          0.462181   \n",
       "\n",
       "   med_tf_tissue_sp  log_avg_tf_tissue_sp  log_med_tf_tissue_sp  \n",
       "0          0.212183              0.222650              0.192423  \n",
       "1          0.412933              0.348478              0.345668  \n",
       "2          0.414182              0.357394              0.346551  \n",
       "3          0.419604              0.361772              0.350378  \n",
       "4          0.481961              0.379929              0.393366  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tissue_sp = avg_sp.merge(med_sp, on=\"unique_id\")\n",
    "tissue_sp[\"log_avg_tf_tissue_sp\"] = np.log(tissue_sp[\"avg_tf_tissue_sp\"]+1)\n",
    "tissue_sp[\"log_med_tf_tissue_sp\"] = np.log(tissue_sp[\"med_tf_tissue_sp\"]+1)\n",
    "tissue_sp.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. find tissue specificity per tile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hepg2_activ[\"overall_mean\"] = hepg2_activ.mean(axis=1)\n",
    "hela_activ[\"overall_mean\"] = hela_activ.mean(axis=1)\n",
    "k562_activ[\"overall_mean\"] = k562_activ.mean(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  \n",
       "0  -3.002306  \n",
       "1  -0.964436  \n",
       "2  -1.394383  \n",
       "3  -1.650828  \n",
       "4  -1.343125  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = hepg2_activ[[\"unique_id\", \"overall_mean\"]].merge(hela_activ[[\"unique_id\", \"overall_mean\"]], on=\"unique_id\").merge(k562_activ[[\"unique_id\", \"overall_mean\"]], on=\"unique_id\")\n",
    "mean_activ.columns = [\"unique_id\", \"hepg2_mean\", \"hela_mean\", \"k562_mean\"]\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>106046</th>\n",
       "      <td>WILDTYPE__mrna.noflip.NA__chr1:153957834..1539...</td>\n",
       "      <td>-1.302625</td>\n",
       "      <td>-1.753467</td>\n",
       "      <td>-2.091155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102926</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__chr19:29...</td>\n",
       "      <td>-0.864368</td>\n",
       "      <td>-1.049161</td>\n",
       "      <td>-1.389210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102576</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__chr1...</td>\n",
       "      <td>0.236499</td>\n",
       "      <td>-0.336387</td>\n",
       "      <td>0.949498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102953</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__chr1:195...</td>\n",
       "      <td>1.843737</td>\n",
       "      <td>1.733347</td>\n",
       "      <td>2.045571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102219</th>\n",
       "      <td>WILDTYPE_BUT_HAS_SNP__Antisense.flip.sense__ch...</td>\n",
       "      <td>2.033647</td>\n",
       "      <td>-0.343059</td>\n",
       "      <td>1.775526</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                unique_id  hepg2_mean  \\\n",
       "106046  WILDTYPE__mrna.noflip.NA__chr1:153957834..1539...   -1.302625   \n",
       "102926  WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__chr19:29...   -0.864368   \n",
       "102576  WILDTYPE_BUT_HAS_SNP__Lincrna.flip.sense__chr1...    0.236499   \n",
       "102953  WILDTYPE_BUT_HAS_SNP__mrna.noflip.NA__chr1:195...    1.843737   \n",
       "102219  WILDTYPE_BUT_HAS_SNP__Antisense.flip.sense__ch...    2.033647   \n",
       "\n",
       "        hela_mean  k562_mean  \n",
       "106046  -1.753467  -2.091155  \n",
       "102926  -1.049161  -1.389210  \n",
       "102576  -0.336387   0.949498  \n",
       "102953   1.733347   2.045571  \n",
       "102219  -0.343059   1.775526  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ[~(mean_activ[\"unique_id\"].str.contains(\"SNP_INDIV\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"SNP_PLUS_HAPLO\")) & \n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"SCRAMBLED\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"FLIPPED\")) &\n",
    "                        ~(mean_activ[\"unique_id\"].str.contains(\"RANDOM\"))]\n",
    "mean_activ.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  \n",
       "0  -3.002306       -1.974712  \n",
       "1  -0.964436       -0.700792  \n",
       "2  -1.394383       -0.886722  \n",
       "3  -1.650828       -1.445209  \n",
       "4  -1.343125       -1.062973  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ[\"tile_mean_expr\"] = mean_activ[[\"hepg2_mean\", \"hela_mean\", \"k562_mean\"]].mean(axis=1)\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  \n",
       "0              0.222650              0.192423  \n",
       "1              0.348478              0.345668  \n",
       "2              0.357394              0.346551  \n",
       "3              0.361772              0.350378  \n",
       "4              0.379929              0.393366  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ.merge(tissue_sp, on=\"unique_id\", how=\"left\")\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# first scale ranges to be positive\n",
    "mean_activ[\"hepg2_scaled\"] = scale_range(mean_activ[\"hepg2_mean\"], 0, 100)\n",
    "mean_activ[\"hela_scaled\"] = scale_range(mean_activ[\"hela_mean\"], 0, 100)\n",
    "mean_activ[\"k562_scaled\"] = scale_range(mean_activ[\"k562_mean\"], 0, 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "      <th>hepg2_scaled</th>\n",
       "      <th>hela_scaled</th>\n",
       "      <th>k562_scaled</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "      <td>20.306424</td>\n",
       "      <td>35.475835</td>\n",
       "      <td>9.487524</td>\n",
       "      <td>0.386721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "      <td>31.843142</td>\n",
       "      <td>44.506588</td>\n",
       "      <td>35.136723</td>\n",
       "      <td>0.165019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "      <td>26.883511</td>\n",
       "      <td>46.540818</td>\n",
       "      <td>29.725285</td>\n",
       "      <td>0.261225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "      <td>19.474389</td>\n",
       "      <td>38.210172</td>\n",
       "      <td>26.497599</td>\n",
       "      <td>0.265622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "      <td>38.747601</td>\n",
       "      <td>33.473407</td>\n",
       "      <td>30.370430</td>\n",
       "      <td>0.117438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  hepg2_scaled  hela_scaled  \\\n",
       "0              0.222650              0.192423     20.306424    35.475835   \n",
       "1              0.348478              0.345668     31.843142    44.506588   \n",
       "2              0.357394              0.346551     26.883511    46.540818   \n",
       "3              0.361772              0.350378     19.474389    38.210172   \n",
       "4              0.379929              0.393366     38.747601    33.473407   \n",
       "\n",
       "   k562_scaled  tile_tissue_sp  \n",
       "0     9.487524        0.386721  \n",
       "1    35.136723        0.165019  \n",
       "2    29.725285        0.261225  \n",
       "3    26.497599        0.265622  \n",
       "4    30.370430        0.117438  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specificities = calculate_tissue_specificity(mean_activ[[\"hepg2_scaled\", \"hela_scaled\", \"k562_scaled\"]])\n",
    "mean_activ[\"tile_tissue_sp\"] = specificities\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>hepg2_mean</th>\n",
       "      <th>hela_mean</th>\n",
       "      <th>k562_mean</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "      <th>hepg2_scaled</th>\n",
       "      <th>hela_scaled</th>\n",
       "      <th>k562_scaled</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>seqID</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>fractionBPcovered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.108018</td>\n",
       "      <td>-1.813811</td>\n",
       "      <td>-3.002306</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "      <td>20.306424</td>\n",
       "      <td>35.475835</td>\n",
       "      <td>9.487524</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>0.394737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.299548</td>\n",
       "      <td>-0.838393</td>\n",
       "      <td>-0.964436</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "      <td>31.843142</td>\n",
       "      <td>44.506588</td>\n",
       "      <td>35.136723</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>24</td>\n",
       "      <td>53</td>\n",
       "      <td>101</td>\n",
       "      <td>0.885965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.647109</td>\n",
       "      <td>-0.618675</td>\n",
       "      <td>-1.394383</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "      <td>26.883511</td>\n",
       "      <td>46.540818</td>\n",
       "      <td>29.725285</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>84</td>\n",
       "      <td>0.736842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.166325</td>\n",
       "      <td>-1.518474</td>\n",
       "      <td>-1.650828</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "      <td>19.474389</td>\n",
       "      <td>38.210172</td>\n",
       "      <td>26.497599</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>54</td>\n",
       "      <td>0.473684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>0.184302</td>\n",
       "      <td>-2.030095</td>\n",
       "      <td>-1.343125</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "      <td>38.747601</td>\n",
       "      <td>33.473407</td>\n",
       "      <td>30.370430</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>99</td>\n",
       "      <td>0.868421</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  hepg2_mean  hela_mean  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...   -1.108018  -1.813811   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...   -0.299548  -0.838393   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -0.647109  -0.618675   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...   -1.166325  -1.518474   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...    0.184302  -2.030095   \n",
       "\n",
       "   k562_mean  tile_mean_expr  avg_tf_tissue_sp  med_tf_tissue_sp  \\\n",
       "0  -3.002306       -1.974712          0.249383          0.212183   \n",
       "1  -0.964436       -0.700792          0.416909          0.412933   \n",
       "2  -1.394383       -0.886722          0.429598          0.414182   \n",
       "3  -1.650828       -1.445209          0.435871          0.419604   \n",
       "4  -1.343125       -1.062973          0.462181          0.481961   \n",
       "\n",
       "   log_avg_tf_tissue_sp  log_med_tf_tissue_sp  hepg2_scaled  hela_scaled  \\\n",
       "0              0.222650              0.192423     20.306424    35.475835   \n",
       "1              0.348478              0.345668     31.843142    44.506588   \n",
       "2              0.357394              0.346551     26.883511    46.540818   \n",
       "3              0.361772              0.350378     19.474389    38.210172   \n",
       "4              0.379929              0.393366     38.747601    33.473407   \n",
       "\n",
       "   k562_scaled  tile_tissue_sp  \\\n",
       "0     9.487524        0.386721   \n",
       "1    35.136723        0.165019   \n",
       "2    29.725285        0.261225   \n",
       "3    26.497599        0.265622   \n",
       "4    30.370430        0.117438   \n",
       "\n",
       "                                               seqID  maxCov  numMotifs  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       6         10   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...      24         53   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...      12         24   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       9         13   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       5         16   \n",
       "\n",
       "   numBPcovered  fractionBPcovered  \n",
       "0            45           0.394737  \n",
       "1           101           0.885965  \n",
       "2            84           0.736842  \n",
       "3            54           0.473684  \n",
       "4            99           0.868421  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mean_activ = mean_activ.merge(cov_map, left_on=\"unique_id\", right_on=\"seqID\", how=\"left\")\n",
    "mean_activ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_activ[\"log_max_cov\"] = np.log(mean_activ[\"maxCov\"]+1)\n",
    "mean_activ[\"log_num_motifs\"] = np.log(mean_activ[\"numMotifs\"]+1)\n",
    "mean_activ[\"log_bp_cov\"] = np.log(mean_activ[\"numBPcovered\"]+1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. plot correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>med_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>log_med_tf_tissue_sp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:5248266..5248380,-__1715.1.0.0</th>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.212183</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>0.192423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:67993585..67993700,+__2318.1.0.0</th>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.412933</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>0.345668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:33571488..33571603,+__2316.1.0.0</th>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.414182</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>0.346551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:76375037..76375152,+__2319.1.0.0</th>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.419604</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>0.350378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:153518247..153518361,-__1714.1.0.0</th>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.481961</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.393366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    tile_mean_expr  \\\n",
       "unique_id                                                            \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...       -1.974712   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...       -0.700792   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...       -0.886722   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...       -1.445209   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...       -1.062973   \n",
       "\n",
       "                                                    tile_tissue_sp  \\\n",
       "unique_id                                                            \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...        0.386721   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...        0.165019   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...        0.261225   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...        0.265622   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...        0.117438   \n",
       "\n",
       "                                                    avg_tf_tissue_sp  \\\n",
       "unique_id                                                              \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...          0.249383   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...          0.416909   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.429598   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.435871   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...          0.462181   \n",
       "\n",
       "                                                    med_tf_tissue_sp  \\\n",
       "unique_id                                                              \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...          0.212183   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...          0.412933   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.414182   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...          0.419604   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...          0.481961   \n",
       "\n",
       "                                                    log_avg_tf_tissue_sp  \\\n",
       "unique_id                                                                  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...              0.222650   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...              0.348478   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.357394   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.361772   \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...              0.379929   \n",
       "\n",
       "                                                    log_med_tf_tissue_sp  \n",
       "unique_id                                                                 \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr11:...              0.192423  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr16:...              0.345668  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.346551  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr17:...              0.350378  \n",
       "CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:1...              0.393366  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for_joint = mean_activ[[\"unique_id\", \"tile_mean_expr\", \"tile_tissue_sp\", \"avg_tf_tissue_sp\", \"med_tf_tissue_sp\", \n",
    "                        \"log_avg_tf_tissue_sp\", \"log_med_tf_tissue_sp\"]]\n",
    "for_joint.set_index(\"unique_id\", inplace=True)\n",
    "for_joint.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## tissue specificity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for_joint.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#cmap = sns.light_palette(\"#8da0cb\", as_cmap=True)\n",
    "cmap = sns.light_palette(\"darkslategray\", as_cmap=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.0909131237771, spearman p: 3.0548323287e-05\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aeba9f2af98>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHMAAABnCAYAAAAzBH1rAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADzJJREFUeJztnXlMFOcbxz8jiyBUBVHAK/GgalBo1eARNNba2opGG6T/\nmUBDtdoKsSCKIqeKFKGNrRo1aT3TatTWRgtpa7DU6qLgsVKj9eBSFFBgW0WEBeb3h2V/sMIMy7E7\nu91PQhhmZ9/3mfnyPO/9jiCKoogNq6CXuQ2w0X3YxLQibGJaETYxrQibmFaETUwrQmVuA2JiYvD0\n9DS3GWanrKyMzZs3dykNs4vp6elJWFiYuc0wO1999VWX0zC7mF0hKSmJZ8+eMXv2bN555x0ATp8+\nTWZmJq6urkRGRrJ27Vr69etHYGAgkyZNMrPFPUuPlpmFhYWsXLmyw9cnJiaSmJhIfX09ADt27CAu\nLo64uDiys7NbXVteXo6HhwcpKSmcPXtWfz4nJ4f09HQ8PDy4ffs2kyZNIikpiRMnTnTPTSmYHvPM\npqYmjh07hqura4e/U1dXR0xMDL179wbgk08+eemaixcvsm/fPpYuXcrgwYMBsLOza5UvgLu7Ozqd\njoqKCrZu3YpOp+vK7VgEsp75xRdfkJeXp39IHeXIkSMsXrwYQRA6/B2VSoWzs3OrvKOjo4mOjiYr\nKwuAKVOmsHPnTgYOHEh5efmLm+j1/9toaGgAoKKiAk9PT2bOnMnKlSvp16+fUfZbIrKeGRwcjFqt\nZuvWrdTU1DB16lTmzJmDo6Oj5Pfy8/MpKCjgypUrXL16lddff91o4z799NN2Pxs+fDgVFRWsX7+e\n+fPnc/HiRerr65k1axZr167F09MTDw8Ptm3bxvHjx1m+fLnR+VscogynTp0SN27cKCYkJIj79+8X\nc3JyxFWrVsl9TU9sbKzk519++WWH07JmuuM5yHpmUVERUVFRODg4kJ+fj4+PD2PHju3wP0tSUlKX\n/tmUQH5Bgf7YZ9QoM1oijaSY8fHxaDQaHj16BICDgwM+Pj64uLiYxDhz0VK8tj5TqqCSYs6bN4/J\nkyejUll0c9QopIRUOpIqubi4UFJSwsiRI01lj2KorK5+6Zzbv80spXqnZNNk3LhxPHnyhIyMDHJz\nc+nbt6+p7DILcl7ZlsBKQradGRwczPz586mqqiI9Pd0UNpmFlkJKidb8mRLDsWxhGBsby7Rp0wgP\nD7fahrcShekMkmLu27eP3r17c/nyZS5fvowgCCQkJJjINNNgKGR7Xtl83s3VlcrqatxcXRVXdkqK\nuWDBAn777Tdmz56NKIr8+uuvprLLLMgJqXQkxTx8+DC5ubmUlZUBL7rQrAm5crK9c25GDB6YEskK\n0MqVK5k6dSr+/v5MnjyZaiP/Qy9dukRERASrVq3i77//7pKhpsZSvLElsrXZkpISJk6cyPTp0yks\nLDQq8fz8fFJTUwkICODy5cudNtLUdERIJYotW5udMGEC69evB8DLy8uoxENCQigpKeHMmTP6NCyF\nxy3EGqjQsGqIrJj+/v4MGDAAgLt37xqVeE5ODmq1moSEBBwcHDpnoQJ4XF1tEYJ2aHA6Pz8fnU7H\n8+fPjUr86NGjlJWVER8fz/Xr1zttpClpL3w+VmBYNUTWMx0cHJg+fTqlpaXU1NQYlbiSe4ykOgqM\nEU5JbU1ZMVevXo29vT0VFRW8/fbbprBJsSg93MqK6eHhAUBQUFCPG2MumkNrWyG2SqvVHw/4dxxX\nqW1No6ZaWkq51xmahbSEsrE9ZD3z5s2bZGRk8Mcff/Dmm28yfvx4U9hlMlp6o6GQLb2y5d9KDbWS\nYgYEBDBt2jSCgoKora01akKzJSHXAdAssqGISgu3kmE2JSUFBwcHdu/ezc2bN3n8+LGp7DI57YVX\nSwq7kp7p6+uLr68voihy4cIFdu7cSVxcnKls63GkPNIwxEqhlOaJpJgnTpwgPz+fsWPHcvXqVf1s\ncWvCUNCWIhp65ePqan2NVolIiqlWq0lJSSEkJIT9+/ebyqYep70Og/aEbNm+rNJqGdhigFpJSJaZ\nvXr1QhAERowY0anENRoN4eHhrFu3Tr+yS+k8rq7WC2l43PKaZpQ0eiIpZlFREQkJCdy9e5eEhASj\np4xkZmaSnp7OzJkzyc3N7Yqd3UazV0qJ0FJEY9M1J7JzgEpKSvDy8jJqNVcz9fX12Nvb4+7uzsOH\nDzttZE/RkQqQkstIQyQ9MyoqivPnz7Nly5ZOJS4Ign6NpBL2LeiI9zyurm5VdhpTqzU3kmI6OTkR\nHBxMbW1tpxJftGgRa9euJS8vT1FL0LurnFNSeQkyYba0tJSMjAz9b3jRK9RRfH19+fzzz7tmoQmw\npI4BKSTFDAsLQxAEli9f3qky05JoL5walpltTSdRSjNFUszExES8vb1xdnZGFEUEQcDPz89Utpmc\ntjy0SqvVCyo3nmnuniDJMvOzzz5j5MiRODs74+PjQ3h4uKnsUhRVWm27nmtYbpqziSIp5oQJE/j4\n448JDQ2luLiY4OBgU9mlWNorX5VQGZIMswcPHuT+/fu4ubmxcOFCIiMjTWWXYjAUb4CLS7t9tOYu\nOyU988aNGzx//pzS0lIOHTpk0YuGOhP+pGq5VVptu5+bK9RKemZycrKp7LAIlD5IbduitJswnBRm\nDu+UnQP09ddfU1xcjCAIRq3PPHDgALdv38bOzs7iwvMAFxf9UJdch4KSpl/Kivn06dNO7eVTV1fH\nxo0b2bRpEzU1Na22UTMHPqNGSXpLywpNcwWno4K2h6nbnbJiZmdnU1ZWpl8r0lEvW7p0KWq1Gjs7\nO7MLaUjL1c+V/3qWYXnYXo11gIsLA11dFTmaIivm0aNH0Wq1iKLIjRs3JK/dvXs3Go0GAG9vb4YN\nG8a6deu6x9Juxq2TswVahtSWx0qoAMmKGRkZSU1NDSNGjECr1TJz5sx2r/3oo4/0x++99x7jxo0j\nJyeH+Ph4+vTp0z0WdwHDUNssaPNxexiWic3e2TKdto5NjayYffv2JTAwEJVKRV5eXocTVupmvW0J\nCrwUdqUY6Or6kmhtiWjqflpZMd966y2GDh3Kd999h7u7uyls6nGaH3JbosrR1nWG58zV2S4rpk6n\n49SpU/j5+VndBhWGDz2/oEBWVKnPzT13VrbT4MKFCzQ2NjJ37lwOHTpkCpvMhs+oUXpB3P4NpYY/\nct8zJ7KeqVKpKC0tJTMzk7q6OlPYZHYMw7AShOoIHarNnjt3jsbGxk5P7LJULEXEZmTFTE5ORhRF\nmpqa+P333y2ua+6/hKyYTk5OrF692hS22OgismIWFxcTExODSqWyyo0QrQlZMceMGcOKFSv+U1t7\nWyqyCj169IiEhAS9mMZ45o8//kh5eTnLli3rtIE2Oo6smJ19lcWjR4/QaDSKWJbwX6HHZhrs37+f\nJUuW9FTyNtqgWwvC5iEwnU6HRqOhsrKSoqIigoKC9PvvGVJWVtYt7460dJr39O0KgiiKYjfY0ib3\n798nIyPDVmaaiB4V04Zpsc3OsyJsYloRNjGtCIsT03AHk/r6eqKiooiKiurRV3W0t3PKBx98oH+L\nobmxODENdzC5d+8eAQEBpKam6l9lbIp8AY4fP66o96NZnJgtdzCpqqpi9OjR+Pv7s2PHDubOnWuy\nfB88eMCTJ0949dVXeyxPY7E4MQ13MKmtrSU+Pp6AgABmz55tsnzVajXFxcWcPXuWH374ocfyNQa7\nBAsb03J1dSUtLY2qqioePnyIKIpkZWVx584dHjx4wGuvvWaSfENDQ3njjTcoKyvj/fffV8SsfVun\ngRVhcWHWRvvYxLQibGJaETYxrQibmFZEjzVN4uLiZNt95eXlZGdnM2bMmC7n9+DBAxITE0lPT0ej\n0dDQ0MCGDRs4f/48P//8M0OHDsXDw4MLFy6g0Wg4d+4cEydO7HK+xvL9998zePBgPvzwQwRBwN7e\nvtXA/d69eyksLKSurq7VlJuTJ0/i7u4uuTSyR6fcVVVVsWXLFkRRZNmyZfz1118UFRVx7tw50tLS\nOHnyJCEhIYSHh+Pl5cWNGzfw8vLCzs6OkJAQtm/fTkNDAwsXLuTu3btcv34dnU7HihUrWLduHRMm\nTGDcuHEsWrSIIUOGkJaWRlxcnH7eUk5OzktzmHJzcykqKsLJyYmCggJ27dqFIAhERESQlpaGIAhE\nRkayY8cOkpKSiIuLIyIiopUtkyZNQqfTERsbiyAIBAcHk5KSot+Xd82aNaSnp9PQ0MDUqVNpbGzk\n9OnT9OvXD1dXV27dusXTp0+prKzk6dOnJCYmUllZSWBgIIWFheh0OoqLi/n2229JTU1lw4YNREdH\nc+TIEUJDQ9t93j0aZjMzM1myZAmJiYkcO3aMvLw8wsLCmDJlCgCVlZX06dMHBwcH/dvmIyMj0Wq1\n/PTTT9TX1+Ps7My1a9cYP348M2bMoODf9R9+fn6sWbOGS5cutZv/rVu3iI6OJjo6Wn/Oz89PHzEq\nKytxcHAgMDCQf/75B51Ox6JFi17qADC0BV7s2aDVann33XcZNGgQbm5ubNiwAZVKRWZmJhUVFTg6\nOvLnn3+SnZ1NWloaCxYsAGDkyJH4+voyYMAAGhsbGT58ONu2baN///6tbOzTpw/Pnj3D0dGRV155\nRXYD5h71TJ1Oh0qloq6uDpVKpX9lY/NvnU4HoA8dzfsmNDU10dDQQGBgIKNGjeLatWscPHiQxYsX\nM2TIEAAcHR1ld9ocM2aM5OzCQYMGERoayuHDh5k7dy5hYWGcOXMGrVarT/vZs2cv2dJsY0REBDdv\n3uT48eP07t1bn66TkxNz5sxh3rx5/PLLL1RUVLS675b06tULe3t7mpqaXlqY5efnx4EDB5g8eTKA\n7NsretQz582bxzfffENqaipBQUF4e3uTnJzMlStXEASh1QMwJCAggL1797JlyxZGjBhB//79UavV\n1NTUdHozYwB3d3fOnj0LvBBq+/btVFZWMnjwYLZv386dO3cYPXo0giCwbds2GhsbX7IFXqyO27Vr\nFxcvXsTb25t79+6xefNmBEFgxowZZGVlERMTw7Bhw5gyZQqxsbGo1eo2bbp9+zZr1qzR/z1kyBCy\nsrLw9/fn0KFD+Pv7A2Bvby99c6IJyc3NFUVRFJOTk0WtVivu3LlTrK2tNaUJPUZsbGyPpLt+/XpR\nFEXxyZMn4p49eySvNWnTpLCwkE2bNmFnZ0f//v1ZvHgxZ86cMaUJFsWePXuYNWsWAKdPnyYwMFDy\neltHuxVh6zSwImxiWhE2Ma0Im5hWhE1MK8ImphXxP6AO3KT2HXF2AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aeba9f2a828>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_mean_expr\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(mean TF tissue specificity)\")\n",
    "\n",
    "r, p = stats.spearmanr(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.0851904541668, spearman p: 9.38739678006e-05\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aeba90f7dd8>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHQAAABrCAYAAACmGqZpAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAD8BJREFUeJztnXtQVOX/x1+Hi1wDASEukomXcbDRrxesBpvQylHGZARt\nbGoKIxvNZCZIQ5AFTEUJNVPLwUinUqHJGjPE1EHJvC6NQVmGN9DkOksYBMIunN8ftvuDdTmHEJbD\nuq8ZBs7us+e8l/d+PvtczvM8giiKIlYsBpv+FmCld7EaamFYDbUwrIZaGFZDLQyroRaG1VALo9uG\nFhUVsWbNmr7UYqUXsJN68sKFCxw+fJhr167R0NDA+++/by5dVnqIZISmpKSg0+lISkpi/PjxBAYG\nmkuXlR4iyHX9Xbp0iby8PE6dOsXzzz/PwoULe11EUlISvr6+vX7egUhVVRVr167t+QlEGVpaWkRR\nFEWNRiOq1Wq54j3iww8/7JPzDkTu938h+R2amJiIm5sbpaWlBAcH4+LiwuTJk3v+6elFdDodCQkJ\niKLISy+9xMSJEwHYu3cvv/zyCwEBAbzwwgukpKQwePBgFi9ezLBhw/pZdd8jW8tNSEjA29ubd955\nhyVLlphDE2lpaaSlpdHa2grA9u3bUalUqFQqCgsLASgpKSEsLIwNGzZw4MABw2uvXr1Keno69fX1\nnD59mqioKBITE8nNzTWL9v5GMkLv3LmDRqNBEAQ0Gg0AXl5efS6qpaWFpKQkBg0aBMDSpUvvKVNZ\nWYmfnx92dp3fgvhvlcDT05OQkBB27dpFSUkJLS0tfa5bCUgaGhQUxL59+xg6dCj79u0D4K233up7\nUXZ2uLi4GI43b95MdXU1ADNmzGD69Ol4enpSXV2NTqczGA93UzHA7du3aWhoYN68eTg5OXHkyJE+\n160EJA2dPHkyhYWFzJ8/n6CgINmTFRcXk52djYuLC2lpadjb25Oenk5jYyOTJk0iKiqqRyLffvtt\nk9pUKhUnTpwgJiaG7777jtGjRxMcHExycjJBQUH4+/uTkpKCjY0NK1eu7NG1BxxSNaYlS5aIt2/f\nFuPj47tVw0pPTxdbW1vFvLw88ccffxSLiorEuLg4MSEhQSwtLe3yddZa7v9zv/8LyUqRu7s7bm5u\nndKfFK2trdjb2+Pj40NdXR23bt0iLCyM5ORksrOze+UDaEUayZRbVlZGamoqV69eJTU1FcDw2xSC\nIKDVaqmpqcHX15fm5mba2tpwdHTExsY6DmAOJA1NSEhAEARDzVGOiIgI3n33XQYPHoyLiwuxsbGo\nVCoOHTrEggULekWwFWkkDU1OTsbV1ZXp06czZMgQAMaPH99l+XHjxrFp06ZOj6Wnp/eCTCvdRdLQ\nb7/9lhs3blBQUMCVK1fw9vY2ly4rPUT2i83W1ha429h/UBrnAxnJCI2MjMTBwYFnn32WCRMmmEuT\nlftAtlJkZWAhaeiUKVPMpWPA8PPlywD8b9SoflZiGklDAdrb26mtraWtrQ0Af3//PhelFPTmdfWc\nEk2VNXT58uWMGjUKGxsbBEFg0aJF5tBlpYfIGvrII4+waNEiQ233QUEqOjuW+d+oUZ3K9nfUyhr6\n119/kZqaajBUquvPUuhoUO2/48AA3ibGgo2N7+9ULGno7t27EUURGxubbnf/DVRMRWRHM42PTZmr\nBCQNjY6O/k8nMx4P1Q88L1y4kIyMDMX2NHXHzIFCrw6B5Ofns3HjRp566inUajUA+/fv56GHHurN\nyygCKcO78/3bV8gampKSwoEDB6irq5M9mfF4aEVFBQ0NDYxSYPVeCr1ZNXV1hp+Ox0pG1lCVSsWw\nYcPYs2cPMTExZGVlUVVVZbKs8XjomTNnKC8v5+TJk3zzzTe9Lr436G40dTRS/7cS07JsLTc7O5vy\n8nJcXV158cUXGTNmDBs2bGDz5s33lDUeD42PjycqKoqtW7cyd+7cPnkDvU3H6JSipq4OH09Pc0j6\nT8ga6urqarg1v7CwkKFDh5o0E0yPhwIsW7bsPmUql1qNRlE1XklDU1JSKCkpobS0FAA/Pz+efvpp\nswjrD3orhSq2HRoSEoK/vz+CIAAYfj9I6E329vK6px2qxLQraegTTzxBeXk5Pj4+tLW10dDQYC5d\nZqGrCpGpSo8SK0CmkKzlDhkyhJ07d+Ls7IyDg4NF34rZE8OUFp3QjUqRi4sL1dXV2NnZdZpyYEmY\nMrNTdHao8Xor0MSOyBqalJREfn4+Op2O+Ph4c2jqV4ybK7UmjpVsqmznfFlZmaFj/uOPP7aY0Rap\nDgV9dBqbaaqM0tKu7KIZDg4OhIWFmUlO/9IxOqXMNEZJbVFJQ5cuXcqRI0cUO0rSF5j6Pq3p8JiP\nl5ei066koadPn6a2tpa8vDxEUUQQBEJCQsylrd/QR2dNF+b6KCQaTSE7Hvrll18yatQow2xuK8pG\ndrRFrVYzYcIEnnzySc6ePStZtri4mNjYWFauXElrayutra0sX76c5cuXc/To0V4TbQ5MRedAQLbZ\nEhgYyLp167C1tcXDw0OyrH6A++jRo6jVanx9fQkPDycsLIzExESee+65XhNuTgbCrSd6ZA2NiIjg\n559/Zvjw4bLT8jsOcFdWVhIaGkpgYCDbt29nxowZvSa6LzA1XCbV4aDUSpFsys3OzubixYt4eHiw\nfv16ybKmJvympKQQHh7OtGnTek10X1JbV0eNRnNPT5H+xxjjD4Lib+O0t7dn0KBBODs709jYKFnW\neIC7qamJGzdukJWVRXBwMK+88kqvCe8t5Drgu9seVUpbVNbQ6OhocnJy2Llzp+xAtakBbksbP62t\nq7s7lPav0UobRpNNuefOnaO+vp47d+7w22+/mUOT4lHysJqsoSUlJaxdu5bU1FSKiorMoclsdPf+\nIVPUaDT/qXvQXMim3LKyMj744APs7e25fPkyO3fuBLBOWlIosoampKQgCAINDQ2EhoZabL+uceo0\nvuXEVHkldgHKptxPPvkEjUbD4cOH2bdvHwEBAQQEBJhDm2JRYqrVI2toe3s7Bw8eZMGCBbS3t5tD\nk9lQQjOjt5FNuYsXLzZ0yr/22mt9LkgJ+Hh5GfpyvT0974lIb0/PTh8GH6Pj/kTW0OHDhzN8+HBz\naOlX9N+ZHQ3s+JjJ1yjISD2yKbe+vh61Ws2tW7csKuV27KIz7hTw9vQ0VHi8vbwMP4bnvbwMz3d8\nbX93+0E3DN20aRN5eXnU1dXd324FCsXYKP2x3lSfDubpn/fx8uoUnUqKUtmUq+/HHTt2LLt37zaD\nJPOhXx+hoyE1/3btSaE3U0nfnXpkDZ0zZw65ubkkJibKrqhpPIPbxsbG5M4NSsKUqaYwft44TSsh\n3UI3DP3hhx/w8/MD7vbrSt1TZDzA7eTkRFhYGDNnzuS9995TpKFAp5VM/kvE6csqxUzohqH6qGxq\naiInJ0eyrPEAt42NjcmdG5SIsSkdo1ZqaExJZkI3DD1//jxwt4NB7iYx4wFunU5ncueGgYDeqK7S\nsdKM1CNr6JAhQ2hvb8fe3l52KkRXK1rrd24YiCjVuK6QNfTQoUNERESQm5uLIAiSK1RbV7Tuf2Tb\noVqtlhMnTvDmm28+kBN+BxqyETp37lwqKipobm5m5syZ5tBk5T6QjdCTJ08yYsQItm7dSkFBgTk0\nWbkPZCO0oaEBtVpNbGwse/bs6RMRVVVVbN26tU/OPdDoag2o7iJr6OTJk/nzzz8RRZFx48bd18W6\nwhL7iPsLyS2bjSf8CoJgMRN+LRXJCJ09ezYnTpxg2rRpiKI44CYcPYhIGpqTk8P58+cNeT0wMNAs\noqz0HMmUa2XgMWC3DOyvuajG19WzcOFCamtr++y63WXAGmq82PLNmzcJDw8nIyOjT9vLSl/kecAa\narzY8ogRIwgNDe3zuahKX+R5wBraX3NRlb7Is23qAG1Yenh4kJmZSV1dHZWVlYiiaNgWs6KiQnKf\n0968bkxMDGFhYVRVVTF//vxub2/dV1hruRbGgE25VkxjNdTCsBpqYVgNtTCshloYfd5sUalUsu3C\n6upqCgsLGT169H1fr6KigrS0NDZu3EhxcTE6nY5Vq1Zx+vRpvv/+ewICAnj44Yc5d+4cxcXFnDp1\nql/2F//666/x8/Pj9ddfRxAE7O3t8exwN/6uXbu4fv06LS0t+Pr6Gh4/ePAgPj4+ODk5mTyvWe6A\nrqurIz09HVEUeeONN/jjjz8oKyvj1KlTZGZmcvDgQaKjo4mNjWXkyJH8/vvvjBw5EltbW6Kjo9m2\nbRs6nY45c+Zw9epVLl68iFarZcmSJaxcuZLHHnuMMWPGEBERgb+/P5mZmahUKlavXg3A2bNnDX/r\nUavVlJWV4ezszLVr19ixYweCIBAXF0dmZiaCIBAfH8/27dtZvXo1KpWKuLi4TlomTpyIVqslOTkZ\nQRB49dVXWb9+PSNHjkQQBFasWMHGjRvR6XQ8/vjjtLW1cezYMdzc3PDw8KC0tJTGxkY0Gg2NjY2k\npaWh0WiIjIzk+vXraLVaysvL2bt3LxkZGaxatYqEhARyc3O7vC3WLCk3Pz+fl19+mbS0NL766iuK\niopYtmyZYY9vjUaDk5MTDg4OxMbG4ubmRnx8PPX19eTl5dHa2oqLiwslJSWMHTuWqVOncu3aNeDu\nViQrVqzgp59+6vL6paWlJCQkdNokPiQkxJA5NBoNDg4OREZG8vfff6PVaomIiLink8BYC0BLSwv1\n9fXMnDkTb29vvLy8WLVqFXZ2duTn51NTU4OjoyO//vorhYWFZGZmMnv2bODu3Ntx48bh6elJW1sb\ngYGBbNmyBXd3904anZycaGpqwtHREVdXVyorK7t8r2aJUK1Wi52dHS0tLdjZ2XHnzh0Aw2+tVgtg\nSCMODg7A3bv1dTodkZGRBAUFUVJSwueff05UVJRhL3BHR0fZ20tHjx59T4R2xNvbm5iYGHJycpgx\nYwbLli3j+PHj1NfXG87d1NR0jxa9xri4OC5dusT+/fs7zRBwdnbmmWeeYdasWRw5coSamppO77sj\nNjY22Nvb097eTktLS6fnQkJC+Oyzz5g0aRIAOp2uy/dilgidNWsWn376KRkZGcybN4/g4GDWrVvH\nhQsXEARBcppEeHg4u3btIj09nUcffRR3d3fOnDnDP//8Q3Nzc481+fj4cPLkSeCuWdu2bUOj0eDn\n58e2bdu4cuUKI0aMQBAEtmzZQltb2z1aAOzs7NixYwfnz58nODiYmzdvsnbtWgRBYOrUqRQUFJCU\nlMTQoUOZMmUKycnJnDlzxqSmy5cvs2LFCsOxv78/BQUFhIaG8sUXXxAaGgrcXa6vS8R+QK1Wi6Io\niuvWrRPr6+vFjz76SGxubu4PKb1OcnJyn5w3MTFRFEVRbGhoELOysros1y/NluvXr7NmzRpsbW1x\nd3cnKiqK48eP94eUAUFWVpZhzcRjx44RGRnZZVlr57yFYe1YsDCshloYVkMtDKuhFobVUAvDaqiF\n8X9omusH558nAwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aeba8fe9a58>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_tissue_sp\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(mean TF tissue specificity)\")\n",
    "\n",
    "r, p = stats.spearmanr(for_joint[\"log_avg_tf_tissue_sp\"], for_joint[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## number of bp covered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mean_activ.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.28149830273, spearman p: 1.70689654122e-39\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aebaa623eb8>"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHkAAABnCAYAAAAkJu2iAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEAtJREFUeJztnXtMU2cfgJ9Cy0XEoTCV6LKaoRiDzE0FL3NxskvATRdh\ni4sumBmdTnQ4dYrVAop4+cQlzulcsk02E3GSoVNBXbzhtDBQYGxTp8glEAuIoliEUjjfH0u7gtAL\n0AvQJyHW9py3v/bp772dc94jEgRBwEGvxsnWATiwPA7JfQCH5D6AQ3IfwCG5D+CQ3AcQ2zoAmUzG\n0KFDbR2GzVEqlWzZssUiZdtc8tChQ1m+fLmtw7A5X375pcXKtrlkc9FoNKxbtw5BEJg3bx4vv/wy\nAD/88AO3bt3C2dmZ9evXExMTg0aj4YMPPmDSpEk2jtq2WLRNLi4uJioqyuTt4+PjiY+PR61WA/DV\nV18hl8uRy+VcvHgRgD/++IPp06ezfft2jh07ptu3sbGRzZs3IxaLKS4uxt/fH5lMhkKh6N4P1QOx\nWCa3tLSQmprKwIEDTd6nsbERmUyGi4sLAMuWLXtqm7t37+Lr64tY3Dr0RYsWoVAocHZ2RiqVEhMT\nw+nTp1m6dGnXPkgvwGgmf/HFF+Tm5tLS0mJWwYcPHyY8PByRSGTyPmKxGA8Pj1bvvW7dOtatW8e5\nc+cAGDRoEJWVlWg0Gt2PAeDAgQNUVlYSExPDpUuXWLZsGUeOHOH8+fNmxd0bMZrJkZGRKBQK/ve/\n/6FSqQgODiYkJAQ3NzeD+xUWFnLnzh3y8vLIz89n3LhxZge3cuXKp56bMGECcrmcCxcusHDhQk6c\nOMGoUaM4evQoo0ePJisri5UrV5KYmEh6ejohISFmv2+vQzDCiRMnhM2bNwtxcXFCcnKykJWVJURH\nRxvbTcfGjRsNvr57926Ty+rNWPJ7MJrJJSUlrFmzBldXVwoLCxk7diz+/v4m/4g2bdrUpR+hg65j\nUHJsbCwFBQVUV1cD4OrqytixY/Hy8rJKcPbMP5X1Jm03akg/C0diHIOSQ0NDGT9+/FM92b6MqXL1\nt7e1aIP2vLy8KCsrY8SIEdaKx64xV7C9YHAINXr0aOrq6khPTycnJwdPT09rxdVjqKlTG/wD2/84\njI6TIyMjmTlzJvfv3ycpKckaMdklbUXpS7R3jDa2GzduZNKkSaxYsYIBAwZYIyYH3YxByQcOHMDF\nxYVr165x7do1RCIRcXFxVgrNfmmbwSV1qna3k3p6UFOnxtvTpd3XrYVByW+//TYXLlzgtddeQxAE\nfv31V2vF1WPoSLA9YVBySkoKOTk5KJVKAJ577jmrBNUbseVQymDHKyoqiuDgYKZOncr48eN58OCB\nWYVfvXqVzz77jOjoaB4+fNilQHsi9pLlRnvXZWVlvPTSS0yePJni4mKzCi8sLGTHjh2EhYVx7dq1\nTgdpr5gq0dZDKaO964CAANavXw+An5+fWYUvWLCAsrIyzp8/ryujJ2GOlJvV/23r/+x/1XJJnQqp\np0d7u1gNo5KnTp3KoEGDACgqKjKr8KysLBQKBXFxcbi6unYuwh7Izer6VqIBm/ayTTppoLCwkKam\nJhoaGswq/MiRIyiVSmJjY/nrr786HaQt6CiL25sA0c9i/ee0z+tX67aoso1msqurK5MnT6aiogKV\nyryORG+eITO1PW4vq62NUcmrV69GIpFQVVXFG2+8YY2YehT6WXy76r/HfoNtf4hRi1HJQ4YMASAi\nIsLiwfRk9AXr/98eZJt1Sm5Pa1fthZvV9ZTUqWx2QMOo5Bs3brBr1y7mzJnTZ858NNY5MmeSo22G\n2wKD1XVYWBiTJk0iIiKCJ0+emHWifG+kbSa216u2Rwxm8rZt23B1dWX//v3cuHGDe/fuWSsuu6ej\nDleJ8pHuz14wmMmBgYEEBgYiCALZ2dns3bsXuVxurdjsAmOHFdsKbrWt8hHSobY/Bm9Q8tGjRyks\nLMTf35/8/Hw0Go214rILOhKszWJDgu0Jg5IVCgXbtm1jwYIFJCcnWyumHkF7gisqWoseNsz2WQxG\n2mQnJydEIhFSqbRThRcUFLBixQpiYmJ0Vyr2VNpmse555SMqKh7pBNfcrbF6bMYwKLmkpIS4uDiK\nioqIi4sz+9SfjIwMkpKSmDZtGjk5OV2J06Z01A7rV9E1d2t0grWPKyrsowNm9ByvsrIy/Pz8zLo6\nUYtarUYikTB48GDu3r3b6SCtibkHECoqHunkNlXc1j0vGWbeYVlLYjCT16xZw5UrV9i6dWunCheJ\nRDQ1NVFVVdUr1gXRr6rbZqi+YHvDoOR+/foRGRnJkydPOlX47NmzWbt2Lbm5ubplH+wZ/SzuKedU\nm4LB6rqiooL09HTdv/DvLJipBAYGsmvXrq5FaGMMjYvb9qbtFYOSly9fjkgkYsmSJZ1qk3sS7WWx\nvmBDVbUh7H4yJD4+njFjxuDh4YEgCIhEIiZOnGit2KxCe5e/tEV/8qOjMbFkmJ/dtssGJW/fvp3M\nzExUKhUjRoxgxowZ1orLKhjqSZfUqTqcnzY2Jtb2rO1lMsSg5ICAAAICArh//z7ff/89kZGRHD9+\n3FqxWZT2BLdXTbed2TIkWH/Y5O3r3W2xdhWDkn/88UfKy8vx9vZm1qxZrFq1ylpxWZ22gm9W17c7\n6QH2OatlCIOSr1+/jkQioaKigoMHD/aaC96MtcPtdbJ6Sk+6PQxKTkxMtFYcNkNfsClnfOhnsbaj\nZU+zW+3RpxYDMWfKsjvHw/7P9kPq6WGzk+uNSv72228pLS1FJBKZVV3rL2hqD1W8oY6WMYYNG9Cu\naGMZLB06wC7O1jQq+fHjx51ai0u7oGlCQgIqlarVcoo9Ab/B/do9Cc/b19tgx0vbq+5o+GSLy1eN\nSr548SJKpVJ3LZOpWam/oGlPE6yPdOgASpSPdNIqKh4ZHR4NGzZAl8W2rqrBBMlHjhyhtrYWQRC4\nfv26wW33799PQUEBAGPGjGH48OHExMR0T6RdZNSQfia3yf7P9uNmdf1TVa2+bEO0Fawfgy0wKnnV\nqlWoVCqkUim1tbVMmzatw20//vhj3eN3331Xt6BpbGws7u7u3RNxF2grWptd2rZZK6SkTqW7fqk9\n2YbQbts2g225YJtRyZ6ensyZMwexWExubq7JBR89erRLgVmK9jLaFNnwX4YbQru9vQgGEyS//vrr\nDBs2jEOHDjF48GBrxGRxOqq69dvNmjp1uxePSz09DI6ntfvYi2AwQXJTUxMnTpxg4sSJvWphGP0v\n35hwc7EnwWCC5OzsbNzd3XnzzTeRyWS9chasIxkdddRM+QHYi2AwQbJYLKaiooKMjAwaGxutEZPd\nYExURz8CexIMJvauL1++THNzc6dP6Out2JvMjjAqOTExEUEQaGlpITMz0y6mKB2Yh1HJ/fr1Y/Xq\n1daIxYGFMCq5tLQUmUyGWCzuNceT+xpGJY8aNYqlS5c6blHQgzFqrrq6mri4OJ1kczL52LFjVFZW\nsnjx4k4H6KDrGJXc2Vv+VFdXU1BQ0Csuj+npWOyGnMnJycyfP99SxTswg25taLWHGpuamigoKKCm\npoaSkhIiIiJ063O2RalUWvTewT0F7ZrilkAkCIJgqcLLy8tJT093tMk2xqKSHdgHFr1JtgP7wCG5\nD+CQ3AfoNZIVCgWrVq0iKiqq0ysjGKO4uLhHLj3ZaySXl5eTlJTEhAkTLDIcaWlpITU1lYEDB3Z7\n2Zam10h+7733OHXqFGlpaRY5F+3w4cOEh4f3yBUXes1Rh/T0dMLCwnB3d0ehUPD66693a/mFhYXc\nuXOHvLw88vPzGTduXLeWb0l6zTg5LS2NrKwsVCoV8fHxeHtb5iJwuVze6fl8W9FrJDvomF7TJjvo\nGIfkPoBDch/AIbkP4JDcB+iSZFPuR1FZWcnJkyfNLjs7O9vofh0NDA4dOsSBAweAf2fCvvnmG7Pf\nvzvQ/366694dcrmczMxMbt++TVRUFLdv30ahUBjcp8uTIffv32fr1q0IgsDixYu5efMmJSUlXL58\nmZ07d3L8+HEWLFhAdHQ0UqmUe/fu8c4773Dv3j2CgoJISUmhuroaHx8f/vnnH55//nlUKhWhoaH8\n/PPPnD9/nrCwMHx8fPjll19Qq9VER0ezePFiIiIimDt3Lmq1mtjYWARBYPbs2aSnpzNr1ixdjOfO\nnaOsrIyAgACqq6t5+PAhSqUSmUyGr68varWaDRs2UFdXx7Jly0hLS6OxsZHg4GB+++03EhISSE5O\nZsaMGfz0009oNBqCg4O5cuUKAB9++CEHDx5Eo9Hw0UcfkZaWRn19PXfu3NHF0NLSQkJCAlVVVXzy\nySdcvHiRhQsXsmnTJt24e86cOQQHB6NUKvH29mbChAkAXLp0CYlEQlhYGAUFBUilUtzd3SkvL8fZ\n2Zns7GwmT57coaMuV9cZGRnMnz+f+Ph4UlNTyc3NZfny5QQFBQFQU1ODu7s7rq6uREdH4+T09Ftq\nNBqWLFmCp6dnqwXhwsLC2LFjBxkZGaSkpODm5gbArVu3GD58OHPnzgXgypUrTJs2jW3btnHy5EmC\ngoKYPn26rpwpU6aQkJCgWz1//vz5fPrpp5w6dUq3f0hICElJSYjFYoYMGUJCQoLuy8vPz6e8vJyy\nsjKqqqpwc3Pjzz//pLGxkdWrV5OamoqbmxsuLi7k5eXR3NyMTCbDx8dHF4NIJOLzzz9n5syZlJaW\nUlpaSmFhYaslol944QXWrFmDm5sbMpmM33//nbNnz7JlyxZGjhxJ//79efHFF/Hy8kIikRAYGMiI\nESNoaGgweBOYLmdyU1MTYrGYxsZGxGKx7va72n+bmpoAWt0/2dnZmZaWFurr/71gTCwW4+LigkQi\nwcnJiZaWFuDfX79IJMLDw4P6+nqio6P5+++/GThwIAMGDHgqBkEQaG5ufipG7XPatUtcXFxQq9W4\nuLjo9pdIJDQ3N+tW29fuN336dPbs2YOPjw8ajYaQkBBCQ0M5c+YMDx48wMPDA41Gw6JFi2hoaKC2\ntlZ3u0Pt59B/rP2M/fv3JzMzk3nz5um2cXd3x8nJCYlEgkgkaiWuoaFBF1dbXF1defz4MV5eXu2+\n3uVMDg0N5bvvvmPHjh1EREQwZswYEhMTycvLQyQS6b5Iffz8/Lh06RIpKSkGyz579ixr165l5syZ\nhIeHs3btWtLS0vD19W213dSpUzlz5gwymYzw8PCnyrl69SpyuZzg4GAAvv76a/bs2cNbb72l2//0\n6dNs2LABd3d3ioqKiI2N5dVXX8XLy4vi4mJmzJjBK6+8wrlz55DJZAwfPlxX/vvvv09CQgL79u1D\nKpUikUh0qx5pkUgkbN++nVOnTjFlyhSCgoLIz89vle3tMWXKFORyOXfv3mXkyJFUVlby8OHDVtuo\n1Wr69+/fcSFCN5OTkyMIgiAkJiYKtbW1wt69e4UnT55099t0mt27dwtVVVW2DkMoKysT9u3b1y1l\n7dq1y+Dr3T6EKi4uJiEhAWdnZ5555hnCw8P7zI08zWHnzp2tOoedpaioyOitHxwHKPoAjsmQPoBD\nch/AIbkP4JDcB3BI7gM4JPcB/g+7amm1fI8fPAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aeba6b09c88>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_mean_expr\"], cmap=\"Blues\", \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(number of bp covered by motif)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.164330518614, spearman p: 3.66858885488e-14\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aeba8fda5c0>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHsAAABrCAYAAABXEf3kAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEUtJREFUeJztnXtMU+f/x98tFKitiloQEKbIyHfRxcuGZYpuiMYLTolW\nFxfd4iUQL+iYykSrLbgq6rwsXuYiw7BkBjRuZBqQaUQRFbks4DUTEAGZtiAI3yLQ0vb8/vDX821r\new5ib9DzSkjL6XOe82nf5/PcP89hEQRBgMElYDvaAAb7wYjtQjBiuxCM2C4EI7YLwYjtQjBiuxA9\nFrusrAwymcyWtjDYGHeqD8vLy5GXl4eamhoolUr88MMP9rKLwQZQerZUKoVGo4FYLMb48eMRFBRk\nL7sYbACLbrj0n3/+QU5ODm7evIn58+dj5cqVVjdCLBbDz8/P6vn2ReRyOXbv3m2bzAkaVCoVQRAE\n0dzcTJSWltIl7xVHjhyxSb59EVv+FpR19vbt2zFo0CBUVlZizJgx4PF4CAsLs81d9w7s2rULHR0d\nmD59OmbPng0AUKvVSEtLQ0REBCZMmIBjx45BoVAgKCgIcXFxDrbYMdC2xpOSkuDj44MtW7Zg7dq1\n9rAJKSkpSElJgVqtBgAcP34cEokEEokEBQUFRmkVCgWGDx+OvXv3orCwkDxeVVWF6upqMs39+/eh\n1WoxceJEu3wHZ4TSs7u6utDc3AwWi4Xm5mYAwLBhw2xulEqlglgshoeHBwBg/fr1b6QpKSlBRkYG\nYmNj4e/vDwBwc3MjPx87diymTZsGAGhoaMCECRMQFxeHTZs2YdKkSTb/Ds4IpdijR49GZmYmAgMD\nkZmZCQCIj4+3vVHu7uDxeOT/hw8fhkKhAADMmjULUVFREAqFEAqFePr0KcrKygAAbLb5gsrb2xt8\nPh9sNhteXl42t99ZoRQ7LCwMBQUFWLJkCUaPHk2b2Z07d5Ceng4ej4eUlBRwOBykpqaivb0dH3/8\nMUQiUa+M/Pbbby1+FhQUhMbGRmzfvh3z5s1DSUkJ1Go1pk6dSqYJCQlBZmYmtm3bhunTp/fKhn4B\nVett7dq1RFtbG7F58+YetfZSU1MJtVpN5OTkEDdu3CDKysqITZs2EUlJSURlZaXF85jW+P+w5W9B\n2UAbPHgwBg0aZFSkUqFWq8HhcODr64uWlhb8+++/iIyMxM6dO5Genm6Vm5Oh91AW47W1tUhOTsbj\nx4+RnJwMAOSrOVgsFrq7u9HY2Ag/Pz90dnZCq9XCy8vLYn3KYD8oxU5KSgKLxQLRwzWJMTEx2Lp1\nK7y9vcHj8bBx40ZIJBLk5uZi6dKlVjGYofdQir1z507w+XxERUVBIBAAAMaPH28x/bhx43Do0CGj\nY6mpqVYwk8EaUIp9/vx51NfXIz8/H9XV1fDx8bGXXQw2gLYi1Q9UqFQqqFQqmxvEYDsoPXvRokXw\n9PTEzJkzXXqYsb9A20Bj6D9Qii0UCu1lR5/jcVOn2eMhPlw7W9JzKMUGAJ1Oh6amJmi1WgBAQECA\nzY1yNiwJS5fW2YSnFTsxMRGhoaFgs9lgsViIjY21h11OwduIbOl8ZxKcVuz33nsPsbGxRtOHrsC7\nCu2M0Ir98uVLJCcnk2JTDZe6Ci3t6jeODeV7mE3rTN5N2c/OyMgAQRBgs9kgCKLHw6Z9HSqvNic0\n1XG6/OwJpWevWLHirTIznc/WrzRZuXIl9u/f32dH4KiEtJTO1NOdwcOtOhV18eJFHDx4ENOmTUNp\naSkA4Pfff8fAgQOteRm7YknoZ+2d5J+l83p6k9gLWrGlUin+/PNPtLS00GZmOp/97NkzKJVKhIaG\nWsVYe9CTItdUYEuCA8Y3i6OLc1qxJRIJRo4cidOnT2P16tU4efIk5HK52bSm89lFRUWoq6tDYWEh\nsrOzrW68I6AS1hLO4uG0rfH09HTU1dWBz+fjyy+/xAcffIB9+/bh8OHDb6Q1nc/evHkzRCIRjh49\nioULF9rkCzgLlm6CAP7rerqlXY2hfA+H1t20YvP5fDIcpaCgAIGBgWaFBszPZwPAhg0b3tFM+xHi\nwzUqbofyPUjP7I1XP2vvJAV3NJRiS6VS3L17F5WVlQAAf39/fPbZZ3YxrC9S1/bmzTByMNdpBKcU\ne9KkSQgICACLxQIA8pXhNebENZdm5GDjotxRUIr9ySefoK6uDr6+vtBqtVAqlfayy2G8S4v5UVOX\n0f//8XGugATK1rhAIEBaWhoGDBgAT0/Pfr8c2JpCmx7rTX1vbWi7XjweDwqFAkqlkhwR649YW2hD\nelLc2wNascViMcrLy1FUVITNmzfbwya705uxcD10Qpv73FGDK5R1dkZGBmpra8kJkBMnTjCzXv+P\nqbc+bWon3wf58O1tTo+g3UDH09MTkZGRdjKnf0AlvCNb5JRir1+/HpcuXeqzs1W2Qu/V+iLaUFxz\nGBbljuxvU4p969YtNDU1IScnBwRBgMViuWwge2952tRutlh3xLAp7Xz22bNnERoaSu7CwEDNs0Zj\nLw/wfVNoRxXltK3x0tJSTJw4EZMnT8bt27cp0965cwcbN27Etm3boFaroVarkZiYiMTERFy+fNlq\nRlsbKg97G1FMhTbkUVMX6tosz3/bA1qxg4KCsGfPHuzbtw9DhgyhTGu6eOHp06eIjo7G/v37kZ+f\nbzWj7Ym5rpd++NMSCvl/yffPGtst1un27oLRznrFxMSgoqICwcHBtFttGC5eeP78OSIiIhAUFITj\nx49j1qxZVjPaFpjOdr0teq/WC61/He436N2NsxK0np2eno4HDx5gyJAh2Lt3L2Vac8H4UqkU0dHR\nfXIvE0OvtlT8GnqtoUcbovdufVHuKGg9m8PhwMPDAwMGDEB7O3UXw3TxQkdHB+rr63Hy5EmMGTMG\nX3/9tdUMtwU99W5TwQzr6hb5/5ZvDfUbCoX8v07j3bRir1ixAllZWUhLS6NdhGBu8UJfmv82FJrO\nqx81ddF6dYu8BUP9hlrZyt5DW4wXFxejtbUVXV1dePjwoT1scmp64tU6eQ108hravOzdz6YV++7d\nu9i9ezeSk5PJzeVcFdORM3MYiqyT1xgV6wB9S96W0BbjtbW1+PHHH8HhcFBVVYW0tDQA6JcBfubq\n7HdpmDkbtGJLpVKwWCwolUpERES47Di5uVa0pYaZJRy9coW2GP/ll1/Q3NyMvLw8ZGZmYsSIERgx\nYoQ9bGOwMrRi63Q6XLhwAUuXLoVOp7OHTU6B6TCpaV37tnPW+vQBfK7DpjhpxV6zZg0WLFgAAFi1\napXNDXI0VC1kveD64thwkkPfxWL7jSZf2X6jLXa9HBEoQFtnBwcHIzg42B62OCUBfK5RI23kYK7F\nUbChfkPRIm8hBdcLrb8pHNkSB3ogdmtrK6qqqhAQEAB/f3+X2INU3yrXR4OYCm6IuSlMPYYjZ45u\nnAE9EPvQoUNgs9kQiUQ4deoUdu7caQ+7nI6erjAxJ74z1NdAD8TWj4uPHTsWGRkZdjDJOTD0bsB4\n+PRtlxb9x8fLqAh32sC+BQsW4MyZM9i+fTvtzsKmOy+w2WwkJSWBIAgsW7YMH330kdUMtzeGHqkv\n2vVYqscNBXa0VwM9EPv69evkA1eKi4sp16DpFy9cvnwZpaWl4HK5iIyMxJw5c/D999/3ObH1Hmg6\nqmZONDpv15/jyK02aMXWe3NHRweysrIo05ouXmCz2fD394e7O+1lnBpDgUzDeXubjyOgVaGkpATA\n68EVugWHposXNBoNFAoFNBpNvwkd6qlgzrjTIa3YAoEAOp0OHA6HNvzH0pMErl27htWrV1vN6L6A\nswhsCK3Yubm5iImJwZkzZ8BisSifDMA8ScC5oR0h6e7uxrVr17Bu3TomGL+PQ+vZCxcuxLNnz9DZ\n2Yk5c+bYwyYGG0Hr2YWFhQgJCcHRo0f77NpvhtfQerZSqURpaSk2btyI06dP28QIuVyOo0eP2iTv\nvoalPeasAa3YYWFhaGhoAEEQGDdunE2MsNmT4BmMYBEUWw2bBuOzWCwmGL8PQ+nZn3/+Oa5du4bp\n06eDIAinDs5joIdS7KysLJSUlJD1SFBQkF2MYrANlMU4Q/+iXy070e/oFB8fj85O2wXQPXnyBPHx\n8TbL31b0K7EbGhpw8OBBhIWF2awLo9PpcO7cOdpYdWekX4m9ZMkS5OXlITs7G76+vja5xpkzZyAS\nifrk0HHfnmg2ITc3F9HR0eByuSgqKsLMmTOtfo179+6hpqYG5eXlqKiowIQJE6x+DVvRrxpo2dnZ\nuH37Nl69eoWUlBQMGzbMZteSSCTYtWuXzfK3Bf1KbAZq+lWdzUANI7YLwYjtQjBiuxCM2C6EVcSW\nSCS0aRQKBXJyct467+LiYtrzLHUoMjMzyZClhoYGnDx58q2vbw0Mf5+e/FY9zfP69euorq5GfHw8\nqqurUVRURHmO1QZVWlpakJqaCoIgEBcXh0ePHqG2thY3b97EgQMHcOHCBaxYsQIJCQkYNWoUXrx4\ngfnz5+PFixcQCoXIyspCU1MTBAIBKisrMXLkSLx69Qpz587FH3/8gatXryI6OhoCgQDnz5+HWq1G\nQkIC4uLisHjxYixduhRqtRpSqRQEQSAmJga5ublkbDkA5Ofno76+Hh9++CGamprQ1tYGuVwOsVgM\nf39/qNVq7NixA0qlEuvXr0d2djZUKhXCw8Nx48YNyGQy/Prrr4iKisLZs2eh0WgQHh6OW7duAQC+\n+uor/Pbbb9BoNFi1ahWys7PR0dGBmhqDTXV0OshkMjQ2NmLdunUoKCjA6tWrsWvXLrLfvmjRIoSH\nh0Mul2PYsGEICwsD8HqJGIfDQXR0NO7cuYNRo0aBy+WioaEBbm5uKC4uxuTJky1qZLVi/OLFi1i+\nfDlSUlJw7tw5lJWVYcOGDRAKhQCA5uZmcLlceHp6IiEhwWzor0ajwZo1azBw4ECjNer6/U8vXryI\nrKwseHm9Dn+tqqpCYGAgGbVy69YtTJs2DXv37kVOTg6EQqHRxvhTpkyBTCYjHwq7fPlyfPPNN8jL\nyyPPnzFjBg4ePAh3d3cMHz4cMpmM/BErKirQ0NCA+vp6NDY2wsvLC/fv34dKpcKWLVtw7tw5eHl5\nwcPDA+Xl5dBqtRCLxRAIBKQNLBYL3333HebNm4e6ujrU1dXh3r17RqFRISEhSExMhJeXF8RiMUpK\nSnDlyhXs3r0boaGh4PP5GD9+PLy9vcHhcDBu3DgEBwejq6sLGo3GokZW8+zu7m64u7tDpVLB3d0d\nXV2vt4/Sv3Z3dwMAPD09yXPc3Nyg0+nQ0dHx2hh3d3h4eIDD4YDNZpPbeuh0OrBYLHLXxISEBDx8\n+BBDhgzBoEGD3rCBIAhotdo3bNQf4/F4AF5HqKrVajJapbu7GxwOB1qtlgxl0p8XGRmJY8eOQSAQ\nQKPRYMaMGZg7dy4uXbqEly9fgsfjQaPRIDY2Fl1dXWhtbcWDBw9I+/Xo3+u/I5/Px/Xr17Fs2TIy\nDZfLBZvNBofDAYvFMhKwq6uLtMsUT09PtLe3w9vb2+znVvPsuXPn4tSpU9i/fz8WL16MMWPGYM+e\nPSgvLweLxTIb/vP++++jsLCQNobsypUr2Lp1K+bNmweRSIStW7ciOzubDDjUExERgUuXLkEsFkMk\nEr2Rz99//w2JRILw8HAAwM8//4xjx45h9uzZ5Pl//fUXduzYAS6Xi8ePH0MqleLTTz+Ft7c3njx5\ngqioKEydOhX5+fkQi8UIDAwk8//iiy8gk8lw4sQJjBo1ChwOBzKZDK9evSLTcDgc7Nu3D3l5eZgy\nZQqEQiEqKiqMvN8cU6ZMgUQiwfPnzxEaGgqFQoG2tjajNGq1Gnw+xV4vhI0oLS0lCIIg9uzZQ7S2\nthI//fQT0dnZaavLvTVHjhwhGhsbHW0GUV9fT5w4ccIqeR06dIjyc5t1vZ48eQKZTAY3NzcMHjwY\nIpEIV69etdXl+iwHDhwwakT2lsePH9OGRDMTIS4EM6jiQjBiuxCM2C4EI7YLwYjtQjBiuxD/B4W0\nhb7TNWh6AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aebaa5f6438>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_tissue_sp\"], cmap=\"Blues\", \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(number of bp covered by motif)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_bp_cov\"], mean_activ[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## max coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmap = sns.light_palette(\"firebrick\", as_cmap=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: 0.329209707519, spearman p: 3.43536181424e-54\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aebaa667898>"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHIAAABnCAYAAADcxhZVAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAD1tJREFUeJztnXlMk+cfwD+VloIML9CJ018w4jEPNuc9Y7I5twR2Rpkj\nmxvEA5WhAdF4ECoogudMdLK4RCM6o0bjsTlYssxzCg5QkW1ivJCJFAFFJ9IC8v7+mG1Kad8elPal\n9vMP0Pd9n37bD9/net/3eWWCIAh46PB0cnUAHhyDR6Sb4BHpJnhEugkekW6CR6SbIHd1AElJSfTu\n3dvVYbgctVrNmjVr7D7e5SJ79+7NggULXB2Gy9m6dWubjne5SFtpampi2bJlCILAF198wRtvvAHA\njh07uHHjBv7+/iQkJJCcnExDQwOffvopkyZNcnHU7U+7tpG3b98mLi7O6v1TU1NJTU2loaEBgG3b\ntqFSqVCpVJw+fRqAK1eu8NZbb7Fu3TqOHTumP9bf35+MjAy0Wi337t0jMjKS1atXk5+f79gPJVHa\nLSObm5s5dOgQ3bt3t/oYrVZLUlIS3t7eAHz99det9qmoqCAoKAi5vGXo06dPZ/PmzTx8+JABAwZw\n//595s6dy6xZs9r2QToIFjNy8+bNFBQU0NzcbFPBBw4cYNq0achkMquPkcvl+Pn5tXjvZcuWsWzZ\nMk6cOAFAjx49qKyspKmpSS8cIDs7m4SEBPr160dxcTEajYZ9+/Zx9uxZm+LuqFjMyKioKHJzc9mw\nYQN1dXWMGzeOd955Bx8fH9HjiouLuXXrFpcuXeLy5cu8/vrrNgeXkJDQ6rXRo0ejUqk4deoUs2bN\n4vjx4wwaNIgrV65w9uxZFAoFvXr1Ys2aNcjlcqZMmWLz+3ZIBAscP35cWL16tZCSkiJkZWUJeXl5\nQnx8vKXD9CQnJ4tu37Jli9VluTNt/R4sZmRpaSlLlixBqVRSXFzMiBEjGDx4sNX/KKtWrWrTP5oU\neFRY2Oq1rqNGuSAS84iKXLlyJUVFRVRVVQGgVCoZMWIE3bp1c0pwrsSUPOPtUpIpKjIsLIxRo0a1\n6iG6M5YEGu8rFZmihrp160ZZWRn9+/d3Vjwuw5xA7fPayBhlz57646QgU3T4MWTIEP7991+ys7PJ\nz8/H39/fWXE5jUeFhSYlaquqzEqUIhbHkVFRUbz//vs8ePCATZs2OSMmp2BOIJjPQlv3cSYWG7/k\n5GTGjx/PwoUL6dKlizNiandsaQc1NTUt/vYJCHB0OA5BVOSuXbvw9vbm4sWLXLx4EZlMRkpKipNC\nczzWCDTMNGOJutekKFNU5AcffMCpU6d4++23EQSBX3/91VlxORRbMrCjIipy//795Ofno1arAejX\nr59TgnIktkiUWrtnC6Kdnbi4OMaNG8fEiRMZNWoUDx8+tKnwwsJCFi1aRHx8PI8ePWpToM7GVLUq\nZSz2WsvKyhg5ciQTJkzg9u3bNhVeXFzM+vXrCQ8P5+LFi3YH2d5YGmpoq6vRVlc7MSLbsdhrHT58\nOCtWrAAgJCTEpsKjo6MpKyvj5MmT+jKkhimBhtlojUApTApYFDlx4kR69OgBwM2bN20qPC8vj9zc\nXFJSUlAqlfZF2AbsGSfaKlEqWHViubi4mMbGRjQajU2FHzx4ELVazcqVK/nrr7/sDtKRWNuhsVWi\nq3vGFjNSqVQyYcIEysvLqaurs6lwV84E2fPFWurgaKurUQYG6seS2qoq/Zyrq7GYkYsXL2b48OEo\nFAreffddZ8TkdDQ1NXZXqYYZ7sqstJiRL7/8MgARERHtHoyjaMsUnCWJxlkpFWy6HFIq7ZyrkWIn\nyGJGlpSUkJ2dze+//87kyZMZNmyYM+JqF4w7OrZUp7rtysDAFsdLJStFRYaHhzN+/HgiIiKor6+3\n6WJjqeEIiVJGVOTatWvJyclh+/btPHjwgOrqagIN/iM7CuYkGgrqCLLEEBUZGhpKaGgogiBw4cIF\nMjMzUalUzorNIViS2NEF6hAVefToUYqLixk8eDCXL1+mqanJWXHZjViP1ZxEjQWZPmZqIeP20ZXT\ndKIic3NzWbt2LdHR0WRlZTkrJrsxlmjqJLGtEg338QkM1A8/pIaoyE6dOiGTyQgODrar8KKiInbs\n2IGfnx+pqakt7tVwJOYuntJhbqyoE2TpSjkxOsTMTmlpKSkpKdy8eZOUlBSbL/PIyclh06ZNTJo0\nqd1ub7NFomF7aEmipW1Sw+I1O2VlZYSEhNh0V5WOhoYG/U01FRUVdgdpC5YkOrpzI5X5VtGMXLJk\nCefPnycjI8OuwmUyGY2Njdy/f79d1gmwpk2E1vKsaRc7GqIiO3fuTFRUFPX19XYV/vHHH7N06VIK\nCgr0t4g7CmslttjHwQKldDmIaNVaXl5Odna2/if8N9tjLaGhoXzzzTdti9AEtkhsizzDKlM3BBHr\nsbrySgFRkQsWLEAmkzFv3jy72sj2wJESHdWZkUI7KSoyNTWVoUOH4ufnhyAIyGQyxowZ46zYWmHT\npY1GEi1KralBaTTAl4IgaxFtI9etW0f//v3x8/NjxIgRLFy40FlxtcLeYYbx38bDDm1NDVrdZIED\n2jxXnVwWFTl8+HBiY2OZNWsWd+7cISoqyllxWcTcrI3xRLjh4N+UxFblGlfPz/c17Onq2kndFJ0U\nsla0at2zZw93794lICCAjz76iMTERGfFJYrY1Jvx74YCDI8Tyz7dNuOqVsqIirx69SoKhYLy8nJ+\n+OEHSdzEY2mYYWoO1ZxAzfPXfezIKHPZKMlea3p6urPiaBPmJsKNe6WmJLYFKVSpOtxucQBTszaO\n6MSYO5VliGRPY8F/i/XduXMHmUxmU9W6e/durl+/jpeXl1Oq41Y9VQsZZ1ydWtMeKgMDUQYGSuo8\npA6LIp88eWLXWjlarZbVq1eTlpZGXV1di6XJnIGyZ0+9TGVAgGhWiklU9uxpMhulVK2CFSJPnz6N\nWq3W37thbXbNmTOH3NxcvLy8HCrRUFCL15+f9PUJDNRXr8Yyxco0h+HUnFSzEawQefDgQWpraxEE\ngatXr4ruu337doqKigAYOnQoffv2Zfny5Y6J1ACdIJ+AADQ1NXqJOnRfvqa62urMMdcGKk3MsUot\nG8EKkYmJidTV1REcHExtba3oIrZz587V//7JJ58wZMgQ8vLyWLlyJb6+vo6J+DmmZBqiy06ry7Mg\n0tT1q1LJRrBCpL+/P1OnTkUul1NQUGB1wUePHm1TYMZ0HTWq1fSXmEzd79ae/TAn0lCgYSZKSSJY\nIXLKlCm88sor7Nu3j169ejkjJrPovjxDoYYyTaGTbAmxK8alWJUaY1FkY2Mjx48fZ8yYMZJZDMIw\nGx4VFrb4ovWdm+eviUk2hSVpUstEHRZFXrhwAV9fX9577z2SkpIkN9tjXOUai2hLNklVmiksipTL\n5ZSXl5OTk4NWq3VGTDZj7RduKLwjSbIGq3qt586d49mzZ3ZfhCUV3E2eIRZFpqenIwgCzc3NnDlz\nxuVnPzyYxqLIzp07s3jxYmfE4qENWBR5584dkpKSkMvlkjgf6cE0FkUOGjSI+fPnv1DLYXdELNqp\nqqoiJSVFL9KWjDx27BiVlZXExMTYHaAH67Ao0t7HPVRVVVFUVOR5pKCTaLeHnGVlZTFjxoz2Kt6D\nEQ5t+HSnsRobGykqKqKmpobS0lIiIiL069kZo1ar2/zsRHdAtyauvcgEQRAcFEsr7t69S3Z2tqeN\ndALtKtKD8/A8LNtN8Ih0Ezwi3YQXQqSrF8m39VnT9vBCiHTlIvn2PGvaHl4IkdHR0dy7d4+TJ08y\nevRop763Pc+atocXYibclYvkO+JZ09bwQowjExMT9afhvvzyS5esOatSqdr1McUvhMgXgReijXwR\n8Ih0Ezwi3QSPSDfBI9JNaFeR1qx/XllZyc8//2xz2YcPH6a2ttaesGxGpVJx5swZbty4QVxcHDdu\n3GhzmbYOFvbs2SO6vd0nBB48eEBGRgaCIBATE8O1a9coLS3l3LlzbNy4kZ9++ono6GgWLlxISEgI\nV69eJSQkBC8vLyIjI9m4cSNarZaEhAS2bt3KokWLyMzMpE+fPpSWlrJ582aCg4NpamqiubmZzz77\njMOHD7Nq1SpUKhUxMTEkJSW12kc3MJ86dSrjxo1DrVYTEBCgn/k5e/YsCoWC8PBwioqKCA4OxtfX\nl7t37+Ll5QVAQUEBe/fuBf5bty8zMxP4b3nU77//nuTkZFasWEFkZCQ//vgjDQ0NxMfHExMTQ0RE\nBI8fP6aiogI/Pz/i4uLIyMige/fuVFVVMXfuXHbt2kVTUxMzZ85EoVCgVqvNXgPV7lVrTk4OM2bM\nIDU1lUOHDlFQUMCCBQsYO3YsADU1Nfj6+qJUKvVPVk9MTKS2thaFQsGHH35Inz59KCkpYf78+cyc\nOZPY2Fh9+SNHjmT27NkEBQURGxtLoYklxMT2GTBgAEuWLMHHx4ekpCT++OMPfvvtN9asWcPAgQN5\n6aWXeO211+jWrRsKhYLQ0FD69+8PwJEjR9i0aRNz5szh4MGDLF26lHnz5pGTk4NGo+Gff/5hwIAB\n7N+/Hx8fHwCuX79O3759iYyMZOzYsYwdO5aSkhLOnz/P5MmTiY2Npbm5WX+Mt7c3f/75J8OGDSMv\nL8/s99zuGdnY2IhcLker1SKXy/WPLtT9bGxsBNDf0aybQmtububEiRM0Njby6quvIggCT58+RaFQ\ntLiZqHPnzshkMry9vZHJZDx79kw/r/n06VOz++jw9fWlU6dOKBQKZDJZiycpaDQaFAqF2c/W3NxM\np06dqK+vp66uDrlcTn19PQqFgiFDhpCVlcVXX33FtWvXiI+P5++//6Z79+506dIFgJ07dzJ79mx6\n9OhBU1MTSqVS/9mampqYM2cOGo2G2tpaunTpIvpo5HbPyLCwMHbu3Mn69euJiIhg6NChpKenc+nS\nJf2Xa46goCBKSkq4du0a169fZ+fOnezevZtt27aJvmdwcDDffvutzQ8u1fHmm2+iUqmoqKhg4MCB\nVFZWmjz9FR4ezvLlyzly5Aiff/45aWlp7N27l7CwMKZMmcKlS5f43//+x7Rp01i6dClHjhwhKChI\nf7yfnx9nzpzh8ePHDBs2jGPHjrFlyxYApk+fTlpaGt999x3BwcE8efJE/AyK4GTy8/MFQRCE9PR0\noba2VsjMzBTq6+udHYbkqKqqEm7duiU8fvxYSEtLa7X9wIEDQnl5udnjnX724/bt2/zyyy94e3vT\ntWtXpk2bxsmTJwkLC3N2KJLC19eXDRs2IAgCM2fObLX9yZMn9OnTx+zxnklzN8EzIeAmeES6CR6R\nboJHpJvgEekmeES6Cf8H2GpIjV2La2kAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aebaa67c518>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_max_cov\"], mean_activ[\"tile_mean_expr\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"mean MPRA activity\")\n",
    "ax.set_xlabel(\"log(maximum motif coverage)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_max_cov\"], mean_activ[\"tile_mean_expr\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r: -0.246344610807, spearman p: 2.32313173595e-30\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Annotation at 0x2aeba8e610b8>"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAHQAAABrCAYAAACmGqZpAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEPhJREFUeJztnXtMU+cbxz9FykUY6pSr4vBC4iDRuSluM8vQOKdszihu\nc5szeMF5JSojXpgFFMV5XWRqooOwLU7dnESNYmKmIvMGOLxMhzfEO4WhKA5KKT2/P7b2V2t7ToG2\nFOwnIUDP2/c8Pd8+73ne570cmSAIAk7aDC4tbYAT6+IUtI3hFLSN4RS0jeEUtI3hFLSN4RS0jWGx\noIWFhaSmptrSFidWwFXsYFFREQcPHqSkpITq6mpWr15tL7ucNBFRD01KSkKj0ZCYmEi/fv0IDg62\nl11OmohMKvVXXFzM/v37OX78OKNGjWLSpElWNyIxMZGAgACr19saKSsrY/ny5U2vQJCgrq5OEARB\nqKysFAoKCqSKN4kNGzbYpN7WSHOvheg9dPHixfj4+HDlyhXCwsLw8vJiwIABTf/22IilS5dSU1PD\nkCFDePfddwHIyckhNzeXhoYGVqxYwbJly1Cr1QwaNIgxY8a0sMW2QzLKXbhwIb6+vnz55ZfMmDHD\nHjaRkpJCSkoKarUagI0bN6JQKFAoFOTm5j5VVqlU4u/vz8qVK8nLy9O/fufOHVauXElYWBglJSVc\nuXIFgF69etnlM7QUoh6qUqmorKxEJpNRWVkJQOfOnW1uVF1dHYmJibi5uQEwa9asZ8rk5+eTlZVF\nbGwsgYGBALRr105/PDY2losXL3L9+nUmTJhAUlISvXv35quvvqJv3742/wwthaigPXv2ZPv27XTr\n1o3t27cDMHv2bNsb5eqKl5eX/v/169ejVCoBGD58OEOHDiUiIoKIiAhu375NYWEhAC4u/29w9u3b\nx99//01KSgpKpZLa2lrkcjkeHh42t78lERV0wIAB5Obm8uGHH9KzZ0/Jys6dO0dGRgZeXl6kpKQg\nl8tJS0vjyZMnvPbaa0RHRzfJyHnz5pk9FhwcTHl5OYsXL+a9994jPz8ftVrN7t278ff3JzExkTlz\n5vDTTz+RnZ3NsGHDmmRDq0EsYpoxY4bw6NEjIT4+3qIIKy0tTVCr1cL+/fuF33//XSgsLBTmz58v\nLFy4ULhy5YrZ9zmj3P/T3GshGhR16NABHx+fp5o/MdRqNXK5HD8/Px48eMDdu3eJjIxkyZIlZGRk\nWOUL6EQc0Sa3tLSU5ORkrl+/TnJyMoD+tylkMhn19fWUl5cTEBBAbW0tDQ0NeHh4PHV/c2I7RAVd\nuHAhMpkMwcJ5ZKNHj2bBggV07NgRLy8v4uLiUCgUHDhwgPHjx1vFYCfiiAq6ZMkSvL29GTp0KF26\ndAGgX79+Zsv37duXdevWPfVaWlqaFcx0Yimigu7du5dbt25x+PBhrl27hq+vr73sctJEJG9sus56\nXV0ddXV1NjfISfMQ9dCxY8fi7u7OsGHD6N+/v71sctIMJIOi54Wq/Pyn/u8YEdFCljQPUUEjWumH\nMkZKLOPjhq+1NmFFBQXQarVUVFTQ0NAAQFBQkM2Nag6mxGlKGcOyrUlUSUETEhIIDQ3FxcUFmUxG\nbGysPexqNI0RqSl1txZRJQXt3r07sbGxTw1NORpSYqr+G6kxxsPfv1HnaA2iSgr68OFDkpOT9YKK\npf4cBXMCmivXGGEdHdF+aFZWFoIg4OLigiAIFqcAWxJLxTR+j+5HDFs269ZC1ENjYmIaVZnxeKhu\nxsGkSZNYtWqVzTNNTRHTXB3mvNbRm16rDoHk5OSwdu1a3nrrLQoKCgD49ddfeeGFF6x5GotQlZej\nKi8XPWb480wZEY91ZE+VFDQpKYk9e/bw4MEDycqMx0Pv3btHdXU1oaGhVjHWUgwFMv5bTGSTr5sR\n1lFFlRRUoVDw0ksvsW3bNqZMmcKWLVsoKyszWdZ4PPTkyZPcvHmTvLw8srOzrW58YzAnmKVlWouo\nklFuRkYGN2/exNvbm08++YQ+ffrw9ddfs379+mfKGo+HxsfHEx0dTXp6ukPOha39TyRPg/ulqrwc\nDz8/k+VbQ1QsKai3t7d+an5ubi7dunUzKSaYHg8FmDNnTjPNNI8pL/Hw89N7mzlxag08rlaptFhU\nU+d3pCBJVNCkpCTOnz+vn6QcGBjI22+/bRfDmoulgugwJaq5+lRKpcN6qaigAwcOJCgoCJlMBqD/\n3VYxFtUQY681FNWRvFRU0Ndff52bN2/i5+dHQ0MD1dXV9rLLImwRlDRGVEdENMrt0qULW7dupX37\n9ri7u7faqZjGzWetRAKiVqk0W8ZcJOwoEa9kUOTl5YVSqcTV1VWf+WlLGApk7H3Gouo8V+epjngv\nleyHJiYmUlRUxMmTJ4mPj7eHTXbDkr6pIaa81rB/6gheKpmc37BhA1euXKGkpITNmzfby64mYYm3\niDW3lgise39jvwz2QnLTDHd3dyIjI+1kjuXYyhukhPLw89MHTqaa3paOeEU9dNasWXTv3h1fX1/9\nz/OOo3qmDlEPPXHiBBUVFezfvx9BEJDJZAwcONBetlmFpgigqqgw+bqHwRdarHvTkkiOh/7888+E\nhobqV3M7Ch0jIprd7BqmCMG8kIbHPXx9Hbo/KhnlFhQU0L9/f9544w1OnTolWvbcuXPExcWxaNEi\n1Go1arWahIQEEhISOHTokNWMbg5mkwYSYprD0ZpgyX5ocHAwK1asoF27dnTq1Em0rG6A+9ChQxQU\nFBAQEEBUVBSRkZEsXryYd955x2qGW4K5i+3p728+cWDidcPoWeeljoqkoKNHj+bs2bP06NFDclm+\n4QD3/fv3GTx4MMHBwWzcuJHhw4dbzWgdTW12jcXUeae5GQrmhs0M76OOkmSQbHIzMjK4ePEinTp1\nYuXKlaJlTS34TUpKIioqiiFDhljNaB1SYpq6zxmK+dT901hkM9NSxI5bapctkfRQuVyOm5sb7du3\n58mTJ6JljQe4a2pquHXrFlu2bCEsLIyJEydazfDmohPE+N5pbvrKUyMtRs2uIwVJkoLGxMSwY8cO\ntm7dKjlQbWqA25HGT6WS8mKYSiLoBHek7otkk3v69GmqqqpQqVRcunTJHjbZHOPmUmVhOk/v1Uql\nyajYGtNIm4ukoOfPn2f58uUkJyfrN3hqjZhMrDehq2JpN6Wl7qOSTW5paSnffPMNcrmcq1evsnXr\nVgCHXbTk4e/vEJ7SUkgKmpSUhEwmo7q6msGDB7eKfK6jiNoSiXrJJve7776jsrKSgwcPsn37drp2\n7UrXrl3tYZvdaUykanwfdZSZDJKCarVa9u3bx/jx49FqtfawySroIlFLRDJMCHj4+TVO2PLyZkXP\n1kayyZ0+fbo+KT958mSbG2QrDNN9uqS8h6+v3ss8DDI+ujLQ/FytvZtcSUF79OhBjx497GFLo2ls\n6s+cqMAzwsJ/6Twz3mpYzlQ/tKUGuSUFraqq4urVqwQFBREYGNiq9uzTBUeGw2S6i19rIJahsOYw\nmbT39dXX0WoyRevWrcPFxYXo6GgyMzNZsmSJPeyyGoaiguXZHSmRTR3TeW1LTkGRFFSXxw0PDycr\nK8sOJjUOw4tnrvk17MYYe5KqvLxJ4urq8fT3/zeQcgAxwQJBP/jgA3bu3MnixYsld9Q0XsHt4uLC\nwoULEQSBzz77jFdffdVqhptCbP8h46EtcwKDZSJ7GkTRjiImWCDosWPH9Jv0nz59WnROkfEAt6en\nJ5GRkYwYMYJly5bZXFBjdBfY5Ao1czMXRAIh/XsNjjuSmGCBoDqvrKmpYceOHaJljQe4XVxcCAwM\nxNVV8jQ2xdzFbozQYmUcRUywQND8/z60VquVnCRmPMCt0WhQKpVoNBqHXEYhJoRYcy313pZEUtAu\nXbqg1WqRy+WSSyHM7Wh99OhRpkyZYjWj7YGjCiaFpKAHDhxg9OjR7Ny5E5lMJrpDtXNH65ZHMktQ\nX1/P0aNHmTlzZptf8NsWkPTQMWPGcO/ePWpraxkxYoQ9bHLSDCQ9NC8vj169epGens7hw4ftYZOT\nZiDpodXV1RQUFBAXF8e2bdtsYkRZWRnp6ek2qbu1YW4PKEuRFHTAgAHcuXMHQRBs9lS/Zj3R1slT\niD6yOSsri9LSUv0unDKZrFVsr/o8I+qh77//PkePHmXIkCEIguAwC46cmEdU0B07dpCfn69v14OD\ng+1ilJOmI9rkOml9tJ7pB1bgzJkzzJ8/n7lz5/Lo0SO7n//GjRs2f0LycyXohQsXWLVqFVFRUfzx\nxx92PbdWq2XXrl2Sa2yby3MlaExMDPfu3ePIkSMMGDDArufeuXMn0dHRNk+ftuxApZ05deoUJ0+e\nJDk5GXd3d7ue+8KFC5SUlFBUVMTZs2d55ZVXbHKe5yooio+Px9XVFZlMxueff054eLjdbVAoFCxd\nutRm9T9Xgj4PPFf30OcBp6BtDKegbQynoG0Mp6BtDJsLqlAoJMsolUr279/f6Lp3795NVVVVU8xq\nNAqFgmPHjnHt2jVmz57NtWvXml1nYzsYP/74o2QZuyQWHjx4QFpaGoIgMG3aNC5fvkxpaSnHjx9n\nzZo17Nu3j5iYGOLi4ujduzd//fUXvXv3pl27dowfP541a9ZQV1fHvHnzSE9PZ/78+WzatImgoCBK\nS0tZv349ISEhaDQatFotH3/8Mbt372bp0qUoFAqmTZtGYmLiM2V0nfuxY8cyaNAgysrK6Ny5sz6L\nlJeXh1wuJyoqinPnzhESEoKnpyd37tzRP36zsLBQP5Njzpw5bNq0Cfj3QbpbtmxhyZIl+mUke/fu\nRa1WM3fuXKZNm8a4ceN4/Pgx9+/fx8vLi9mzZ5OWlkanTp2oqKjgiy++ICsrC41Gw+TJk5HL5ZSV\nlREQEGD2Wtulyc3JyWHChAmkpKSwa9cuCgsLmTNnjv4Z35WVlXh6euLu7k5cXBw+Pj7Ex8dTVVWF\nXC5n1KhRBAUFUVxczIwZM5g8eTIzZ87U19+/f3+mTp1KYGAgM2fO5MyZM8/YIFamV69eJCQk4OHh\nQWJiIvn5+fz2228sX76c0NBQvL296devHx07dkQul9O3b1/9mtns7GzWrl1LbGwsv/zyCwsWLGD6\n9Onk5OSgUqm4ffs2vXr1YseOHXh4eABw9epVunXrxvjx44mIiCAiIoLi4mJOnDjB0KFDmTlzJlqt\nVv8eNzc3/vzzT8LDwyU30LSLh9bX1+Pq6kpdXR2urq6oVCoA/e/6+noAPD09AfRpOa1Wy+HDh6mv\nr+fll19GEARqamqQy+XU1dXp62/fvj0ymQw3NzdkMhkNDQ36nGlNTY3ZMjo8PT1xcXFBLpcjk8nQ\naDT6YyqVCrlcbvazabVaXFxcqK2t5Z9//sHV1ZXa2lrkcjl9+vTh+++/Z+LEiVy+fJm5c+dy6dIl\nOnXqhI+PDwCZmZlMnTqVF198EY1Gg7u7u/6zaTQaYmNjUalUVFVV4ePjw8OHD0WvtV08dOTIkWRm\nZrJq1SrGjRtHWFgYK1asoKioSH+RzREYGEhxcTGXL1/m6tWrZGZm8sMPP7Bx40bRc4aEhPDtt99y\n/fr1Jtn85ptvolAouH//PqGhoSiVSpNDblFRUSxatIjs7Gw+/fRTUlNT2bZtGyNHjmTYsGEUFRXR\nvXt3oqOjWbBgAdnZ2frFX/DvUzeOHTvG48ePCQ8PZ8+ePWzYsAGAjz76iNTUVDZv3kxISAhPnjyR\nHq0RWoCCggJBEARhxYoVQlVVlbBp0yahtra2JUxxKCoqKoSSkhLh8ePHQmpq6jPHd+7cKdy9e1e0\njhYZbblx4wYHDx7Ezc2NDh06EB0dzZEjRxg5cmRLmOMweHp6snr1agRBMLlByZMnTwgKChKtw5mc\nb2M4EwttDKegbQynoG0Mp6BtDKegbQynoG2M/wFGO2yxh0tEKQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2aebaa716978>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig = plt.figure(figsize=(1.2, 1.2))\n",
    "ax = sns.kdeplot(mean_activ[\"log_max_cov\"], mean_activ[\"tile_tissue_sp\"], cmap=cmap, \n",
    "                 shade=True, shade_lowest=False)\n",
    "ax.set_ylabel(\"tissue specificity in MPRA\")\n",
    "ax.set_xlabel(\"log(maximum motif coverage)\")\n",
    "\n",
    "r, p = stats.spearmanr(mean_activ[\"log_max_cov\"], mean_activ[\"tile_tissue_sp\"])\n",
    "print(\"r: %s, spearman p: %s\" % (r, p))\n",
    "ax.annotate(\"r = {:.2f}\".format(r), xy=(.05, .9), xycoords=ax.transAxes, fontsize=fontsize)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. write files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>tile_mean_expr</th>\n",
       "      <th>tile_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>numMotifs</th>\n",
       "      <th>log_num_motifs</th>\n",
       "      <th>numBPcovered</th>\n",
       "      <th>log_bp_cov</th>\n",
       "      <th>maxCov</th>\n",
       "      <th>log_max_cov</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>10</td>\n",
       "      <td>2.397895</td>\n",
       "      <td>45</td>\n",
       "      <td>3.828641</td>\n",
       "      <td>6</td>\n",
       "      <td>1.945910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>53</td>\n",
       "      <td>3.988984</td>\n",
       "      <td>101</td>\n",
       "      <td>4.624973</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "      <td>84</td>\n",
       "      <td>4.442651</td>\n",
       "      <td>12</td>\n",
       "      <td>2.564949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>13</td>\n",
       "      <td>2.639057</td>\n",
       "      <td>54</td>\n",
       "      <td>4.007333</td>\n",
       "      <td>9</td>\n",
       "      <td>2.302585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>16</td>\n",
       "      <td>2.833213</td>\n",
       "      <td>99</td>\n",
       "      <td>4.605170</td>\n",
       "      <td>5</td>\n",
       "      <td>1.791759</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  tile_mean_expr  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...       -1.974712   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...       -0.700792   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       -0.886722   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...       -1.445209   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...       -1.062973   \n",
       "\n",
       "   tile_tissue_sp  avg_tf_tissue_sp  log_avg_tf_tissue_sp  numMotifs  \\\n",
       "0        0.386721          0.249383              0.222650         10   \n",
       "1        0.165019          0.416909              0.348478         53   \n",
       "2        0.261225          0.429598              0.357394         24   \n",
       "3        0.265622          0.435871              0.361772         13   \n",
       "4        0.117438          0.462181              0.379929         16   \n",
       "\n",
       "   log_num_motifs  numBPcovered  log_bp_cov  maxCov  log_max_cov  \n",
       "0        2.397895            45    3.828641       6     1.945910  \n",
       "1        3.988984           101    4.624973      24     3.218876  \n",
       "2        3.218876            84    4.442651      12     2.564949  \n",
       "3        2.639057            54    4.007333       9     2.302585  \n",
       "4        2.833213            99    4.605170       5     1.791759  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final = mean_activ[[\"unique_id\", \"tile_mean_expr\", \"tile_tissue_sp\", \"avg_tf_tissue_sp\", \"log_avg_tf_tissue_sp\",\n",
    "                    \"numMotifs\", \"log_num_motifs\", \"numBPcovered\", \"log_bp_cov\", \"maxCov\", \"log_max_cov\"]]\n",
    "final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_id</th>\n",
       "      <th>MPRA_mean_activ</th>\n",
       "      <th>MPRA_tissue_sp</th>\n",
       "      <th>avg_tf_tissue_sp</th>\n",
       "      <th>log_avg_tf_tissue_sp</th>\n",
       "      <th>num_motifs</th>\n",
       "      <th>log_num_motifs</th>\n",
       "      <th>num_bp_covered</th>\n",
       "      <th>log_num_bp_covered</th>\n",
       "      <th>max_coverage</th>\n",
       "      <th>log_max_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...</td>\n",
       "      <td>-1.974712</td>\n",
       "      <td>0.386721</td>\n",
       "      <td>0.249383</td>\n",
       "      <td>0.222650</td>\n",
       "      <td>10</td>\n",
       "      <td>2.397895</td>\n",
       "      <td>45</td>\n",
       "      <td>3.828641</td>\n",
       "      <td>6</td>\n",
       "      <td>1.945910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...</td>\n",
       "      <td>-0.700792</td>\n",
       "      <td>0.165019</td>\n",
       "      <td>0.416909</td>\n",
       "      <td>0.348478</td>\n",
       "      <td>53</td>\n",
       "      <td>3.988984</td>\n",
       "      <td>101</td>\n",
       "      <td>4.624973</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-0.886722</td>\n",
       "      <td>0.261225</td>\n",
       "      <td>0.429598</td>\n",
       "      <td>0.357394</td>\n",
       "      <td>24</td>\n",
       "      <td>3.218876</td>\n",
       "      <td>84</td>\n",
       "      <td>4.442651</td>\n",
       "      <td>12</td>\n",
       "      <td>2.564949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...</td>\n",
       "      <td>-1.445209</td>\n",
       "      <td>0.265622</td>\n",
       "      <td>0.435871</td>\n",
       "      <td>0.361772</td>\n",
       "      <td>13</td>\n",
       "      <td>2.639057</td>\n",
       "      <td>54</td>\n",
       "      <td>4.007333</td>\n",
       "      <td>9</td>\n",
       "      <td>2.302585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...</td>\n",
       "      <td>-1.062973</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.462181</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>16</td>\n",
       "      <td>2.833213</td>\n",
       "      <td>99</td>\n",
       "      <td>4.605170</td>\n",
       "      <td>5</td>\n",
       "      <td>1.791759</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           unique_id  MPRA_mean_activ  \\\n",
       "0  CONTROL_BUT_HAS_SNP__control.flip.sense__chr11...        -1.974712   \n",
       "1  CONTROL_BUT_HAS_SNP__control.flip.sense__chr16...        -0.700792   \n",
       "2  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...        -0.886722   \n",
       "3  CONTROL_BUT_HAS_SNP__control.flip.sense__chr17...        -1.445209   \n",
       "4  CONTROL_BUT_HAS_SNP__control.flip.sense__chr1:...        -1.062973   \n",
       "\n",
       "   MPRA_tissue_sp  avg_tf_tissue_sp  log_avg_tf_tissue_sp  num_motifs  \\\n",
       "0        0.386721          0.249383              0.222650          10   \n",
       "1        0.165019          0.416909              0.348478          53   \n",
       "2        0.261225          0.429598              0.357394          24   \n",
       "3        0.265622          0.435871              0.361772          13   \n",
       "4        0.117438          0.462181              0.379929          16   \n",
       "\n",
       "   log_num_motifs  num_bp_covered  log_num_bp_covered  max_coverage  \\\n",
       "0        2.397895              45            3.828641             6   \n",
       "1        3.988984             101            4.624973            24   \n",
       "2        3.218876              84            4.442651            12   \n",
       "3        2.639057              54            4.007333             9   \n",
       "4        2.833213              99            4.605170             5   \n",
       "\n",
       "   log_max_coverage  \n",
       "0          1.945910  \n",
       "1          3.218876  \n",
       "2          2.564949  \n",
       "3          2.302585  \n",
       "4          1.791759  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final.columns = [\"unique_id\", \"MPRA_mean_activ\", \"MPRA_tissue_sp\", \"avg_tf_tissue_sp\", \"log_avg_tf_tissue_sp\",\n",
    "                 \"num_motifs\", \"log_num_motifs\", \"num_bp_covered\", \"log_num_bp_covered\", \"max_coverage\",\n",
    "                 \"log_max_coverage\"]\n",
    "final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_dir = \"../../data/04__coverage\"\n",
    "!mkdir -p $out_dir\n",
    "final.to_csv(\"%s/motif_coverage.txt\" % out_dir, sep=\"\\t\", index=False)\n",
    "tf_expr.to_csv(\"%s/tf_tissue_sp.txt\" % out_dir, sep=\"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
