{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.patheffects\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sys\n",
    "\n",
    "from decimal import Decimal\n",
    "from matplotlib import gridspec\n",
    "from mpl_toolkits.axes_grid1 import make_axes_locatable\n",
    "from matplotlib.font_manager import FontProperties\n",
    "from matplotlib import transforms\n",
    "from scipy import stats\n",
    "from scipy.spatial import distance\n",
    "from scipy.cluster import hierarchy\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../utils\")\n",
    "from misc_utils import *\n",
    "from norm_utils import *\n",
    "from snp_utils import *\n",
    "from del_utils import *\n",
    "\n",
    "mpl.rcParams['figure.dpi'] = 90"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## style pre-sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NOTEBOOK_PRESET = {\"style\": \"white\", \"font\": \"Helvetica\", \"font_scale\": 1.2, \"context\": \"notebook\"}\n",
    "NOTEBOOK_FONTSIZE = 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PAPER_PRESET = {\"style\": \"ticks\", \"font\": \"Helvetica\", \"context\": \"paper\", \n",
    "                \"rc\": {\"font.size\":7,\"axes.titlesize\":7,\n",
    "                       \"axes.labelsize\":7, 'axes.linewidth':0.5,\n",
    "                       \"legend.fontsize\":6, \"xtick.labelsize\":6,\n",
    "                       \"ytick.labelsize\":6, \"xtick.major.size\": 3.0,\n",
    "                       \"ytick.major.size\": 3.0, \"axes.edgecolor\": \"black\",\n",
    "                       \"xtick.major.pad\": 3.0, \"ytick.major.pad\": 3.0}}\n",
    "PAPER_FONTSIZE = 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## palette pre-sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "husl = sns.color_palette(\"husl\", 9)\n",
    "BETTER_TYPE_PALETTE = {\"CONTROL\": husl[3], \"CONTROL_SNP\": husl[4], \"WILDTYPE\": husl[5], \"FLIPPED\": husl[6], \n",
    "                       \"SNP\": husl[7], \"DELETION\": husl[0], \"SCRAMBLED\": \"lightgray\", \"RANDOM\": \"darkgray\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "TSS_CLASS_PALETTE = {\"Enhancer\": sns.color_palette(\"deep\")[1], \n",
    "                     \"intergenic\": sns.color_palette(\"deep\")[2], \"protein_coding\": sns.color_palette(\"deep\")[5], \n",
    "                     \"div_lnc\": sns.color_palette(\"deep\")[3], \"div_pc\": sns.color_palette(\"deep\")[0]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "COLOR_DICT = {\"A\": \"crimson\", \"C\": \"mediumblue\", \"G\": \"orange\", \"T\": \"forestgreen\"}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## label pre-sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "BETTER_TYPE_ORDER1 = [\"CONTROL\", \"CONTROL_SNP\", \"WILDTYPE\", \"FLIPPED\", \"SNP\", \"SCRAMBLED\", \"RANDOM\"]\n",
    "BETTER_TYPE_ORDER2 = [\"CONTROL\", \"CONTROL_SNP\", \"WILDTYPE\", \"FLIPPED\", \"SNP\", \"DELETION\", \"SCRAMBLED\", \"RANDOM\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "TSS_CLASS_ORDER = [\"Enhancer\", \"intergenic\", \"div_lnc\", \"protein_coding\", \"div_pc\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "class Scale(matplotlib.patheffects.RendererBase):\n",
    "    def __init__(self, sx, sy=None):\n",
    "        self._sx = sx\n",
    "        self._sy = sy\n",
    "\n",
    "    def draw_path(self, renderer, gc, tpath, affine, rgbFace):\n",
    "        affine = affine.identity().scale(self._sx, self._sy)+affine\n",
    "        renderer.draw_path(gc, tpath, affine, rgbFace)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## plotting functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def mimic_r_boxplot(ax):\n",
    "    for i, patch in enumerate(ax.artists):\n",
    "        r, g, b, a = patch.get_facecolor()\n",
    "        col = (r, g, b, 1)\n",
    "        patch.set_facecolor((r, g, b, .5))\n",
    "        patch.set_edgecolor((r, g, b, 1))\n",
    "\n",
    "        # Each box has 6 associated Line2D objects (to make the whiskers, fliers, etc.)\n",
    "        # Loop over them here, and use the same colour as above\n",
    "        line_order = [\"lower\", \"upper\", \"whisker_1\", \"whisker_2\", \"med\", \"fliers\"]\n",
    "        for j in range(i*6,i*6+6):\n",
    "            elem = line_order[j%6]\n",
    "            line = ax.lines[j]\n",
    "            if \"whisker\" in elem:\n",
    "                line.set_visible(False)\n",
    "            line.set_color(col)\n",
    "            line.set_mfc(col)\n",
    "            line.set_mec(col)\n",
    "            if \"fliers\" in elem:\n",
    "                line.set_alpha(0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def annotate_pval(ax, x1, x2, y, h, text_y, val, fontsize):\n",
    "    from decimal import Decimal\n",
    "    ax.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=1, c=\"black\", linewidth=0.5)\n",
    "    if val < 0.0001:\n",
    "        #text = \"{:.2e}\".format(Decimal(val))\n",
    "        text = \"**\"\n",
    "    elif val < 0.05:\n",
    "        #text = \"%.4f\" % val\n",
    "        text = \"*\"\n",
    "    else:\n",
    "        text = \"n.s.\"\n",
    "    ax.text((x1+x2)*.5, text_y, text, ha='center', va='bottom', color=\"black\", size=fontsize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def neg_control_plot(df, order, palette, fontsize, cell_type, ax, figsize, ylabel, sharey, title, save, plotname):\n",
    "    df_sub = df[df[\"better_type\"].isin([\"WILDTYPE\", \"RANDOM\", \"SCRAMBLED\"])].drop_duplicates()\n",
    "    \n",
    "    if ax == None:\n",
    "        plt.figure(figsize=figsize)\n",
    "        ax = sns.boxplot(data=df_sub, x=\"better_type\", y=\"overall_mean\", order=order, palette=palette, linewidth=1,\n",
    "                         saturation=1, flierprops = dict(marker='o', markersize=5))\n",
    "    else:\n",
    "        sns.boxplot(data=df_sub, x=\"better_type\", y=\"overall_mean\", order=order, palette=palette, linewidth=1,\n",
    "                    saturation=1, flierprops = dict(marker='o', markersize=5), ax=ax)\n",
    "    ax.set_xticklabels([\"random\", \"scrambled\", \"reference\"], rotation=30)\n",
    "\n",
    "    mimic_r_boxplot(ax)\n",
    "\n",
    "    # calc p-vals b/w dists\n",
    "    rand_dist = np.asarray(df[df[\"better_type\"] == \"RANDOM\"][\"overall_mean\"])\n",
    "    scram_dist = np.asarray(df[df[\"better_type\"] == \"SCRAMBLED\"][\"overall_mean\"])\n",
    "    wt_dist = np.asarray(df[df[\"better_type\"] == \"WILDTYPE\"][\"overall_mean\"])\n",
    "\n",
    "    rand_dist = rand_dist[~np.isnan(rand_dist)]\n",
    "    scram_dist = scram_dist[~np.isnan(scram_dist)]\n",
    "    wt_dist = wt_dist[~np.isnan(wt_dist)]\n",
    "\n",
    "    rand_u, rand_pval = stats.mannwhitneyu(rand_dist, wt_dist, alternative=\"two-sided\", use_continuity=False)\n",
    "    scram_u, scram_pval = stats.mannwhitneyu(scram_dist, wt_dist, alternative=\"two-sided\", use_continuity=False)\n",
    "    \n",
    "    if sharey:\n",
    "        ax.set_ylim((-10, 10))\n",
    "        ax.yaxis.set_ticks(np.arange(-10, 11, 5))\n",
    "        y_1 = 8\n",
    "        y_2 = 6\n",
    "        text_y_1 = 7.5\n",
    "        text_y_2 = 5.5\n",
    "    else:\n",
    "        ax.set_ylim((np.min(rand_dist)-2, np.max(wt_dist)+3.5))\n",
    "        y_1 = np.max(wt_dist)+1.85\n",
    "        y_2 = np.max(wt_dist)+0.75\n",
    "        text_y_1 = np.max(wt_dist)+1.65\n",
    "        text_y_2 = np.max(wt_dist)+0.55\n",
    "    \n",
    "    # statistical annotation\n",
    "    annotate_pval(ax, 0, 2, y_1, 0, text_y_1, rand_pval, fontsize)\n",
    "    annotate_pval(ax, 1, 2, y_2, 0, text_y_2, scram_pval, fontsize)\n",
    "    \n",
    "    ax.set_ylabel(ylabel)\n",
    "    ax.set_xlabel(\"\")\n",
    "    if title:\n",
    "        ax.set_title(\"%s\" % (cell_type))\n",
    "    if save:\n",
    "        plt.savefig(\"%s/%s.pdf\" % (figs_dir, plotname), dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def promtype_plot(df, order, palette, fontsize, cell_type, ax, figsize, ylabel, sharey, title, save, plotname):  \n",
    "    palette = {\"Enhancer\": sns.color_palette(\"deep\")[1], \n",
    "               \"intergenic\": sns.color_palette(\"deep\")[2], \"protein_coding\": sns.color_palette(\"deep\")[5], \n",
    "               \"div_lnc\": sns.color_palette(\"deep\")[3], \"div_pc\": sns.color_palette(\"deep\")[0]}\n",
    "    order = [\"Enhancer\", \"intergenic\", \"div_lnc\", \"protein_coding\", \"div_pc\"]\n",
    "    \n",
    "    if ax == None:\n",
    "        plt.figure(figsize=figsize)\n",
    "        ax = sns.boxplot(data=df, x=\"PromType2\", y=\"overall_mean\", order=order, palette=palette, linewidth=1,\n",
    "                         saturation=1, flierprops = dict(marker='o', markersize=5))\n",
    "    else:\n",
    "        sns.boxplot(data=df, x=\"PromType2\", y=\"overall_mean\", order=order, palette=palette, linewidth=1,\n",
    "                    saturation=1, flierprops = dict(marker='o', markersize=5), ax=ax)\n",
    "    ax.set_xticklabels([\"enhancers\", \"lincRNAs\", \"div. lncRNAs\", \"mRNAs\", \"div. mRNAs\"], rotation=30)\n",
    "\n",
    "    mimic_r_boxplot(ax)\n",
    "\n",
    "    # calc p-vals b/w dists\n",
    "    enh_dist = np.asarray(df[df[\"PromType2\"] == \"Enhancer\"][\"overall_mean\"])\n",
    "    int_dist = np.asarray(df[df[\"PromType2\"] == \"intergenic\"][\"overall_mean\"])\n",
    "    div_lnc_dist = np.asarray(df[df[\"PromType2\"] == \"div_lnc\"][\"overall_mean\"])\n",
    "    pc_dist = np.asarray(df[df[\"PromType2\"] == \"protein_coding\"][\"overall_mean\"])\n",
    "    div_pc_dist = np.asarray(df[df[\"PromType2\"] == \"div_pc\"][\"overall_mean\"])\n",
    "\n",
    "    enh_dist = enh_dist[~np.isnan(enh_dist)]\n",
    "    int_dist = int_dist[~np.isnan(int_dist)]\n",
    "    div_lnc_dist = div_lnc_dist[~np.isnan(div_lnc_dist)]\n",
    "    pc_dist = pc_dist[~np.isnan(pc_dist)]\n",
    "    div_pc_dist = div_pc_dist[~np.isnan(div_pc_dist)]\n",
    "    \n",
    "    all_dists = list(enh_dist) + list(int_dist) + list(div_lnc_dist) + list(pc_dist) + list(div_pc_dist)\n",
    "\n",
    "    lnc_u, lnc_pval = stats.mannwhitneyu(int_dist, div_lnc_dist, alternative=\"two-sided\", use_continuity=False)\n",
    "    pc_u, pc_pval = stats.mannwhitneyu(pc_dist, div_pc_dist, alternative=\"two-sided\", use_continuity=False)\n",
    "\n",
    "    \n",
    "    if sharey:\n",
    "        ax.set_ylim((-10, 10))\n",
    "        ax.yaxis.set_ticks(np.arange(-10, 11, 5))\n",
    "        y_1 = 6\n",
    "        y_2 = 6\n",
    "        text_y_1 = 5.75\n",
    "        text_y_2 = 5.75\n",
    "    else:\n",
    "        ax.set_ylim((np.min(all_dists)-2, np.max(all_dists)+3.5))\n",
    "        y_1 = np.max(pc_dist)+1\n",
    "        y_2 = np.max(pc_dist)+1\n",
    "        text_y_1 = np.max(pc_dist)+0.75\n",
    "        text_y_2 = np.max(pc_dist)+0.75\n",
    "    \n",
    "    # statistical annotation\n",
    "    annotate_pval(ax, 1, 2, y_1, 0, text_y_1, lnc_pval, fontsize)\n",
    "    annotate_pval(ax, 3, 4, y_2, 0, text_y_2, pc_pval, fontsize)\n",
    "    \n",
    "    ax.set_ylabel(ylabel)\n",
    "    ax.set_xlabel(\"\")\n",
    "    if title:\n",
    "        ax.set_title(\"%s\" % (cell_type))\n",
    "    if save:\n",
    "        plt.savefig(\"%s/%s.pdf\" % (figs_dir, plotname), dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def plot_dendrogram(linkage, max_dist, title):\n",
    "    \n",
    "    plt.figure(figsize=(25, 8))\n",
    "    dg = hierarchy.dendrogram(linkage, show_leaf_counts=True)\n",
    "\n",
    "    dists = []\n",
    "    for i, d, c in zip(dg['icoord'], dg['dcoord'], dg['color_list']):\n",
    "                x = 0.5 * sum(i[1:3])\n",
    "                y = d[1]\n",
    "                plt.plot(x, y, 'o', c=c)\n",
    "                if y > max_dist:\n",
    "                    plt.annotate(\"%.3g\" % y, (x, y), xytext=(0, -5),\n",
    "                                 textcoords='offset points',\n",
    "                                 va='top', ha='center')\n",
    "                dists.append(y)\n",
    "\n",
    "    plt.axhline(y=max_dist)\n",
    "    plt.title(title)\n",
    "    plt.show()\n",
    "    return dists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def pearsonfunc(x, y, **kws):\n",
    "    r, p = stats.pearsonr(x, y)\n",
    "    ax = plt.gca()\n",
    "    ax.annotate(\"pearson r = {:.2f}\\np = {:.2e}\".format(r, Decimal(p)),\n",
    "                xy=(.1, .9), xycoords=ax.transAxes)\n",
    "\n",
    "def spearmanfunc(x, y, **kws):\n",
    "    r, p = stats.spearmanr(x, y)\n",
    "    ax = plt.gca()\n",
    "    ax.annotate(\"spearman r = {:.2f}\\np = {:.2e}\".format(r, Decimal(p)),\n",
    "                xy=(.1, .9), xycoords=ax.transAxes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def plot_peaks_and_tfbs(figsize, seq_len, seq_name, cell, scores, yerrs, motif_vals, bases, plotname, save):\n",
    "    \n",
    "    fig = plt.figure(figsize=figsize)\n",
    "    gs = gridspec.GridSpec(3, 1, height_ratios=[4, 3, 1], hspace=0.2)\n",
    "    peak_ax = plt.subplot(gs[0])\n",
    "    motif_ax = plt.subplot(gs[1])\n",
    "    \n",
    "    # plot deletion values\n",
    "    xs = list(range(0, seq_len))\n",
    "    peak_ax.bar(xs, scores, yerr=yerrs, color=\"lightgray\", edgecolor=\"gray\", linewidth=0.5, ecolor=\"gray\", \n",
    "                error_kw={\"elinewidth\": 0.75})\n",
    "    \n",
    "    # labels\n",
    "    peak_ax.set_xlim((-0.5, seq_len))\n",
    "    peak_ax.set_xlabel(\"\")\n",
    "    peak_ax.set_ylabel(\"log2(del/WT)\", fontsize=5)\n",
    "    peak_ax.xaxis.set_visible(False)\n",
    "    peak_ax.set_title(\"filtered scores and peaks: %s (%s)\" % (seq_name, cell))\n",
    "    \n",
    "    # plot motif nums\n",
    "    xs = list(range(0, seq_len))\n",
    "    max_motif_val = np.nanmax(np.abs(motif_vals))\n",
    "    motif_ax.axhline(y=0, color=\"darkgrey\", linewidth=0.5, linestyle=\"dashed\")\n",
    "    motif_ax.plot(xs, motif_vals, color=\"black\", linewidth=0.75, zorder=10)\n",
    "    \n",
    "    # labels\n",
    "    motif_ax.set_xlim((-0.5, seq_len))\n",
    "    motif_ax.set_ylim((-max_motif_val-1, max_motif_val+1))\n",
    "    motif_ax.set_xlabel(\"nucleotide number\")\n",
    "    motif_ax.set_ylabel(r'$\\Delta$ motifs', fontsize=5)\n",
    "    motif_ax.xaxis.set_visible(False)\n",
    "    \n",
    "    plt.show()\n",
    "    if save:\n",
    "        fig.savefig(\"%s.pdf\" % (plotname), dpi=\"figure\", bbox_inches=\"tight\", transparent=True)\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def paired_swarmplots_w_pval(n_rows, n_cols, figsize, snp_df, data_df, fontsize, figs_dir, plotname, save):\n",
    "    fig, axarr = plt.subplots(figsize=figsize, squeeze=False)\n",
    "    pal = {\"ref\": \"grey\", \"alt\": sns.color_palette()[2]}\n",
    "    median_width = 0.3\n",
    "    \n",
    "    # make axes objects\n",
    "    axes = []\n",
    "    counter = 0\n",
    "    for r in range(n_rows):\n",
    "        for c in range(n_cols):\n",
    "            if counter < len(snp_df):\n",
    "                ax = plt.subplot2grid((n_rows, n_cols), (r, c))\n",
    "                axes.append(ax)\n",
    "            counter += 1\n",
    "\n",
    "    # add plots\n",
    "    counter = 0\n",
    "    for i, row in snp_df.iterrows():\n",
    "        ax = axes[counter]\n",
    "        wt_id = row.wt_id\n",
    "        snp_id = row.unique_id\n",
    "        df = data_df[data_df[\"unique_id\"].isin([wt_id, snp_id])]\n",
    "        df = df.sort_values(by=\"wt_or_snp\", ascending=False)\n",
    "        if not \"NA\" in str(row.combined_padj) and not pd.isnull(row.combined_padj):\n",
    "            sns.swarmplot(data=df, x=\"wt_or_snp\", y=\"rep_mean\", ax=ax, palette=pal)\n",
    "            \n",
    "            for tick, text in zip(ax.get_xticks(), ax.get_xticklabels()):\n",
    "                snp = text.get_text()\n",
    "\n",
    "                # calculate the median value for all replicates of either X or Y\n",
    "                median_val = df[df[\"wt_or_snp\"]==snp][\"rep_mean\"].median()\n",
    "\n",
    "                # plot horizontal lines across the column, centered on the tick\n",
    "                ax.plot([tick-median_width/2, tick+median_width/2], [median_val, median_val],\n",
    "                        lw=2, color='k', zorder=10)\n",
    "            \n",
    "        else:\n",
    "            sns.swarmplot(data=df, x=\"wt_or_snp\", y=\"rep_mean\", ax=ax, color=\"lightgray\")\n",
    "            \n",
    "            for tick, text in zip(ax.get_xticks(), ax.get_xticklabels()):\n",
    "                snp = text.get_text()\n",
    "\n",
    "                # calculate the median value for all replicates of either X or Y\n",
    "                median_val = df[df[\"wt_or_snp\"]==snp][\"rep_mean\"].median()\n",
    "\n",
    "                # plot horizontal lines across the column, centered on the tick\n",
    "                ax.plot([tick-median_width/2, tick+median_width/2], [median_val, median_val],\n",
    "                        lw=2, color='k', zorder=10)\n",
    "        if len(row.SNP) > 50:\n",
    "            ax.set_title(\"SNP: long haplotype\", fontsize=fontsize)\n",
    "        else:\n",
    "            ax.set_title(\"SNP: %s\" % row.SNP, fontsize=fontsize)\n",
    "        ax.set_ylim((df.rep_mean.min()-2, df.rep_mean.max()+3))\n",
    "        ax.set_ylabel(\"\")\n",
    "        ax.set_xlabel(\"\")\n",
    "\n",
    "        # statistical annotation\n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h, col = df[\"rep_mean\"].max() + 0.75, 0, \"black\"\n",
    "        ax.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=0.5, c=col)\n",
    "        if not \"NA\" in str(row.combined_padj) and not pd.isnull(row.combined_padj):\n",
    "            if row.combined_padj < 0.0001:\n",
    "                #text = \"{:.2e}\".format(Decimal(val))\n",
    "                text = \"**\"\n",
    "            elif row.combined_padj < 0.05:\n",
    "                #text = \"%.4f\" % val\n",
    "                text = \"*\"\n",
    "            else:\n",
    "                text = \"n.s.\"\n",
    "        else:\n",
    "            text = \"tile activities not sig\"\n",
    "        ax.text((x1+x2)*.5, y+h, text, ha='center', va='bottom', color=col, size=fontsize)\n",
    "            \n",
    "\n",
    "        counter += 1\n",
    "\n",
    "    plt.tight_layout()\n",
    "    plt.subplots_adjust(wspace = .3)\n",
    "    plt.show()\n",
    "    if save:\n",
    "        fig.savefig(\"%s/%s.pdf\" % (figs_dir, plotname), dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def plot_peaks_and_snps(figsize, seq_len, seq_name, widths, scores, yerrs, scaled_scores, snp_vals, snp_sigs, bases, plotname, figs_dir, save):\n",
    "    sns.set(style=\"ticks\", font=\"Helvetica\", context=\"paper\", rc={\"font.size\":7,\"axes.titlesize\":7,\n",
    "                                                              \"axes.labelsize\":7, 'axes.linewidth':0.5,\n",
    "                                                              \"legend.fontsize\":6, \"xtick.labelsize\":6,\n",
    "                                                              \"ytick.labelsize\":6, \"xtick.major.size\": 3.0,\n",
    "                                                              \"ytick.major.size\": 3.0, \"axes.edgecolor\": \"black\",\n",
    "                                                              \"xtick.major.pad\": 3.0, \"ytick.major.pad\": 3.0})\n",
    "    \n",
    "    snp_pal = {\"sig\": \"firebrick\", \"not sig\": \"darkgray\", \"NA__too_many_rep_NAs\": \"darkgray\", \"NA\": \"white\"}\n",
    "    \n",
    "    fig = plt.figure(figsize=figsize)\n",
    "    gs = gridspec.GridSpec(3, 1, height_ratios=[5, 1, 2], hspace=0.1)\n",
    "    peak_ax = plt.subplot(gs[0])\n",
    "    snp_ax = plt.subplot(gs[2])\n",
    "    seq_ax = plt.subplot(gs[1])\n",
    "    \n",
    "    ### peaks figure ###\n",
    "    # plot peak locations\n",
    "    for w in widths:\n",
    "        peak_ax.axvline(x=w[0], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        snp_ax.axvline(x=w[0], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        peak_ax.axvline(x=w[1], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        snp_ax.axvline(x=w[1], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        peak_ax.axvspan(w[0], w[1], alpha=0.5, color=\"gainsboro\", zorder=1)\n",
    "        snp_ax.axvspan(w[0], w[1], alpha=0.5, color=\"gainsboro\", zorder=1)\n",
    "    \n",
    "    # plot deletion values\n",
    "    xs = list(range(0, seq_len))\n",
    "    peak_ax.bar(xs, scores, yerr=yerrs, color=\"lightgray\", edgecolor=\"gray\", linewidth=0.5, ecolor=\"gray\", \n",
    "                error_kw={\"elinewidth\": 0.75})\n",
    "    \n",
    "    # labels\n",
    "    peak_ax.set_xlim((-0.5, seq_len))\n",
    "    peak_ax.set_xlabel(\"\")\n",
    "    peak_ax.set_ylabel(\"log2(del/WT)\")\n",
    "    peak_ax.xaxis.set_visible(False)\n",
    "    peak_ax.set_title(seq_name)\n",
    "    \n",
    "    # plot snp values\n",
    "    xs = list(range(0, seq_len))\n",
    "    snp_colors = [snp_pal[x] for x in snp_sigs]\n",
    "    snp_ax.scatter(xs, snp_vals, s=12, color=snp_colors, edgecolor=\"black\", linewidth=0.5, zorder=10)\n",
    "    for i in range(seq_len):\n",
    "        l2fc = snp_vals[i]\n",
    "        snp_ax.plot([i, i], [0, l2fc], lw=1, color=\"k\")\n",
    "    snp_ax.axhline(y=0, lw=1, color=\"k\", zorder=1)\n",
    "    \n",
    "    # labels\n",
    "    snp_ax.set_xlim((-0.5, seq_len))\n",
    "    snp_ax.set_xlabel(\"nucleotide number\")\n",
    "    snp_ax.set_ylabel(\"log2(alt/ref)\")\n",
    "    snp_ax.xaxis.set_visible(False)\n",
    "    \n",
    "    ### seq logo ###\n",
    "    mpl.rcParams[\"font.family\"] = \"Arial\"\n",
    "    scaled_scores = scale_range(scaled_scores, 0.5, 2.0)\n",
    "    \n",
    "    font = FontProperties()\n",
    "    font.set_size(6)\n",
    "    font.set_weight(\"bold\")\n",
    "    \n",
    "    seq_ax.set_xticks(range(1,len(scaled_scores)+1))\n",
    "    seq_ax.set_ylim((0, 2))\n",
    "    seq_ax.axis(\"off\")\n",
    "    trans_offset = transforms.offset_copy(seq_ax.transData, \n",
    "                                          fig=fig, \n",
    "                                          x=1, \n",
    "                                          y=0, \n",
    "                                          units=\"dots\")\n",
    "    \n",
    "    for i in range(0, len(scaled_scores)):\n",
    "        score = scaled_scores[i]\n",
    "        base = bases[i]\n",
    "        color = COLOR_DICT[base]\n",
    "        txt = seq_ax.text(i+0.25, 0, base, transform=trans_offset,fontsize=6, color=color, \n",
    "                          ha=\"center\", fontproperties=font)\n",
    "        txt.set_path_effects([Scale(1.0, score)])\n",
    "        fig.canvas.draw()\n",
    "        trans_offset = transforms.offset_copy(seq_ax.transData, fig=fig, x=1, y=0, units='points')\n",
    "    \n",
    "    #plt.tight_layout()\n",
    "    plt.show()\n",
    "    if save:\n",
    "        fig.savefig(\"%s/%s\" % (figs_dir, plotname), dpi=\"figure\", bbox_inches=\"tight\")\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def getOverlap(a, b):\n",
    "    return max(a[0], b[0]) - min(a[1], b[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def plot_peaks_and_fimo(figsize, seq_len, seq_name, widths, scores, yerrs, scaled_scores, bases, motif_pos, motif_names, plotname, figs_dir, save):\n",
    "     \n",
    "    fig = plt.figure(figsize=figsize)\n",
    "    gs = gridspec.GridSpec(3, 1, height_ratios=[5, 1, 2], hspace=0.1)\n",
    "    peak_ax = plt.subplot(gs[0])\n",
    "    seq_ax = plt.subplot(gs[1])\n",
    "    motif_ax = plt.subplot(gs[2])\n",
    "    \n",
    "    ### peaks figure ###\n",
    "    # plot peak locations\n",
    "    for w in widths:\n",
    "        peak_ax.axvline(x=w[0], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        peak_ax.axvline(x=w[1], color=\"gray\", linestyle=\"solid\", linewidth=0.5, zorder=1)\n",
    "        peak_ax.axvspan(w[0], w[1], alpha=0.5, color=\"gainsboro\", zorder=1)\n",
    "    \n",
    "    # plot deletion values\n",
    "    xs = list(range(0, seq_len))\n",
    "    peak_ax.bar(xs, scores, yerr=yerrs, color=\"lightgray\", edgecolor=\"gray\", linewidth=0.5, ecolor=\"gray\", \n",
    "                error_kw={\"elinewidth\": 0.75})\n",
    "    \n",
    "    # labels\n",
    "    peak_ax.set_xlim((-0.5, seq_len))\n",
    "    peak_ax.set_xlabel(\"\")\n",
    "    peak_ax.set_ylabel(\"log2(del/WT)\")\n",
    "    peak_ax.xaxis.set_visible(False)\n",
    "    peak_ax.set_title(seq_name)\n",
    "    \n",
    "    # plot motif locations\n",
    "    xs = list(range(0, seq_len))\n",
    "    prev_plotted = {}\n",
    "    \n",
    "    # iterate through things plotted at each prev_y value\n",
    "    # if any overlaps, move\n",
    "    for i, pos in enumerate(motif_pos):\n",
    "        #print(\"\")\n",
    "        #print(\"i: %s, pos: %s\" % (i, pos))\n",
    "        plotted = False\n",
    "        if i == 0:\n",
    "            #print(\"first motif, plotting at y=0\")\n",
    "            motif_ax.plot([pos[0], pos[1]], [0, 0], color=\"darkgrey\", linewidth=2, solid_capstyle=\"butt\")\n",
    "            plotted = True\n",
    "            prev_plotted[0] = [pos]\n",
    "            continue\n",
    "        for prev_y in sorted(prev_plotted.keys(), reverse=True):\n",
    "            vals = prev_plotted[prev_y]\n",
    "            overlaps = []\n",
    "            for prev_pos in vals:\n",
    "                overlaps.append(getOverlap(prev_pos, pos))\n",
    "            if any(x < 0 for x in overlaps):\n",
    "                #print(\"motif overlaps w/ %s, continuing\" % (prev_y))\n",
    "                continue\n",
    "            else:\n",
    "                if not plotted:\n",
    "                    #print(\"motif doesn't overlap anything at y=%s, plotting\" % prev_y)\n",
    "                    motif_ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=2, \n",
    "                                  solid_capstyle=\"butt\")\n",
    "                    if prev_y not in prev_plotted:\n",
    "                        prev_plotted[prev_y] = [pos]\n",
    "                    else:\n",
    "                        new_vals = list(prev_plotted[prev_y])\n",
    "                        new_vals.extend([pos])\n",
    "                        prev_plotted[prev_y] = new_vals\n",
    "                    plotted = True\n",
    "        if not plotted:\n",
    "            prev_y -= 0.25\n",
    "            #print(\"motif overlaps at all prev_y, plotting at %s\" % prev_y)\n",
    "            motif_ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=2, \n",
    "                          solid_capstyle=\"butt\")\n",
    "            if prev_y not in prev_plotted:\n",
    "                prev_plotted[prev_y] = [pos]\n",
    "            else:\n",
    "                new_vals = list(prev_plotted[prev_y])\n",
    "                new_vals.extend([pos])\n",
    "                prev_plotted[prev_y] = new_vals\n",
    "            plotted = True\n",
    "        #print(prev_plotted)\n",
    "        \n",
    "    min_y = np.min(list(prev_plotted.keys()))\n",
    "\n",
    "    # labels\n",
    "    motif_ax.set_xlim((-0.5, seq_len))\n",
    "    motif_ax.set_ylim((min_y - 0.25, 0.25))\n",
    "    motif_ax.set_xlabel(\"nucleotide number\")\n",
    "    motif_ax.set_ylabel(\"\")\n",
    "    motif_ax.xaxis.set_visible(False)\n",
    "    motif_ax.yaxis.set_visible(False)\n",
    "    motif_ax.axis(\"off\")\n",
    "    \n",
    "    ### seq logo ###\n",
    "    mpl.rcParams[\"font.family\"] = \"Arial\"\n",
    "    scaled_scores = scale_range(scaled_scores, 0.5, 2.0)\n",
    "    \n",
    "    font = FontProperties()\n",
    "    font.set_size(6)\n",
    "    font.set_weight(\"bold\")\n",
    "    \n",
    "    seq_ax.set_xticks(range(1,len(scaled_scores)+1))\n",
    "    seq_ax.set_ylim((0, 2))\n",
    "    seq_ax.axis(\"off\")\n",
    "    trans_offset = transforms.offset_copy(seq_ax.transData, \n",
    "                                          fig=fig, \n",
    "                                          x=1, \n",
    "                                          y=0, \n",
    "                                          units=\"dots\")\n",
    "    \n",
    "    for i in range(0, len(scaled_scores)):\n",
    "        score = scaled_scores[i]\n",
    "        base = bases[i]\n",
    "        color = COLOR_DICT[base]\n",
    "        txt = seq_ax.text(i+0.25, 0, base, transform=trans_offset,fontsize=6, color=color, \n",
    "                          ha=\"center\", fontproperties=font)\n",
    "        txt.set_path_effects([Scale(1.0, score)])\n",
    "        fig.canvas.draw()\n",
    "        trans_offset = transforms.offset_copy(seq_ax.transData, fig=fig, x=1, y=0, units='points')\n",
    "    \n",
    "    #plt.tight_layout()\n",
    "    plt.show()\n",
    "    if save:\n",
    "        fig.savefig(\"%s/%s\" % (figs_dir, plotname), dpi=\"figure\", bbox_inches=\"tight\")\n",
    "    plt.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
